<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Dataset1_paired</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="Dataset1_paired_files/libs/clipboard/clipboard.min.js"></script>
<script src="Dataset1_paired_files/libs/quarto-html/quarto.js" type="module"></script>
<script src="Dataset1_paired_files/libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="Dataset1_paired_files/libs/quarto-html/popper.min.js"></script>
<script src="Dataset1_paired_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Dataset1_paired_files/libs/quarto-html/anchor.min.js"></script>
<link href="Dataset1_paired_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Dataset1_paired_files/libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Dataset1_paired_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Dataset1_paired_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Dataset1_paired_files/libs/bootstrap/bootstrap-bb462d781dde1847d9e3ccf7736099dd.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent quarto-light">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Dataset1_paired</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Setup</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GEOquery)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Lade nötiges Paket: Biobase</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Lade nötiges Paket: BiocGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Lade nötiges Paket: generics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attache Paket: 'generics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Die folgenden Objekte sind maskiert von 'package:base':

    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,
    setequal, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attache Paket: 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Die folgenden Objekte sind maskiert von 'package:stats':

    IQR, mad, sd, var, xtabs</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Die folgenden Objekte sind maskiert von 'package:base':

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,
    unsplit, which.max, which.min</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Setting options('download.file.method.GEOquery'='auto')</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Setting options('GEOquery.inmemory.gpl'=FALSE)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(limma)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attache Paket: 'limma'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Das folgende Objekt ist maskiert 'package:BiocGenerics':

    plotMA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(WGCNA)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Paket 'WGCNA' wurde unter R Version 4.5.2 erstellt</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Lade nötiges Paket: dynamicTreeCut</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Paket 'dynamicTreeCut' wurde unter R Version 4.5.2 erstellt</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Lade nötiges Paket: fastcluster</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Paket 'fastcluster' wurde unter R Version 4.5.2 erstellt</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attache Paket: 'fastcluster'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Das folgende Objekt ist maskiert 'package:stats':

    hclust</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attache Paket: 'WGCNA'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Das folgende Objekt ist maskiert 'package:stats':

    cor</code></pre>
</div>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(iCheck)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Lade nötiges Paket: lumi</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Keine Methoden in Paket 'RSQLite' gefunden für Anforderung: 'dbListFields' beim Laden von 'lumi'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Lade nötiges Paket: gplots</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
---------------------
gplots 3.3.0 loaded:
  * Use citation('gplots') for citation info.
  * Homepage: https://talgalili.github.io/gplots/
  * Report issues: https://github.com/talgalili/gplots/issues
  * Ask questions: https://stackoverflow.com/questions/tagged/gplots
  * Suppress this message with: suppressPackageStartupMessages(library(gplots))
---------------------</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attache Paket: 'gplots'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Das folgende Objekt ist maskiert 'package:stats':

    lowess</code></pre>
</div>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(nlme)</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attache Paket: 'dplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Das folgende Objekt ist maskiert 'package:nlme':

    collapse</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Das folgende Objekt ist maskiert 'package:lumi':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Das folgende Objekt ist maskiert 'package:Biobase':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Die folgenden Objekte sind maskiert von 'package:BiocGenerics':

    combine, intersect, setdiff, setequal, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Das folgende Objekt ist maskiert 'package:generics':

    explain</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Die folgenden Objekte sind maskiert von 'package:stats':

    filter, lag</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Die folgenden Objekte sind maskiert von 'package:base':

    intersect, setdiff, setequal, union</code></pre>
</div>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a><span class="fu">disableWGCNAThreads</span>()</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Load data</span></span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>gse   <span class="ot">&lt;-</span> <span class="fu">getGEO</span>(<span class="st">"GSE45238"</span>, <span class="at">GSEMatrix =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found 1 file(s)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>GSE45238_series_matrix.txt.gz</code></pre>
</div>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>eset  <span class="ot">&lt;-</span> gse[[<span class="dv">1</span>]]</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>expr  <span class="ot">&lt;-</span> <span class="fu">exprs</span>(eset)</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> <span class="fu">pData</span>(eset)</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>annot <span class="ot">&lt;-</span> <span class="fu">fData</span>(eset)</span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(pheno) <span class="ot">&lt;-</span> pheno<span class="sc">$</span>geo_accession</span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(expr); <span class="fu">head</span>(pheno); <span class="fu">head</span>(annot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 858  80</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>                  title geo_accession                status submission_date
GSM1099620 non-OSCC_01N    GSM1099620 Public on Jan 07 2015     Mar 18 2013
GSM1099621 non-OSCC_02N    GSM1099621 Public on Jan 07 2015     Mar 18 2013
GSM1099622 non-OSCC_03N    GSM1099622 Public on Jan 07 2015     Mar 18 2013
GSM1099623 non-OSCC_04N    GSM1099623 Public on Jan 07 2015     Mar 18 2013
GSM1099624 non-OSCC_05N    GSM1099624 Public on Jan 07 2015     Mar 18 2013
GSM1099625 non-OSCC_06N    GSM1099625 Public on Jan 07 2015     Mar 18 2013
           last_update_date type channel_count              source_name_ch1
GSM1099620      Jan 07 2015  RNA             1 adjacent non-tumor epithelia
GSM1099621      Jan 07 2015  RNA             1 adjacent non-tumor epithelia
GSM1099622      Jan 07 2015  RNA             1 adjacent non-tumor epithelia
GSM1099623      Jan 07 2015  RNA             1 adjacent non-tumor epithelia
GSM1099624      Jan 07 2015  RNA             1 adjacent non-tumor epithelia
GSM1099625      Jan 07 2015  RNA             1 adjacent non-tumor epithelia
           organism_ch1 characteristics_ch1 characteristics_ch1.1
GSM1099620 Homo sapiens        gender: male               age: 46
GSM1099621 Homo sapiens        gender: male               age: 49
GSM1099622 Homo sapiens        gender: male               age: 56
GSM1099623 Homo sapiens        gender: male               age: 50
GSM1099624 Homo sapiens        gender: male               age: 36
GSM1099625 Homo sapiens        gender: male               age: 48
           characteristics_ch1.2  characteristics_ch1.3 molecule_ch1
GSM1099620             Stage: II tissue: oral epithelia    total RNA
GSM1099621             Stage: II tissue: oral epithelia    total RNA
GSM1099622            Stage: III tissue: oral epithelia    total RNA
GSM1099623            Stage: III tissue: oral epithelia    total RNA
GSM1099624             Stage: II tissue: oral epithelia    total RNA
GSM1099625            Stage: IVA tissue: oral epithelia    total RNA
                                                                                                                                      extract_protocol_ch1
GSM1099620 Total RNA was extracted  from 40 pairs of OSCC patients using the miRNeasy Mini Kit (Qiagen, #217004) according to the manufacturer’s protocol.
GSM1099621 Total RNA was extracted  from 40 pairs of OSCC patients using the miRNeasy Mini Kit (Qiagen, #217004) according to the manufacturer’s protocol.
GSM1099622 Total RNA was extracted  from 40 pairs of OSCC patients using the miRNeasy Mini Kit (Qiagen, #217004) according to the manufacturer’s protocol.
GSM1099623 Total RNA was extracted  from 40 pairs of OSCC patients using the miRNeasy Mini Kit (Qiagen, #217004) according to the manufacturer’s protocol.
GSM1099624 Total RNA was extracted  from 40 pairs of OSCC patients using the miRNeasy Mini Kit (Qiagen, #217004) according to the manufacturer’s protocol.
GSM1099625 Total RNA was extracted  from 40 pairs of OSCC patients using the miRNeasy Mini Kit (Qiagen, #217004) according to the manufacturer’s protocol.
           label_ch1
GSM1099620       Cy3
GSM1099621       Cy3
GSM1099622       Cy3
GSM1099623       Cy3
GSM1099624       Cy3
GSM1099625       Cy3
                                                                   label_protocol_ch1
GSM1099620 Cy3 labled cRNA (PCR product) were prepared with illumina GoldenGate assay
GSM1099621 Cy3 labled cRNA (PCR product) were prepared with illumina GoldenGate assay
GSM1099622 Cy3 labled cRNA (PCR product) were prepared with illumina GoldenGate assay
GSM1099623 Cy3 labled cRNA (PCR product) were prepared with illumina GoldenGate assay
GSM1099624 Cy3 labled cRNA (PCR product) were prepared with illumina GoldenGate assay
GSM1099625 Cy3 labled cRNA (PCR product) were prepared with illumina GoldenGate assay
           taxid_ch1                             hyb_protocol
GSM1099620      9606 Standard Illumina hybridization protocol
GSM1099621      9606 Standard Illumina hybridization protocol
GSM1099622      9606 Standard Illumina hybridization protocol
GSM1099623      9606 Standard Illumina hybridization protocol
GSM1099624      9606 Standard Illumina hybridization protocol
GSM1099625      9606 Standard Illumina hybridization protocol
                                 scan_protocol description
GSM1099620 Standard Illumina scanning protocol replicate 1
GSM1099621 Standard Illumina scanning protocol replicate 1
GSM1099622 Standard Illumina scanning protocol replicate 1
GSM1099623 Standard Illumina scanning protocol replicate 1
GSM1099624 Standard Illumina scanning protocol replicate 1
GSM1099625 Standard Illumina scanning protocol replicate 1
                                                                       data_processing
GSM1099620 The data were normalised using quantile normalisation with IlluminaGUI in R
GSM1099621 The data were normalised using quantile normalisation with IlluminaGUI in R
GSM1099622 The data were normalised using quantile normalisation with IlluminaGUI in R
GSM1099623 The data were normalised using quantile normalisation with IlluminaGUI in R
GSM1099624 The data were normalised using quantile normalisation with IlluminaGUI in R
GSM1099625 The data were normalised using quantile normalisation with IlluminaGUI in R
           platform_id     contact_name        contact_email
GSM1099620     GPL8179 Shine-Gwo,,Shiah davidssg@nhri.org.tw
GSM1099621     GPL8179 Shine-Gwo,,Shiah davidssg@nhri.org.tw
GSM1099622     GPL8179 Shine-Gwo,,Shiah davidssg@nhri.org.tw
GSM1099623     GPL8179 Shine-Gwo,,Shiah davidssg@nhri.org.tw
GSM1099624     GPL8179 Shine-Gwo,,Shiah davidssg@nhri.org.tw
GSM1099625     GPL8179 Shine-Gwo,,Shiah davidssg@nhri.org.tw
                              contact_department
GSM1099620 National Institute of Cancer Research
GSM1099621 National Institute of Cancer Research
GSM1099622 National Institute of Cancer Research
GSM1099623 National Institute of Cancer Research
GSM1099624 National Institute of Cancer Research
GSM1099625 National Institute of Cancer Research
                             contact_institute   contact_address contact_city
GSM1099620 National Health Research Institutes No. 35, Keyan Rd.       Zhunan
GSM1099621 National Health Research Institutes No. 35, Keyan Rd.       Zhunan
GSM1099622 National Health Research Institutes No. 35, Keyan Rd.       Zhunan
GSM1099623 National Health Research Institutes No. 35, Keyan Rd.       Zhunan
GSM1099624 National Health Research Institutes No. 35, Keyan Rd.       Zhunan
GSM1099625 National Health Research Institutes No. 35, Keyan Rd.       Zhunan
           contact_state contact_zip/postal_code contact_country
GSM1099620 Miaoli County                     350          Taiwan
GSM1099621 Miaoli County                     350          Taiwan
GSM1099622 Miaoli County                     350          Taiwan
GSM1099623 Miaoli County                     350          Taiwan
GSM1099624 Miaoli County                     350          Taiwan
GSM1099625 Miaoli County                     350          Taiwan
           supplementary_file data_row_count age:ch1 gender:ch1 Stage:ch1
GSM1099620               NONE            858      46       male        II
GSM1099621               NONE            858      49       male        II
GSM1099622               NONE            858      56       male       III
GSM1099623               NONE            858      50       male       III
GSM1099624               NONE            858      36       male        II
GSM1099625               NONE            858      48       male       IVA
               tissue:ch1
GSM1099620 oral epithelia
GSM1099621 oral epithelia
GSM1099622 oral epithelia
GSM1099623 oral epithelia
GSM1099624 oral epithelia
GSM1099625 oral epithelia</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>                       ID       SYMBOL    ILMN_Gene   Search_Key
ILMN_3166935 ILMN_3166935 ILMN_3166935  hsa-miR-329  hsa-miR-329
ILMN_3166938 ILMN_3166938 ILMN_3166938  hsa-miR-424  hsa-miR-424
ILMN_3166940 ILMN_3166940 ILMN_3166940  hsa-miR-137  hsa-miR-137
ILMN_3166941 ILMN_3166941 ILMN_3166941 hsa-miR-376c hsa-miR-376c
ILMN_3166943 ILMN_3166943 ILMN_3166943  hsa-miR-585  hsa-miR-585
ILMN_3166944 ILMN_3166944 ILMN_3166944 hsa-miR-302c hsa-miR-302c
                        SEQUENCE        TargetMatureSeqs TargetMatureName
ILMN_3166935 AACACACCTGGTTAACCTC  AACACACCTGGTTAACCTCTTT      hsa-miR-329
ILMN_3166938  CAGCAGCAATTCATGTTT  CAGCAGCAATTCATGTTTTGAA      hsa-miR-424
ILMN_3166940 TATTGCTTAAGAATACGCG TTATTGCTTAAGAATACGCGTAG      hsa-miR-137
ILMN_3166941 ACATAGAGGAAATTCCACG   AACATAGAGGAAATTCCACGT     hsa-miR-376c
ILMN_3166943   TGGGCGTATCTGTATGC     TGGGCGTATCTGTATGCTA      hsa-miR-585
ILMN_3166944  GCTTCCATGTTTCAGTGG TAAGTGCTTCCATGTTTCAGTGG     hsa-miR-302c
                 miRNA_ID SPOT_ID NumTargets TargetMatureVersion
ILMN_3166935  hsa-miR-329                  1                  12
ILMN_3166938  hsa-miR-424                  1                  12
ILMN_3166940  hsa-miR-137                  1                  12
ILMN_3166941 hsa-miR-376c                  1                  12
ILMN_3166943  hsa-miR-585                  1                  12
ILMN_3166944 hsa-miR-302c                  1                  12
                   OriginalMatureSeq OriginalMatureName
ILMN_3166935  AACACACCTGGTTAACCTCTTT        hsa-miR-329
ILMN_3166938  CAGCAGCAATTCATGTTTTGAA        hsa-miR-424
ILMN_3166940  TATTGCTTAAGAATACGCGTAG        hsa-miR-137
ILMN_3166941  ACATAGAGGAAATTCCACGTTT        hsa-miR-368
ILMN_3166943     TGGGCGTATCTGTATGCTA        hsa-miR-585
ILMN_3166944 TAAGTGCTTCCATGTTTCAGTGG       hsa-miR-302c
                                        Source Array_Address_Id
ILMN_3166935 Sanger miRNAdb (v_8.2, July 2006)              931
ILMN_3166938 Sanger miRNAdb (v_8.2, July 2006)             1051
ILMN_3166940 Sanger miRNAdb (v_8.2, July 2006)              233
ILMN_3166941 Sanger miRNAdb (v_8.2, July 2006)              976
ILMN_3166943 Sanger miRNAdb (v_8.2, July 2006)             1761
ILMN_3166944 Sanger miRNAdb (v_8.2, July 2006)              864
                         Illumicode
ILMN_3166935 CGGCAGATAGCACGTTTCGACT
ILMN_3166938 GGGCACTATTACTCGCTAACCG
ILMN_3166940 GTACAGTCTCGCGGTGCCTAAG
ILMN_3166941 GTTTCCACGACTTGCAGTACGG
ILMN_3166943 GTCGGTTAGTGATGTACGGCCA
ILMN_3166944 CCGTTTGGACGTAAGCAGGGCC
                                                                   Oligo
ILMN_3166935 ACTTCGTCAGTAACGGACCGGCAGATAGCACGTTTCGACTAACACACCTGGTTAACCTC
ILMN_3166938  ACTTCGTCAGTAACGGACGGGCACTATTACTCGCTAACCGCAGCAGCAATTCATGTTT
ILMN_3166940 ACTTCGTCAGTAACGGACGTACAGTCTCGCGGTGCCTAAGTATTGCTTAAGAATACGCG
ILMN_3166941 ACTTCGTCAGTAACGGACGTTTCCACGACTTGCAGTACGGACATAGAGGAAATTCCACG
ILMN_3166943   ACTTCGTCAGTAACGGACGTCGGTTAGTGATGTACGGCCATGGGCGTATCTGTATGC
ILMN_3166944  ACTTCGTCAGTAACGGACCCGTTTGGACGTAAGCAGGGCCGCTTCCATGTTTCAGTGG
                         U3_Seq  Ploidy Species Probe_MatchOrder Chromosome
ILMN_3166935 ACTTCGTCAGTAACGGAC diploid   human                2      14,14
ILMN_3166938 ACTTCGTCAGTAACGGAC diploid   human                1          X
ILMN_3166940 ACTTCGTCAGTAACGGAC diploid   human                1          1
ILMN_3166941 ACTTCGTCAGTAACGGAC diploid   human                1         14
ILMN_3166943 ACTTCGTCAGTAACGGAC diploid   human                1          5
ILMN_3166944 ACTTCGTCAGTAACGGAC diploid   human                1          4
                   Probe_Coordinates Probe_Chr_Orientation
ILMN_3166935 100,562,923,100,563,000                   +,+
ILMN_3166938               133508398                     -
ILMN_3166940                98284258                     -
ILMN_3166941               100575821                     +
ILMN_3166943               168623217                     -
ILMN_3166944               113788994                     -</code></pre>
</div>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Probe filtering</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(annot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "ID"                    "SYMBOL"                "ILMN_Gene"            
 [4] "Search_Key"            "SEQUENCE"              "TargetMatureSeqs"     
 [7] "TargetMatureName"      "miRNA_ID"              "SPOT_ID"              
[10] "NumTargets"            "TargetMatureVersion"   "OriginalMatureSeq"    
[13] "OriginalMatureName"    "Source"                "Array_Address_Id"     
[16] "Illumicode"            "Oligo"                 "U3_Seq"               
[19] "Ploidy"                "Species"               "Probe_MatchOrder"     
[22] "Chromosome"            "Probe_Coordinates"     "Probe_Chr_Orientation"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(annot<span class="sc">$</span>miRNA_version)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>NULL</code></pre>
</div>
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>is_mature_v12 <span class="ot">&lt;-</span> annot<span class="sc">$</span>TargetMatureVersion <span class="sc">==</span> <span class="dv">12</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>is_control <span class="ot">&lt;-</span> <span class="fu">grepl</span>(<span class="st">"control|spike|U6|snoR"</span>, annot<span class="sc">$</span>SYMBOL, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>)</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>keep_probes <span class="ot">&lt;-</span> is_mature_v12 <span class="sc">&amp;</span> <span class="sc">!</span>is_control</span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a>expr  <span class="ot">&lt;-</span> expr[keep_probes, ]</span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>annot <span class="ot">&lt;-</span> annot[keep_probes, ]</span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(expr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 830  80</code></pre>
</div>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(annot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                       ID       SYMBOL    ILMN_Gene   Search_Key
ILMN_3166935 ILMN_3166935 ILMN_3166935  hsa-miR-329  hsa-miR-329
ILMN_3166938 ILMN_3166938 ILMN_3166938  hsa-miR-424  hsa-miR-424
ILMN_3166940 ILMN_3166940 ILMN_3166940  hsa-miR-137  hsa-miR-137
ILMN_3166941 ILMN_3166941 ILMN_3166941 hsa-miR-376c hsa-miR-376c
ILMN_3166943 ILMN_3166943 ILMN_3166943  hsa-miR-585  hsa-miR-585
ILMN_3166944 ILMN_3166944 ILMN_3166944 hsa-miR-302c hsa-miR-302c
                        SEQUENCE        TargetMatureSeqs TargetMatureName
ILMN_3166935 AACACACCTGGTTAACCTC  AACACACCTGGTTAACCTCTTT      hsa-miR-329
ILMN_3166938  CAGCAGCAATTCATGTTT  CAGCAGCAATTCATGTTTTGAA      hsa-miR-424
ILMN_3166940 TATTGCTTAAGAATACGCG TTATTGCTTAAGAATACGCGTAG      hsa-miR-137
ILMN_3166941 ACATAGAGGAAATTCCACG   AACATAGAGGAAATTCCACGT     hsa-miR-376c
ILMN_3166943   TGGGCGTATCTGTATGC     TGGGCGTATCTGTATGCTA      hsa-miR-585
ILMN_3166944  GCTTCCATGTTTCAGTGG TAAGTGCTTCCATGTTTCAGTGG     hsa-miR-302c
                 miRNA_ID SPOT_ID NumTargets TargetMatureVersion
ILMN_3166935  hsa-miR-329                  1                  12
ILMN_3166938  hsa-miR-424                  1                  12
ILMN_3166940  hsa-miR-137                  1                  12
ILMN_3166941 hsa-miR-376c                  1                  12
ILMN_3166943  hsa-miR-585                  1                  12
ILMN_3166944 hsa-miR-302c                  1                  12
                   OriginalMatureSeq OriginalMatureName
ILMN_3166935  AACACACCTGGTTAACCTCTTT        hsa-miR-329
ILMN_3166938  CAGCAGCAATTCATGTTTTGAA        hsa-miR-424
ILMN_3166940  TATTGCTTAAGAATACGCGTAG        hsa-miR-137
ILMN_3166941  ACATAGAGGAAATTCCACGTTT        hsa-miR-368
ILMN_3166943     TGGGCGTATCTGTATGCTA        hsa-miR-585
ILMN_3166944 TAAGTGCTTCCATGTTTCAGTGG       hsa-miR-302c
                                        Source Array_Address_Id
ILMN_3166935 Sanger miRNAdb (v_8.2, July 2006)              931
ILMN_3166938 Sanger miRNAdb (v_8.2, July 2006)             1051
ILMN_3166940 Sanger miRNAdb (v_8.2, July 2006)              233
ILMN_3166941 Sanger miRNAdb (v_8.2, July 2006)              976
ILMN_3166943 Sanger miRNAdb (v_8.2, July 2006)             1761
ILMN_3166944 Sanger miRNAdb (v_8.2, July 2006)              864
                         Illumicode
ILMN_3166935 CGGCAGATAGCACGTTTCGACT
ILMN_3166938 GGGCACTATTACTCGCTAACCG
ILMN_3166940 GTACAGTCTCGCGGTGCCTAAG
ILMN_3166941 GTTTCCACGACTTGCAGTACGG
ILMN_3166943 GTCGGTTAGTGATGTACGGCCA
ILMN_3166944 CCGTTTGGACGTAAGCAGGGCC
                                                                   Oligo
ILMN_3166935 ACTTCGTCAGTAACGGACCGGCAGATAGCACGTTTCGACTAACACACCTGGTTAACCTC
ILMN_3166938  ACTTCGTCAGTAACGGACGGGCACTATTACTCGCTAACCGCAGCAGCAATTCATGTTT
ILMN_3166940 ACTTCGTCAGTAACGGACGTACAGTCTCGCGGTGCCTAAGTATTGCTTAAGAATACGCG
ILMN_3166941 ACTTCGTCAGTAACGGACGTTTCCACGACTTGCAGTACGGACATAGAGGAAATTCCACG
ILMN_3166943   ACTTCGTCAGTAACGGACGTCGGTTAGTGATGTACGGCCATGGGCGTATCTGTATGC
ILMN_3166944  ACTTCGTCAGTAACGGACCCGTTTGGACGTAAGCAGGGCCGCTTCCATGTTTCAGTGG
                         U3_Seq  Ploidy Species Probe_MatchOrder Chromosome
ILMN_3166935 ACTTCGTCAGTAACGGAC diploid   human                2      14,14
ILMN_3166938 ACTTCGTCAGTAACGGAC diploid   human                1          X
ILMN_3166940 ACTTCGTCAGTAACGGAC diploid   human                1          1
ILMN_3166941 ACTTCGTCAGTAACGGAC diploid   human                1         14
ILMN_3166943 ACTTCGTCAGTAACGGAC diploid   human                1          5
ILMN_3166944 ACTTCGTCAGTAACGGAC diploid   human                1          4
                   Probe_Coordinates Probe_Chr_Orientation
ILMN_3166935 100,562,923,100,563,000                   +,+
ILMN_3166938               133508398                     -
ILMN_3166940                98284258                     -
ILMN_3166941               100575821                     +
ILMN_3166943               168623217                     -
ILMN_3166944               113788994                     -</code></pre>
</div>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalization</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>expr_log2 <span class="ot">&lt;-</span> <span class="fu">log2</span>(expr <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>expr_norm <span class="ot">&lt;-</span> <span class="fu">normalizeBetweenArrays</span>(expr_log2, <span class="at">method =</span> <span class="st">"quantile"</span>)</span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(expr_norm) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(expr)</span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(expr_norm) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(expr)</span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a><span class="co"># QC: PCA + sample clustering (label only outliers in dendrogram)</span></span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>op <span class="ot">&lt;-</span> <span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">10</span>), <span class="at">xpd =</span> <span class="cn">TRUE</span>)</span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a>expr_t <span class="ot">&lt;-</span> <span class="fu">t</span>(expr_norm)</span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(expr_t, <span class="at">scale. =</span> <span class="cn">TRUE</span>)</span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a>sample_ids    <span class="ot">&lt;-</span> <span class="fu">colnames</span>(expr_norm)</span>
<span id="cb58-15"><a href="#cb58-15" aria-hidden="true" tabindex="-1"></a>highlight_ids <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"GSM1099641"</span>, <span class="st">"GSM1099655"</span>)</span>
<span id="cb58-16"><a href="#cb58-16" aria-hidden="true" tabindex="-1"></a>point_col     <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_ids <span class="sc">%in%</span> highlight_ids, <span class="st">"red"</span>, <span class="st">"black"</span>)</span>
<span id="cb58-17"><a href="#cb58-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-18"><a href="#cb58-18" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(pca<span class="sc">$</span>x[,<span class="dv">1</span>], pca<span class="sc">$</span>x[,<span class="dv">2</span>],</span>
<span id="cb58-19"><a href="#cb58-19" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"PC1"</span>, <span class="at">ylab =</span> <span class="st">"PC2"</span>,</span>
<span id="cb58-20"><a href="#cb58-20" aria-hidden="true" tabindex="-1"></a>     <span class="at">pch =</span> <span class="dv">19</span>,</span>
<span id="cb58-21"><a href="#cb58-21" aria-hidden="true" tabindex="-1"></a>     <span class="at">col =</span> point_col,</span>
<span id="cb58-22"><a href="#cb58-22" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"PCA after normalization"</span>)</span>
<span id="cb58-23"><a href="#cb58-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-24"><a href="#cb58-24" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(pca<span class="sc">$</span>x[,<span class="dv">1</span>], pca<span class="sc">$</span>x[,<span class="dv">2</span>],</span>
<span id="cb58-25"><a href="#cb58-25" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels =</span> sample_ids,</span>
<span id="cb58-26"><a href="#cb58-26" aria-hidden="true" tabindex="-1"></a>     <span class="at">pos =</span> <span class="dv">3</span>, <span class="at">cex =</span> <span class="fl">0.6</span>,</span>
<span id="cb58-27"><a href="#cb58-27" aria-hidden="true" tabindex="-1"></a>     <span class="at">col =</span> point_col)</span>
<span id="cb58-28"><a href="#cb58-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-29"><a href="#cb58-29" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"topright"</span>,</span>
<span id="cb58-30"><a href="#cb58-30" aria-hidden="true" tabindex="-1"></a>       <span class="at">inset =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">0.55</span>, <span class="dv">0</span>),</span>
<span id="cb58-31"><a href="#cb58-31" aria-hidden="true" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">"Other samples"</span>, <span class="st">"Outliers (to remove)"</span>),</span>
<span id="cb58-32"><a href="#cb58-32" aria-hidden="true" tabindex="-1"></a>       <span class="at">col    =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"red"</span>),</span>
<span id="cb58-33"><a href="#cb58-33" aria-hidden="true" tabindex="-1"></a>       <span class="at">pch    =</span> <span class="dv">19</span>,</span>
<span id="cb58-34"><a href="#cb58-34" aria-hidden="true" tabindex="-1"></a>       <span class="at">bty    =</span> <span class="st">"n"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_paired_files/figure-html/unnamed-chunk-1-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(op)</span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>sampleTree <span class="ot">&lt;-</span> <span class="fu">hclust</span>(<span class="fu">dist</span>(expr_t), <span class="at">method =</span> <span class="st">"average"</span>)</span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a>outliers <span class="ot">&lt;-</span> highlight_ids</span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a>leaf_order <span class="ot">&lt;-</span> sampleTree<span class="sc">$</span>labels[sampleTree<span class="sc">$</span>order]</span>
<span id="cb59-7"><a href="#cb59-7" aria-hidden="true" tabindex="-1"></a>out_pos <span class="ot">&lt;-</span> <span class="fu">match</span>(outliers, leaf_order)</span>
<span id="cb59-8"><a href="#cb59-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-9"><a href="#cb59-9" aria-hidden="true" tabindex="-1"></a>op <span class="ot">&lt;-</span> <span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">12</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">2</span>) <span class="sc">+</span> <span class="fl">0.1</span>, <span class="at">xpd =</span> <span class="cn">NA</span>)</span>
<span id="cb59-10"><a href="#cb59-10" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(sampleTree, <span class="at">labels =</span> <span class="cn">FALSE</span>, <span class="at">xaxt =</span> <span class="st">"n"</span>,</span>
<span id="cb59-11"><a href="#cb59-11" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"Sample clustering dendrogram"</span>, <span class="at">xlab =</span> <span class="st">""</span>, <span class="at">sub =</span> <span class="st">""</span>)</span>
<span id="cb59-12"><a href="#cb59-12" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">35</span>, <span class="at">col =</span> <span class="st">"red"</span>)</span>
<span id="cb59-13"><a href="#cb59-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-14"><a href="#cb59-14" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(outliers)) {</span>
<span id="cb59-15"><a href="#cb59-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">axis</span>(<span class="dv">1</span>, <span class="at">at =</span> out_pos[i], <span class="at">labels =</span> outliers[i],</span>
<span id="cb59-16"><a href="#cb59-16" aria-hidden="true" tabindex="-1"></a>       <span class="at">tick =</span> <span class="cn">TRUE</span>, <span class="at">line =</span> <span class="sc">-</span><span class="dv">20</span>, <span class="at">las =</span> <span class="dv">2</span>, <span class="at">cex.axis =</span> <span class="fl">1.2</span>, <span class="at">col.axis =</span> <span class="st">"red"</span>)</span>
<span id="cb59-17"><a href="#cb59-17" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_paired_files/figure-html/unnamed-chunk-1-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(op)</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"SampleClustering.pdf"</span>, <span class="at">width =</span> <span class="dv">14</span>, <span class="at">height =</span> <span class="dv">10</span>)</span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>op <span class="ot">&lt;-</span> <span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">12</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">2</span>) <span class="sc">+</span> <span class="fl">0.1</span>, <span class="at">xpd =</span> <span class="cn">NA</span>)</span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(sampleTree, <span class="at">labels =</span> <span class="cn">FALSE</span>, <span class="at">xaxt =</span> <span class="st">"n"</span>,</span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"Sample clustering dendrogram"</span>, <span class="at">xlab =</span> <span class="st">""</span>, <span class="at">sub =</span> <span class="st">""</span>)</span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">35</span>, <span class="at">col =</span> <span class="st">"red"</span>)</span>
<span id="cb60-8"><a href="#cb60-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-9"><a href="#cb60-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(outliers)) {</span>
<span id="cb60-10"><a href="#cb60-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">axis</span>(<span class="dv">1</span>, <span class="at">at =</span> out_pos[i], <span class="at">labels =</span> outliers[i],</span>
<span id="cb60-11"><a href="#cb60-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">tick =</span> <span class="cn">FALSE</span>, <span class="at">line =</span> <span class="sc">-</span><span class="dv">22</span>, <span class="at">las =</span> <span class="dv">2</span>, <span class="at">cex.axis =</span> <span class="dv">1</span>, <span class="at">col.axis =</span> <span class="st">"red"</span>)</span>
<span id="cb60-12"><a href="#cb60-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb60-13"><a href="#cb60-13" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(op)</span>
<span id="cb60-14"><a href="#cb60-14" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Patient IDs + remove outlier pairs</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>patient_id <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">".*_(</span><span class="sc">\\</span><span class="st">d+)[NT]$"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, pheno<span class="sc">$</span>title)</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>patient_id <span class="ot">&lt;-</span> <span class="fu">as.character</span>(pheno<span class="sc">$</span>patient_id)</span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a>outlier_samples <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"GSM1099641"</span>, <span class="st">"GSM1099655"</span>)</span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a>outlier_patient_ids <span class="ot">&lt;-</span> <span class="fu">unique</span>(pheno<span class="sc">$</span>patient_id[pheno<span class="sc">$</span>geo_accession <span class="sc">%in%</span> outlier_samples])</span>
<span id="cb62-7"><a href="#cb62-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-8"><a href="#cb62-8" aria-hidden="true" tabindex="-1"></a>keep_samples <span class="ot">&lt;-</span> <span class="sc">!</span>pheno<span class="sc">$</span>patient_id <span class="sc">%in%</span> outlier_patient_ids</span>
<span id="cb62-9"><a href="#cb62-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-10"><a href="#cb62-10" aria-hidden="true" tabindex="-1"></a>expr_norm <span class="ot">&lt;-</span> expr_norm[, keep_samples]</span>
<span id="cb62-11"><a href="#cb62-11" aria-hidden="true" tabindex="-1"></a>pheno     <span class="ot">&lt;-</span> pheno[keep_samples, ]</span>
<span id="cb62-12"><a href="#cb62-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-13"><a href="#cb62-13" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(expr_norm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 830  76</code></pre>
</div>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(pheno<span class="sc">$</span>patient_id)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
01 02 03 04 05 06 07 08 09 10 11 12 13 14 15 16 17 18 19 20 21 23 24 25 26 27 
 2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2 
28 29 30 31 32 33 34 35 37 38 39 40 
 2  2  2  2  2  2  2  2  2  2  2  2 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Li et al. pipeline: WGCNA on all samples (no Δ)</span></span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>condition <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"T$"</span>, pheno<span class="sc">$</span>title), <span class="st">"tumor"</span>, <span class="st">"normal"</span>)</span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a>datExpr <span class="ot">&lt;-</span> <span class="fu">t</span>(expr_norm)   <span class="co"># samples × miRNAs</span></span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(datExpr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  76 830</code></pre>
</div>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>gsg <span class="ot">&lt;-</span> <span class="fu">goodSamplesGenes</span>(datExpr, <span class="at">verbose =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Flagging genes and samples with too many missing values...
  ..step 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>gsg<span class="sc">$</span>allOK</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Soft-threshold selection (unsigned; Li)</span></span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>powers <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span></span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a>sft <span class="ot">&lt;-</span> <span class="fu">pickSoftThreshold</span>(datExpr,</span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">powerVector =</span> powers,</span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">networkType =</span> <span class="st">"unsigned"</span>,</span>
<span id="cb72-6"><a href="#cb72-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">verbose =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>pickSoftThreshold: will use block size 830.
 pickSoftThreshold: calculating connectivity for given powers...
   ..working on genes 1 through 830 of 830
   Power SFT.R.sq  slope truncated.R.sq  mean.k. median.k. max.k.
1      1    0.132 -0.619          0.813 138.0000  1.34e+02 231.00
2      2    0.613 -1.070          0.839  38.7000  3.33e+01  97.60
3      3    0.850 -1.150          0.920  14.7000  1.03e+01  52.20
4      4    0.922 -1.280          0.967   6.7900  3.72e+00  34.30
5      5    0.960 -1.310          0.988   3.6400  1.51e+00  24.70
6      6    0.957 -1.330          0.968   2.1700  6.61e-01  19.10
7      7    0.970 -1.320          0.983   1.4000  3.06e-01  15.20
8      8    0.954 -1.300          0.948   0.9560  1.47e-01  12.40
9      9    0.959 -1.280          0.952   0.6850  7.33e-02  10.40
10    10    0.940 -1.290          0.927   0.5100  3.94e-02   8.91
11    11    0.960 -1.260          0.953   0.3910  2.21e-02   7.73
12    12    0.971 -1.250          0.966   0.3070  1.25e-02   6.77
13    13    0.976 -1.240          0.972   0.2460  7.14e-03   5.98
14    14    0.979 -1.230          0.980   0.2010  4.08e-03   5.32
15    15    0.977 -1.210          0.975   0.1660  2.40e-03   4.75
16    16    0.979 -1.190          0.976   0.1400  1.39e-03   4.27
17    17    0.956 -1.200          0.957   0.1180  8.06e-04   3.86
18    18    0.947 -1.200          0.937   0.1020  4.92e-04   3.50
19    19    0.964 -1.180          0.961   0.0878  2.94e-04   3.18
20    20    0.964 -1.170          0.962   0.0765  1.75e-04   2.91</code></pre>
</div>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a>     <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">3</span>]) <span class="sc">*</span> sft<span class="sc">$</span>fitIndices[,<span class="dv">2</span>],</span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"Soft Threshold (power)"</span>,</span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">"Scale Free Topology Model Fit, signed R^2"</span>,</span>
<span id="cb74-6"><a href="#cb74-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">type =</span> <span class="st">"n"</span>,</span>
<span id="cb74-7"><a href="#cb74-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"Scale independence"</span>)</span>
<span id="cb74-8"><a href="#cb74-8" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb74-9"><a href="#cb74-9" aria-hidden="true" tabindex="-1"></a>     <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">3</span>]) <span class="sc">*</span> sft<span class="sc">$</span>fitIndices[,<span class="dv">2</span>],</span>
<span id="cb74-10"><a href="#cb74-10" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels =</span> powers, <span class="at">cex =</span> <span class="fl">0.8</span>, <span class="at">col =</span> <span class="st">"red"</span>)</span>
<span id="cb74-11"><a href="#cb74-11" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="fl">0.97</span>, <span class="at">col =</span> <span class="st">"red"</span>)</span>
<span id="cb74-12"><a href="#cb74-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-13"><a href="#cb74-13" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb74-14"><a href="#cb74-14" aria-hidden="true" tabindex="-1"></a>     sft<span class="sc">$</span>fitIndices[,<span class="dv">5</span>],</span>
<span id="cb74-15"><a href="#cb74-15" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"Soft Threshold (power)"</span>,</span>
<span id="cb74-16"><a href="#cb74-16" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">"Mean Connectivity"</span>,</span>
<span id="cb74-17"><a href="#cb74-17" aria-hidden="true" tabindex="-1"></a>     <span class="at">type =</span> <span class="st">"n"</span>,</span>
<span id="cb74-18"><a href="#cb74-18" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"Mean connectivity"</span>)</span>
<span id="cb74-19"><a href="#cb74-19" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb74-20"><a href="#cb74-20" aria-hidden="true" tabindex="-1"></a>     sft<span class="sc">$</span>fitIndices[,<span class="dv">5</span>],</span>
<span id="cb74-21"><a href="#cb74-21" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels =</span> powers, <span class="at">cex =</span> <span class="fl">0.8</span>, <span class="at">col =</span> <span class="st">"red"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_paired_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>softPower <span class="ot">&lt;-</span> <span class="dv">7</span></span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>softPower</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 7</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Network construction (unsigned)</span></span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a>adjacency <span class="ot">&lt;-</span> <span class="fu">adjacency</span>(datExpr, <span class="at">power =</span> softPower, <span class="at">type =</span> <span class="st">"unsigned"</span>)</span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a>TOM     <span class="ot">&lt;-</span> <span class="fu">TOMsimilarity</span>(adjacency, <span class="at">TOMType =</span> <span class="st">"unsigned"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>..connectivity..
..matrix multiplication (system BLAS)..
..normalization..
..done.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>dissTOM <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">-</span> TOM</span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>geneTree <span class="ot">&lt;-</span> <span class="fu">hclust</span>(<span class="fu">as.dist</span>(dissTOM), <span class="at">method =</span> <span class="st">"average"</span>)</span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(geneTree,</span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"Clustering of miRNAs based on TOM (unsigned, Li pipeline)"</span>,</span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">""</span>, <span class="at">sub =</span> <span class="st">""</span>, <span class="at">cex =</span> <span class="fl">0.4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_paired_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Module detection + merging</span></span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>minModuleSize <span class="ot">&lt;-</span> <span class="dv">30</span></span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a>dynamicMods <span class="ot">&lt;-</span> <span class="fu">cutreeDynamic</span>(</span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">dendro =</span> geneTree,</span>
<span id="cb80-6"><a href="#cb80-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">distM =</span> dissTOM,</span>
<span id="cb80-7"><a href="#cb80-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">deepSplit =</span> <span class="dv">2</span>,</span>
<span id="cb80-8"><a href="#cb80-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">pamRespectsDendro =</span> <span class="cn">FALSE</span>,</span>
<span id="cb80-9"><a href="#cb80-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">minClusterSize =</span> minModuleSize</span>
<span id="cb80-10"><a href="#cb80-10" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> ..cutHeight not given, setting it to 0.998  ===&gt;  99% of the (truncated) height range in dendro.
 ..done.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>dynamicColors <span class="ot">&lt;-</span> <span class="fu">labels2colors</span>(dynamicMods)</span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(dynamicColors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>dynamicColors
     blue     brown      grey turquoise 
      201        85       250       294 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>  geneTree,</span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>  dynamicColors,</span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Dynamic Tree Cut"</span>,</span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">dendroLabels =</span> <span class="cn">FALSE</span>,</span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">hang =</span> <span class="fl">0.03</span>,</span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">addGuide =</span> <span class="cn">TRUE</span>,</span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">guideHang =</span> <span class="fl">0.05</span>,</span>
<span id="cb84-9"><a href="#cb84-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">"miRNA dendrogram and module colors (Li pipeline)"</span></span>
<span id="cb84-10"><a href="#cb84-10" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_paired_files/figure-html/unnamed-chunk-3-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>moduleColors0 <span class="ot">&lt;-</span> dynamicColors</span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a>MElist <span class="ot">&lt;-</span> <span class="fu">moduleEigengenes</span>(datExpr, <span class="at">colors =</span> moduleColors0)</span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a>MEs    <span class="ot">&lt;-</span> <span class="fu">orderMEs</span>(MElist<span class="sc">$</span>eigengenes)</span>
<span id="cb85-5"><a href="#cb85-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-6"><a href="#cb85-6" aria-hidden="true" tabindex="-1"></a>MEDiss <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">-</span> <span class="fu">cor</span>(MEs)</span>
<span id="cb85-7"><a href="#cb85-7" aria-hidden="true" tabindex="-1"></a>METree <span class="ot">&lt;-</span> <span class="fu">hclust</span>(<span class="fu">as.dist</span>(MEDiss), <span class="at">method =</span> <span class="st">"average"</span>)</span>
<span id="cb85-8"><a href="#cb85-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-9"><a href="#cb85-9" aria-hidden="true" tabindex="-1"></a>mergeCutHeight <span class="ot">&lt;-</span> <span class="fl">0.25</span></span>
<span id="cb85-10"><a href="#cb85-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-11"><a href="#cb85-11" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(METree,</span>
<span id="cb85-12"><a href="#cb85-12" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"Clustering of module eigengenes (before merging)"</span>,</span>
<span id="cb85-13"><a href="#cb85-13" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">""</span>, <span class="at">sub =</span> <span class="st">""</span>)</span>
<span id="cb85-14"><a href="#cb85-14" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> mergeCutHeight, <span class="at">col =</span> <span class="st">"red"</span>)</span>
<span id="cb85-15"><a href="#cb85-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-16"><a href="#cb85-16" aria-hidden="true" tabindex="-1"></a>merge <span class="ot">&lt;-</span> <span class="fu">mergeCloseModules</span>(</span>
<span id="cb85-17"><a href="#cb85-17" aria-hidden="true" tabindex="-1"></a>  datExpr,</span>
<span id="cb85-18"><a href="#cb85-18" aria-hidden="true" tabindex="-1"></a>  moduleColors0,</span>
<span id="cb85-19"><a href="#cb85-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">cutHeight =</span> mergeCutHeight,</span>
<span id="cb85-20"><a href="#cb85-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="dv">3</span></span>
<span id="cb85-21"><a href="#cb85-21" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> mergeCloseModules: Merging modules whose distance is less than 0.25
   multiSetMEs: Calculating module MEs.
     Working on set 1 ...
     moduleEigengenes: Calculating 4 module eigengenes in given set.
   Calculating new MEs...
   multiSetMEs: Calculating module MEs.
     Working on set 1 ...
     moduleEigengenes: Calculating 4 module eigengenes in given set.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a>moduleColors <span class="ot">&lt;-</span> merge<span class="sc">$</span>colors</span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>MEs          <span class="ot">&lt;-</span> <span class="fu">orderMEs</span>(merge<span class="sc">$</span>newMEs)</span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(</span>
<span id="cb87-5"><a href="#cb87-5" aria-hidden="true" tabindex="-1"></a>  geneTree,</span>
<span id="cb87-6"><a href="#cb87-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cbind</span>(moduleColors0, moduleColors),</span>
<span id="cb87-7"><a href="#cb87-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">"Dynamic Tree Cut"</span>, <span class="st">"Merged modules"</span>),</span>
<span id="cb87-8"><a href="#cb87-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">dendroLabels =</span> <span class="cn">FALSE</span>,</span>
<span id="cb87-9"><a href="#cb87-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">hang =</span> <span class="fl">0.03</span>,</span>
<span id="cb87-10"><a href="#cb87-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">addGuide =</span> <span class="cn">TRUE</span>,</span>
<span id="cb87-11"><a href="#cb87-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">guideHang =</span> <span class="fl">0.05</span>,</span>
<span id="cb87-12"><a href="#cb87-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">"Modules before and after merging"</span></span>
<span id="cb87-13"><a href="#cb87-13" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_paired_files/figure-html/unnamed-chunk-3-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_paired_files/figure-html/unnamed-chunk-3-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(moduleColors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>moduleColors
     blue     brown      grey turquoise 
      201        85       250       294 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sample-level traits for LMM</span></span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">rownames</span>(pheno), <span class="fu">rownames</span>(datExpr)))</span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>tumour <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(pheno<span class="sc">$</span>condition <span class="sc">==</span> <span class="st">"tumor"</span>, <span class="dv">1</span>, <span class="dv">0</span>)</span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>Age <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(pheno[,<span class="st">"age:ch1"</span>])</span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>Stage_num <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">factor</span>(pheno[,<span class="st">"Stage:ch1"</span>], <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"I"</span>,<span class="st">"II"</span>,<span class="st">"III"</span>,<span class="st">"IVA"</span>)))</span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a>sampleTraits <span class="ot">&lt;-</span> pheno[, <span class="fu">c</span>(<span class="st">"patient_id"</span>, <span class="st">"tumour"</span>, <span class="st">"Age"</span>, <span class="st">"Stage_num"</span>)]</span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sampleTraits) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(pheno)</span>
<span id="cb90-10"><a href="#cb90-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-11"><a href="#cb90-11" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">rownames</span>(sampleTraits), <span class="fu">rownames</span>(datExpr))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="co"># LMM on module eigengenes</span></span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a>datLMM <span class="ot">&lt;-</span> <span class="fu">cbind</span>(sampleTraits[<span class="fu">rownames</span>(MEs), ],</span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a>                <span class="fu">as.data.frame</span>(MEs))</span>
<span id="cb92-4"><a href="#cb92-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-5"><a href="#cb92-5" aria-hidden="true" tabindex="-1"></a>modNames <span class="ot">&lt;-</span> <span class="fu">colnames</span>(MEs)</span>
<span id="cb92-6"><a href="#cb92-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-7"><a href="#cb92-7" aria-hidden="true" tabindex="-1"></a>lmm_results <span class="ot">&lt;-</span> <span class="fu">lapply</span>(modNames, <span class="cf">function</span>(mn) {</span>
<span id="cb92-8"><a href="#cb92-8" aria-hidden="true" tabindex="-1"></a>  form <span class="ot">&lt;-</span> <span class="fu">as.formula</span>(<span class="fu">paste</span>(mn, <span class="st">"~ tumour + Age + Stage_num"</span>))</span>
<span id="cb92-9"><a href="#cb92-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-10"><a href="#cb92-10" aria-hidden="true" tabindex="-1"></a>  fit <span class="ot">&lt;-</span> <span class="fu">lme</span>(form,</span>
<span id="cb92-11"><a href="#cb92-11" aria-hidden="true" tabindex="-1"></a>             <span class="at">random =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> patient_id,</span>
<span id="cb92-12"><a href="#cb92-12" aria-hidden="true" tabindex="-1"></a>             <span class="at">data   =</span> datLMM,</span>
<span id="cb92-13"><a href="#cb92-13" aria-hidden="true" tabindex="-1"></a>             <span class="at">na.action =</span> na.omit,</span>
<span id="cb92-14"><a href="#cb92-14" aria-hidden="true" tabindex="-1"></a>             <span class="at">method =</span> <span class="st">"REML"</span>)</span>
<span id="cb92-15"><a href="#cb92-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-16"><a href="#cb92-16" aria-hidden="true" tabindex="-1"></a>  sm <span class="ot">&lt;-</span> <span class="fu">summary</span>(fit)</span>
<span id="cb92-17"><a href="#cb92-17" aria-hidden="true" tabindex="-1"></a>  tumour_row <span class="ot">&lt;-</span> sm<span class="sc">$</span>tTable[<span class="st">"tumour"</span>, ]</span>
<span id="cb92-18"><a href="#cb92-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-19"><a href="#cb92-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb92-20"><a href="#cb92-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">module      =</span> mn,</span>
<span id="cb92-21"><a href="#cb92-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">beta_tumour =</span> <span class="fu">round</span>(tumour_row[<span class="st">"Value"</span>], <span class="dv">2</span>),</span>
<span id="cb92-22"><a href="#cb92-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">t_tumour    =</span> <span class="fu">round</span>(tumour_row[<span class="st">"t-value"</span>], <span class="dv">2</span>),</span>
<span id="cb92-23"><a href="#cb92-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">p_tumour    =</span> <span class="fu">formatC</span>(<span class="fu">as.numeric</span>(tumour_row[<span class="st">"p-value"</span>]), <span class="at">format =</span> <span class="st">"e"</span>, <span class="at">digits =</span> <span class="dv">2</span>),</span>
<span id="cb92-24"><a href="#cb92-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">row.names   =</span> <span class="cn">NULL</span></span>
<span id="cb92-25"><a href="#cb92-25" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb92-26"><a href="#cb92-26" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb92-27"><a href="#cb92-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-28"><a href="#cb92-28" aria-hidden="true" tabindex="-1"></a>lmm_results_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, lmm_results)</span>
<span id="cb92-29"><a href="#cb92-29" aria-hidden="true" tabindex="-1"></a>lmm_results_df[<span class="fu">order</span>(<span class="fu">abs</span>(lmm_results_df<span class="sc">$</span>t_tumour), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       module beta_tumour t_tumour p_tumour
1 MEturquoise       -0.21   -19.28 6.87e-21
4      MEgrey        0.11     6.32 2.29e-07
2      MEblue       -0.01    -0.38 7.02e-01
3     MEbrown        0.00    -0.11 9.15e-01</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Heatmap of tumour t-values</span></span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a>lmm_results_df<span class="sc">$</span>module_clean <span class="ot">&lt;-</span> <span class="fu">substring</span>(lmm_results_df<span class="sc">$</span>module, <span class="dv">3</span>)</span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a>tmat <span class="ot">&lt;-</span> <span class="fu">matrix</span>(lmm_results_df<span class="sc">$</span>t_tumour,</span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">nrow =</span> <span class="fu">nrow</span>(lmm_results_df),</span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">dimnames =</span> <span class="fu">list</span>(lmm_results_df<span class="sc">$</span>module_clean, <span class="st">"Tumour (t)"</span>))</span>
<span id="cb94-7"><a href="#cb94-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-8"><a href="#cb94-8" aria-hidden="true" tabindex="-1"></a>pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(tmat,</span>
<span id="cb94-9"><a href="#cb94-9" aria-hidden="true" tabindex="-1"></a>                   <span class="at">main =</span> <span class="st">"Module–tumour associations (LMM t-values)"</span>,</span>
<span id="cb94-10"><a href="#cb94-10" aria-hidden="true" tabindex="-1"></a>                   <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb94-11"><a href="#cb94-11" aria-hidden="true" tabindex="-1"></a>                   <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb94-12"><a href="#cb94-12" aria-hidden="true" tabindex="-1"></a>                   <span class="at">display_numbers =</span> <span class="fu">round</span>(tmat, <span class="dv">2</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_paired_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Gene significance (GS) via per-miRNA LMM t-stat</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a>expr_forGS <span class="ot">&lt;-</span> datExpr</span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a>datGS <span class="ot">&lt;-</span> <span class="fu">cbind</span>(sampleTraits[<span class="fu">rownames</span>(expr_forGS), ],</span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a>               <span class="fu">as.data.frame</span>(expr_forGS))</span>
<span id="cb95-6"><a href="#cb95-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-7"><a href="#cb95-7" aria-hidden="true" tabindex="-1"></a>miRNAs <span class="ot">&lt;-</span> <span class="fu">colnames</span>(expr_forGS)</span>
<span id="cb95-8"><a href="#cb95-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-9"><a href="#cb95-9" aria-hidden="true" tabindex="-1"></a>gs_list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(miRNAs, <span class="cf">function</span>(g) {</span>
<span id="cb95-10"><a href="#cb95-10" aria-hidden="true" tabindex="-1"></a>  form <span class="ot">&lt;-</span> <span class="fu">as.formula</span>(<span class="fu">paste</span>(g, <span class="st">"~ tumour + Age + Stage_num"</span>))</span>
<span id="cb95-11"><a href="#cb95-11" aria-hidden="true" tabindex="-1"></a>  fit  <span class="ot">&lt;-</span> <span class="fu">lme</span>(form,</span>
<span id="cb95-12"><a href="#cb95-12" aria-hidden="true" tabindex="-1"></a>              <span class="at">random =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> patient_id,</span>
<span id="cb95-13"><a href="#cb95-13" aria-hidden="true" tabindex="-1"></a>              <span class="at">data   =</span> datGS,</span>
<span id="cb95-14"><a href="#cb95-14" aria-hidden="true" tabindex="-1"></a>              <span class="at">na.action =</span> na.omit,</span>
<span id="cb95-15"><a href="#cb95-15" aria-hidden="true" tabindex="-1"></a>              <span class="at">method =</span> <span class="st">"REML"</span>)</span>
<span id="cb95-16"><a href="#cb95-16" aria-hidden="true" tabindex="-1"></a>  sm <span class="ot">&lt;-</span> <span class="fu">summary</span>(fit)</span>
<span id="cb95-17"><a href="#cb95-17" aria-hidden="true" tabindex="-1"></a>  t_tumour <span class="ot">&lt;-</span> sm<span class="sc">$</span>tTable[<span class="st">"tumour"</span>,<span class="st">"t-value"</span>]</span>
<span id="cb95-18"><a href="#cb95-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">miRNA =</span> g, <span class="at">GS =</span> <span class="fu">abs</span>(t_tumour), <span class="at">row.names =</span> <span class="cn">NULL</span>)</span>
<span id="cb95-19"><a href="#cb95-19" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb95-20"><a href="#cb95-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-21"><a href="#cb95-21" aria-hidden="true" tabindex="-1"></a>geneGS <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, gs_list)</span>
<span id="cb95-22"><a href="#cb95-22" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(geneGS) <span class="ot">&lt;-</span> geneGS<span class="sc">$</span>miRNA</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Module membership (MM) + hub table</span></span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a>modNames_clean <span class="ot">&lt;-</span> <span class="fu">substring</span>(<span class="fu">colnames</span>(MEs), <span class="dv">3</span>)</span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a>geneModuleMembership <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">cor</span>(datExpr, MEs, <span class="at">use =</span> <span class="st">"p"</span>))</span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(geneModuleMembership) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"MM"</span>, modNames_clean)</span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(geneModuleMembership) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(datExpr)</span>
<span id="cb96-7"><a href="#cb96-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-8"><a href="#cb96-8" aria-hidden="true" tabindex="-1"></a>hubData <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb96-9"><a href="#cb96-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">miRNA  =</span> <span class="fu">colnames</span>(datExpr),</span>
<span id="cb96-10"><a href="#cb96-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">module =</span> moduleColors,</span>
<span id="cb96-11"><a href="#cb96-11" aria-hidden="true" tabindex="-1"></a>  geneModuleMembership,</span>
<span id="cb96-12"><a href="#cb96-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">GS     =</span> geneGS[<span class="fu">colnames</span>(datExpr), <span class="st">"GS"</span>],</span>
<span id="cb96-13"><a href="#cb96-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">row.names =</span> <span class="cn">NULL</span></span>
<span id="cb96-14"><a href="#cb96-14" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb96-15"><a href="#cb96-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-16"><a href="#cb96-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Intramodular connectivity + top hubs</span></span>
<span id="cb96-17"><a href="#cb96-17" aria-hidden="true" tabindex="-1"></a>IMConn <span class="ot">&lt;-</span> <span class="fu">intramodularConnectivity</span>(adjacency, moduleColors)</span>
<span id="cb96-18"><a href="#cb96-18" aria-hidden="true" tabindex="-1"></a>IMConn_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(IMConn)</span>
<span id="cb96-19"><a href="#cb96-19" aria-hidden="true" tabindex="-1"></a>IMConn_df<span class="sc">$</span>miRNA <span class="ot">&lt;-</span> <span class="fu">rownames</span>(IMConn_df)</span>
<span id="cb96-20"><a href="#cb96-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-21"><a href="#cb96-21" aria-hidden="true" tabindex="-1"></a>hubData <span class="ot">&lt;-</span> hubData <span class="sc">%&gt;%</span></span>
<span id="cb96-22"><a href="#cb96-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(IMConn_df[, <span class="fu">c</span>(<span class="st">"miRNA"</span>, <span class="st">"kWithin"</span>)], <span class="at">by =</span> <span class="st">"miRNA"</span>)</span>
<span id="cb96-23"><a href="#cb96-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-24"><a href="#cb96-24" aria-hidden="true" tabindex="-1"></a>hubList <span class="ot">&lt;-</span> hubData <span class="sc">%&gt;%</span></span>
<span id="cb96-25"><a href="#cb96-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(module) <span class="sc">%&gt;%</span></span>
<span id="cb96-26"><a href="#cb96-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(kWithin)) <span class="sc">%&gt;%</span></span>
<span id="cb96-27"><a href="#cb96-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb96-28"><a href="#cb96-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb96-29"><a href="#cb96-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-30"><a href="#cb96-30" aria-hidden="true" tabindex="-1"></a>hubList</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 40 × 8
   miRNA        module MMturquoise MMblue MMbrown  MMgrey    GS kWithin
   &lt;chr&gt;        &lt;chr&gt;        &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt;
 1 ILMN_3168603 blue       0.0672   0.930   0.302 -0.0774 1.72     15.0
 2 ILMN_3167275 blue      -0.0461   0.911   0.261  0.0774 0.621    14.6
 3 ILMN_3168541 blue      -0.199    0.935   0.409  0.117  0.554    14.5
 4 ILMN_3167184 blue      -0.0164   0.916   0.215  0.0661 0.805    14.2
 5 ILMN_3168633 blue       0.0325   0.908   0.227  0.0416 1.28     13.7
 6 ILMN_3168736 blue      -0.145    0.910   0.308  0.0969 0.265    13.4
 7 ILMN_3168183 blue      -0.0273   0.925   0.353 -0.0624 0.878    13.4
 8 ILMN_3168848 blue      -0.0341   0.905   0.257  0.0677 0.740    13.4
 9 ILMN_3166996 blue      -0.0278   0.895   0.292  0.0182 0.861    12.5
10 ILMN_3167242 blue       0.00604  0.885   0.120  0.0544 1.24     11.8
# ℹ 30 more rows</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Example plot: GS vs MM in turquoise</span></span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>targetModule <span class="ot">&lt;-</span> <span class="st">"turquoise"</span></span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a>inMod <span class="ot">&lt;-</span> hubData<span class="sc">$</span>module <span class="sc">==</span> targetModule</span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-5"><a href="#cb98-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(hubData<span class="sc">$</span>MMturquoise[inMod],</span>
<span id="cb98-6"><a href="#cb98-6" aria-hidden="true" tabindex="-1"></a>     hubData<span class="sc">$</span>GS[inMod],</span>
<span id="cb98-7"><a href="#cb98-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"Module membership (turquoise)"</span>,</span>
<span id="cb98-8"><a href="#cb98-8" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">"Gene significance (|t| for tumour)"</span>,</span>
<span id="cb98-9"><a href="#cb98-9" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">"GS vs MM in"</span>, targetModule, <span class="st">"module"</span>))</span>
<span id="cb98-10"><a href="#cb98-10" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="fu">lm</span>(GS <span class="sc">~</span> MMturquoise, <span class="at">data =</span> hubData[inMod, ]), <span class="at">lty =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_paired_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Export module lists + top-10 hubs</span></span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a>moduleGenes <span class="ot">&lt;-</span> <span class="fu">split</span>(hubData<span class="sc">$</span>miRNA, hubData<span class="sc">$</span>module)</span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (mod <span class="cf">in</span> <span class="fu">names</span>(moduleGenes)) {</span>
<span id="cb99-5"><a href="#cb99-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(moduleGenes[[mod]],</span>
<span id="cb99-6"><a href="#cb99-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"module_"</span>, mod, <span class="st">"_miRNAs.txt"</span>),</span>
<span id="cb99-7"><a href="#cb99-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb99-8"><a href="#cb99-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">col.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb99-9"><a href="#cb99-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb99-10"><a href="#cb99-10" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb99-11"><a href="#cb99-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-12"><a href="#cb99-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (mod <span class="cf">in</span> <span class="fu">unique</span>(hubList<span class="sc">$</span>module)) {</span>
<span id="cb99-13"><a href="#cb99-13" aria-hidden="true" tabindex="-1"></a>  sub <span class="ot">&lt;-</span> hubList[hubList<span class="sc">$</span>module <span class="sc">==</span> mod, ]</span>
<span id="cb99-14"><a href="#cb99-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(sub,</span>
<span id="cb99-15"><a href="#cb99-15" aria-hidden="true" tabindex="-1"></a>              <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"module_"</span>, mod, <span class="st">"_top10_hubs.txt"</span>),</span>
<span id="cb99-16"><a href="#cb99-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb99-17"><a href="#cb99-17" aria-hidden="true" tabindex="-1"></a>              <span class="at">quote =</span> <span class="cn">FALSE</span>,</span>
<span id="cb99-18"><a href="#cb99-18" aria-hidden="true" tabindex="-1"></a>              <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb99-19"><a href="#cb99-19" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb99-20"><a href="#cb99-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-21"><a href="#cb99-21" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(moduleColors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>moduleColors
     blue     brown      grey turquoise 
      201        85       250       294 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Export adjacency + export full paired-WGCNA object</span></span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(adjacency) <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">colnames</span>(datExpr), <span class="fu">colnames</span>(datExpr))</span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(adjacency, <span class="at">file =</span> <span class="st">"paired_adjacency_matrix.rds"</span>)</span>
<span id="cb101-4"><a href="#cb101-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-5"><a href="#cb101-5" aria-hidden="true" tabindex="-1"></a>adj_thresh <span class="ot">&lt;-</span> <span class="fl">0.10</span></span>
<span id="cb101-6"><a href="#cb101-6" aria-hidden="true" tabindex="-1"></a>idx <span class="ot">&lt;-</span> <span class="fu">which</span>(adjacency <span class="sc">&gt;</span> adj_thresh <span class="sc">&amp;</span> <span class="fu">upper.tri</span>(adjacency), <span class="at">arr.ind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb101-7"><a href="#cb101-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-8"><a href="#cb101-8" aria-hidden="true" tabindex="-1"></a>adj_edge_list <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb101-9"><a href="#cb101-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene1 =</span> <span class="fu">rownames</span>(adjacency)[idx[,<span class="dv">1</span>]],</span>
<span id="cb101-10"><a href="#cb101-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene2 =</span> <span class="fu">colnames</span>(adjacency)[idx[,<span class="dv">2</span>]],</span>
<span id="cb101-11"><a href="#cb101-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">adjacency =</span> adjacency[idx]</span>
<span id="cb101-12"><a href="#cb101-12" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb101-13"><a href="#cb101-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-14"><a href="#cb101-14" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(adj_edge_list,</span>
<span id="cb101-15"><a href="#cb101-15" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> <span class="st">"paired_adjacency_edges_thresh0.10.tsv"</span>,</span>
<span id="cb101-16"><a href="#cb101-16" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb101-17"><a href="#cb101-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-18"><a href="#cb101-18" aria-hidden="true" tabindex="-1"></a>moduleColors_paired <span class="ot">&lt;-</span> moduleColors</span>
<span id="cb101-19"><a href="#cb101-19" aria-hidden="true" tabindex="-1"></a>TOM_paired          <span class="ot">&lt;-</span> TOM</span>
<span id="cb101-20"><a href="#cb101-20" aria-hidden="true" tabindex="-1"></a>MEs_paired          <span class="ot">&lt;-</span> MEs</span>
<span id="cb101-21"><a href="#cb101-21" aria-hidden="true" tabindex="-1"></a>hubData_paired      <span class="ot">&lt;-</span> hubData</span>
<span id="cb101-22"><a href="#cb101-22" aria-hidden="true" tabindex="-1"></a>lmm_results_paired  <span class="ot">&lt;-</span> lmm_results_df</span>
<span id="cb101-23"><a href="#cb101-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-24"><a href="#cb101-24" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(moduleColors_paired) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(datExpr)</span>
<span id="cb101-25"><a href="#cb101-25" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(TOM_paired) <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">colnames</span>(datExpr), <span class="fu">colnames</span>(datExpr))</span>
<span id="cb101-26"><a href="#cb101-26" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(hubData_paired) <span class="ot">&lt;-</span> hubData_paired<span class="sc">$</span>miRNA</span>
<span id="cb101-27"><a href="#cb101-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-28"><a href="#cb101-28" aria-hidden="true" tabindex="-1"></a>paired_wgcna_export <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb101-29"><a href="#cb101-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">"paired_WGCNA_Li_pipeline"</span>,</span>
<span id="cb101-30"><a href="#cb101-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">date   =</span> <span class="fu">Sys.Date</span>(),</span>
<span id="cb101-31"><a href="#cb101-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">softPower =</span> softPower,</span>
<span id="cb101-32"><a href="#cb101-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">networkType =</span> <span class="st">"unsigned"</span>,</span>
<span id="cb101-33"><a href="#cb101-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">miRNAs =</span> <span class="fu">colnames</span>(datExpr),</span>
<span id="cb101-34"><a href="#cb101-34" aria-hidden="true" tabindex="-1"></a>  <span class="at">moduleColors =</span> moduleColors_paired,</span>
<span id="cb101-35"><a href="#cb101-35" aria-hidden="true" tabindex="-1"></a>  <span class="at">adjacency    =</span> adjacency,</span>
<span id="cb101-36"><a href="#cb101-36" aria-hidden="true" tabindex="-1"></a>  <span class="at">TOM          =</span> TOM_paired,</span>
<span id="cb101-37"><a href="#cb101-37" aria-hidden="true" tabindex="-1"></a>  <span class="at">MEs          =</span> MEs_paired,</span>
<span id="cb101-38"><a href="#cb101-38" aria-hidden="true" tabindex="-1"></a>  <span class="at">hubData      =</span> hubData_paired,</span>
<span id="cb101-39"><a href="#cb101-39" aria-hidden="true" tabindex="-1"></a>  <span class="at">lmm_results  =</span> lmm_results_paired</span>
<span id="cb101-40"><a href="#cb101-40" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb101-41"><a href="#cb101-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-42"><a href="#cb101-42" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(paired_wgcna_export, <span class="at">file =</span> <span class="st">"paired_wgcna_export.rds"</span>)</span>
<span id="cb101-43"><a href="#cb101-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-44"><a href="#cb101-44" aria-hidden="true" tabindex="-1"></a><span class="co"># Module sizes table</span></span>
<span id="cb101-45"><a href="#cb101-45" aria-hidden="true" tabindex="-1"></a>mod_sizes <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">table</span>(moduleColors_paired), <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span>
<span id="cb101-46"><a href="#cb101-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-47"><a href="#cb101-47" aria-hidden="true" tabindex="-1"></a>modules_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb101-48"><a href="#cb101-48" aria-hidden="true" tabindex="-1"></a>  <span class="at">Module  =</span> <span class="fu">names</span>(mod_sizes),</span>
<span id="cb101-49"><a href="#cb101-49" aria-hidden="true" tabindex="-1"></a>  <span class="at">Number_of_Genes =</span> <span class="fu">as.integer</span>(mod_sizes),</span>
<span id="cb101-50"><a href="#cb101-50" aria-hidden="true" tabindex="-1"></a>  <span class="at">Relative_Amount =</span> <span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> <span class="fu">as.integer</span>(mod_sizes) <span class="sc">/</span> <span class="fu">length</span>(moduleColors_paired), <span class="dv">1</span>),</span>
<span id="cb101-51"><a href="#cb101-51" aria-hidden="true" tabindex="-1"></a>  <span class="at">row.names =</span> <span class="cn">NULL</span></span>
<span id="cb101-52"><a href="#cb101-52" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb101-53"><a href="#cb101-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-54"><a href="#cb101-54" aria-hidden="true" tabindex="-1"></a>modules_df</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     Module Number_of_Genes Relative_Amount
1 turquoise             294            35.4
2      grey             250            30.1
3      blue             201            24.2
4     brown              85            10.2</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Top 10 turquoise by kWithin (with MM + GS)</span></span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a>top10_turquoise <span class="ot">&lt;-</span> hubData <span class="sc">%&gt;%</span></span>
<span id="cb103-3"><a href="#cb103-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(module <span class="sc">==</span> <span class="st">"turquoise"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb103-4"><a href="#cb103-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(kWithin)) <span class="sc">%&gt;%</span></span>
<span id="cb103-5"><a href="#cb103-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">transmute</span>(</span>
<span id="cb103-6"><a href="#cb103-6" aria-hidden="true" tabindex="-1"></a>    miRNA,</span>
<span id="cb103-7"><a href="#cb103-7" aria-hidden="true" tabindex="-1"></a>    module,</span>
<span id="cb103-8"><a href="#cb103-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">kWithin =</span> <span class="fu">round</span>(kWithin, <span class="dv">2</span>),</span>
<span id="cb103-9"><a href="#cb103-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">MMturquoise =</span> <span class="fu">round</span>(MMturquoise, <span class="dv">3</span>),</span>
<span id="cb103-10"><a href="#cb103-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">GS =</span> <span class="fu">round</span>(GS, <span class="dv">3</span>)</span>
<span id="cb103-11"><a href="#cb103-11" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb103-12"><a href="#cb103-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">10</span>)</span>
<span id="cb103-13"><a href="#cb103-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-14"><a href="#cb103-14" aria-hidden="true" tabindex="-1"></a>top10_turquoise</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          miRNA    module kWithin MMturquoise     GS
1  ILMN_3167805 turquoise    7.48       0.867 11.228
2  ILMN_3167455 turquoise    7.08       0.887 12.730
3  ILMN_3166941 turquoise    6.53       0.858  9.727
4  ILMN_3167624 turquoise    6.31       0.845  9.378
5  ILMN_3167988 turquoise    6.22       0.843 10.057
6  ILMN_3167522 turquoise    6.22       0.826  9.131
7  ILMN_3168513 turquoise    6.04       0.881 14.642
8  ILMN_3168646 turquoise    5.83      -0.844 14.660
9  ILMN_3167031 turquoise    5.68       0.793  8.073
10 ILMN_3168716 turquoise    5.43       0.809  8.304</code></pre>
</div>
</div>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>