<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Dataset1_separate</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="Dataset1_separate_files/libs/clipboard/clipboard.min.js"></script>
<script src="Dataset1_separate_files/libs/quarto-html/quarto.js" type="module"></script>
<script src="Dataset1_separate_files/libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="Dataset1_separate_files/libs/quarto-html/popper.min.js"></script>
<script src="Dataset1_separate_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Dataset1_separate_files/libs/quarto-html/anchor.min.js"></script>
<link href="Dataset1_separate_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Dataset1_separate_files/libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Dataset1_separate_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Dataset1_separate_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Dataset1_separate_files/libs/bootstrap/bootstrap-bb462d781dde1847d9e3ccf7736099dd.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent quarto-light">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Dataset1_separate</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Setup</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GEOquery)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Lade nötiges Paket: Biobase</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Lade nötiges Paket: BiocGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Lade nötiges Paket: generics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attache Paket: 'generics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Die folgenden Objekte sind maskiert von 'package:base':

    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,
    setequal, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attache Paket: 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Die folgenden Objekte sind maskiert von 'package:stats':

    IQR, mad, sd, var, xtabs</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Die folgenden Objekte sind maskiert von 'package:base':

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,
    unsplit, which.max, which.min</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Setting options('download.file.method.GEOquery'='auto')</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Setting options('GEOquery.inmemory.gpl'=FALSE)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(limma)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attache Paket: 'limma'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Das folgende Objekt ist maskiert 'package:BiocGenerics':

    plotMA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(WGCNA)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Paket 'WGCNA' wurde unter R Version 4.5.2 erstellt</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Lade nötiges Paket: dynamicTreeCut</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Paket 'dynamicTreeCut' wurde unter R Version 4.5.2 erstellt</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Lade nötiges Paket: fastcluster</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Paket 'fastcluster' wurde unter R Version 4.5.2 erstellt</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attache Paket: 'fastcluster'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Das folgende Objekt ist maskiert 'package:stats':

    hclust</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attache Paket: 'WGCNA'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Das folgende Objekt ist maskiert 'package:stats':

    cor</code></pre>
</div>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(nlme)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attache Paket: 'dplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Das folgende Objekt ist maskiert 'package:nlme':

    collapse</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Das folgende Objekt ist maskiert 'package:Biobase':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Die folgenden Objekte sind maskiert von 'package:BiocGenerics':

    combine, intersect, setdiff, setequal, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Das folgende Objekt ist maskiert 'package:generics':

    explain</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Die folgenden Objekte sind maskiert von 'package:stats':

    filter, lag</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Die folgenden Objekte sind maskiert von 'package:base':

    intersect, setdiff, setequal, union</code></pre>
</div>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">disableWGCNAThreads</span>()</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Load data</span></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>gse   <span class="ot">&lt;-</span> <span class="fu">getGEO</span>(<span class="st">"GSE45238"</span>, <span class="at">GSEMatrix =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found 1 file(s)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>GSE45238_series_matrix.txt.gz</code></pre>
</div>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>eset  <span class="ot">&lt;-</span> gse[[<span class="dv">1</span>]]</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>expr  <span class="ot">&lt;-</span> <span class="fu">exprs</span>(eset)</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> <span class="fu">pData</span>(eset)</span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>annot <span class="ot">&lt;-</span> <span class="fu">fData</span>(eset)</span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(pheno) <span class="ot">&lt;-</span> pheno<span class="sc">$</span>geo_accession</span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(expr); <span class="fu">head</span>(pheno); <span class="fu">head</span>(annot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 858  80</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>                  title geo_accession                status submission_date
GSM1099620 non-OSCC_01N    GSM1099620 Public on Jan 07 2015     Mar 18 2013
GSM1099621 non-OSCC_02N    GSM1099621 Public on Jan 07 2015     Mar 18 2013
GSM1099622 non-OSCC_03N    GSM1099622 Public on Jan 07 2015     Mar 18 2013
GSM1099623 non-OSCC_04N    GSM1099623 Public on Jan 07 2015     Mar 18 2013
GSM1099624 non-OSCC_05N    GSM1099624 Public on Jan 07 2015     Mar 18 2013
GSM1099625 non-OSCC_06N    GSM1099625 Public on Jan 07 2015     Mar 18 2013
           last_update_date type channel_count              source_name_ch1
GSM1099620      Jan 07 2015  RNA             1 adjacent non-tumor epithelia
GSM1099621      Jan 07 2015  RNA             1 adjacent non-tumor epithelia
GSM1099622      Jan 07 2015  RNA             1 adjacent non-tumor epithelia
GSM1099623      Jan 07 2015  RNA             1 adjacent non-tumor epithelia
GSM1099624      Jan 07 2015  RNA             1 adjacent non-tumor epithelia
GSM1099625      Jan 07 2015  RNA             1 adjacent non-tumor epithelia
           organism_ch1 characteristics_ch1 characteristics_ch1.1
GSM1099620 Homo sapiens        gender: male               age: 46
GSM1099621 Homo sapiens        gender: male               age: 49
GSM1099622 Homo sapiens        gender: male               age: 56
GSM1099623 Homo sapiens        gender: male               age: 50
GSM1099624 Homo sapiens        gender: male               age: 36
GSM1099625 Homo sapiens        gender: male               age: 48
           characteristics_ch1.2  characteristics_ch1.3 molecule_ch1
GSM1099620             Stage: II tissue: oral epithelia    total RNA
GSM1099621             Stage: II tissue: oral epithelia    total RNA
GSM1099622            Stage: III tissue: oral epithelia    total RNA
GSM1099623            Stage: III tissue: oral epithelia    total RNA
GSM1099624             Stage: II tissue: oral epithelia    total RNA
GSM1099625            Stage: IVA tissue: oral epithelia    total RNA
                                                                                                                                      extract_protocol_ch1
GSM1099620 Total RNA was extracted  from 40 pairs of OSCC patients using the miRNeasy Mini Kit (Qiagen, #217004) according to the manufacturer’s protocol.
GSM1099621 Total RNA was extracted  from 40 pairs of OSCC patients using the miRNeasy Mini Kit (Qiagen, #217004) according to the manufacturer’s protocol.
GSM1099622 Total RNA was extracted  from 40 pairs of OSCC patients using the miRNeasy Mini Kit (Qiagen, #217004) according to the manufacturer’s protocol.
GSM1099623 Total RNA was extracted  from 40 pairs of OSCC patients using the miRNeasy Mini Kit (Qiagen, #217004) according to the manufacturer’s protocol.
GSM1099624 Total RNA was extracted  from 40 pairs of OSCC patients using the miRNeasy Mini Kit (Qiagen, #217004) according to the manufacturer’s protocol.
GSM1099625 Total RNA was extracted  from 40 pairs of OSCC patients using the miRNeasy Mini Kit (Qiagen, #217004) according to the manufacturer’s protocol.
           label_ch1
GSM1099620       Cy3
GSM1099621       Cy3
GSM1099622       Cy3
GSM1099623       Cy3
GSM1099624       Cy3
GSM1099625       Cy3
                                                                   label_protocol_ch1
GSM1099620 Cy3 labled cRNA (PCR product) were prepared with illumina GoldenGate assay
GSM1099621 Cy3 labled cRNA (PCR product) were prepared with illumina GoldenGate assay
GSM1099622 Cy3 labled cRNA (PCR product) were prepared with illumina GoldenGate assay
GSM1099623 Cy3 labled cRNA (PCR product) were prepared with illumina GoldenGate assay
GSM1099624 Cy3 labled cRNA (PCR product) were prepared with illumina GoldenGate assay
GSM1099625 Cy3 labled cRNA (PCR product) were prepared with illumina GoldenGate assay
           taxid_ch1                             hyb_protocol
GSM1099620      9606 Standard Illumina hybridization protocol
GSM1099621      9606 Standard Illumina hybridization protocol
GSM1099622      9606 Standard Illumina hybridization protocol
GSM1099623      9606 Standard Illumina hybridization protocol
GSM1099624      9606 Standard Illumina hybridization protocol
GSM1099625      9606 Standard Illumina hybridization protocol
                                 scan_protocol description
GSM1099620 Standard Illumina scanning protocol replicate 1
GSM1099621 Standard Illumina scanning protocol replicate 1
GSM1099622 Standard Illumina scanning protocol replicate 1
GSM1099623 Standard Illumina scanning protocol replicate 1
GSM1099624 Standard Illumina scanning protocol replicate 1
GSM1099625 Standard Illumina scanning protocol replicate 1
                                                                       data_processing
GSM1099620 The data were normalised using quantile normalisation with IlluminaGUI in R
GSM1099621 The data were normalised using quantile normalisation with IlluminaGUI in R
GSM1099622 The data were normalised using quantile normalisation with IlluminaGUI in R
GSM1099623 The data were normalised using quantile normalisation with IlluminaGUI in R
GSM1099624 The data were normalised using quantile normalisation with IlluminaGUI in R
GSM1099625 The data were normalised using quantile normalisation with IlluminaGUI in R
           platform_id     contact_name        contact_email
GSM1099620     GPL8179 Shine-Gwo,,Shiah davidssg@nhri.org.tw
GSM1099621     GPL8179 Shine-Gwo,,Shiah davidssg@nhri.org.tw
GSM1099622     GPL8179 Shine-Gwo,,Shiah davidssg@nhri.org.tw
GSM1099623     GPL8179 Shine-Gwo,,Shiah davidssg@nhri.org.tw
GSM1099624     GPL8179 Shine-Gwo,,Shiah davidssg@nhri.org.tw
GSM1099625     GPL8179 Shine-Gwo,,Shiah davidssg@nhri.org.tw
                              contact_department
GSM1099620 National Institute of Cancer Research
GSM1099621 National Institute of Cancer Research
GSM1099622 National Institute of Cancer Research
GSM1099623 National Institute of Cancer Research
GSM1099624 National Institute of Cancer Research
GSM1099625 National Institute of Cancer Research
                             contact_institute   contact_address contact_city
GSM1099620 National Health Research Institutes No. 35, Keyan Rd.       Zhunan
GSM1099621 National Health Research Institutes No. 35, Keyan Rd.       Zhunan
GSM1099622 National Health Research Institutes No. 35, Keyan Rd.       Zhunan
GSM1099623 National Health Research Institutes No. 35, Keyan Rd.       Zhunan
GSM1099624 National Health Research Institutes No. 35, Keyan Rd.       Zhunan
GSM1099625 National Health Research Institutes No. 35, Keyan Rd.       Zhunan
           contact_state contact_zip/postal_code contact_country
GSM1099620 Miaoli County                     350          Taiwan
GSM1099621 Miaoli County                     350          Taiwan
GSM1099622 Miaoli County                     350          Taiwan
GSM1099623 Miaoli County                     350          Taiwan
GSM1099624 Miaoli County                     350          Taiwan
GSM1099625 Miaoli County                     350          Taiwan
           supplementary_file data_row_count age:ch1 gender:ch1 Stage:ch1
GSM1099620               NONE            858      46       male        II
GSM1099621               NONE            858      49       male        II
GSM1099622               NONE            858      56       male       III
GSM1099623               NONE            858      50       male       III
GSM1099624               NONE            858      36       male        II
GSM1099625               NONE            858      48       male       IVA
               tissue:ch1
GSM1099620 oral epithelia
GSM1099621 oral epithelia
GSM1099622 oral epithelia
GSM1099623 oral epithelia
GSM1099624 oral epithelia
GSM1099625 oral epithelia</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>                       ID       SYMBOL    ILMN_Gene   Search_Key
ILMN_3166935 ILMN_3166935 ILMN_3166935  hsa-miR-329  hsa-miR-329
ILMN_3166938 ILMN_3166938 ILMN_3166938  hsa-miR-424  hsa-miR-424
ILMN_3166940 ILMN_3166940 ILMN_3166940  hsa-miR-137  hsa-miR-137
ILMN_3166941 ILMN_3166941 ILMN_3166941 hsa-miR-376c hsa-miR-376c
ILMN_3166943 ILMN_3166943 ILMN_3166943  hsa-miR-585  hsa-miR-585
ILMN_3166944 ILMN_3166944 ILMN_3166944 hsa-miR-302c hsa-miR-302c
                        SEQUENCE        TargetMatureSeqs TargetMatureName
ILMN_3166935 AACACACCTGGTTAACCTC  AACACACCTGGTTAACCTCTTT      hsa-miR-329
ILMN_3166938  CAGCAGCAATTCATGTTT  CAGCAGCAATTCATGTTTTGAA      hsa-miR-424
ILMN_3166940 TATTGCTTAAGAATACGCG TTATTGCTTAAGAATACGCGTAG      hsa-miR-137
ILMN_3166941 ACATAGAGGAAATTCCACG   AACATAGAGGAAATTCCACGT     hsa-miR-376c
ILMN_3166943   TGGGCGTATCTGTATGC     TGGGCGTATCTGTATGCTA      hsa-miR-585
ILMN_3166944  GCTTCCATGTTTCAGTGG TAAGTGCTTCCATGTTTCAGTGG     hsa-miR-302c
                 miRNA_ID SPOT_ID NumTargets TargetMatureVersion
ILMN_3166935  hsa-miR-329                  1                  12
ILMN_3166938  hsa-miR-424                  1                  12
ILMN_3166940  hsa-miR-137                  1                  12
ILMN_3166941 hsa-miR-376c                  1                  12
ILMN_3166943  hsa-miR-585                  1                  12
ILMN_3166944 hsa-miR-302c                  1                  12
                   OriginalMatureSeq OriginalMatureName
ILMN_3166935  AACACACCTGGTTAACCTCTTT        hsa-miR-329
ILMN_3166938  CAGCAGCAATTCATGTTTTGAA        hsa-miR-424
ILMN_3166940  TATTGCTTAAGAATACGCGTAG        hsa-miR-137
ILMN_3166941  ACATAGAGGAAATTCCACGTTT        hsa-miR-368
ILMN_3166943     TGGGCGTATCTGTATGCTA        hsa-miR-585
ILMN_3166944 TAAGTGCTTCCATGTTTCAGTGG       hsa-miR-302c
                                        Source Array_Address_Id
ILMN_3166935 Sanger miRNAdb (v_8.2, July 2006)              931
ILMN_3166938 Sanger miRNAdb (v_8.2, July 2006)             1051
ILMN_3166940 Sanger miRNAdb (v_8.2, July 2006)              233
ILMN_3166941 Sanger miRNAdb (v_8.2, July 2006)              976
ILMN_3166943 Sanger miRNAdb (v_8.2, July 2006)             1761
ILMN_3166944 Sanger miRNAdb (v_8.2, July 2006)              864
                         Illumicode
ILMN_3166935 CGGCAGATAGCACGTTTCGACT
ILMN_3166938 GGGCACTATTACTCGCTAACCG
ILMN_3166940 GTACAGTCTCGCGGTGCCTAAG
ILMN_3166941 GTTTCCACGACTTGCAGTACGG
ILMN_3166943 GTCGGTTAGTGATGTACGGCCA
ILMN_3166944 CCGTTTGGACGTAAGCAGGGCC
                                                                   Oligo
ILMN_3166935 ACTTCGTCAGTAACGGACCGGCAGATAGCACGTTTCGACTAACACACCTGGTTAACCTC
ILMN_3166938  ACTTCGTCAGTAACGGACGGGCACTATTACTCGCTAACCGCAGCAGCAATTCATGTTT
ILMN_3166940 ACTTCGTCAGTAACGGACGTACAGTCTCGCGGTGCCTAAGTATTGCTTAAGAATACGCG
ILMN_3166941 ACTTCGTCAGTAACGGACGTTTCCACGACTTGCAGTACGGACATAGAGGAAATTCCACG
ILMN_3166943   ACTTCGTCAGTAACGGACGTCGGTTAGTGATGTACGGCCATGGGCGTATCTGTATGC
ILMN_3166944  ACTTCGTCAGTAACGGACCCGTTTGGACGTAAGCAGGGCCGCTTCCATGTTTCAGTGG
                         U3_Seq  Ploidy Species Probe_MatchOrder Chromosome
ILMN_3166935 ACTTCGTCAGTAACGGAC diploid   human                2      14,14
ILMN_3166938 ACTTCGTCAGTAACGGAC diploid   human                1          X
ILMN_3166940 ACTTCGTCAGTAACGGAC diploid   human                1          1
ILMN_3166941 ACTTCGTCAGTAACGGAC diploid   human                1         14
ILMN_3166943 ACTTCGTCAGTAACGGAC diploid   human                1          5
ILMN_3166944 ACTTCGTCAGTAACGGAC diploid   human                1          4
                   Probe_Coordinates Probe_Chr_Orientation
ILMN_3166935 100,562,923,100,563,000                   +,+
ILMN_3166938               133508398                     -
ILMN_3166940                98284258                     -
ILMN_3166941               100575821                     +
ILMN_3166943               168623217                     -
ILMN_3166944               113788994                     -</code></pre>
</div>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Probe filtering</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(annot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "ID"                    "SYMBOL"                "ILMN_Gene"            
 [4] "Search_Key"            "SEQUENCE"              "TargetMatureSeqs"     
 [7] "TargetMatureName"      "miRNA_ID"              "SPOT_ID"              
[10] "NumTargets"            "TargetMatureVersion"   "OriginalMatureSeq"    
[13] "OriginalMatureName"    "Source"                "Array_Address_Id"     
[16] "Illumicode"            "Oligo"                 "U3_Seq"               
[19] "Ploidy"                "Species"               "Probe_MatchOrder"     
[22] "Chromosome"            "Probe_Coordinates"     "Probe_Chr_Orientation"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(annot<span class="sc">$</span>miRNA_version)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>NULL</code></pre>
</div>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>is_mature_v12 <span class="ot">&lt;-</span> annot<span class="sc">$</span>TargetMatureVersion <span class="sc">==</span> <span class="dv">12</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>is_control <span class="ot">&lt;-</span> <span class="fu">grepl</span>(<span class="st">"control|spike|U6|snoR"</span>, annot<span class="sc">$</span>SYMBOL, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>)</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>keep_probes <span class="ot">&lt;-</span> is_mature_v12 <span class="sc">&amp;</span> <span class="sc">!</span>is_control</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>expr  <span class="ot">&lt;-</span> expr[keep_probes, ]</span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>annot <span class="ot">&lt;-</span> annot[keep_probes, ]</span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(expr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 830  80</code></pre>
</div>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(annot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                       ID       SYMBOL    ILMN_Gene   Search_Key
ILMN_3166935 ILMN_3166935 ILMN_3166935  hsa-miR-329  hsa-miR-329
ILMN_3166938 ILMN_3166938 ILMN_3166938  hsa-miR-424  hsa-miR-424
ILMN_3166940 ILMN_3166940 ILMN_3166940  hsa-miR-137  hsa-miR-137
ILMN_3166941 ILMN_3166941 ILMN_3166941 hsa-miR-376c hsa-miR-376c
ILMN_3166943 ILMN_3166943 ILMN_3166943  hsa-miR-585  hsa-miR-585
ILMN_3166944 ILMN_3166944 ILMN_3166944 hsa-miR-302c hsa-miR-302c
                        SEQUENCE        TargetMatureSeqs TargetMatureName
ILMN_3166935 AACACACCTGGTTAACCTC  AACACACCTGGTTAACCTCTTT      hsa-miR-329
ILMN_3166938  CAGCAGCAATTCATGTTT  CAGCAGCAATTCATGTTTTGAA      hsa-miR-424
ILMN_3166940 TATTGCTTAAGAATACGCG TTATTGCTTAAGAATACGCGTAG      hsa-miR-137
ILMN_3166941 ACATAGAGGAAATTCCACG   AACATAGAGGAAATTCCACGT     hsa-miR-376c
ILMN_3166943   TGGGCGTATCTGTATGC     TGGGCGTATCTGTATGCTA      hsa-miR-585
ILMN_3166944  GCTTCCATGTTTCAGTGG TAAGTGCTTCCATGTTTCAGTGG     hsa-miR-302c
                 miRNA_ID SPOT_ID NumTargets TargetMatureVersion
ILMN_3166935  hsa-miR-329                  1                  12
ILMN_3166938  hsa-miR-424                  1                  12
ILMN_3166940  hsa-miR-137                  1                  12
ILMN_3166941 hsa-miR-376c                  1                  12
ILMN_3166943  hsa-miR-585                  1                  12
ILMN_3166944 hsa-miR-302c                  1                  12
                   OriginalMatureSeq OriginalMatureName
ILMN_3166935  AACACACCTGGTTAACCTCTTT        hsa-miR-329
ILMN_3166938  CAGCAGCAATTCATGTTTTGAA        hsa-miR-424
ILMN_3166940  TATTGCTTAAGAATACGCGTAG        hsa-miR-137
ILMN_3166941  ACATAGAGGAAATTCCACGTTT        hsa-miR-368
ILMN_3166943     TGGGCGTATCTGTATGCTA        hsa-miR-585
ILMN_3166944 TAAGTGCTTCCATGTTTCAGTGG       hsa-miR-302c
                                        Source Array_Address_Id
ILMN_3166935 Sanger miRNAdb (v_8.2, July 2006)              931
ILMN_3166938 Sanger miRNAdb (v_8.2, July 2006)             1051
ILMN_3166940 Sanger miRNAdb (v_8.2, July 2006)              233
ILMN_3166941 Sanger miRNAdb (v_8.2, July 2006)              976
ILMN_3166943 Sanger miRNAdb (v_8.2, July 2006)             1761
ILMN_3166944 Sanger miRNAdb (v_8.2, July 2006)              864
                         Illumicode
ILMN_3166935 CGGCAGATAGCACGTTTCGACT
ILMN_3166938 GGGCACTATTACTCGCTAACCG
ILMN_3166940 GTACAGTCTCGCGGTGCCTAAG
ILMN_3166941 GTTTCCACGACTTGCAGTACGG
ILMN_3166943 GTCGGTTAGTGATGTACGGCCA
ILMN_3166944 CCGTTTGGACGTAAGCAGGGCC
                                                                   Oligo
ILMN_3166935 ACTTCGTCAGTAACGGACCGGCAGATAGCACGTTTCGACTAACACACCTGGTTAACCTC
ILMN_3166938  ACTTCGTCAGTAACGGACGGGCACTATTACTCGCTAACCGCAGCAGCAATTCATGTTT
ILMN_3166940 ACTTCGTCAGTAACGGACGTACAGTCTCGCGGTGCCTAAGTATTGCTTAAGAATACGCG
ILMN_3166941 ACTTCGTCAGTAACGGACGTTTCCACGACTTGCAGTACGGACATAGAGGAAATTCCACG
ILMN_3166943   ACTTCGTCAGTAACGGACGTCGGTTAGTGATGTACGGCCATGGGCGTATCTGTATGC
ILMN_3166944  ACTTCGTCAGTAACGGACCCGTTTGGACGTAAGCAGGGCCGCTTCCATGTTTCAGTGG
                         U3_Seq  Ploidy Species Probe_MatchOrder Chromosome
ILMN_3166935 ACTTCGTCAGTAACGGAC diploid   human                2      14,14
ILMN_3166938 ACTTCGTCAGTAACGGAC diploid   human                1          X
ILMN_3166940 ACTTCGTCAGTAACGGAC diploid   human                1          1
ILMN_3166941 ACTTCGTCAGTAACGGAC diploid   human                1         14
ILMN_3166943 ACTTCGTCAGTAACGGAC diploid   human                1          5
ILMN_3166944 ACTTCGTCAGTAACGGAC diploid   human                1          4
                   Probe_Coordinates Probe_Chr_Orientation
ILMN_3166935 100,562,923,100,563,000                   +,+
ILMN_3166938               133508398                     -
ILMN_3166940                98284258                     -
ILMN_3166941               100575821                     +
ILMN_3166943               168623217                     -
ILMN_3166944               113788994                     -</code></pre>
</div>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalization</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>expr_log2 <span class="ot">&lt;-</span> <span class="fu">log2</span>(expr <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>expr_norm <span class="ot">&lt;-</span> <span class="fu">normalizeBetweenArrays</span>(expr_log2, <span class="at">method =</span> <span class="st">"quantile"</span>)</span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(expr_norm) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(expr)</span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(expr_norm) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(expr)</span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a><span class="co"># QC: PCA + sample clustering</span></span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a>op <span class="ot">&lt;-</span> <span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">10</span>), <span class="at">xpd =</span> <span class="cn">TRUE</span>)</span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a>expr_t <span class="ot">&lt;-</span> <span class="fu">t</span>(expr_norm)</span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(expr_t, <span class="at">scale. =</span> <span class="cn">TRUE</span>)</span>
<span id="cb50-13"><a href="#cb50-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-14"><a href="#cb50-14" aria-hidden="true" tabindex="-1"></a>sample_ids    <span class="ot">&lt;-</span> <span class="fu">colnames</span>(expr_norm)</span>
<span id="cb50-15"><a href="#cb50-15" aria-hidden="true" tabindex="-1"></a>highlight_ids <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"GSM1099641"</span>, <span class="st">"GSM1099655"</span>)</span>
<span id="cb50-16"><a href="#cb50-16" aria-hidden="true" tabindex="-1"></a>point_col     <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_ids <span class="sc">%in%</span> highlight_ids, <span class="st">"red"</span>, <span class="st">"black"</span>)</span>
<span id="cb50-17"><a href="#cb50-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-18"><a href="#cb50-18" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(pca<span class="sc">$</span>x[,<span class="dv">1</span>], pca<span class="sc">$</span>x[,<span class="dv">2</span>],</span>
<span id="cb50-19"><a href="#cb50-19" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"PC1"</span>, <span class="at">ylab =</span> <span class="st">"PC2"</span>,</span>
<span id="cb50-20"><a href="#cb50-20" aria-hidden="true" tabindex="-1"></a>     <span class="at">pch =</span> <span class="dv">19</span>,</span>
<span id="cb50-21"><a href="#cb50-21" aria-hidden="true" tabindex="-1"></a>     <span class="at">col =</span> point_col,</span>
<span id="cb50-22"><a href="#cb50-22" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"PCA after normalization"</span>)</span>
<span id="cb50-23"><a href="#cb50-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-24"><a href="#cb50-24" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(pca<span class="sc">$</span>x[,<span class="dv">1</span>], pca<span class="sc">$</span>x[,<span class="dv">2</span>],</span>
<span id="cb50-25"><a href="#cb50-25" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels =</span> sample_ids,</span>
<span id="cb50-26"><a href="#cb50-26" aria-hidden="true" tabindex="-1"></a>     <span class="at">pos =</span> <span class="dv">3</span>, <span class="at">cex =</span> <span class="fl">0.6</span>,</span>
<span id="cb50-27"><a href="#cb50-27" aria-hidden="true" tabindex="-1"></a>     <span class="at">col =</span> point_col)</span>
<span id="cb50-28"><a href="#cb50-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-29"><a href="#cb50-29" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"topright"</span>,</span>
<span id="cb50-30"><a href="#cb50-30" aria-hidden="true" tabindex="-1"></a>       <span class="at">inset =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">0.55</span>, <span class="dv">0</span>),</span>
<span id="cb50-31"><a href="#cb50-31" aria-hidden="true" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">"Other samples"</span>, <span class="st">"Outliers (to remove)"</span>),</span>
<span id="cb50-32"><a href="#cb50-32" aria-hidden="true" tabindex="-1"></a>       <span class="at">col    =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"red"</span>),</span>
<span id="cb50-33"><a href="#cb50-33" aria-hidden="true" tabindex="-1"></a>       <span class="at">pch    =</span> <span class="dv">19</span>,</span>
<span id="cb50-34"><a href="#cb50-34" aria-hidden="true" tabindex="-1"></a>       <span class="at">bty    =</span> <span class="st">"n"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_separate_files/figure-html/unnamed-chunk-1-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(op)</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>sampleTree <span class="ot">&lt;-</span> <span class="fu">hclust</span>(<span class="fu">dist</span>(expr_t), <span class="at">method =</span> <span class="st">"average"</span>)</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(sampleTree, <span class="at">labels =</span> <span class="fu">colnames</span>(expr_norm), <span class="at">cex =</span> <span class="fl">0.6</span>,</span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"Sample clustering"</span>)</span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">35</span>, <span class="at">col =</span> <span class="st">"red"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_separate_files/figure-html/unnamed-chunk-1-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"SampleClustering.pdf"</span>, <span class="at">width =</span> <span class="dv">14</span>, <span class="at">height =</span> <span class="dv">10</span>)</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(sampleTree,</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels =</span> <span class="fu">colnames</span>(expr_norm),</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">cex =</span> <span class="fl">0.6</span>,</span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"Sample clustering dendrogram"</span>)</span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">35</span>, <span class="at">col =</span> <span class="st">"red"</span>)</span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Patient IDs + remove outlier pairs (order-safe)</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>patient_id <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">".*_(</span><span class="sc">\\</span><span class="st">d+)[NT]$"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, pheno<span class="sc">$</span>title)</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>patient_id <span class="ot">&lt;-</span> <span class="fu">as.character</span>(pheno<span class="sc">$</span>patient_id)</span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a>outlier_samples <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"GSM1099641"</span>, <span class="st">"GSM1099655"</span>)</span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>outlier_patient_ids <span class="ot">&lt;-</span> <span class="fu">unique</span>(pheno<span class="sc">$</span>patient_id[pheno<span class="sc">$</span>geo_accession <span class="sc">%in%</span> outlier_samples])</span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a>keep_ids <span class="ot">&lt;-</span> pheno<span class="sc">$</span>geo_accession[<span class="sc">!</span>pheno<span class="sc">$</span>patient_id <span class="sc">%in%</span> outlier_patient_ids]</span>
<span id="cb54-9"><a href="#cb54-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-10"><a href="#cb54-10" aria-hidden="true" tabindex="-1"></a>expr_norm <span class="ot">&lt;-</span> expr_norm[, keep_ids]</span>
<span id="cb54-11"><a href="#cb54-11" aria-hidden="true" tabindex="-1"></a>pheno     <span class="ot">&lt;-</span> pheno[keep_ids, ]</span>
<span id="cb54-12"><a href="#cb54-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-13"><a href="#cb54-13" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">colnames</span>(expr_norm), <span class="fu">rownames</span>(pheno)))</span>
<span id="cb54-14"><a href="#cb54-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-15"><a href="#cb54-15" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(expr_norm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 830  76</code></pre>
</div>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(pheno<span class="sc">$</span>patient_id)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
01 02 03 04 05 06 07 08 09 10 11 12 13 14 15 16 17 18 19 20 21 23 24 25 26 27 
 2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2 
28 29 30 31 32 33 34 35 37 38 39 40 
 2  2  2  2  2  2  2  2  2  2  2  2 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Expression matrix + condition split</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>condition <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"T$"</span>, pheno<span class="sc">$</span>title), <span class="st">"tumor"</span>, <span class="st">"normal"</span>)</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>datExpr <span class="ot">&lt;-</span> <span class="fu">t</span>(expr_norm)</span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(datExpr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  76 830</code></pre>
</div>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>gsg <span class="ot">&lt;-</span> <span class="fu">goodSamplesGenes</span>(datExpr, <span class="at">verbose =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Flagging genes and samples with too many missing values...
  ..step 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>gsg<span class="sc">$</span>allOK</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Separate tumour vs normal matrices</span></span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">rownames</span>(pheno), <span class="fu">rownames</span>(datExpr)))</span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>datExpr_tumor  <span class="ot">&lt;-</span> datExpr[pheno<span class="sc">$</span>condition <span class="sc">==</span> <span class="st">"tumor"</span>,  , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>datExpr_normal <span class="ot">&lt;-</span> datExpr[pheno<span class="sc">$</span>condition <span class="sc">==</span> <span class="st">"normal"</span>, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(datExpr_tumor)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  38 830</code></pre>
</div>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(datExpr_normal)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  38 830</code></pre>
</div>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>gsg_t <span class="ot">&lt;-</span> <span class="fu">goodSamplesGenes</span>(datExpr_tumor,  <span class="at">verbose =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Flagging genes and samples with too many missing values...
  ..step 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>gsg_n <span class="ot">&lt;-</span> <span class="fu">goodSamplesGenes</span>(datExpr_normal, <span class="at">verbose =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Flagging genes and samples with too many missing values...
  ..step 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(gsg_t<span class="sc">$</span>allOK, gsg_n<span class="sc">$</span>allOK)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Soft-threshold power per subset</span></span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>powers <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span></span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a>sft_tumor <span class="ot">&lt;-</span> <span class="fu">pickSoftThreshold</span>(datExpr_tumor,</span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a>                               <span class="at">powerVector =</span> powers,</span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a>                               <span class="at">networkType =</span> <span class="st">"unsigned"</span>,</span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a>                               <span class="at">verbose =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>pickSoftThreshold: will use block size 830.
 pickSoftThreshold: calculating connectivity for given powers...
   ..working on genes 1 through 830 of 830
   Power SFT.R.sq slope truncated.R.sq mean.k. median.k. max.k.
1      1    0.676 -2.50          0.876 147.000  1.38e+02 255.00
2      2    0.903 -1.98          0.953  42.500  3.51e+01 120.00
3      3    0.972 -1.72          0.979  16.200  1.13e+01  68.20
4      4    0.993 -1.55          0.992   7.540  4.27e+00  43.60
5      5    0.965 -1.47          0.956   4.050  1.80e+00  31.70
6      6    0.948 -1.44          0.939   2.430  8.12e-01  24.50
7      7    0.913 -1.42          0.910   1.590  3.82e-01  19.60
8      8    0.919 -1.36          0.913   1.110  2.02e-01  16.20
9      9    0.933 -1.30          0.931   0.810  1.11e-01  13.60
10    10    0.931 -1.29          0.931   0.617  6.30e-02  11.60
11    11    0.890 -1.30          0.887   0.484  3.77e-02  10.00
12    12    0.894 -1.26          0.887   0.389  2.28e-02   8.73
13    13    0.939 -1.21          0.939   0.319  1.38e-02   7.67
14    14    0.949 -1.18          0.944   0.266  8.23e-03   6.78
15    15    0.931 -1.19          0.923   0.224  5.15e-03   6.02
16    16    0.953 -1.16          0.952   0.192  3.17e-03   5.38
17    17    0.965 -1.14          0.967   0.165  2.00e-03   4.82
18    18    0.957 -1.13          0.954   0.144  1.27e-03   4.33
19    19    0.952 -1.12          0.945   0.126  7.72e-04   3.91
20    20    0.944 -1.11          0.936   0.111  4.82e-04   3.54</code></pre>
</div>
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>sft_normal <span class="ot">&lt;-</span> <span class="fu">pickSoftThreshold</span>(datExpr_normal,</span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">powerVector =</span> powers,</span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">networkType =</span> <span class="st">"unsigned"</span>,</span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a>                                <span class="at">verbose =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>pickSoftThreshold: will use block size 830.
 pickSoftThreshold: calculating connectivity for given powers...
   ..working on genes 1 through 830 of 830
   Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.
1      1    0.242 -0.991          0.833 158.000  1.52e+02 257.00
2      2    0.708 -1.630          0.824  48.900  4.28e+01 125.00
3      3    0.869 -1.630          0.880  19.900  1.47e+01  75.20
4      4    0.909 -1.550          0.890   9.710  5.89e+00  51.10
5      5    0.951 -1.440          0.937   5.450  2.61e+00  37.30
6      6    0.931 -1.390          0.911   3.390  1.27e+00  28.70
7      7    0.945 -1.280          0.936   2.280  6.51e-01  22.80
8      8    0.971 -1.220          0.976   1.620  3.53e-01  18.50
9      9    0.960 -1.160          0.969   1.210  2.04e-01  15.30
10    10    0.931 -1.140          0.931   0.934  1.16e-01  13.10
11    11    0.954 -1.120          0.949   0.741  6.95e-02  11.50
12    12    0.961 -1.110          0.953   0.601  4.18e-02  10.20
13    13    0.957 -1.110          0.945   0.496  2.63e-02   9.18
14    14    0.916 -1.130          0.904   0.415  1.68e-02   8.28
15    15    0.928 -1.120          0.922   0.352  1.08e-02   7.51
16    16    0.931 -1.120          0.931   0.301  6.98e-03   6.84
17    17    0.936 -1.100          0.943   0.260  4.38e-03   6.25
18    18    0.928 -1.110          0.935   0.226  2.88e-03   5.73
19    19    0.936 -1.090          0.954   0.199  1.98e-03   5.27
20    20    0.920 -1.100          0.935   0.175  1.29e-03   4.86</code></pre>
</div>
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(sft_tumor<span class="sc">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb77-4"><a href="#cb77-4" aria-hidden="true" tabindex="-1"></a>     <span class="sc">-</span><span class="fu">sign</span>(sft_tumor<span class="sc">$</span>fitIndices[,<span class="dv">3</span>]) <span class="sc">*</span> sft_tumor<span class="sc">$</span>fitIndices[,<span class="dv">2</span>],</span>
<span id="cb77-5"><a href="#cb77-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">"Power"</span>, <span class="at">ylab=</span><span class="st">"Scale-free R^2"</span>,</span>
<span id="cb77-6"><a href="#cb77-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">type=</span><span class="st">"n"</span>, <span class="at">main=</span><span class="st">"Tumor: scale independence"</span>)</span>
<span id="cb77-7"><a href="#cb77-7" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(sft_tumor<span class="sc">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb77-8"><a href="#cb77-8" aria-hidden="true" tabindex="-1"></a>     <span class="sc">-</span><span class="fu">sign</span>(sft_tumor<span class="sc">$</span>fitIndices[,<span class="dv">3</span>]) <span class="sc">*</span> sft_tumor<span class="sc">$</span>fitIndices[,<span class="dv">2</span>],</span>
<span id="cb77-9"><a href="#cb77-9" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels=</span>powers, <span class="at">cex=</span><span class="fl">0.8</span>, <span class="at">col=</span><span class="st">"red"</span>)</span>
<span id="cb77-10"><a href="#cb77-10" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="fl">0.9</span>, <span class="at">col=</span><span class="st">"red"</span>)</span>
<span id="cb77-11"><a href="#cb77-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-12"><a href="#cb77-12" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(sft_tumor<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], sft_tumor<span class="sc">$</span>fitIndices[,<span class="dv">5</span>],</span>
<span id="cb77-13"><a href="#cb77-13" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">"Power"</span>, <span class="at">ylab=</span><span class="st">"Mean connectivity"</span>,</span>
<span id="cb77-14"><a href="#cb77-14" aria-hidden="true" tabindex="-1"></a>     <span class="at">type=</span><span class="st">"n"</span>, <span class="at">main=</span><span class="st">"Tumor: mean connectivity"</span>)</span>
<span id="cb77-15"><a href="#cb77-15" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(sft_tumor<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], sft_tumor<span class="sc">$</span>fitIndices[,<span class="dv">5</span>],</span>
<span id="cb77-16"><a href="#cb77-16" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels=</span>powers, <span class="at">cex=</span><span class="fl">0.8</span>, <span class="at">col=</span><span class="st">"red"</span>)</span>
<span id="cb77-17"><a href="#cb77-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-18"><a href="#cb77-18" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(sft_normal<span class="sc">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb77-19"><a href="#cb77-19" aria-hidden="true" tabindex="-1"></a>     <span class="sc">-</span><span class="fu">sign</span>(sft_normal<span class="sc">$</span>fitIndices[,<span class="dv">3</span>]) <span class="sc">*</span> sft_normal<span class="sc">$</span>fitIndices[,<span class="dv">2</span>],</span>
<span id="cb77-20"><a href="#cb77-20" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">"Power"</span>, <span class="at">ylab=</span><span class="st">"Scale-free R^2"</span>,</span>
<span id="cb77-21"><a href="#cb77-21" aria-hidden="true" tabindex="-1"></a>     <span class="at">type=</span><span class="st">"n"</span>, <span class="at">main=</span><span class="st">"Normal: scale independence"</span>)</span>
<span id="cb77-22"><a href="#cb77-22" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(sft_normal<span class="sc">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb77-23"><a href="#cb77-23" aria-hidden="true" tabindex="-1"></a>     <span class="sc">-</span><span class="fu">sign</span>(sft_normal<span class="sc">$</span>fitIndices[,<span class="dv">3</span>]) <span class="sc">*</span> sft_normal<span class="sc">$</span>fitIndices[,<span class="dv">2</span>],</span>
<span id="cb77-24"><a href="#cb77-24" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels=</span>powers, <span class="at">cex=</span><span class="fl">0.8</span>, <span class="at">col=</span><span class="st">"red"</span>)</span>
<span id="cb77-25"><a href="#cb77-25" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="fl">0.9</span>, <span class="at">col=</span><span class="st">"red"</span>)</span>
<span id="cb77-26"><a href="#cb77-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-27"><a href="#cb77-27" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(sft_normal<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], sft_normal<span class="sc">$</span>fitIndices[,<span class="dv">5</span>],</span>
<span id="cb77-28"><a href="#cb77-28" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">"Power"</span>, <span class="at">ylab=</span><span class="st">"Mean connectivity"</span>,</span>
<span id="cb77-29"><a href="#cb77-29" aria-hidden="true" tabindex="-1"></a>     <span class="at">type=</span><span class="st">"n"</span>, <span class="at">main=</span><span class="st">"Normal: mean connectivity"</span>)</span>
<span id="cb77-30"><a href="#cb77-30" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(sft_normal<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], sft_normal<span class="sc">$</span>fitIndices[,<span class="dv">5</span>],</span>
<span id="cb77-31"><a href="#cb77-31" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels=</span>powers, <span class="at">cex=</span><span class="fl">0.8</span>, <span class="at">col=</span><span class="st">"red"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_separate_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>softPower_tumor  <span class="ot">&lt;-</span> <span class="dv">4</span></span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a>softPower_normal <span class="ot">&lt;-</span> <span class="dv">4</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Helper: single-network WGCNA run (subset)</span></span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>run_single_wgcna <span class="ot">&lt;-</span> <span class="cf">function</span>(datExpr_subset,</span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>                             <span class="at">softPower =</span> <span class="dv">7</span>,</span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>                             <span class="at">minModuleSize =</span> <span class="dv">30</span>,</span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a>                             <span class="at">deepSplit =</span> <span class="dv">2</span>,</span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a>                             <span class="at">mergeCutHeight =</span> <span class="fl">0.25</span>,</span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a>                             <span class="at">networkType =</span> <span class="st">"unsigned"</span>) {</span>
<span id="cb79-8"><a href="#cb79-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-9"><a href="#cb79-9" aria-hidden="true" tabindex="-1"></a>  adjacency_mat <span class="ot">&lt;-</span> <span class="fu">adjacency</span>(datExpr_subset,</span>
<span id="cb79-10"><a href="#cb79-10" aria-hidden="true" tabindex="-1"></a>                             <span class="at">power =</span> softPower,</span>
<span id="cb79-11"><a href="#cb79-11" aria-hidden="true" tabindex="-1"></a>                             <span class="at">type  =</span> networkType)</span>
<span id="cb79-12"><a href="#cb79-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-13"><a href="#cb79-13" aria-hidden="true" tabindex="-1"></a>  TOM <span class="ot">&lt;-</span> <span class="fu">TOMsimilarity</span>(adjacency_mat, <span class="at">TOMType =</span> networkType)</span>
<span id="cb79-14"><a href="#cb79-14" aria-hidden="true" tabindex="-1"></a>  dissTOM <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">-</span> TOM</span>
<span id="cb79-15"><a href="#cb79-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-16"><a href="#cb79-16" aria-hidden="true" tabindex="-1"></a>  geneTree <span class="ot">&lt;-</span> <span class="fu">hclust</span>(<span class="fu">as.dist</span>(dissTOM), <span class="at">method =</span> <span class="st">"average"</span>)</span>
<span id="cb79-17"><a href="#cb79-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-18"><a href="#cb79-18" aria-hidden="true" tabindex="-1"></a>  dynamicMods <span class="ot">&lt;-</span> <span class="fu">cutreeDynamic</span>(</span>
<span id="cb79-19"><a href="#cb79-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">dendro =</span> geneTree,</span>
<span id="cb79-20"><a href="#cb79-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">distM =</span> dissTOM,</span>
<span id="cb79-21"><a href="#cb79-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">deepSplit =</span> deepSplit,</span>
<span id="cb79-22"><a href="#cb79-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">pamRespectsDendro =</span> <span class="cn">FALSE</span>,</span>
<span id="cb79-23"><a href="#cb79-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">minClusterSize =</span> minModuleSize</span>
<span id="cb79-24"><a href="#cb79-24" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb79-25"><a href="#cb79-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-26"><a href="#cb79-26" aria-hidden="true" tabindex="-1"></a>  dynamicColors <span class="ot">&lt;-</span> <span class="fu">labels2colors</span>(dynamicMods)</span>
<span id="cb79-27"><a href="#cb79-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-28"><a href="#cb79-28" aria-hidden="true" tabindex="-1"></a>  MElist <span class="ot">&lt;-</span> <span class="fu">moduleEigengenes</span>(datExpr_subset, <span class="at">colors =</span> dynamicColors)</span>
<span id="cb79-29"><a href="#cb79-29" aria-hidden="true" tabindex="-1"></a>  MEs    <span class="ot">&lt;-</span> <span class="fu">orderMEs</span>(MElist<span class="sc">$</span>eigengenes)</span>
<span id="cb79-30"><a href="#cb79-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-31"><a href="#cb79-31" aria-hidden="true" tabindex="-1"></a>  merge <span class="ot">&lt;-</span> <span class="fu">mergeCloseModules</span>(</span>
<span id="cb79-32"><a href="#cb79-32" aria-hidden="true" tabindex="-1"></a>    datExpr_subset,</span>
<span id="cb79-33"><a href="#cb79-33" aria-hidden="true" tabindex="-1"></a>    dynamicColors,</span>
<span id="cb79-34"><a href="#cb79-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">cutHeight =</span> mergeCutHeight,</span>
<span id="cb79-35"><a href="#cb79-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">verbose =</span> <span class="dv">0</span></span>
<span id="cb79-36"><a href="#cb79-36" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb79-37"><a href="#cb79-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-38"><a href="#cb79-38" aria-hidden="true" tabindex="-1"></a>  moduleColors <span class="ot">&lt;-</span> merge<span class="sc">$</span>colors</span>
<span id="cb79-39"><a href="#cb79-39" aria-hidden="true" tabindex="-1"></a>  MEs_merged   <span class="ot">&lt;-</span> <span class="fu">orderMEs</span>(merge<span class="sc">$</span>newMEs)</span>
<span id="cb79-40"><a href="#cb79-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-41"><a href="#cb79-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(</span>
<span id="cb79-42"><a href="#cb79-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">adjacency =</span> adjacency_mat,</span>
<span id="cb79-43"><a href="#cb79-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">TOM =</span> TOM,</span>
<span id="cb79-44"><a href="#cb79-44" aria-hidden="true" tabindex="-1"></a>    <span class="at">dissTOM =</span> dissTOM,</span>
<span id="cb79-45"><a href="#cb79-45" aria-hidden="true" tabindex="-1"></a>    <span class="at">geneTree =</span> geneTree,</span>
<span id="cb79-46"><a href="#cb79-46" aria-hidden="true" tabindex="-1"></a>    <span class="at">dynamicColors =</span> dynamicColors,</span>
<span id="cb79-47"><a href="#cb79-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">moduleColors =</span> moduleColors,</span>
<span id="cb79-48"><a href="#cb79-48" aria-hidden="true" tabindex="-1"></a>    <span class="at">MEs =</span> MEs_merged</span>
<span id="cb79-49"><a href="#cb79-49" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb79-50"><a href="#cb79-50" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Consensus WGCNA across tumour + normal</span></span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">colnames</span>(datExpr_tumor), <span class="fu">colnames</span>(datExpr_normal)))</span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a>multiExpr_cons <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">Tumor  =</span> <span class="fu">list</span>(<span class="at">data =</span> datExpr_tumor),</span>
<span id="cb80-6"><a href="#cb80-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">Normal =</span> <span class="fu">list</span>(<span class="at">data =</span> datExpr_normal)</span>
<span id="cb80-7"><a href="#cb80-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb80-8"><a href="#cb80-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-9"><a href="#cb80-9" aria-hidden="true" tabindex="-1"></a>softPower_cons <span class="ot">&lt;-</span> <span class="dv">7</span></span>
<span id="cb80-10"><a href="#cb80-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-11"><a href="#cb80-11" aria-hidden="true" tabindex="-1"></a>consensus_net <span class="ot">&lt;-</span> <span class="fu">blockwiseConsensusModules</span>(</span>
<span id="cb80-12"><a href="#cb80-12" aria-hidden="true" tabindex="-1"></a>  multiExpr_cons,</span>
<span id="cb80-13"><a href="#cb80-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">power =</span> softPower_cons,</span>
<span id="cb80-14"><a href="#cb80-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">networkType =</span> <span class="st">"unsigned"</span>,</span>
<span id="cb80-15"><a href="#cb80-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">TOMType =</span> <span class="st">"unsigned"</span>,</span>
<span id="cb80-16"><a href="#cb80-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">minModuleSize =</span> <span class="dv">30</span>,</span>
<span id="cb80-17"><a href="#cb80-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">deepSplit =</span> <span class="dv">2</span>,</span>
<span id="cb80-18"><a href="#cb80-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">pamRespectsDendro =</span> <span class="cn">FALSE</span>,</span>
<span id="cb80-19"><a href="#cb80-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">mergeCutHeight =</span> <span class="fl">0.25</span>,</span>
<span id="cb80-20"><a href="#cb80-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">numericLabels =</span> <span class="cn">FALSE</span>,</span>
<span id="cb80-21"><a href="#cb80-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">maxBlockSize =</span> <span class="fu">ncol</span>(datExpr),</span>
<span id="cb80-22"><a href="#cb80-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">saveTOMs =</span> <span class="cn">FALSE</span>,</span>
<span id="cb80-23"><a href="#cb80-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="dv">3</span></span>
<span id="cb80-24"><a href="#cb80-24" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Calculating consensus modules and module eigengenes block-wise from all genes
 Calculating topological overlaps block-wise from all genes
   Flagging genes and samples with too many missing values...
    ..step 1
 ....Working on set 1
    TOM calculation: adjacency..
    ..will not use multithreading.
     Fraction of slow calculations: 0.000000
    ..connectivity..
    ..matrix multiplication (system BLAS)..
    ..normalization..
    ..done.
 ....Working on set 2
    TOM calculation: adjacency..
    ..will not use multithreading.
     Fraction of slow calculations: 0.000000
    ..connectivity..
    ..matrix multiplication (system BLAS)..
    ..normalization..
    ..done.
 ..Working on block 1 .
 ....Working on set 1
 ....Working on set 2
 ....Calculating consensus network
 ..Working on block 1 .
 ....clustering and detecting modules..
 ....calculating eigengenes..
 ....checking consensus modules for statistical meaningfulness..
 ....checking for genes that should be reassigned..
 ..merging consensus modules that are too close..
     mergeCloseModules: Merging modules whose distance is less than 0.25
       Calculating new MEs...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>consensus_colors <span class="ot">&lt;-</span> <span class="fu">setNames</span>(consensus_net<span class="sc">$</span>colors, <span class="fu">colnames</span>(datExpr_tumor))</span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">names</span>(consensus_colors), <span class="fu">colnames</span>(datExpr_normal)))</span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(consensus_colors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>consensus_colors
     blue      grey turquoise 
       74       563       193 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>ME_cons_tumor <span class="ot">&lt;-</span> <span class="fu">moduleEigengenes</span>(datExpr_tumor, <span class="at">colors =</span> consensus_colors)<span class="sc">$</span>eigengenes</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>ME_cons_tumor <span class="ot">&lt;-</span> <span class="fu">orderMEs</span>(ME_cons_tumor)</span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>ME_cons_normal <span class="ot">&lt;-</span> <span class="fu">moduleEigengenes</span>(datExpr_normal, <span class="at">colors =</span> consensus_colors)<span class="sc">$</span>eigengenes</span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>ME_cons_normal <span class="ot">&lt;-</span> <span class="fu">orderMEs</span>(ME_cons_normal)</span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ME_cons_tumor)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                  MEblue MEturquoise      MEgrey
GSM1099660  0.0548917583 -0.03740008  0.31015112
GSM1099661  0.0591315498 -0.25232898  0.19402797
GSM1099662  0.1822054169  0.09786067 -0.22442552
GSM1099663 -0.0459951615 -0.04540917 -0.04480272
GSM1099664 -0.0001909041 -0.13986857  0.01260188
GSM1099665 -0.1082688489 -0.08390461  0.35609689</code></pre>
</div>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ME_cons_normal)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                MEblue  MEturquoise      MEgrey
GSM1099620  0.18616365  0.005367248 -0.34661164
GSM1099621  0.22122964 -0.135612241  0.16820012
GSM1099622  0.08221267 -0.008165824  0.02871443
GSM1099623 -0.22261968 -0.214292835  0.02834398
GSM1099624  0.02955216 -0.091863423  0.11488301
GSM1099625 -0.08291760  0.048178735  0.02202740</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Paired LMM on consensus module eigengenes</span></span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>tumour <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(pheno<span class="sc">$</span>condition <span class="sc">==</span> <span class="st">"tumor"</span>, <span class="dv">1</span>, <span class="dv">0</span>)</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>Age <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(pheno[,<span class="st">"age:ch1"</span>])</span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>Stage_num <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">factor</span>(pheno[,<span class="st">"Stage:ch1"</span>], <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"I"</span>,<span class="st">"II"</span>,<span class="st">"III"</span>,<span class="st">"IVA"</span>)))</span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a>sampleTraits <span class="ot">&lt;-</span> pheno[, <span class="fu">c</span>(<span class="st">"patient_id"</span>, <span class="st">"tumour"</span>, <span class="st">"Age"</span>, <span class="st">"Stage_num"</span>)]</span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sampleTraits) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(pheno)</span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-9"><a href="#cb88-9" aria-hidden="true" tabindex="-1"></a>ME_cons_all <span class="ot">&lt;-</span> <span class="fu">moduleEigengenes</span>(datExpr, <span class="at">colors =</span> consensus_colors)<span class="sc">$</span>eigengenes</span>
<span id="cb88-10"><a href="#cb88-10" aria-hidden="true" tabindex="-1"></a>ME_cons_all <span class="ot">&lt;-</span> <span class="fu">orderMEs</span>(ME_cons_all)</span>
<span id="cb88-11"><a href="#cb88-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-12"><a href="#cb88-12" aria-hidden="true" tabindex="-1"></a>datLMM_cons <span class="ot">&lt;-</span> <span class="fu">cbind</span>(sampleTraits[<span class="fu">rownames</span>(ME_cons_all), ],</span>
<span id="cb88-13"><a href="#cb88-13" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">as.data.frame</span>(ME_cons_all))</span>
<span id="cb88-14"><a href="#cb88-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-15"><a href="#cb88-15" aria-hidden="true" tabindex="-1"></a>modNames_cons <span class="ot">&lt;-</span> <span class="fu">colnames</span>(ME_cons_all)</span>
<span id="cb88-16"><a href="#cb88-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-17"><a href="#cb88-17" aria-hidden="true" tabindex="-1"></a>cons_lmm <span class="ot">&lt;-</span> <span class="fu">lapply</span>(modNames_cons, <span class="cf">function</span>(mn) {</span>
<span id="cb88-18"><a href="#cb88-18" aria-hidden="true" tabindex="-1"></a>  form <span class="ot">&lt;-</span> <span class="fu">as.formula</span>(<span class="fu">paste</span>(mn, <span class="st">"~ tumour + Age + Stage_num"</span>))</span>
<span id="cb88-19"><a href="#cb88-19" aria-hidden="true" tabindex="-1"></a>  fit  <span class="ot">&lt;-</span> <span class="fu">lme</span>(form,</span>
<span id="cb88-20"><a href="#cb88-20" aria-hidden="true" tabindex="-1"></a>              <span class="at">random =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> patient_id,</span>
<span id="cb88-21"><a href="#cb88-21" aria-hidden="true" tabindex="-1"></a>              <span class="at">data   =</span> datLMM_cons,</span>
<span id="cb88-22"><a href="#cb88-22" aria-hidden="true" tabindex="-1"></a>              <span class="at">na.action =</span> na.omit,</span>
<span id="cb88-23"><a href="#cb88-23" aria-hidden="true" tabindex="-1"></a>              <span class="at">method =</span> <span class="st">"REML"</span>)</span>
<span id="cb88-24"><a href="#cb88-24" aria-hidden="true" tabindex="-1"></a>  sm <span class="ot">&lt;-</span> <span class="fu">summary</span>(fit)</span>
<span id="cb88-25"><a href="#cb88-25" aria-hidden="true" tabindex="-1"></a>  tumour_row <span class="ot">&lt;-</span> sm<span class="sc">$</span>tTable[<span class="st">"tumour"</span>, ]</span>
<span id="cb88-26"><a href="#cb88-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb88-27"><a href="#cb88-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">module =</span> mn,</span>
<span id="cb88-28"><a href="#cb88-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">beta_tumour =</span> tumour_row[<span class="st">"Value"</span>],</span>
<span id="cb88-29"><a href="#cb88-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">t_tumour =</span> tumour_row[<span class="st">"t-value"</span>],</span>
<span id="cb88-30"><a href="#cb88-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">p_tumour =</span> tumour_row[<span class="st">"p-value"</span>],</span>
<span id="cb88-31"><a href="#cb88-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">NULL</span></span>
<span id="cb88-32"><a href="#cb88-32" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb88-33"><a href="#cb88-33" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb88-34"><a href="#cb88-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-35"><a href="#cb88-35" aria-hidden="true" tabindex="-1"></a>cons_lmm_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, cons_lmm)</span>
<span id="cb88-36"><a href="#cb88-36" aria-hidden="true" tabindex="-1"></a>cons_lmm_df[<span class="fu">order</span>(cons_lmm_df<span class="sc">$</span>p_tumour), ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       module beta_tumour  t_tumour     p_tumour
3      MEgrey 0.210578431 19.801951 2.770233e-21
1      MEblue 0.012396502  1.124946 2.678567e-01
2 MEturquoise 0.005973147  0.248842 8.048591e-01</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Preservation of consensus modules across conditions</span></span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>multiColor_cons <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">Tumor  =</span> consensus_colors,</span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">Normal =</span> consensus_colors</span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">12345</span>)</span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a>mp_cons <span class="ot">&lt;-</span> <span class="fu">modulePreservation</span>(</span>
<span id="cb90-10"><a href="#cb90-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">multiData =</span> multiExpr_cons,</span>
<span id="cb90-11"><a href="#cb90-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">multiColor =</span> multiColor_cons,</span>
<span id="cb90-12"><a href="#cb90-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">dataIsExpr =</span> <span class="cn">TRUE</span>,</span>
<span id="cb90-13"><a href="#cb90-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">networkType =</span> <span class="st">"unsigned"</span>,</span>
<span id="cb90-14"><a href="#cb90-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">referenceNetworks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>),</span>
<span id="cb90-15"><a href="#cb90-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">nPermutations =</span> <span class="dv">200</span>,</span>
<span id="cb90-16"><a href="#cb90-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">randomSeed =</span> <span class="dv">12345</span>,</span>
<span id="cb90-17"><a href="#cb90-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="dv">3</span>,</span>
<span id="cb90-18"><a href="#cb90-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">parallelCalculation =</span> <span class="cn">FALSE</span></span>
<span id="cb90-19"><a href="#cb90-19" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  ..checking data for excessive amounts of missing data..
     Flagging genes and samples with too many missing values...
      ..step 1
     Flagging genes and samples with too many missing values...
      ..step 1
  ..unassigned 'module' name: grey 
  ..all network sample 'module' name: gold
  ..calculating observed preservation values
  ..calculating permutation Z scores
 ..Working with set 1 as reference set
 ....working with set 2 as test set
  ......working on permutation 1
  ......working on permutation 2
  ......working on permutation 3
  ......working on permutation 4
  ......working on permutation 5
  ......working on permutation 6
  ......working on permutation 7
  ......working on permutation 8
  ......working on permutation 9
  ......working on permutation 10
  ......working on permutation 11
  ......working on permutation 12
  ......working on permutation 13
  ......working on permutation 14
  ......working on permutation 15
  ......working on permutation 16
  ......working on permutation 17
  ......working on permutation 18
  ......working on permutation 19
  ......working on permutation 20
  ......working on permutation 21
  ......working on permutation 22
  ......working on permutation 23
  ......working on permutation 24
  ......working on permutation 25
  ......working on permutation 26
  ......working on permutation 27
  ......working on permutation 28
  ......working on permutation 29
  ......working on permutation 30
  ......working on permutation 31
  ......working on permutation 32
  ......working on permutation 33
  ......working on permutation 34
  ......working on permutation 35
  ......working on permutation 36
  ......working on permutation 37
  ......working on permutation 38
  ......working on permutation 39
  ......working on permutation 40
  ......working on permutation 41
  ......working on permutation 42
  ......working on permutation 43
  ......working on permutation 44
  ......working on permutation 45
  ......working on permutation 46
  ......working on permutation 47
  ......working on permutation 48
  ......working on permutation 49
  ......working on permutation 50
  ......working on permutation 51
  ......working on permutation 52
  ......working on permutation 53
  ......working on permutation 54
  ......working on permutation 55
  ......working on permutation 56
  ......working on permutation 57
  ......working on permutation 58
  ......working on permutation 59
  ......working on permutation 60
  ......working on permutation 61
  ......working on permutation 62
  ......working on permutation 63
  ......working on permutation 64
  ......working on permutation 65
  ......working on permutation 66
  ......working on permutation 67
  ......working on permutation 68
  ......working on permutation 69
  ......working on permutation 70
  ......working on permutation 71
  ......working on permutation 72
  ......working on permutation 73
  ......working on permutation 74
  ......working on permutation 75
  ......working on permutation 76
  ......working on permutation 77
  ......working on permutation 78
  ......working on permutation 79
  ......working on permutation 80
  ......working on permutation 81
  ......working on permutation 82
  ......working on permutation 83
  ......working on permutation 84
  ......working on permutation 85
  ......working on permutation 86
  ......working on permutation 87
  ......working on permutation 88
  ......working on permutation 89
  ......working on permutation 90
  ......working on permutation 91
  ......working on permutation 92
  ......working on permutation 93
  ......working on permutation 94
  ......working on permutation 95
  ......working on permutation 96
  ......working on permutation 97
  ......working on permutation 98
  ......working on permutation 99
  ......working on permutation 100
  ......working on permutation 101
  ......working on permutation 102
  ......working on permutation 103
  ......working on permutation 104
  ......working on permutation 105
  ......working on permutation 106
  ......working on permutation 107
  ......working on permutation 108
  ......working on permutation 109
  ......working on permutation 110
  ......working on permutation 111
  ......working on permutation 112
  ......working on permutation 113
  ......working on permutation 114
  ......working on permutation 115
  ......working on permutation 116
  ......working on permutation 117
  ......working on permutation 118
  ......working on permutation 119
  ......working on permutation 120
  ......working on permutation 121
  ......working on permutation 122
  ......working on permutation 123
  ......working on permutation 124
  ......working on permutation 125
  ......working on permutation 126
  ......working on permutation 127
  ......working on permutation 128
  ......working on permutation 129
  ......working on permutation 130
  ......working on permutation 131
  ......working on permutation 132
  ......working on permutation 133
  ......working on permutation 134
  ......working on permutation 135
  ......working on permutation 136
  ......working on permutation 137
  ......working on permutation 138
  ......working on permutation 139
  ......working on permutation 140
  ......working on permutation 141
  ......working on permutation 142
  ......working on permutation 143
  ......working on permutation 144
  ......working on permutation 145
  ......working on permutation 146
  ......working on permutation 147
  ......working on permutation 148
  ......working on permutation 149
  ......working on permutation 150
  ......working on permutation 151
  ......working on permutation 152
  ......working on permutation 153
  ......working on permutation 154
  ......working on permutation 155
  ......working on permutation 156
  ......working on permutation 157
  ......working on permutation 158
  ......working on permutation 159
  ......working on permutation 160
  ......working on permutation 161
  ......working on permutation 162
  ......working on permutation 163
  ......working on permutation 164
  ......working on permutation 165
  ......working on permutation 166
  ......working on permutation 167
  ......working on permutation 168
  ......working on permutation 169
  ......working on permutation 170
  ......working on permutation 171
  ......working on permutation 172
  ......working on permutation 173
  ......working on permutation 174
  ......working on permutation 175
  ......working on permutation 176
  ......working on permutation 177
  ......working on permutation 178
  ......working on permutation 179
  ......working on permutation 180
  ......working on permutation 181
  ......working on permutation 182
  ......working on permutation 183
  ......working on permutation 184
  ......working on permutation 185
  ......working on permutation 186
  ......working on permutation 187
  ......working on permutation 188
  ......working on permutation 189
  ......working on permutation 190
  ......working on permutation 191
  ......working on permutation 192
  ......working on permutation 193
  ......working on permutation 194
  ......working on permutation 195
  ......working on permutation 196
  ......working on permutation 197
  ......working on permutation 198
  ......working on permutation 199
  ......working on permutation 200
 ..Working with set 2 as reference set
 ....working with set 1 as test set
  ......working on permutation 1
  ......working on permutation 2
  ......working on permutation 3
  ......working on permutation 4
  ......working on permutation 5
  ......working on permutation 6
  ......working on permutation 7
  ......working on permutation 8
  ......working on permutation 9
  ......working on permutation 10
  ......working on permutation 11
  ......working on permutation 12
  ......working on permutation 13
  ......working on permutation 14
  ......working on permutation 15
  ......working on permutation 16
  ......working on permutation 17
  ......working on permutation 18
  ......working on permutation 19
  ......working on permutation 20
  ......working on permutation 21
  ......working on permutation 22
  ......working on permutation 23
  ......working on permutation 24
  ......working on permutation 25
  ......working on permutation 26
  ......working on permutation 27
  ......working on permutation 28
  ......working on permutation 29
  ......working on permutation 30
  ......working on permutation 31
  ......working on permutation 32
  ......working on permutation 33
  ......working on permutation 34
  ......working on permutation 35
  ......working on permutation 36
  ......working on permutation 37
  ......working on permutation 38
  ......working on permutation 39
  ......working on permutation 40
  ......working on permutation 41
  ......working on permutation 42
  ......working on permutation 43
  ......working on permutation 44
  ......working on permutation 45
  ......working on permutation 46
  ......working on permutation 47
  ......working on permutation 48
  ......working on permutation 49
  ......working on permutation 50
  ......working on permutation 51
  ......working on permutation 52
  ......working on permutation 53
  ......working on permutation 54
  ......working on permutation 55
  ......working on permutation 56
  ......working on permutation 57
  ......working on permutation 58
  ......working on permutation 59
  ......working on permutation 60
  ......working on permutation 61
  ......working on permutation 62
  ......working on permutation 63
  ......working on permutation 64
  ......working on permutation 65
  ......working on permutation 66
  ......working on permutation 67
  ......working on permutation 68
  ......working on permutation 69
  ......working on permutation 70
  ......working on permutation 71
  ......working on permutation 72
  ......working on permutation 73
  ......working on permutation 74
  ......working on permutation 75
  ......working on permutation 76
  ......working on permutation 77
  ......working on permutation 78
  ......working on permutation 79
  ......working on permutation 80
  ......working on permutation 81
  ......working on permutation 82
  ......working on permutation 83
  ......working on permutation 84
  ......working on permutation 85
  ......working on permutation 86
  ......working on permutation 87
  ......working on permutation 88
  ......working on permutation 89
  ......working on permutation 90
  ......working on permutation 91
  ......working on permutation 92
  ......working on permutation 93
  ......working on permutation 94
  ......working on permutation 95
  ......working on permutation 96
  ......working on permutation 97
  ......working on permutation 98
  ......working on permutation 99
  ......working on permutation 100
  ......working on permutation 101
  ......working on permutation 102
  ......working on permutation 103
  ......working on permutation 104
  ......working on permutation 105
  ......working on permutation 106
  ......working on permutation 107
  ......working on permutation 108
  ......working on permutation 109
  ......working on permutation 110
  ......working on permutation 111
  ......working on permutation 112
  ......working on permutation 113
  ......working on permutation 114
  ......working on permutation 115
  ......working on permutation 116
  ......working on permutation 117
  ......working on permutation 118
  ......working on permutation 119
  ......working on permutation 120
  ......working on permutation 121
  ......working on permutation 122
  ......working on permutation 123
  ......working on permutation 124
  ......working on permutation 125
  ......working on permutation 126
  ......working on permutation 127
  ......working on permutation 128
  ......working on permutation 129
  ......working on permutation 130
  ......working on permutation 131
  ......working on permutation 132
  ......working on permutation 133
  ......working on permutation 134
  ......working on permutation 135
  ......working on permutation 136
  ......working on permutation 137
  ......working on permutation 138
  ......working on permutation 139
  ......working on permutation 140
  ......working on permutation 141
  ......working on permutation 142
  ......working on permutation 143
  ......working on permutation 144
  ......working on permutation 145
  ......working on permutation 146
  ......working on permutation 147
  ......working on permutation 148
  ......working on permutation 149
  ......working on permutation 150
  ......working on permutation 151
  ......working on permutation 152
  ......working on permutation 153
  ......working on permutation 154
  ......working on permutation 155
  ......working on permutation 156
  ......working on permutation 157
  ......working on permutation 158
  ......working on permutation 159
  ......working on permutation 160
  ......working on permutation 161
  ......working on permutation 162
  ......working on permutation 163
  ......working on permutation 164
  ......working on permutation 165
  ......working on permutation 166
  ......working on permutation 167
  ......working on permutation 168
  ......working on permutation 169
  ......working on permutation 170
  ......working on permutation 171
  ......working on permutation 172
  ......working on permutation 173
  ......working on permutation 174
  ......working on permutation 175
  ......working on permutation 176
  ......working on permutation 177
  ......working on permutation 178
  ......working on permutation 179
  ......working on permutation 180
  ......working on permutation 181
  ......working on permutation 182
  ......working on permutation 183
  ......working on permutation 184
  ......working on permutation 185
  ......working on permutation 186
  ......working on permutation 187
  ......working on permutation 188
  ......working on permutation 189
  ......working on permutation 190
  ......working on permutation 191
  ......working on permutation 192
  ......working on permutation 193
  ......working on permutation 194
  ......working on permutation 195
  ......working on permutation 196
  ......working on permutation 197
  ......working on permutation 198
  ......working on permutation 199
  ......working on permutation 200</code></pre>
</div>
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(mp_cons, <span class="at">file =</span> <span class="st">"modulePreservation_consensusColors_expr.RData"</span>)</span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a>Z_tum_in_norm <span class="ot">&lt;-</span> mp_cons<span class="sc">$</span>preservation<span class="sc">$</span>Z<span class="sc">$</span>ref.Tumor<span class="sc">$</span>inColumnsAlsoPresentIn.Normal</span>
<span id="cb92-4"><a href="#cb92-4" aria-hidden="true" tabindex="-1"></a>Z_norm_in_tum <span class="ot">&lt;-</span> mp_cons<span class="sc">$</span>preservation<span class="sc">$</span>Z<span class="sc">$</span>ref.Normal<span class="sc">$</span>inColumnsAlsoPresentIn.Tumor</span>
<span id="cb92-5"><a href="#cb92-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-6"><a href="#cb92-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Z_tum_in_norm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          moduleSize Zsummary.pres Zdensity.pres Zconnectivity.pres
blue              74     18.107774     24.115128          12.100420
gold             415     17.760303      9.945712          25.574895
grey             563      1.785345     -2.846835           6.417524
turquoise        193     30.070029     39.095233          21.044824
          Z.propVarExplained.pres Z.meanSignAwareKME.pres
blue                   14.6369896               19.416534
gold                    0.2689681               19.622455
grey                  -13.0872252                3.454089
turquoise              21.6594260               33.940021
          Z.meanSignAwareCorDat.pres Z.meanAdj.pres Z.meanMAR.pres Z.cor.kIM
blue                        81.25186     28.8137226      6.1001038  4.938508
gold                        81.66774     -0.2442315     -0.6668698 14.432990
grey                        37.34064     -9.1477597     -8.1323706  3.884539
turquoise                  217.30816     44.2504459     10.4130802 11.694737
          Z.cor.kME Z.cor.kMEall Z.cor.cor Z.cor.MAR
blue      12.100420     42.63956  44.33804  5.299918
gold      25.574895     45.59403 126.24594 13.690298
grey       6.417524     28.83112  70.33515  8.460863
turquoise 21.044824     47.72660 109.73817 11.966431</code></pre>
</div>
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Z_norm_in_tum)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          moduleSize Zsummary.pres Zdensity.pres Zconnectivity.pres
blue              74     18.811422     25.661768           11.96108
gold             415     14.059812      7.353892           20.76573
grey             563      1.605536     -2.663037            5.87411
turquoise        193     27.167058     33.489471           20.84464
          Z.propVarExplained.pres Z.meanSignAwareKME.pres
blue                    14.928482               16.856042
gold                    -1.889136               16.100028
grey                   -16.695916                2.863902
turquoise               19.872110               31.474808
          Z.meanSignAwareCorDat.pres Z.meanAdj.pres Z.meanMAR.pres Z.cor.kIM
blue                        68.29600      34.467494      7.2090366  5.784688
gold                        73.61362      -1.392243      0.8402624 14.997267
grey                        36.24295      -8.189976     -5.0431037  3.903081
turquoise                  193.48848      35.504135      8.1994651 12.459841
          Z.cor.kME Z.cor.kMEall Z.cor.cor Z.cor.MAR
blue       11.96108     39.27381  33.42047  6.097196
gold       20.76573     38.14396 100.43312 11.338653
grey        5.87411     25.42942  60.66192 10.211534
turquoise  20.84464     41.85636 106.77627 12.870295</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Separate networks (tumour and normal)</span></span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a>tumor_net  <span class="ot">&lt;-</span> <span class="fu">run_single_wgcna</span>(datExpr_tumor,  <span class="at">softPower =</span> softPower_tumor)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>..connectivity..
..matrix multiplication (system BLAS)..
..normalization..
..done.
 ..cutHeight not given, setting it to 0.993  ===&gt;  99% of the (truncated) height range in dendro.
 ..done.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>normal_net <span class="ot">&lt;-</span> <span class="fu">run_single_wgcna</span>(datExpr_normal, <span class="at">softPower =</span> softPower_normal)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>..connectivity..
..matrix multiplication (system BLAS)..
..normalization..
..done.
 ..cutHeight not given, setting it to 0.992  ===&gt;  99% of the (truncated) height range in dendro.
 ..done.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>tumor_colors  <span class="ot">&lt;-</span> <span class="fu">setNames</span>(tumor_net<span class="sc">$</span>moduleColors,  <span class="fu">colnames</span>(datExpr_tumor))</span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>normal_colors <span class="ot">&lt;-</span> <span class="fu">setNames</span>(normal_net<span class="sc">$</span>moduleColors, <span class="fu">colnames</span>(datExpr_normal))</span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">names</span>(tumor_colors),  <span class="fu">colnames</span>(datExpr_tumor)))</span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">names</span>(normal_colors), <span class="fu">colnames</span>(datExpr_normal)))</span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-7"><a href="#cb100-7" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(tumor_net<span class="sc">$</span>moduleColors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
     blue     brown      grey turquoise    yellow 
      215        58        48       451        58 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(normal_net<span class="sc">$</span>moduleColors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
     blue     brown     green      grey turquoise    yellow 
      166       138        46        25       408        47 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tumor_net<span class="sc">$</span>geneTree,  <span class="at">main =</span> <span class="st">"Tumor miRNA clustering"</span>,  <span class="at">xlab =</span> <span class="st">""</span>, <span class="at">sub =</span> <span class="st">""</span>, <span class="at">cex =</span> <span class="fl">0.4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_separate_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(normal_net<span class="sc">$</span>geneTree, <span class="at">main =</span> <span class="st">"Normal miRNA clustering"</span>, <span class="at">xlab =</span> <span class="st">""</span>, <span class="at">sub =</span> <span class="st">""</span>, <span class="at">cex =</span> <span class="fl">0.4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_separate_files/figure-html/unnamed-chunk-8-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Helper: extract preservation stats (Zsummary + medianRank)</span></span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a>extract_preservation <span class="ot">&lt;-</span> <span class="cf">function</span>(mp_obj,</span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">ref =</span> <span class="fu">c</span>(<span class="st">"Tumor"</span>, <span class="st">"Normal"</span>),</span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">test =</span> <span class="fu">c</span>(<span class="st">"Tumor"</span>, <span class="st">"Normal"</span>)) {</span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a>  ref  <span class="ot">&lt;-</span> <span class="fu">match.arg</span>(ref)</span>
<span id="cb106-6"><a href="#cb106-6" aria-hidden="true" tabindex="-1"></a>  test <span class="ot">&lt;-</span> <span class="fu">match.arg</span>(test)</span>
<span id="cb106-7"><a href="#cb106-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-8"><a href="#cb106-8" aria-hidden="true" tabindex="-1"></a>  ref_key  <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"ref."</span>, ref)</span>
<span id="cb106-9"><a href="#cb106-9" aria-hidden="true" tabindex="-1"></a>  test_key <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"inColumnsAlsoPresentIn."</span>, test)</span>
<span id="cb106-10"><a href="#cb106-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-11"><a href="#cb106-11" aria-hidden="true" tabindex="-1"></a>  zlist <span class="ot">&lt;-</span> mp_obj<span class="sc">$</span>preservation<span class="sc">$</span>Z</span>
<span id="cb106-12"><a href="#cb106-12" aria-hidden="true" tabindex="-1"></a>  statsZ <span class="ot">&lt;-</span> zlist[[ref_key]][[test_key]]</span>
<span id="cb106-13"><a href="#cb106-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-14"><a href="#cb106-14" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(statsZ)) {</span>
<span id="cb106-15"><a href="#cb106-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">"Could not find Z preservation stats for: "</span>, ref_key, <span class="st">" / "</span>, test_key)</span>
<span id="cb106-16"><a href="#cb106-16" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb106-17"><a href="#cb106-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-18"><a href="#cb106-18" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="st">"gold"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(statsZ)) {</span>
<span id="cb106-19"><a href="#cb106-19" aria-hidden="true" tabindex="-1"></a>    statsZ <span class="ot">&lt;-</span> statsZ[<span class="fu">rownames</span>(statsZ) <span class="sc">!=</span> <span class="st">"gold"</span>, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb106-20"><a href="#cb106-20" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb106-21"><a href="#cb106-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-22"><a href="#cb106-22" aria-hidden="true" tabindex="-1"></a>  zsum_col <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">c</span>(<span class="st">"Zsummary.pres"</span>, <span class="st">"Zsummary"</span>), <span class="fu">colnames</span>(statsZ))[<span class="dv">1</span>]</span>
<span id="cb106-23"><a href="#cb106-23" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.na</span>(zsum_col)) {</span>
<span id="cb106-24"><a href="#cb106-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">"No Zsummary column found. Available columns: "</span>,</span>
<span id="cb106-25"><a href="#cb106-25" aria-hidden="true" tabindex="-1"></a>         <span class="fu">paste</span>(<span class="fu">colnames</span>(statsZ), <span class="at">collapse =</span> <span class="st">", "</span>))</span>
<span id="cb106-26"><a href="#cb106-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb106-27"><a href="#cb106-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-28"><a href="#cb106-28" aria-hidden="true" tabindex="-1"></a>  Zsummary_pres <span class="ot">&lt;-</span> statsZ[, zsum_col]</span>
<span id="cb106-29"><a href="#cb106-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-30"><a href="#cb106-30" aria-hidden="true" tabindex="-1"></a>  medianRank_pres <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb106-31"><a href="#cb106-31" aria-hidden="true" tabindex="-1"></a>  obslist <span class="ot">&lt;-</span> mp_obj<span class="sc">$</span>preservation<span class="sc">$</span>observed</span>
<span id="cb106-32"><a href="#cb106-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-33"><a href="#cb106-33" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(obslist)) {</span>
<span id="cb106-34"><a href="#cb106-34" aria-hidden="true" tabindex="-1"></a>    statsO <span class="ot">&lt;-</span> obslist[[ref_key]][[test_key]]</span>
<span id="cb106-35"><a href="#cb106-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-36"><a href="#cb106-36" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(statsO) <span class="sc">&amp;&amp;</span> <span class="st">"gold"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(statsO)) {</span>
<span id="cb106-37"><a href="#cb106-37" aria-hidden="true" tabindex="-1"></a>      statsO <span class="ot">&lt;-</span> statsO[<span class="fu">rownames</span>(statsO) <span class="sc">!=</span> <span class="st">"gold"</span>, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb106-38"><a href="#cb106-38" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb106-39"><a href="#cb106-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-40"><a href="#cb106-40" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(statsO)) {</span>
<span id="cb106-41"><a href="#cb106-41" aria-hidden="true" tabindex="-1"></a>      mcol <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">c</span>(<span class="st">"medianRank.pres"</span>, <span class="st">"medianRank"</span>), <span class="fu">colnames</span>(statsO))[<span class="dv">1</span>]</span>
<span id="cb106-42"><a href="#cb106-42" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.na</span>(mcol)) {</span>
<span id="cb106-43"><a href="#cb106-43" aria-hidden="true" tabindex="-1"></a>        medianRank_pres <span class="ot">&lt;-</span> statsO[<span class="fu">rownames</span>(statsZ), mcol]</span>
<span id="cb106-44"><a href="#cb106-44" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb106-45"><a href="#cb106-45" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb106-46"><a href="#cb106-46" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb106-47"><a href="#cb106-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-48"><a href="#cb106-48" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(medianRank_pres)) {</span>
<span id="cb106-49"><a href="#cb106-49" aria-hidden="true" tabindex="-1"></a>    medianRank_pres <span class="ot">&lt;-</span> <span class="fu">rank</span>(<span class="sc">-</span>Zsummary_pres, <span class="at">ties.method =</span> <span class="st">"average"</span>)</span>
<span id="cb106-50"><a href="#cb106-50" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb106-51"><a href="#cb106-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-52"><a href="#cb106-52" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb106-53"><a href="#cb106-53" aria-hidden="true" tabindex="-1"></a>    <span class="at">module =</span> <span class="fu">rownames</span>(statsZ),</span>
<span id="cb106-54"><a href="#cb106-54" aria-hidden="true" tabindex="-1"></a>    <span class="at">Zsummary_pres =</span> <span class="fu">as.numeric</span>(Zsummary_pres),</span>
<span id="cb106-55"><a href="#cb106-55" aria-hidden="true" tabindex="-1"></a>    <span class="at">medianRank_pres =</span> <span class="fu">as.numeric</span>(medianRank_pres),</span>
<span id="cb106-56"><a href="#cb106-56" aria-hidden="true" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">NULL</span></span>
<span id="cb106-57"><a href="#cb106-57" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb106-58"><a href="#cb106-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-59"><a href="#cb106-59" aria-hidden="true" tabindex="-1"></a>  out[<span class="fu">order</span>(<span class="sc">-</span>out<span class="sc">$</span>Zsummary_pres), ]</span>
<span id="cb106-60"><a href="#cb106-60" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Preservation stats: separate networks</span></span>
<span id="cb107-2"><a href="#cb107-2" aria-hidden="true" tabindex="-1"></a>multiExpr <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb107-3"><a href="#cb107-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">Tumor  =</span> <span class="fu">list</span>(<span class="at">data =</span> datExpr_tumor),</span>
<span id="cb107-4"><a href="#cb107-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">Normal =</span> <span class="fu">list</span>(<span class="at">data =</span> datExpr_normal)</span>
<span id="cb107-5"><a href="#cb107-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb107-6"><a href="#cb107-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-7"><a href="#cb107-7" aria-hidden="true" tabindex="-1"></a>multiColor <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb107-8"><a href="#cb107-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">Tumor  =</span> tumor_colors,</span>
<span id="cb107-9"><a href="#cb107-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">Normal =</span> normal_colors</span>
<span id="cb107-10"><a href="#cb107-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb107-11"><a href="#cb107-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-12"><a href="#cb107-12" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">12345</span>)</span>
<span id="cb107-13"><a href="#cb107-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-14"><a href="#cb107-14" aria-hidden="true" tabindex="-1"></a>mp_expr <span class="ot">&lt;-</span> <span class="fu">modulePreservation</span>(</span>
<span id="cb107-15"><a href="#cb107-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">multiData =</span> multiExpr,</span>
<span id="cb107-16"><a href="#cb107-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">multiColor =</span> multiColor,</span>
<span id="cb107-17"><a href="#cb107-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">dataIsExpr =</span> <span class="cn">TRUE</span>,</span>
<span id="cb107-18"><a href="#cb107-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">networkType =</span> <span class="st">"unsigned"</span>,</span>
<span id="cb107-19"><a href="#cb107-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">referenceNetworks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>),</span>
<span id="cb107-20"><a href="#cb107-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">nPermutations =</span> <span class="dv">200</span>,</span>
<span id="cb107-21"><a href="#cb107-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">randomSeed =</span> <span class="dv">12345</span>,</span>
<span id="cb107-22"><a href="#cb107-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="dv">3</span>,</span>
<span id="cb107-23"><a href="#cb107-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">parallelCalculation =</span> <span class="cn">FALSE</span></span>
<span id="cb107-24"><a href="#cb107-24" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  ..checking data for excessive amounts of missing data..
     Flagging genes and samples with too many missing values...
      ..step 1
     Flagging genes and samples with too many missing values...
      ..step 1
  ..unassigned 'module' name: grey 
  ..all network sample 'module' name: gold
  ..calculating observed preservation values
  ..calculating permutation Z scores
 ..Working with set 1 as reference set
 ....working with set 2 as test set
  ......working on permutation 1
  ......working on permutation 2
  ......working on permutation 3
  ......working on permutation 4
  ......working on permutation 5
  ......working on permutation 6
  ......working on permutation 7
  ......working on permutation 8
  ......working on permutation 9
  ......working on permutation 10
  ......working on permutation 11
  ......working on permutation 12
  ......working on permutation 13
  ......working on permutation 14
  ......working on permutation 15
  ......working on permutation 16
  ......working on permutation 17
  ......working on permutation 18
  ......working on permutation 19
  ......working on permutation 20
  ......working on permutation 21
  ......working on permutation 22
  ......working on permutation 23
  ......working on permutation 24
  ......working on permutation 25
  ......working on permutation 26
  ......working on permutation 27
  ......working on permutation 28
  ......working on permutation 29
  ......working on permutation 30
  ......working on permutation 31
  ......working on permutation 32
  ......working on permutation 33
  ......working on permutation 34
  ......working on permutation 35
  ......working on permutation 36
  ......working on permutation 37
  ......working on permutation 38
  ......working on permutation 39
  ......working on permutation 40
  ......working on permutation 41
  ......working on permutation 42
  ......working on permutation 43
  ......working on permutation 44
  ......working on permutation 45
  ......working on permutation 46
  ......working on permutation 47
  ......working on permutation 48
  ......working on permutation 49
  ......working on permutation 50
  ......working on permutation 51
  ......working on permutation 52
  ......working on permutation 53
  ......working on permutation 54
  ......working on permutation 55
  ......working on permutation 56
  ......working on permutation 57
  ......working on permutation 58
  ......working on permutation 59
  ......working on permutation 60
  ......working on permutation 61
  ......working on permutation 62
  ......working on permutation 63
  ......working on permutation 64
  ......working on permutation 65
  ......working on permutation 66
  ......working on permutation 67
  ......working on permutation 68
  ......working on permutation 69
  ......working on permutation 70
  ......working on permutation 71
  ......working on permutation 72
  ......working on permutation 73
  ......working on permutation 74
  ......working on permutation 75
  ......working on permutation 76
  ......working on permutation 77
  ......working on permutation 78
  ......working on permutation 79
  ......working on permutation 80
  ......working on permutation 81
  ......working on permutation 82
  ......working on permutation 83
  ......working on permutation 84
  ......working on permutation 85
  ......working on permutation 86
  ......working on permutation 87
  ......working on permutation 88
  ......working on permutation 89
  ......working on permutation 90
  ......working on permutation 91
  ......working on permutation 92
  ......working on permutation 93
  ......working on permutation 94
  ......working on permutation 95
  ......working on permutation 96
  ......working on permutation 97
  ......working on permutation 98
  ......working on permutation 99
  ......working on permutation 100
  ......working on permutation 101
  ......working on permutation 102
  ......working on permutation 103
  ......working on permutation 104
  ......working on permutation 105
  ......working on permutation 106
  ......working on permutation 107
  ......working on permutation 108
  ......working on permutation 109
  ......working on permutation 110
  ......working on permutation 111
  ......working on permutation 112
  ......working on permutation 113
  ......working on permutation 114
  ......working on permutation 115
  ......working on permutation 116
  ......working on permutation 117
  ......working on permutation 118
  ......working on permutation 119
  ......working on permutation 120
  ......working on permutation 121
  ......working on permutation 122
  ......working on permutation 123
  ......working on permutation 124
  ......working on permutation 125
  ......working on permutation 126
  ......working on permutation 127
  ......working on permutation 128
  ......working on permutation 129
  ......working on permutation 130
  ......working on permutation 131
  ......working on permutation 132
  ......working on permutation 133
  ......working on permutation 134
  ......working on permutation 135
  ......working on permutation 136
  ......working on permutation 137
  ......working on permutation 138
  ......working on permutation 139
  ......working on permutation 140
  ......working on permutation 141
  ......working on permutation 142
  ......working on permutation 143
  ......working on permutation 144
  ......working on permutation 145
  ......working on permutation 146
  ......working on permutation 147
  ......working on permutation 148
  ......working on permutation 149
  ......working on permutation 150
  ......working on permutation 151
  ......working on permutation 152
  ......working on permutation 153
  ......working on permutation 154
  ......working on permutation 155
  ......working on permutation 156
  ......working on permutation 157
  ......working on permutation 158
  ......working on permutation 159
  ......working on permutation 160
  ......working on permutation 161
  ......working on permutation 162
  ......working on permutation 163
  ......working on permutation 164
  ......working on permutation 165
  ......working on permutation 166
  ......working on permutation 167
  ......working on permutation 168
  ......working on permutation 169
  ......working on permutation 170
  ......working on permutation 171
  ......working on permutation 172
  ......working on permutation 173
  ......working on permutation 174
  ......working on permutation 175
  ......working on permutation 176
  ......working on permutation 177
  ......working on permutation 178
  ......working on permutation 179
  ......working on permutation 180
  ......working on permutation 181
  ......working on permutation 182
  ......working on permutation 183
  ......working on permutation 184
  ......working on permutation 185
  ......working on permutation 186
  ......working on permutation 187
  ......working on permutation 188
  ......working on permutation 189
  ......working on permutation 190
  ......working on permutation 191
  ......working on permutation 192
  ......working on permutation 193
  ......working on permutation 194
  ......working on permutation 195
  ......working on permutation 196
  ......working on permutation 197
  ......working on permutation 198
  ......working on permutation 199
  ......working on permutation 200
 ..Working with set 2 as reference set
 ....working with set 1 as test set
  ......working on permutation 1
  ......working on permutation 2
  ......working on permutation 3
  ......working on permutation 4
  ......working on permutation 5
  ......working on permutation 6
  ......working on permutation 7
  ......working on permutation 8
  ......working on permutation 9
  ......working on permutation 10
  ......working on permutation 11
  ......working on permutation 12
  ......working on permutation 13
  ......working on permutation 14
  ......working on permutation 15
  ......working on permutation 16
  ......working on permutation 17
  ......working on permutation 18
  ......working on permutation 19
  ......working on permutation 20
  ......working on permutation 21
  ......working on permutation 22
  ......working on permutation 23
  ......working on permutation 24
  ......working on permutation 25
  ......working on permutation 26
  ......working on permutation 27
  ......working on permutation 28
  ......working on permutation 29
  ......working on permutation 30
  ......working on permutation 31
  ......working on permutation 32
  ......working on permutation 33
  ......working on permutation 34
  ......working on permutation 35
  ......working on permutation 36
  ......working on permutation 37
  ......working on permutation 38
  ......working on permutation 39
  ......working on permutation 40
  ......working on permutation 41
  ......working on permutation 42
  ......working on permutation 43
  ......working on permutation 44
  ......working on permutation 45
  ......working on permutation 46
  ......working on permutation 47
  ......working on permutation 48
  ......working on permutation 49
  ......working on permutation 50
  ......working on permutation 51
  ......working on permutation 52
  ......working on permutation 53
  ......working on permutation 54
  ......working on permutation 55
  ......working on permutation 56
  ......working on permutation 57
  ......working on permutation 58
  ......working on permutation 59
  ......working on permutation 60
  ......working on permutation 61
  ......working on permutation 62
  ......working on permutation 63
  ......working on permutation 64
  ......working on permutation 65
  ......working on permutation 66
  ......working on permutation 67
  ......working on permutation 68
  ......working on permutation 69
  ......working on permutation 70
  ......working on permutation 71
  ......working on permutation 72
  ......working on permutation 73
  ......working on permutation 74
  ......working on permutation 75
  ......working on permutation 76
  ......working on permutation 77
  ......working on permutation 78
  ......working on permutation 79
  ......working on permutation 80
  ......working on permutation 81
  ......working on permutation 82
  ......working on permutation 83
  ......working on permutation 84
  ......working on permutation 85
  ......working on permutation 86
  ......working on permutation 87
  ......working on permutation 88
  ......working on permutation 89
  ......working on permutation 90
  ......working on permutation 91
  ......working on permutation 92
  ......working on permutation 93
  ......working on permutation 94
  ......working on permutation 95
  ......working on permutation 96
  ......working on permutation 97
  ......working on permutation 98
  ......working on permutation 99
  ......working on permutation 100
  ......working on permutation 101
  ......working on permutation 102
  ......working on permutation 103
  ......working on permutation 104
  ......working on permutation 105
  ......working on permutation 106
  ......working on permutation 107
  ......working on permutation 108
  ......working on permutation 109
  ......working on permutation 110
  ......working on permutation 111
  ......working on permutation 112
  ......working on permutation 113
  ......working on permutation 114
  ......working on permutation 115
  ......working on permutation 116
  ......working on permutation 117
  ......working on permutation 118
  ......working on permutation 119
  ......working on permutation 120
  ......working on permutation 121
  ......working on permutation 122
  ......working on permutation 123
  ......working on permutation 124
  ......working on permutation 125
  ......working on permutation 126
  ......working on permutation 127
  ......working on permutation 128
  ......working on permutation 129
  ......working on permutation 130
  ......working on permutation 131
  ......working on permutation 132
  ......working on permutation 133
  ......working on permutation 134
  ......working on permutation 135
  ......working on permutation 136
  ......working on permutation 137
  ......working on permutation 138
  ......working on permutation 139
  ......working on permutation 140
  ......working on permutation 141
  ......working on permutation 142
  ......working on permutation 143
  ......working on permutation 144
  ......working on permutation 145
  ......working on permutation 146
  ......working on permutation 147
  ......working on permutation 148
  ......working on permutation 149
  ......working on permutation 150
  ......working on permutation 151
  ......working on permutation 152
  ......working on permutation 153
  ......working on permutation 154
  ......working on permutation 155
  ......working on permutation 156
  ......working on permutation 157
  ......working on permutation 158
  ......working on permutation 159
  ......working on permutation 160
  ......working on permutation 161
  ......working on permutation 162
  ......working on permutation 163
  ......working on permutation 164
  ......working on permutation 165
  ......working on permutation 166
  ......working on permutation 167
  ......working on permutation 168
  ......working on permutation 169
  ......working on permutation 170
  ......working on permutation 171
  ......working on permutation 172
  ......working on permutation 173
  ......working on permutation 174
  ......working on permutation 175
  ......working on permutation 176
  ......working on permutation 177
  ......working on permutation 178
  ......working on permutation 179
  ......working on permutation 180
  ......working on permutation 181
  ......working on permutation 182
  ......working on permutation 183
  ......working on permutation 184
  ......working on permutation 185
  ......working on permutation 186
  ......working on permutation 187
  ......working on permutation 188
  ......working on permutation 189
  ......working on permutation 190
  ......working on permutation 191
  ......working on permutation 192
  ......working on permutation 193
  ......working on permutation 194
  ......working on permutation 195
  ......working on permutation 196
  ......working on permutation 197
  ......working on permutation 198
  ......working on permutation 199
  ......working on permutation 200</code></pre>
</div>
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(mp_expr, <span class="at">file =</span> <span class="st">"modulePreservation_expr.RData"</span>)</span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-3"><a href="#cb109-3" aria-hidden="true" tabindex="-1"></a>tumor_in_normal <span class="ot">&lt;-</span> <span class="fu">extract_preservation</span>(mp_expr, <span class="at">ref =</span> <span class="st">"Tumor"</span>,  <span class="at">test =</span> <span class="st">"Normal"</span>)</span>
<span id="cb109-4"><a href="#cb109-4" aria-hidden="true" tabindex="-1"></a>normal_in_tumor <span class="ot">&lt;-</span> <span class="fu">extract_preservation</span>(mp_expr, <span class="at">ref =</span> <span class="st">"Normal"</span>, <span class="at">test =</span> <span class="st">"Tumor"</span>)</span>
<span id="cb109-5"><a href="#cb109-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-6"><a href="#cb109-6" aria-hidden="true" tabindex="-1"></a>tumor_in_normal</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     module Zsummary_pres medianRank_pres
4 turquoise     23.383008               2
1      blue     14.691577               1
2     brown      6.173669               3
5    yellow      3.175626               5
3      grey     -1.222979               6</code></pre>
</div>
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a>normal_in_tumor</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     module Zsummary_pres medianRank_pres
5 turquoise    22.7468241               2
1      blue    13.2755273               3
2     brown     6.8061073               6
6    yellow     6.5732793               1
3     green     4.3199755               4
4      grey    -0.6841512               7</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Dot plots: preservation Zsummary</span></span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tumor_in_normal<span class="sc">$</span>Zsummary_pres,</span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">xaxt =</span> <span class="st">"n"</span>, <span class="at">pch =</span> <span class="dv">19</span>,</span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"Tumor modules"</span>,</span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">"Preservation Zsummary (in Normal)"</span>,</span>
<span id="cb113-6"><a href="#cb113-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"Tumor \u2192 Normal module preservation"</span>)</span>
<span id="cb113-7"><a href="#cb113-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-8"><a href="#cb113-8" aria-hidden="true" tabindex="-1"></a><span class="fu">axis</span>(<span class="dv">1</span>, <span class="at">at =</span> <span class="fu">seq_along</span>(tumor_in_normal<span class="sc">$</span>module),</span>
<span id="cb113-9"><a href="#cb113-9" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels =</span> tumor_in_normal<span class="sc">$</span>module, <span class="at">las =</span> <span class="dv">2</span>, <span class="at">cex.axis =</span> <span class="fl">0.7</span>)</span>
<span id="cb113-10"><a href="#cb113-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-11"><a href="#cb113-11" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">2</span>,  <span class="at">col =</span> <span class="st">"red"</span>,  <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb113-12"><a href="#cb113-12" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">10</span>, <span class="at">col =</span> <span class="st">"blue"</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb113-13"><a href="#cb113-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-14"><a href="#cb113-14" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"topright"</span>,</span>
<span id="cb113-15"><a href="#cb113-15" aria-hidden="true" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">"Zsummary = 2 (weak / not preserved)"</span>, <span class="st">"Zsummary = 10 (strongly preserved)"</span>),</span>
<span id="cb113-16"><a href="#cb113-16" aria-hidden="true" tabindex="-1"></a>       <span class="at">col    =</span> <span class="fu">c</span>(<span class="st">"red"</span>, <span class="st">"blue"</span>),</span>
<span id="cb113-17"><a href="#cb113-17" aria-hidden="true" tabindex="-1"></a>       <span class="at">lty    =</span> <span class="dv">2</span>,</span>
<span id="cb113-18"><a href="#cb113-18" aria-hidden="true" tabindex="-1"></a>       <span class="at">lwd    =</span> <span class="dv">2</span>,</span>
<span id="cb113-19"><a href="#cb113-19" aria-hidden="true" tabindex="-1"></a>       <span class="at">bty    =</span> <span class="st">"n"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_separate_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(normal_in_tumor<span class="sc">$</span>Zsummary_pres,</span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a>     <span class="at">xaxt =</span> <span class="st">"n"</span>, <span class="at">pch =</span> <span class="dv">19</span>,</span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"Normal modules"</span>,</span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">"Preservation Zsummary (in Tumour)"</span>,</span>
<span id="cb114-5"><a href="#cb114-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"Normal \u2192 Tumour module preservation"</span>)</span>
<span id="cb114-6"><a href="#cb114-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-7"><a href="#cb114-7" aria-hidden="true" tabindex="-1"></a><span class="fu">axis</span>(<span class="dv">1</span>, <span class="at">at =</span> <span class="fu">seq_along</span>(normal_in_tumor<span class="sc">$</span>module),</span>
<span id="cb114-8"><a href="#cb114-8" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels =</span> normal_in_tumor<span class="sc">$</span>module, <span class="at">las =</span> <span class="dv">2</span>, <span class="at">cex.axis =</span> <span class="fl">0.7</span>)</span>
<span id="cb114-9"><a href="#cb114-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-10"><a href="#cb114-10" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">2</span>,  <span class="at">col =</span> <span class="st">"red"</span>,  <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb114-11"><a href="#cb114-11" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">10</span>, <span class="at">col =</span> <span class="st">"blue"</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb114-12"><a href="#cb114-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-13"><a href="#cb114-13" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"topright"</span>,</span>
<span id="cb114-14"><a href="#cb114-14" aria-hidden="true" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">"Zsummary = 2 (weak / not preserved)"</span>, <span class="st">"Zsummary = 10 (strongly preserved)"</span>),</span>
<span id="cb114-15"><a href="#cb114-15" aria-hidden="true" tabindex="-1"></a>       <span class="at">col    =</span> <span class="fu">c</span>(<span class="st">"red"</span>, <span class="st">"blue"</span>),</span>
<span id="cb114-16"><a href="#cb114-16" aria-hidden="true" tabindex="-1"></a>       <span class="at">lty    =</span> <span class="dv">2</span>,</span>
<span id="cb114-17"><a href="#cb114-17" aria-hidden="true" tabindex="-1"></a>       <span class="at">lwd    =</span> <span class="dv">2</span>,</span>
<span id="cb114-18"><a href="#cb114-18" aria-hidden="true" tabindex="-1"></a>       <span class="at">bty    =</span> <span class="st">"n"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_separate_files/figure-html/unnamed-chunk-11-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare separate networks to paired WGCNA export (Li)</span></span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb115-3"><a href="#cb115-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mclust)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Paket 'mclust' wurde unter R Version 4.5.2 erstellt</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Package 'mclust' version 6.1.2
Type 'citation("mclust")' for citing this R package in publications.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attache Paket: 'mclust'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Das folgende Objekt ist maskiert 'package:dplyr':

    count</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Das folgende Objekt ist maskiert 'package:limma':

    logsumexp</code></pre>
</div>
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(nlme)</span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb121-3"><a href="#cb121-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-4"><a href="#cb121-4" aria-hidden="true" tabindex="-1"></a>paired <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"paired_wgcna_export.rds"</span>)</span>
<span id="cb121-5"><a href="#cb121-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-6"><a href="#cb121-6" aria-hidden="true" tabindex="-1"></a>moduleColors_wgcna <span class="ot">&lt;-</span> paired<span class="sc">$</span>moduleColors</span>
<span id="cb121-7"><a href="#cb121-7" aria-hidden="true" tabindex="-1"></a>TOM_wgcna          <span class="ot">&lt;-</span> paired<span class="sc">$</span>TOM</span>
<span id="cb121-8"><a href="#cb121-8" aria-hidden="true" tabindex="-1"></a>lmm_results_wgcna  <span class="ot">&lt;-</span> paired<span class="sc">$</span>lmm_results</span>
<span id="cb121-9"><a href="#cb121-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-10"><a href="#cb121-10" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="sc">!</span><span class="fu">is.null</span>(<span class="fu">names</span>(moduleColors_wgcna)))</span>
<span id="cb121-11"><a href="#cb121-11" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="sc">!</span><span class="fu">is.null</span>(<span class="fu">dimnames</span>(TOM_wgcna)))</span>
<span id="cb121-12"><a href="#cb121-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-13"><a href="#cb121-13" aria-hidden="true" tabindex="-1"></a>moduleColors_tumor  <span class="ot">&lt;-</span> <span class="fu">setNames</span>(tumor_net<span class="sc">$</span>moduleColors,  <span class="fu">colnames</span>(datExpr_tumor))</span>
<span id="cb121-14"><a href="#cb121-14" aria-hidden="true" tabindex="-1"></a>moduleColors_normal <span class="ot">&lt;-</span> <span class="fu">setNames</span>(normal_net<span class="sc">$</span>moduleColors, <span class="fu">colnames</span>(datExpr_normal))</span>
<span id="cb121-15"><a href="#cb121-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-16"><a href="#cb121-16" aria-hidden="true" tabindex="-1"></a>TOM_tumor  <span class="ot">&lt;-</span> tumor_net<span class="sc">$</span>TOM</span>
<span id="cb121-17"><a href="#cb121-17" aria-hidden="true" tabindex="-1"></a>TOM_normal <span class="ot">&lt;-</span> normal_net<span class="sc">$</span>TOM</span>
<span id="cb121-18"><a href="#cb121-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-19"><a href="#cb121-19" aria-hidden="true" tabindex="-1"></a>miRNAs <span class="ot">&lt;-</span> <span class="fu">colnames</span>(datExpr)</span>
<span id="cb121-20"><a href="#cb121-20" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(TOM_wgcna)  <span class="ot">&lt;-</span> <span class="fu">list</span>(miRNAs, miRNAs)</span>
<span id="cb121-21"><a href="#cb121-21" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(TOM_tumor)  <span class="ot">&lt;-</span> <span class="fu">list</span>(miRNAs, miRNAs)</span>
<span id="cb121-22"><a href="#cb121-22" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(TOM_normal) <span class="ot">&lt;-</span> <span class="fu">list</span>(miRNAs, miRNAs)</span>
<span id="cb121-23"><a href="#cb121-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-24"><a href="#cb121-24" aria-hidden="true" tabindex="-1"></a>compare_module_overlap <span class="ot">&lt;-</span> <span class="cf">function</span>(colors_A, colors_B,</span>
<span id="cb121-25"><a href="#cb121-25" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">name_A =</span> <span class="st">"A"</span>, <span class="at">name_B =</span> <span class="st">"B"</span>,</span>
<span id="cb121-26"><a href="#cb121-26" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>, <span class="at">cluster_cols =</span> <span class="cn">TRUE</span>) {</span>
<span id="cb121-27"><a href="#cb121-27" aria-hidden="true" tabindex="-1"></a>  common <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">names</span>(colors_A), <span class="fu">names</span>(colors_B))</span>
<span id="cb121-28"><a href="#cb121-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>(<span class="fu">length</span>(common) <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb121-29"><a href="#cb121-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-30"><a href="#cb121-30" aria-hidden="true" tabindex="-1"></a>  ca <span class="ot">&lt;-</span> colors_A[common]</span>
<span id="cb121-31"><a href="#cb121-31" aria-hidden="true" tabindex="-1"></a>  cb <span class="ot">&lt;-</span> colors_B[common]</span>
<span id="cb121-32"><a href="#cb121-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-33"><a href="#cb121-33" aria-hidden="true" tabindex="-1"></a>  tab <span class="ot">&lt;-</span> <span class="fu">table</span>(<span class="at">A =</span> ca, <span class="at">B =</span> cb)</span>
<span id="cb121-34"><a href="#cb121-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-35"><a href="#cb121-35" aria-hidden="true" tabindex="-1"></a>  a_sizes <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(tab)</span>
<span id="cb121-36"><a href="#cb121-36" aria-hidden="true" tabindex="-1"></a>  b_sizes <span class="ot">&lt;-</span> <span class="fu">colSums</span>(tab)</span>
<span id="cb121-37"><a href="#cb121-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-38"><a href="#cb121-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(tab) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">rownames</span>(tab), <span class="st">" (n="</span>, a_sizes, <span class="st">")"</span>)</span>
<span id="cb121-39"><a href="#cb121-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(tab) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">colnames</span>(tab), <span class="st">" (n="</span>, b_sizes, <span class="st">")"</span>)</span>
<span id="cb121-40"><a href="#cb121-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-41"><a href="#cb121-41" aria-hidden="true" tabindex="-1"></a>  pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(</span>
<span id="cb121-42"><a href="#cb121-42" aria-hidden="true" tabindex="-1"></a>    tab,</span>
<span id="cb121-43"><a href="#cb121-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_rows =</span> cluster_rows,</span>
<span id="cb121-44"><a href="#cb121-44" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_cols =</span> cluster_cols,</span>
<span id="cb121-45"><a href="#cb121-45" aria-hidden="true" tabindex="-1"></a>    <span class="at">main =</span> <span class="fu">paste0</span>(<span class="st">"Module overlap: "</span>, name_A, <span class="st">" vs "</span>, name_B),</span>
<span id="cb121-46"><a href="#cb121-46" aria-hidden="true" tabindex="-1"></a>    <span class="at">display_numbers =</span> tab,</span>
<span id="cb121-47"><a href="#cb121-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">number_format  =</span> <span class="st">"%.0f"</span>,</span>
<span id="cb121-48"><a href="#cb121-48" aria-hidden="true" tabindex="-1"></a>    <span class="at">fontsize_number =</span> <span class="dv">8</span></span>
<span id="cb121-49"><a href="#cb121-49" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb121-50"><a href="#cb121-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-51"><a href="#cb121-51" aria-hidden="true" tabindex="-1"></a>  ari <span class="ot">&lt;-</span> mclust<span class="sc">::</span><span class="fu">adjustedRandIndex</span>(<span class="fu">as.integer</span>(<span class="fu">factor</span>(ca)),</span>
<span id="cb121-52"><a href="#cb121-52" aria-hidden="true" tabindex="-1"></a>                                   <span class="fu">as.integer</span>(<span class="fu">factor</span>(cb)))</span>
<span id="cb121-53"><a href="#cb121-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-54"><a href="#cb121-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">ARI("</span>, name_A, <span class="st">" vs "</span>, name_B, <span class="st">") ="</span>, ari, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb121-55"><a href="#cb121-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">invisible</span>(<span class="fu">list</span>(<span class="at">tab =</span> tab, <span class="at">ari =</span> ari, <span class="at">common =</span> common))</span>
<span id="cb121-56"><a href="#cb121-56" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb121-57"><a href="#cb121-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-58"><a href="#cb121-58" aria-hidden="true" tabindex="-1"></a>compare_edge_weights <span class="ot">&lt;-</span> <span class="cf">function</span>(TOM_A, TOM_B, miRNA_ids,</span>
<span id="cb121-59"><a href="#cb121-59" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">name_A =</span> <span class="st">"A"</span>, <span class="at">name_B =</span> <span class="st">"B"</span>) {</span>
<span id="cb121-60"><a href="#cb121-60" aria-hidden="true" tabindex="-1"></a>  common <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(TOM_A), <span class="fu">rownames</span>(TOM_B))</span>
<span id="cb121-61"><a href="#cb121-61" aria-hidden="true" tabindex="-1"></a>  common <span class="ot">&lt;-</span> <span class="fu">intersect</span>(common, miRNA_ids)</span>
<span id="cb121-62"><a href="#cb121-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>(<span class="fu">length</span>(common) <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb121-63"><a href="#cb121-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-64"><a href="#cb121-64" aria-hidden="true" tabindex="-1"></a>  A <span class="ot">&lt;-</span> TOM_A[common, common]</span>
<span id="cb121-65"><a href="#cb121-65" aria-hidden="true" tabindex="-1"></a>  B <span class="ot">&lt;-</span> TOM_B[common, common]</span>
<span id="cb121-66"><a href="#cb121-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-67"><a href="#cb121-67" aria-hidden="true" tabindex="-1"></a>  v1 <span class="ot">&lt;-</span> A[<span class="fu">upper.tri</span>(A)]</span>
<span id="cb121-68"><a href="#cb121-68" aria-hidden="true" tabindex="-1"></a>  v2 <span class="ot">&lt;-</span> B[<span class="fu">upper.tri</span>(B)]</span>
<span id="cb121-69"><a href="#cb121-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-70"><a href="#cb121-70" aria-hidden="true" tabindex="-1"></a>  test <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(v1, v2, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb121-71"><a href="#cb121-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-72"><a href="#cb121-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(test)</span>
<span id="cb121-73"><a href="#cb121-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(v1, v2, <span class="at">pch =</span> <span class="dv">20</span>,</span>
<span id="cb121-74"><a href="#cb121-74" aria-hidden="true" tabindex="-1"></a>       <span class="at">xlab =</span> <span class="fu">paste0</span>(name_A, <span class="st">" TOM edge weight"</span>),</span>
<span id="cb121-75"><a href="#cb121-75" aria-hidden="true" tabindex="-1"></a>       <span class="at">ylab =</span> <span class="fu">paste0</span>(name_B, <span class="st">" TOM edge weight"</span>),</span>
<span id="cb121-76"><a href="#cb121-76" aria-hidden="true" tabindex="-1"></a>       <span class="at">main =</span> <span class="fu">paste0</span>(<span class="st">"Edge-weight agreement (TOM): "</span>, name_A, <span class="st">" vs "</span>, name_B))</span>
<span id="cb121-77"><a href="#cb121-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-78"><a href="#cb121-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">Spearman rho ="</span>, <span class="fu">unname</span>(test<span class="sc">$</span>estimate),</span>
<span id="cb121-79"><a href="#cb121-79" aria-hidden="true" tabindex="-1"></a>      <span class="st">" | p ="</span>, test<span class="sc">$</span>p.value, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb121-80"><a href="#cb121-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-81"><a href="#cb121-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">invisible</span>(test)</span>
<span id="cb121-82"><a href="#cb121-82" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb121-83"><a href="#cb121-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-84"><a href="#cb121-84" aria-hidden="true" tabindex="-1"></a>compare_hubs <span class="ot">&lt;-</span> <span class="cf">function</span>(TOM_A, TOM_B, miRNA_ids,</span>
<span id="cb121-85"><a href="#cb121-85" aria-hidden="true" tabindex="-1"></a>                         <span class="at">name_A =</span> <span class="st">"A"</span>, <span class="at">name_B =</span> <span class="st">"B"</span>,</span>
<span id="cb121-86"><a href="#cb121-86" aria-hidden="true" tabindex="-1"></a>                         <span class="at">topK =</span> <span class="fu">c</span>(<span class="dv">25</span>, <span class="dv">50</span>, <span class="dv">75</span>, <span class="dv">100</span>)) {</span>
<span id="cb121-87"><a href="#cb121-87" aria-hidden="true" tabindex="-1"></a>  common <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(TOM_A), <span class="fu">rownames</span>(TOM_B))</span>
<span id="cb121-88"><a href="#cb121-88" aria-hidden="true" tabindex="-1"></a>  common <span class="ot">&lt;-</span> <span class="fu">intersect</span>(common, miRNA_ids)</span>
<span id="cb121-89"><a href="#cb121-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>(<span class="fu">length</span>(common) <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb121-90"><a href="#cb121-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-91"><a href="#cb121-91" aria-hidden="true" tabindex="-1"></a>  A <span class="ot">&lt;-</span> TOM_A[common, common]</span>
<span id="cb121-92"><a href="#cb121-92" aria-hidden="true" tabindex="-1"></a>  B <span class="ot">&lt;-</span> TOM_B[common, common]</span>
<span id="cb121-93"><a href="#cb121-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-94"><a href="#cb121-94" aria-hidden="true" tabindex="-1"></a>  kA <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(A) <span class="sc">-</span> <span class="fu">diag</span>(A)</span>
<span id="cb121-95"><a href="#cb121-95" aria-hidden="true" tabindex="-1"></a>  kB <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(B) <span class="sc">-</span> <span class="fu">diag</span>(B)</span>
<span id="cb121-96"><a href="#cb121-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-97"><a href="#cb121-97" aria-hidden="true" tabindex="-1"></a>  test <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(kA, kB, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb121-98"><a href="#cb121-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(test)</span>
<span id="cb121-99"><a href="#cb121-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-100"><a href="#cb121-100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">Top-hub overlap ("</span>, name_A, <span class="st">" vs "</span>, name_B, <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb121-101"><a href="#cb121-101" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (K <span class="cf">in</span> topK) {</span>
<span id="cb121-102"><a href="#cb121-102" aria-hidden="true" tabindex="-1"></a>    hubsA <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">sort</span>(kA, <span class="at">decreasing =</span> <span class="cn">TRUE</span>))[<span class="dv">1</span><span class="sc">:</span>K]</span>
<span id="cb121-103"><a href="#cb121-103" aria-hidden="true" tabindex="-1"></a>    hubsB <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">sort</span>(kB, <span class="at">decreasing =</span> <span class="cn">TRUE</span>))[<span class="dv">1</span><span class="sc">:</span>K]</span>
<span id="cb121-104"><a href="#cb121-104" aria-hidden="true" tabindex="-1"></a>    n_ol  <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">intersect</span>(hubsA, hubsB))</span>
<span id="cb121-105"><a href="#cb121-105" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"  K ="</span>, K, <span class="st">": overlap ="</span>, n_ol,</span>
<span id="cb121-106"><a href="#cb121-106" aria-hidden="true" tabindex="-1"></a>        <span class="st">" ("</span>, <span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> n_ol <span class="sc">/</span> K, <span class="dv">1</span>), <span class="st">"%)</span><span class="sc">\n</span><span class="st">"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb121-107"><a href="#cb121-107" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb121-108"><a href="#cb121-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-109"><a href="#cb121-109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">invisible</span>(<span class="fu">list</span>(<span class="at">test =</span> test, <span class="at">kA =</span> kA, <span class="at">kB =</span> kB))</span>
<span id="cb121-110"><a href="#cb121-110" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb121-111"><a href="#cb121-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-112"><a href="#cb121-112" aria-hidden="true" tabindex="-1"></a>fit_module_lmm <span class="ot">&lt;-</span> <span class="cf">function</span>(MEs_mat, sampleTraits_df) {</span>
<span id="cb121-113"><a href="#cb121-113" aria-hidden="true" tabindex="-1"></a>  dat <span class="ot">&lt;-</span> <span class="fu">cbind</span>(sampleTraits_df[<span class="fu">rownames</span>(MEs_mat), ], <span class="fu">as.data.frame</span>(MEs_mat))</span>
<span id="cb121-114"><a href="#cb121-114" aria-hidden="true" tabindex="-1"></a>  mods <span class="ot">&lt;-</span> <span class="fu">colnames</span>(MEs_mat)</span>
<span id="cb121-115"><a href="#cb121-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-116"><a href="#cb121-116" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">lapply</span>(mods, <span class="cf">function</span>(mn) {</span>
<span id="cb121-117"><a href="#cb121-117" aria-hidden="true" tabindex="-1"></a>    form <span class="ot">&lt;-</span> <span class="fu">as.formula</span>(<span class="fu">paste</span>(mn, <span class="st">"~ tumour + Age + Stage_num"</span>))</span>
<span id="cb121-118"><a href="#cb121-118" aria-hidden="true" tabindex="-1"></a>    fit  <span class="ot">&lt;-</span> <span class="fu">lme</span>(form, <span class="at">random =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> patient_id, <span class="at">data =</span> dat,</span>
<span id="cb121-119"><a href="#cb121-119" aria-hidden="true" tabindex="-1"></a>                <span class="at">na.action =</span> na.omit, <span class="at">method =</span> <span class="st">"REML"</span>)</span>
<span id="cb121-120"><a href="#cb121-120" aria-hidden="true" tabindex="-1"></a>    sm <span class="ot">&lt;-</span> <span class="fu">summary</span>(fit)</span>
<span id="cb121-121"><a href="#cb121-121" aria-hidden="true" tabindex="-1"></a>    tumour_row <span class="ot">&lt;-</span> sm<span class="sc">$</span>tTable[<span class="st">"tumour"</span>, ]</span>
<span id="cb121-122"><a href="#cb121-122" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(</span>
<span id="cb121-123"><a href="#cb121-123" aria-hidden="true" tabindex="-1"></a>      <span class="at">module =</span> mn,</span>
<span id="cb121-124"><a href="#cb121-124" aria-hidden="true" tabindex="-1"></a>      <span class="at">beta_tumour =</span> tumour_row[<span class="st">"Value"</span>],</span>
<span id="cb121-125"><a href="#cb121-125" aria-hidden="true" tabindex="-1"></a>      <span class="at">t_tumour =</span> tumour_row[<span class="st">"t-value"</span>],</span>
<span id="cb121-126"><a href="#cb121-126" aria-hidden="true" tabindex="-1"></a>      <span class="at">p_tumour =</span> tumour_row[<span class="st">"p-value"</span>],</span>
<span id="cb121-127"><a href="#cb121-127" aria-hidden="true" tabindex="-1"></a>      <span class="at">row.names =</span> <span class="cn">NULL</span></span>
<span id="cb121-128"><a href="#cb121-128" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb121-129"><a href="#cb121-129" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb121-130"><a href="#cb121-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-131"><a href="#cb121-131" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, res)</span>
<span id="cb121-132"><a href="#cb121-132" aria-hidden="true" tabindex="-1"></a>  out<span class="sc">$</span>FDR <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(out<span class="sc">$</span>p_tumour, <span class="at">method =</span> <span class="st">"BH"</span>)</span>
<span id="cb121-133"><a href="#cb121-133" aria-hidden="true" tabindex="-1"></a>  out[<span class="fu">order</span>(out<span class="sc">$</span>FDR), ]</span>
<span id="cb121-134"><a href="#cb121-134" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb121-135"><a href="#cb121-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-136"><a href="#cb121-136" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(<span class="fu">c</span>(<span class="st">"patient_id"</span>,<span class="st">"tumour"</span>,<span class="st">"Age"</span>,<span class="st">"Stage_num"</span>) <span class="sc">%in%</span> <span class="fu">colnames</span>(sampleTraits)))</span>
<span id="cb121-137"><a href="#cb121-137" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">rownames</span>(sampleTraits), <span class="fu">rownames</span>(datExpr)))</span>
<span id="cb121-138"><a href="#cb121-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-139"><a href="#cb121-139" aria-hidden="true" tabindex="-1"></a>ME_tumor_all  <span class="ot">&lt;-</span> <span class="fu">moduleEigengenes</span>(datExpr, <span class="at">colors =</span> moduleColors_tumor)<span class="sc">$</span>eigengenes</span>
<span id="cb121-140"><a href="#cb121-140" aria-hidden="true" tabindex="-1"></a>ME_tumor_all  <span class="ot">&lt;-</span> <span class="fu">orderMEs</span>(ME_tumor_all)</span>
<span id="cb121-141"><a href="#cb121-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-142"><a href="#cb121-142" aria-hidden="true" tabindex="-1"></a>ME_normal_all <span class="ot">&lt;-</span> <span class="fu">moduleEigengenes</span>(datExpr, <span class="at">colors =</span> moduleColors_normal)<span class="sc">$</span>eigengenes</span>
<span id="cb121-143"><a href="#cb121-143" aria-hidden="true" tabindex="-1"></a>ME_normal_all <span class="ot">&lt;-</span> <span class="fu">orderMEs</span>(ME_normal_all)</span>
<span id="cb121-144"><a href="#cb121-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-145"><a href="#cb121-145" aria-hidden="true" tabindex="-1"></a>lmm_tumor_modules  <span class="ot">&lt;-</span> <span class="fu">fit_module_lmm</span>(ME_tumor_all,  sampleTraits)</span>
<span id="cb121-146"><a href="#cb121-146" aria-hidden="true" tabindex="-1"></a>lmm_normal_modules <span class="ot">&lt;-</span> <span class="fu">fit_module_lmm</span>(ME_normal_all, sampleTraits)</span>
<span id="cb121-147"><a href="#cb121-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-148"><a href="#cb121-148" aria-hidden="true" tabindex="-1"></a>top_wgcna_me    <span class="ot">&lt;-</span> lmm_results_wgcna<span class="sc">$</span>module[<span class="fu">order</span>(lmm_results_wgcna<span class="sc">$</span>p_tumour)][<span class="dv">1</span>]</span>
<span id="cb121-149"><a href="#cb121-149" aria-hidden="true" tabindex="-1"></a>top_wgcna_color <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"^ME"</span>, <span class="st">""</span>, top_wgcna_me)</span>
<span id="cb121-150"><a href="#cb121-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-151"><a href="#cb121-151" aria-hidden="true" tabindex="-1"></a>top_tumor_me    <span class="ot">&lt;-</span> lmm_tumor_modules<span class="sc">$</span>module[<span class="dv">1</span>]</span>
<span id="cb121-152"><a href="#cb121-152" aria-hidden="true" tabindex="-1"></a>top_tumor_color <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"^ME"</span>, <span class="st">""</span>, top_tumor_me)</span>
<span id="cb121-153"><a href="#cb121-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-154"><a href="#cb121-154" aria-hidden="true" tabindex="-1"></a>top_normal_me    <span class="ot">&lt;-</span> lmm_normal_modules<span class="sc">$</span>module[<span class="dv">1</span>]</span>
<span id="cb121-155"><a href="#cb121-155" aria-hidden="true" tabindex="-1"></a>top_normal_color <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"^ME"</span>, <span class="st">""</span>, top_normal_me)</span>
<span id="cb121-156"><a href="#cb121-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-157"><a href="#cb121-157" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">Top modules (by FDR/p):</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Top modules (by FDR/p):</code></pre>
</div>
<div class="sourceCode cell-code" id="cb123"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"  Paired WGCNA:"</span>, top_wgcna_me, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  Paired WGCNA: MEgrey </code></pre>
</div>
<div class="sourceCode cell-code" id="cb125"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"  Tumor-net   :"</span>, top_tumor_me, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  Tumor-net   : MEgrey </code></pre>
</div>
<div class="sourceCode cell-code" id="cb127"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"  Normal-net  :"</span>, top_normal_me, <span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  Normal-net  : MEgreen </code></pre>
</div>
<div class="sourceCode cell-code" id="cb129"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a>jaccard_set <span class="ot">&lt;-</span> <span class="cf">function</span>(a, b) <span class="fu">length</span>(<span class="fu">intersect</span>(a,b)) <span class="sc">/</span> <span class="fu">length</span>(<span class="fu">union</span>(a,b))</span>
<span id="cb129-2"><a href="#cb129-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-3"><a href="#cb129-3" aria-hidden="true" tabindex="-1"></a>genes_w <span class="ot">&lt;-</span> <span class="fu">names</span>(moduleColors_wgcna)[moduleColors_wgcna <span class="sc">==</span> top_wgcna_color]</span>
<span id="cb129-4"><a href="#cb129-4" aria-hidden="true" tabindex="-1"></a>genes_t <span class="ot">&lt;-</span> <span class="fu">names</span>(moduleColors_tumor)[moduleColors_tumor <span class="sc">==</span> top_tumor_color]</span>
<span id="cb129-5"><a href="#cb129-5" aria-hidden="true" tabindex="-1"></a>genes_n <span class="ot">&lt;-</span> <span class="fu">names</span>(moduleColors_normal)[moduleColors_normal <span class="sc">==</span> top_normal_color]</span>
<span id="cb129-6"><a href="#cb129-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-7"><a href="#cb129-7" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Jaccard (Paired top vs Tumor top)  ="</span>, <span class="fu">jaccard_set</span>(genes_w, genes_t), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Jaccard (Paired top vs Tumor top)  = 0.1461538 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb131"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Jaccard (Paired top vs Normal top) ="</span>, <span class="fu">jaccard_set</span>(genes_w, genes_n), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Jaccard (Paired top vs Normal top) = 0.03496503 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a>ov_w_t <span class="ot">&lt;-</span> <span class="fu">compare_module_overlap</span>(moduleColors_wgcna, moduleColors_tumor,</span>
<span id="cb133-2"><a href="#cb133-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">name_A =</span> <span class="st">"Paired WGCNA"</span>, <span class="at">name_B =</span> <span class="st">"Tumor network"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_separate_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
ARI( Paired WGCNA  vs  Tumor network ) = 0.07993446 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb135"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" aria-hidden="true" tabindex="-1"></a>ov_w_n <span class="ot">&lt;-</span> <span class="fu">compare_module_overlap</span>(moduleColors_wgcna, moduleColors_normal,</span>
<span id="cb135-2"><a href="#cb135-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">name_A =</span> <span class="st">"Paired WGCNA"</span>, <span class="at">name_B =</span> <span class="st">"Normal network"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_separate_files/figure-html/unnamed-chunk-12-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
ARI( Paired WGCNA  vs  Normal network ) = 0.09487845 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb137"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb137-1"><a href="#cb137-1" aria-hidden="true" tabindex="-1"></a>ov_t_n <span class="ot">&lt;-</span> <span class="fu">compare_module_overlap</span>(moduleColors_tumor, moduleColors_normal,</span>
<span id="cb137-2"><a href="#cb137-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">name_A =</span> <span class="st">"Tumor network"</span>, <span class="at">name_B =</span> <span class="st">"Normal network"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_separate_files/figure-html/unnamed-chunk-12-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
ARI( Tumor network  vs  Normal network ) = 0.1123883 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb139"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb139-1"><a href="#cb139-1" aria-hidden="true" tabindex="-1"></a>ew_w_t <span class="ot">&lt;-</span> <span class="fu">compare_edge_weights</span>(TOM_wgcna, TOM_tumor, miRNAs,</span>
<span id="cb139-2"><a href="#cb139-2" aria-hidden="true" tabindex="-1"></a>                               <span class="at">name_A =</span> <span class="st">"Paired WGCNA"</span>, <span class="at">name_B =</span> <span class="st">"Tumor network"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Spearman's rank correlation rho

data:  v1 and v2
S = 2.8984e+15, p-value &lt; 2.2e-16
alternative hypothesis: true rho is not equal to 0
sample estimates:
      rho 
0.5729314 </code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_separate_files/figure-html/unnamed-chunk-12-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
Spearman rho = 0.5729314  | p = 0 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb142"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1" aria-hidden="true" tabindex="-1"></a>ew_w_n <span class="ot">&lt;-</span> <span class="fu">compare_edge_weights</span>(TOM_wgcna, TOM_normal, miRNAs,</span>
<span id="cb142-2"><a href="#cb142-2" aria-hidden="true" tabindex="-1"></a>                               <span class="at">name_A =</span> <span class="st">"Paired WGCNA"</span>, <span class="at">name_B =</span> <span class="st">"Normal network"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Spearman's rank correlation rho

data:  v1 and v2
S = 3.0269e+15, p-value &lt; 2.2e-16
alternative hypothesis: true rho is not equal to 0
sample estimates:
      rho 
0.5540002 </code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_separate_files/figure-html/unnamed-chunk-12-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
Spearman rho = 0.5540002  | p = 0 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb145"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb145-1"><a href="#cb145-1" aria-hidden="true" tabindex="-1"></a>ew_t_n <span class="ot">&lt;-</span> <span class="fu">compare_edge_weights</span>(TOM_tumor, TOM_normal, miRNAs,</span>
<span id="cb145-2"><a href="#cb145-2" aria-hidden="true" tabindex="-1"></a>                               <span class="at">name_A =</span> <span class="st">"Tumor network"</span>, <span class="at">name_B =</span> <span class="st">"Normal network"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Spearman's rank correlation rho

data:  v1 and v2
S = 4.2104e+15, p-value &lt; 2.2e-16
alternative hypothesis: true rho is not equal to 0
sample estimates:
      rho 
0.3796092 </code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_separate_files/figure-html/unnamed-chunk-12-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
Spearman rho = 0.3796092  | p = 0 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb148"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb148-1"><a href="#cb148-1" aria-hidden="true" tabindex="-1"></a>hb_w_t <span class="ot">&lt;-</span> <span class="fu">compare_hubs</span>(TOM_wgcna, TOM_tumor, miRNAs,</span>
<span id="cb148-2"><a href="#cb148-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">name_A =</span> <span class="st">"Paired WGCNA"</span>, <span class="at">name_B =</span> <span class="st">"Tumor network"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Spearman's rank correlation rho

data:  kA and kB
S = 28572670, p-value &lt; 2.2e-16
alternative hypothesis: true rho is not equal to 0
sample estimates:
      rho 
0.7001746 


Top-hub overlap (Paired WGCNA vs Tumor network)
  K =25: overlap =22 (88%)
  K =50: overlap =38 (76%)
  K =75: overlap =58 (77.3%)
  K =100: overlap =78 (78%)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb150"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb150-1"><a href="#cb150-1" aria-hidden="true" tabindex="-1"></a>hb_w_n <span class="ot">&lt;-</span> <span class="fu">compare_hubs</span>(TOM_wgcna, TOM_normal, miRNAs,</span>
<span id="cb150-2"><a href="#cb150-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">name_A =</span> <span class="st">"Paired WGCNA"</span>, <span class="at">name_B =</span> <span class="st">"Normal network"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Spearman's rank correlation rho

data:  kA and kB
S = 31518566, p-value &lt; 2.2e-16
alternative hypothesis: true rho is not equal to 0
sample estimates:
     rho 
0.669262 


Top-hub overlap (Paired WGCNA vs Normal network)
  K =25: overlap =19 (76%)
  K =50: overlap =41 (82%)
  K =75: overlap =60 (80%)
  K =100: overlap =74 (74%)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb152"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb152-1"><a href="#cb152-1" aria-hidden="true" tabindex="-1"></a>hb_t_n <span class="ot">&lt;-</span> <span class="fu">compare_hubs</span>(TOM_tumor, TOM_normal, miRNAs,</span>
<span id="cb152-2"><a href="#cb152-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">name_A =</span> <span class="st">"Tumor network"</span>, <span class="at">name_B =</span> <span class="st">"Normal network"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Spearman's rank correlation rho

data:  kA and kB
S = 49968868, p-value &lt; 2.2e-16
alternative hypothesis: true rho is not equal to 0
sample estimates:
     rho 
0.475655 


Top-hub overlap (Tumor network vs Normal network)
  K =25: overlap =19 (76%)
  K =50: overlap =35 (70%)
  K =75: overlap =53 (70.7%)
  K =100: overlap =67 (67%)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb154"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb154-1"><a href="#cb154-1" aria-hidden="true" tabindex="-1"></a>common_h <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(TOM_tumor), <span class="fu">rownames</span>(TOM_normal))</span>
<span id="cb154-2"><a href="#cb154-2" aria-hidden="true" tabindex="-1"></a>common_h <span class="ot">&lt;-</span> <span class="fu">intersect</span>(common_h, miRNAs)</span>
<span id="cb154-3"><a href="#cb154-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb154-4"><a href="#cb154-4" aria-hidden="true" tabindex="-1"></a>A <span class="ot">&lt;-</span> TOM_tumor[common_h, common_h]</span>
<span id="cb154-5"><a href="#cb154-5" aria-hidden="true" tabindex="-1"></a>B <span class="ot">&lt;-</span> TOM_normal[common_h, common_h]</span>
<span id="cb154-6"><a href="#cb154-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb154-7"><a href="#cb154-7" aria-hidden="true" tabindex="-1"></a>k_t <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(A) <span class="sc">-</span> <span class="fu">diag</span>(A)</span>
<span id="cb154-8"><a href="#cb154-8" aria-hidden="true" tabindex="-1"></a>k_n <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(B) <span class="sc">-</span> <span class="fu">diag</span>(B)</span>
<span id="cb154-9"><a href="#cb154-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb154-10"><a href="#cb154-10" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(k_t, k_n, <span class="at">pch =</span> <span class="dv">20</span>,</span>
<span id="cb154-11"><a href="#cb154-11" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"Tumor node strength (sum TOM)"</span>,</span>
<span id="cb154-12"><a href="#cb154-12" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">"Normal node strength (sum TOM)"</span>,</span>
<span id="cb154-13"><a href="#cb154-13" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"Hub-strength agreement: Tumor vs Normal"</span>)</span>
<span id="cb154-14"><a href="#cb154-14" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="fu">lm</span>(k_n <span class="sc">~</span> k_t), <span class="at">lty =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_separate_files/figure-html/unnamed-chunk-12-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>