<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>ReproducingWGCNA_exactly</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="Dataset1_Limma_DiffCoExp_Glasso_files/libs/clipboard/clipboard.min.js"></script>
<script src="Dataset1_Limma_DiffCoExp_Glasso_files/libs/quarto-html/quarto.js" type="module"></script>
<script src="Dataset1_Limma_DiffCoExp_Glasso_files/libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="Dataset1_Limma_DiffCoExp_Glasso_files/libs/quarto-html/popper.min.js"></script>
<script src="Dataset1_Limma_DiffCoExp_Glasso_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Dataset1_Limma_DiffCoExp_Glasso_files/libs/quarto-html/anchor.min.js"></script>
<link href="Dataset1_Limma_DiffCoExp_Glasso_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Dataset1_Limma_DiffCoExp_Glasso_files/libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Dataset1_Limma_DiffCoExp_Glasso_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Dataset1_Limma_DiffCoExp_Glasso_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Dataset1_Limma_DiffCoExp_Glasso_files/libs/bootstrap/bootstrap-81267100e462c21b3d6c0d5bf76a3417.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent quarto-light">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">ReproducingWGCNA_exactly</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>This file includes methods: (i) WGCNA on paired data (vi) Limma (vii) DiffCoExp (viii) Glasso</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Setup ----</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GEOquery)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(WGCNA)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(iCheck)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'gplots' was built under R version 4.5.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in rgl.init(initValue, onlyNULL): RGL: unable to open X11 display</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: 'rgl.init' failed, will use the null device.
See '?rgl.useNULL' for ways to avoid this warning.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(nlme)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="fu">allowWGCNAThreads</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Allowing multi-threading with up to 8 threads.</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load data ----</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>gse   <span class="ot">&lt;-</span> <span class="fu">getGEO</span>(<span class="st">"GSE45238"</span>, <span class="at">GSEMatrix =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found 1 file(s)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>GSE45238_series_matrix.txt.gz</code></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>eset  <span class="ot">&lt;-</span> gse[[<span class="dv">1</span>]]</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>expr  <span class="ot">&lt;-</span> <span class="fu">exprs</span>(eset)        <span class="co"># probes (858) × samples (80 = 2x40)</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> <span class="fu">pData</span>(eset)        <span class="co"># sample annotations -&gt; gender etc. (80 x 38)</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>annot <span class="ot">&lt;-</span> <span class="fu">fData</span>(eset)        <span class="co"># probe / miRNA annotation (858 x 24)</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Make sure sample annotation rows match expression columns</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(pheno) <span class="ot">&lt;-</span> pheno<span class="sc">$</span>geo_accession</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Probe filtering (Target Mature Version 12 &amp; remove controls) ----</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(annot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "ID"                    "SYMBOL"                "ILMN_Gene"            
 [4] "Search_Key"            "SEQUENCE"              "TargetMatureSeqs"     
 [7] "TargetMatureName"      "miRNA_ID"              "SPOT_ID"              
[10] "NumTargets"            "TargetMatureVersion"   "OriginalMatureSeq"    
[13] "OriginalMatureName"    "Source"                "Array_Address_Id"     
[16] "Illumicode"            "Oligo"                 "U3_Seq"               
[19] "Ploidy"                "Species"               "Probe_MatchOrder"     
[22] "Chromosome"            "Probe_Coordinates"     "Probe_Chr_Orientation"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(annot<span class="sc">$</span>miRNA_version)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>NULL</code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>is_mature_v12 <span class="ot">&lt;-</span> annot<span class="sc">$</span>TargetMatureVersion <span class="sc">==</span> <span class="dv">12</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="co"># TargetMatureVersion tells which version of that miRNA target library the probe’s annotation is based on -&gt; lose 28 miRNA</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>is_control <span class="ot">&lt;-</span> <span class="fu">grepl</span>(<span class="st">"control|spike|U6|snoR"</span>,</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>                    annot<span class="sc">$</span>SYMBOL, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="co"># To identify control miRNA to remove (actually none)</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>keep_probes <span class="ot">&lt;-</span> is_mature_v12 <span class="sc">&amp;</span> <span class="sc">!</span>is_control</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>expr  <span class="ot">&lt;-</span> expr[keep_probes, ]</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>annot <span class="ot">&lt;-</span> annot[keep_probes, ]</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(expr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 830  80</code></pre>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(annot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                       ID       SYMBOL    ILMN_Gene   Search_Key
ILMN_3166935 ILMN_3166935 ILMN_3166935  hsa-miR-329  hsa-miR-329
ILMN_3166938 ILMN_3166938 ILMN_3166938  hsa-miR-424  hsa-miR-424
ILMN_3166940 ILMN_3166940 ILMN_3166940  hsa-miR-137  hsa-miR-137
ILMN_3166941 ILMN_3166941 ILMN_3166941 hsa-miR-376c hsa-miR-376c
ILMN_3166943 ILMN_3166943 ILMN_3166943  hsa-miR-585  hsa-miR-585
ILMN_3166944 ILMN_3166944 ILMN_3166944 hsa-miR-302c hsa-miR-302c
                        SEQUENCE        TargetMatureSeqs TargetMatureName
ILMN_3166935 AACACACCTGGTTAACCTC  AACACACCTGGTTAACCTCTTT      hsa-miR-329
ILMN_3166938  CAGCAGCAATTCATGTTT  CAGCAGCAATTCATGTTTTGAA      hsa-miR-424
ILMN_3166940 TATTGCTTAAGAATACGCG TTATTGCTTAAGAATACGCGTAG      hsa-miR-137
ILMN_3166941 ACATAGAGGAAATTCCACG   AACATAGAGGAAATTCCACGT     hsa-miR-376c
ILMN_3166943   TGGGCGTATCTGTATGC     TGGGCGTATCTGTATGCTA      hsa-miR-585
ILMN_3166944  GCTTCCATGTTTCAGTGG TAAGTGCTTCCATGTTTCAGTGG     hsa-miR-302c
                 miRNA_ID SPOT_ID NumTargets TargetMatureVersion
ILMN_3166935  hsa-miR-329                  1                  12
ILMN_3166938  hsa-miR-424                  1                  12
ILMN_3166940  hsa-miR-137                  1                  12
ILMN_3166941 hsa-miR-376c                  1                  12
ILMN_3166943  hsa-miR-585                  1                  12
ILMN_3166944 hsa-miR-302c                  1                  12
                   OriginalMatureSeq OriginalMatureName
ILMN_3166935  AACACACCTGGTTAACCTCTTT        hsa-miR-329
ILMN_3166938  CAGCAGCAATTCATGTTTTGAA        hsa-miR-424
ILMN_3166940  TATTGCTTAAGAATACGCGTAG        hsa-miR-137
ILMN_3166941  ACATAGAGGAAATTCCACGTTT        hsa-miR-368
ILMN_3166943     TGGGCGTATCTGTATGCTA        hsa-miR-585
ILMN_3166944 TAAGTGCTTCCATGTTTCAGTGG       hsa-miR-302c
                                        Source Array_Address_Id
ILMN_3166935 Sanger miRNAdb (v_8.2, July 2006)              931
ILMN_3166938 Sanger miRNAdb (v_8.2, July 2006)             1051
ILMN_3166940 Sanger miRNAdb (v_8.2, July 2006)              233
ILMN_3166941 Sanger miRNAdb (v_8.2, July 2006)              976
ILMN_3166943 Sanger miRNAdb (v_8.2, July 2006)             1761
ILMN_3166944 Sanger miRNAdb (v_8.2, July 2006)              864
                         Illumicode
ILMN_3166935 CGGCAGATAGCACGTTTCGACT
ILMN_3166938 GGGCACTATTACTCGCTAACCG
ILMN_3166940 GTACAGTCTCGCGGTGCCTAAG
ILMN_3166941 GTTTCCACGACTTGCAGTACGG
ILMN_3166943 GTCGGTTAGTGATGTACGGCCA
ILMN_3166944 CCGTTTGGACGTAAGCAGGGCC
                                                                   Oligo
ILMN_3166935 ACTTCGTCAGTAACGGACCGGCAGATAGCACGTTTCGACTAACACACCTGGTTAACCTC
ILMN_3166938  ACTTCGTCAGTAACGGACGGGCACTATTACTCGCTAACCGCAGCAGCAATTCATGTTT
ILMN_3166940 ACTTCGTCAGTAACGGACGTACAGTCTCGCGGTGCCTAAGTATTGCTTAAGAATACGCG
ILMN_3166941 ACTTCGTCAGTAACGGACGTTTCCACGACTTGCAGTACGGACATAGAGGAAATTCCACG
ILMN_3166943   ACTTCGTCAGTAACGGACGTCGGTTAGTGATGTACGGCCATGGGCGTATCTGTATGC
ILMN_3166944  ACTTCGTCAGTAACGGACCCGTTTGGACGTAAGCAGGGCCGCTTCCATGTTTCAGTGG
                         U3_Seq  Ploidy Species Probe_MatchOrder Chromosome
ILMN_3166935 ACTTCGTCAGTAACGGAC diploid   human                2      14,14
ILMN_3166938 ACTTCGTCAGTAACGGAC diploid   human                1          X
ILMN_3166940 ACTTCGTCAGTAACGGAC diploid   human                1          1
ILMN_3166941 ACTTCGTCAGTAACGGAC diploid   human                1         14
ILMN_3166943 ACTTCGTCAGTAACGGAC diploid   human                1          5
ILMN_3166944 ACTTCGTCAGTAACGGAC diploid   human                1          4
                   Probe_Coordinates Probe_Chr_Orientation
ILMN_3166935 100,562,923,100,563,000                   +,+
ILMN_3166938               133508398                     -
ILMN_3166940                98284258                     -
ILMN_3166941               100575821                     +
ILMN_3166943               168623217                     -
ILMN_3166944               113788994                     -</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalization ----</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="co"># needed as </span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; summary(as.numeric(expr))</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="co">#    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. </span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   26.27    50.04   128.60  1653.56   661.65 32331.24 </span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>expr_log2 <span class="ot">&lt;-</span> <span class="fu">log2</span>(expr <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>expr_norm <span class="ot">&lt;-</span> <span class="fu">normalizeBetweenArrays</span>(expr_log2, <span class="at">method =</span> <span class="st">"quantile"</span>)</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(expr_norm) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(expr)</span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(expr_norm) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(expr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># QC: PCA &amp; sample clustering ----</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="co"># QC: PCA &amp; sample clustering ----</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Make room on the right side for the legend</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>op <span class="ot">&lt;-</span> <span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">10</span>), <span class="at">xpd =</span> <span class="cn">TRUE</span>)  <span class="co"># bigger right margin + allow drawing outside</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>expr_t <span class="ot">&lt;-</span> <span class="fu">t</span>(expr_norm)  <span class="co"># samples × miRNAs</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(expr_t, <span class="at">scale. =</span> <span class="cn">TRUE</span>)</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>sample_ids    <span class="ot">&lt;-</span> <span class="fu">colnames</span>(expr_norm)</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>highlight_ids <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"GSM1099641"</span>, <span class="st">"GSM1099655"</span>)</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>point_col     <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_ids <span class="sc">%in%</span> highlight_ids, <span class="st">"red"</span>, <span class="st">"black"</span>)</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(pca<span class="sc">$</span>x[,<span class="dv">1</span>], pca<span class="sc">$</span>x[,<span class="dv">2</span>],</span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"PC1"</span>, <span class="at">ylab =</span> <span class="st">"PC2"</span>,</span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>     <span class="at">pch =</span> <span class="dv">19</span>,</span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>     <span class="at">col =</span> point_col,</span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"PCA after normalization"</span>)</span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(pca<span class="sc">$</span>x[,<span class="dv">1</span>], pca<span class="sc">$</span>x[,<span class="dv">2</span>], <span class="co"># Plots the names of the sample_ids</span></span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels =</span> sample_ids,</span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>     <span class="at">pos =</span> <span class="dv">3</span>, <span class="at">cex =</span> <span class="fl">0.6</span>,</span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a>     <span class="at">col =</span> point_col)</span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Legend outside (to the right)</span></span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"topright"</span>,</span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a>       <span class="at">inset =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">0.55</span>, <span class="dv">0</span>),   <span class="co"># move it outside into the right margin</span></span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">"Other samples"</span>, <span class="st">"Outliers (to remove)"</span>),</span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a>       <span class="at">col    =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"red"</span>),</span>
<span id="cb20-29"><a href="#cb20-29" aria-hidden="true" tabindex="-1"></a>       <span class="at">pch    =</span> <span class="dv">19</span>,</span>
<span id="cb20-30"><a href="#cb20-30" aria-hidden="true" tabindex="-1"></a>       <span class="at">bty    =</span> <span class="st">"n"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_Limma_DiffCoExp_Glasso_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># (Optional) reset par</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(op)</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>sampleTree <span class="ot">&lt;-</span> <span class="fu">hclust</span>(<span class="fu">dist</span>(expr_t), <span class="at">method =</span> <span class="st">"average"</span>)</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(sampleTree, <span class="at">labels =</span> <span class="fu">colnames</span>(expr_norm), <span class="at">cex =</span> <span class="fl">0.6</span>,</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"Sample clustering"</span>)</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">35</span>, <span class="at">col =</span> <span class="st">"red"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_Limma_DiffCoExp_Glasso_files/figure-html/unnamed-chunk-5-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"SampleClustering.pdf"</span>, <span class="at">width =</span> <span class="dv">14</span>, <span class="at">height =</span> <span class="dv">10</span>)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(sampleTree,</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels =</span> <span class="fu">colnames</span>(expr_norm),</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">cex =</span> <span class="fl">0.6</span>,</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"Sample clustering dendrogram"</span>)</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">35</span>, <span class="at">col =</span> <span class="st">"red"</span>)</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create patient IDs &amp; remove outlier pairs ----</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract patient ID from title, e.g. "non-OSCC_01N" -&gt; "01"</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>patient_id <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">".*_(</span><span class="sc">\\</span><span class="st">d+)[NT]$"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, pheno<span class="sc">$</span>title)</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>patient_id <span class="ot">&lt;-</span> <span class="fu">as.character</span>(pheno<span class="sc">$</span>patient_id)</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Outlier control samples (from dendrogram)</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>outlier_samples <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"GSM1099641"</span>, <span class="st">"GSM1099655"</span>)</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Patients corresponding to those outliers</span></span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>outlier_patient_ids <span class="ot">&lt;-</span> <span class="fu">unique</span>(pheno<span class="sc">$</span>patient_id[</span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>  pheno<span class="sc">$</span>geo_accession <span class="sc">%in%</span> outlier_samples</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>])</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove ALL samples (normal + tumor) for those patients</span></span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>keep_samples <span class="ot">&lt;-</span> <span class="sc">!</span>pheno<span class="sc">$</span>patient_id <span class="sc">%in%</span> outlier_patient_ids</span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>expr_norm <span class="ot">&lt;-</span> expr_norm[, keep_samples]</span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>pheno     <span class="ot">&lt;-</span> pheno[keep_samples, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Preprocessing done, we now start with the new method!</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># -------------------------------</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Li et al. pipeline: WGCNA on all samples (no Δ) ----</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="co"># -------------------------------</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Define condition from title suffix (as done before)</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>condition <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"T$"</span>, pheno<span class="sc">$</span>title), <span class="st">"tumor"</span>, <span class="st">"normal"</span>)</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a><span class="co"># WGCNA expression data:</span></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a><span class="co"># rows = samples (arrays), columns = miRNAs</span></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>datExpr <span class="ot">&lt;-</span> <span class="fu">t</span>(expr_norm)   <span class="co"># 76 samples × 830 miRNAs</span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(datExpr)  <span class="co"># 76 x 830</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  76 830</code></pre>
</div>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Check for bad samples/genes</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>gsg <span class="ot">&lt;-</span> <span class="fu">goodSamplesGenes</span>(datExpr, <span class="at">verbose =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Flagging genes and samples with too many missing values...
  ..step 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>gsg<span class="sc">$</span>allOK  <span class="co"># TRUE</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Soft-threshold power selection (unsigned, as in Li et al.) ----</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>powers <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>sft <span class="ot">&lt;-</span> <span class="fu">pickSoftThreshold</span>(datExpr,</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">powerVector =</span> powers,</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">networkType =</span> <span class="st">"unsigned"</span>,</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">verbose =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>pickSoftThreshold: will use block size 830.
 pickSoftThreshold: calculating connectivity for given powers...
   ..working on genes 1 through 830 of 830
   Power SFT.R.sq  slope truncated.R.sq  mean.k. median.k. max.k.
1      1    0.132 -0.619          0.813 138.0000  1.34e+02 231.00
2      2    0.613 -1.070          0.839  38.7000  3.33e+01  97.60
3      3    0.850 -1.150          0.920  14.7000  1.03e+01  52.20
4      4    0.922 -1.280          0.967   6.7900  3.72e+00  34.30
5      5    0.960 -1.310          0.988   3.6400  1.51e+00  24.70
6      6    0.957 -1.330          0.968   2.1700  6.61e-01  19.10
7      7    0.970 -1.320          0.983   1.4000  3.06e-01  15.20
8      8    0.954 -1.300          0.948   0.9560  1.47e-01  12.40
9      9    0.959 -1.280          0.952   0.6850  7.33e-02  10.40
10    10    0.940 -1.290          0.927   0.5100  3.94e-02   8.91
11    11    0.960 -1.260          0.953   0.3910  2.21e-02   7.73
12    12    0.971 -1.250          0.966   0.3070  1.25e-02   6.77
13    13    0.976 -1.240          0.972   0.2460  7.14e-03   5.98
14    14    0.979 -1.230          0.980   0.2010  4.08e-03   5.32
15    15    0.977 -1.210          0.975   0.1660  2.40e-03   4.75
16    16    0.979 -1.190          0.976   0.1400  1.39e-03   4.27
17    17    0.956 -1.200          0.957   0.1180  8.06e-04   3.86
18    18    0.947 -1.200          0.937   0.1020  4.92e-04   3.50
19    19    0.964 -1.180          0.961   0.0878  2.94e-04   3.18
20    20    0.964 -1.170          0.962   0.0765  1.75e-04   2.91</code></pre>
</div>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>     <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">3</span>]) <span class="sc">*</span> sft<span class="sc">$</span>fitIndices[,<span class="dv">2</span>],</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"Soft Threshold (power)"</span>,</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">"Scale Free Topology Model Fit, signed R^2"</span>,</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">type =</span> <span class="st">"n"</span>,</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"Scale independence"</span>)</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>     <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">3</span>]) <span class="sc">*</span> sft<span class="sc">$</span>fitIndices[,<span class="dv">2</span>],</span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels =</span> powers, <span class="at">cex =</span> <span class="fl">0.8</span>, <span class="at">col =</span> <span class="st">"red"</span>)</span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="fl">0.97</span>, <span class="at">col =</span> <span class="st">"red"</span>)</span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>     sft<span class="sc">$</span>fitIndices[,<span class="dv">5</span>],</span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"Soft Threshold (power)"</span>,</span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">"Mean Connectivity"</span>,</span>
<span id="cb33-17"><a href="#cb33-17" aria-hidden="true" tabindex="-1"></a>     <span class="at">type =</span> <span class="st">"n"</span>,</span>
<span id="cb33-18"><a href="#cb33-18" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"Mean connectivity"</span>)</span>
<span id="cb33-19"><a href="#cb33-19" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb33-20"><a href="#cb33-20" aria-hidden="true" tabindex="-1"></a>     sft<span class="sc">$</span>fitIndices[,<span class="dv">5</span>],</span>
<span id="cb33-21"><a href="#cb33-21" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels =</span> powers, <span class="at">cex =</span> <span class="fl">0.8</span>, <span class="at">col =</span> <span class="st">"red"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_Limma_DiffCoExp_Glasso_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Li et al. chose β = 7 as first peak in R^2 (≈0.98)</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>softPower <span class="ot">&lt;-</span> <span class="dv">7</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>softPower</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 7</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Build adjacency, TOM, and cluster miRNAs (unsigned) ----</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>adjacency <span class="ot">&lt;-</span> <span class="fu">adjacency</span>(datExpr,</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">power =</span> softPower,</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">type  =</span> <span class="st">"unsigned"</span>)</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>TOM     <span class="ot">&lt;-</span> <span class="fu">TOMsimilarity</span>(adjacency, <span class="at">TOMType =</span> <span class="st">"unsigned"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>..connectivity..
..matrix multiplication (system BLAS)..
..normalization..
..done.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>dissTOM <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">-</span> TOM</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>geneTree <span class="ot">&lt;-</span> <span class="fu">hclust</span>(<span class="fu">as.dist</span>(dissTOM), <span class="at">method =</span> <span class="st">"average"</span>)</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(geneTree,</span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"Clustering of miRNAs based on TOM (unsigned, Li pipeline)"</span>,</span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">""</span>, <span class="at">sub =</span> <span class="st">""</span>, <span class="at">cex =</span> <span class="fl">0.4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_Limma_DiffCoExp_Glasso_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Dynamic tree cut: detect modules ----</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>minModuleSize <span class="ot">&lt;-</span> <span class="dv">30</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>dynamicMods <span class="ot">&lt;-</span> <span class="fu">cutreeDynamic</span>(</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">dendro =</span> geneTree,</span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">distM =</span> dissTOM,</span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">deepSplit =</span> <span class="dv">2</span>,</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">pamRespectsDendro =</span> <span class="cn">FALSE</span>,</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">minClusterSize =</span> minModuleSize</span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> ..cutHeight not given, setting it to 0.998  ===&gt;  99% of the (truncated) height range in dendro.
 ..done.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>dynamicColors <span class="ot">&lt;-</span> <span class="fu">labels2colors</span>(dynamicMods)</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(dynamicColors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>dynamicColors
     blue     brown      grey turquoise 
      201        85       250       294 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>  geneTree,</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>  dynamicColors,</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Dynamic Tree Cut"</span>,</span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">dendroLabels =</span> <span class="cn">FALSE</span>,</span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">hang =</span> <span class="fl">0.03</span>,</span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">addGuide =</span> <span class="cn">TRUE</span>,</span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">guideHang =</span> <span class="fl">0.05</span>,</span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">"miRNA dendrogram and module colors (Li pipeline)"</span></span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_Limma_DiffCoExp_Glasso_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Module eigengenes + merging ----</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>moduleColors0 <span class="ot">&lt;-</span> dynamicColors</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>MElist <span class="ot">&lt;-</span> <span class="fu">moduleEigengenes</span>(datExpr, <span class="at">colors =</span> moduleColors0)</span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>MEs    <span class="ot">&lt;-</span> <span class="fu">orderMEs</span>(MElist<span class="sc">$</span>eigengenes)</span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Eigengene clustering</span></span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a>MEDiss <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">-</span> <span class="fu">cor</span>(MEs)</span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a>METree <span class="ot">&lt;-</span> <span class="fu">hclust</span>(<span class="fu">as.dist</span>(MEDiss), <span class="at">method =</span> <span class="st">"average"</span>)</span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-12"><a href="#cb44-12" aria-hidden="true" tabindex="-1"></a>mergeCutHeight <span class="ot">&lt;-</span> <span class="fl">0.25</span>  <span class="co"># standard WGCNA choice</span></span>
<span id="cb44-13"><a href="#cb44-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-14"><a href="#cb44-14" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(METree,</span>
<span id="cb44-15"><a href="#cb44-15" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"Clustering of module eigengenes (before merging)"</span>,</span>
<span id="cb44-16"><a href="#cb44-16" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">""</span>, <span class="at">sub =</span> <span class="st">""</span>)</span>
<span id="cb44-17"><a href="#cb44-17" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> mergeCutHeight, <span class="at">col =</span> <span class="st">"red"</span>)</span>
<span id="cb44-18"><a href="#cb44-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Actually the merging will do nothing as the modules are far apart</span></span>
<span id="cb44-19"><a href="#cb44-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-20"><a href="#cb44-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge similar modules</span></span>
<span id="cb44-21"><a href="#cb44-21" aria-hidden="true" tabindex="-1"></a>merge <span class="ot">&lt;-</span> <span class="fu">mergeCloseModules</span>(</span>
<span id="cb44-22"><a href="#cb44-22" aria-hidden="true" tabindex="-1"></a>  datExpr,</span>
<span id="cb44-23"><a href="#cb44-23" aria-hidden="true" tabindex="-1"></a>  moduleColors0,</span>
<span id="cb44-24"><a href="#cb44-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">cutHeight =</span> mergeCutHeight,</span>
<span id="cb44-25"><a href="#cb44-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="dv">3</span></span>
<span id="cb44-26"><a href="#cb44-26" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> mergeCloseModules: Merging modules whose distance is less than 0.25
   multiSetMEs: Calculating module MEs.
     Working on set 1 ...
     moduleEigengenes: Calculating 4 module eigengenes in given set.
   Calculating new MEs...
   multiSetMEs: Calculating module MEs.
     Working on set 1 ...
     moduleEigengenes: Calculating 4 module eigengenes in given set.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>moduleColors <span class="ot">&lt;-</span> merge<span class="sc">$</span>colors</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>MEs          <span class="ot">&lt;-</span> <span class="fu">orderMEs</span>(merge<span class="sc">$</span>newMEs)</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>  geneTree,</span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cbind</span>(moduleColors0, moduleColors),</span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">"Dynamic Tree Cut"</span>, <span class="st">"Merged modules"</span>),</span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">dendroLabels =</span> <span class="cn">FALSE</span>,</span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">hang =</span> <span class="fl">0.03</span>,</span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">addGuide =</span> <span class="cn">TRUE</span>,</span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">guideHang =</span> <span class="fl">0.05</span>,</span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">"Modules before and after merging"</span></span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_Limma_DiffCoExp_Glasso_files/figure-html/unnamed-chunk-10-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_Limma_DiffCoExp_Glasso_files/figure-html/unnamed-chunk-10-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>moduleColors_wgcna <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">miRNA        =</span> <span class="fu">colnames</span>(datExpr),</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">moduleColor  =</span> moduleColors</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sample-level traits for LMM (Li et al.) ----</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="co"># We already have pheno$patient_id and pheno$condition</span></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Make sure rownames(pheno) are sample IDs and match datExpr rows</span></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">rownames</span>(pheno), <span class="fu">rownames</span>(datExpr)))</span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Numeric tumour indicator: 0 = normal, 1 = tumour</span></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>tumour <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(pheno<span class="sc">$</span>condition <span class="sc">==</span> <span class="st">"tumor"</span>, <span class="dv">1</span>, <span class="dv">0</span>)</span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Age and stage (same for both samples of a patient)</span></span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>Age <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(pheno[,<span class="st">"age:ch1"</span>])</span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>Stage_num <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">factor</span>(pheno[,<span class="st">"Stage:ch1"</span>],</span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"I"</span>,<span class="st">"II"</span>,<span class="st">"III"</span>,<span class="st">"IVA"</span>)))</span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only the columns we need and align with datExpr</span></span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a>sampleTraits <span class="ot">&lt;-</span> pheno[, <span class="fu">c</span>(<span class="st">"patient_id"</span>, <span class="st">"tumour"</span>, <span class="st">"Age"</span>, <span class="st">"Stage_num"</span>)]</span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sampleTraits) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(pheno)</span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-20"><a href="#cb48-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Check alignment</span></span>
<span id="cb48-21"><a href="#cb48-21" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">rownames</span>(sampleTraits), <span class="fu">rownames</span>(datExpr))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Data for LMM: traits + eigengenes</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>datLMM <span class="ot">&lt;-</span> <span class="fu">cbind</span>(sampleTraits[<span class="fu">rownames</span>(MEs), ],</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>                <span class="fu">as.data.frame</span>(MEs))</span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(datLMM)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           patient_id tumour Age Stage_num MEturquoise       MEblue     MEbrown
GSM1099620         01      0  46         2  0.07980279  0.016102121  0.13424746
GSM1099621         02      0  49         2  0.10611460 -0.093689168  0.15966022
GSM1099622         03      0  56         3  0.14360387  0.004774269  0.05513379
GSM1099623         04      0  50         3  0.10902236 -0.151306809 -0.14964605
GSM1099624         05      0  36         2  0.13971935 -0.048803596  0.02222138
GSM1099625         06      0  48         4  0.10628017  0.049414716 -0.05235323
                MEgrey
GSM1099620  0.26244882
GSM1099621 -0.12313062
GSM1099622 -0.10426881
GSM1099623 -0.03464253
GSM1099624 -0.07848453
GSM1099625 -0.11756807</code></pre>
</div>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>modNames <span class="ot">&lt;-</span> <span class="fu">colnames</span>(MEs)</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>lmm_results <span class="ot">&lt;-</span> <span class="fu">lapply</span>(modNames, <span class="cf">function</span>(mn) {</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>  form <span class="ot">&lt;-</span> <span class="fu">as.formula</span>(<span class="fu">paste</span>(mn, <span class="st">"~ tumour + Age + Stage_num"</span>))</span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>  fit  <span class="ot">&lt;-</span> <span class="fu">lme</span>(form,</span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">random =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> patient_id,</span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">data   =</span> datLMM,</span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">na.action =</span> na.omit,</span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a>              <span class="at">method =</span> <span class="st">"REML"</span>)</span>
<span id="cb52-11"><a href="#cb52-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb52-12"><a href="#cb52-12" aria-hidden="true" tabindex="-1"></a>  sm <span class="ot">&lt;-</span> <span class="fu">summary</span>(fit)</span>
<span id="cb52-13"><a href="#cb52-13" aria-hidden="true" tabindex="-1"></a>  tumour_row <span class="ot">&lt;-</span> sm<span class="sc">$</span>tTable[<span class="st">"tumour"</span>, ]  <span class="co"># fixed effect for tumour status</span></span>
<span id="cb52-14"><a href="#cb52-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb52-15"><a href="#cb52-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb52-16"><a href="#cb52-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">module     =</span> mn,</span>
<span id="cb52-17"><a href="#cb52-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">beta_tumour =</span> tumour_row[<span class="st">"Value"</span>],</span>
<span id="cb52-18"><a href="#cb52-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">t_tumour   =</span> tumour_row[<span class="st">"t-value"</span>],</span>
<span id="cb52-19"><a href="#cb52-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">p_tumour   =</span> tumour_row[<span class="st">"p-value"</span>],</span>
<span id="cb52-20"><a href="#cb52-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">row.names  =</span> <span class="cn">NULL</span></span>
<span id="cb52-21"><a href="#cb52-21" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb52-22"><a href="#cb52-22" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb52-23"><a href="#cb52-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-24"><a href="#cb52-24" aria-hidden="true" tabindex="-1"></a>lmm_results_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, lmm_results)</span>
<span id="cb52-25"><a href="#cb52-25" aria-hidden="true" tabindex="-1"></a>lmm_results_df[<span class="fu">order</span>(lmm_results_df<span class="sc">$</span>p_tumour), ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       module  beta_tumour    t_tumour     p_tumour
1 MEturquoise -0.209887568 -19.2768639 6.873471e-21
4      MEgrey  0.107948034   6.3234950 2.293120e-07
2      MEblue -0.009391942  -0.3849917 7.024478e-01
3     MEbrown -0.001226233  -0.1078641 9.146863e-01</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Simple heatmap of t-values for tumour effect ----</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>lmm_results_df<span class="sc">$</span>module_clean <span class="ot">&lt;-</span> <span class="fu">substring</span>(lmm_results_df<span class="sc">$</span>module, <span class="dv">3</span>)</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>tmat <span class="ot">&lt;-</span> <span class="fu">matrix</span>(lmm_results_df<span class="sc">$</span>t_tumour,</span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">nrow =</span> <span class="fu">nrow</span>(lmm_results_df),</span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">dimnames =</span> <span class="fu">list</span>(lmm_results_df<span class="sc">$</span>module_clean, <span class="st">"Tumour (t)"</span>))</span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a>pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(tmat,</span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a>                   <span class="at">main =</span> <span class="st">"Module–tumour associations (LMM t-values)"</span>,</span>
<span id="cb54-9"><a href="#cb54-9" aria-hidden="true" tabindex="-1"></a>                   <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb54-10"><a href="#cb54-10" aria-hidden="true" tabindex="-1"></a>                   <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb54-11"><a href="#cb54-11" aria-hidden="true" tabindex="-1"></a>                   <span class="at">display_numbers =</span> <span class="fu">round</span>(tmat, <span class="dv">2</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_Limma_DiffCoExp_Glasso_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>lmm_results_df</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       module  beta_tumour    t_tumour     p_tumour module_clean
1 MEturquoise -0.209887568 -19.2768639 6.873471e-21    turquoise
2      MEblue -0.009391942  -0.3849917 7.024478e-01         blue
3     MEbrown -0.001226233  -0.1078641 9.146863e-01        brown
4      MEgrey  0.107948034   6.3234950 2.293120e-07         grey</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Gene significance (GS) based on LMM per miRNA ----</span></span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>expr_forGS <span class="ot">&lt;-</span> datExpr  <span class="co"># samples × miRNAs</span></span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a>datGS <span class="ot">&lt;-</span> <span class="fu">cbind</span>(sampleTraits[<span class="fu">rownames</span>(expr_forGS), ],</span>
<span id="cb57-5"><a href="#cb57-5" aria-hidden="true" tabindex="-1"></a>               <span class="fu">as.data.frame</span>(expr_forGS))</span>
<span id="cb57-6"><a href="#cb57-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-7"><a href="#cb57-7" aria-hidden="true" tabindex="-1"></a>miRNAs <span class="ot">&lt;-</span> <span class="fu">colnames</span>(expr_forGS)</span>
<span id="cb57-8"><a href="#cb57-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-9"><a href="#cb57-9" aria-hidden="true" tabindex="-1"></a>gs_list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(miRNAs, <span class="cf">function</span>(g) {</span>
<span id="cb57-10"><a href="#cb57-10" aria-hidden="true" tabindex="-1"></a>  form <span class="ot">&lt;-</span> <span class="fu">as.formula</span>(<span class="fu">paste</span>(g, <span class="st">"~ tumour + Age + Stage_num"</span>))</span>
<span id="cb57-11"><a href="#cb57-11" aria-hidden="true" tabindex="-1"></a>  fit  <span class="ot">&lt;-</span> <span class="fu">lme</span>(form,</span>
<span id="cb57-12"><a href="#cb57-12" aria-hidden="true" tabindex="-1"></a>              <span class="at">random =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> patient_id,</span>
<span id="cb57-13"><a href="#cb57-13" aria-hidden="true" tabindex="-1"></a>              <span class="at">data   =</span> datGS,</span>
<span id="cb57-14"><a href="#cb57-14" aria-hidden="true" tabindex="-1"></a>              <span class="at">na.action =</span> na.omit,</span>
<span id="cb57-15"><a href="#cb57-15" aria-hidden="true" tabindex="-1"></a>              <span class="at">method =</span> <span class="st">"REML"</span>)</span>
<span id="cb57-16"><a href="#cb57-16" aria-hidden="true" tabindex="-1"></a>  sm <span class="ot">&lt;-</span> <span class="fu">summary</span>(fit)</span>
<span id="cb57-17"><a href="#cb57-17" aria-hidden="true" tabindex="-1"></a>  t_tumour <span class="ot">&lt;-</span> sm<span class="sc">$</span>tTable[<span class="st">"tumour"</span>,<span class="st">"t-value"</span>]</span>
<span id="cb57-18"><a href="#cb57-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">miRNA =</span> g, <span class="at">GS =</span> <span class="fu">abs</span>(t_tumour), <span class="at">row.names =</span> <span class="cn">NULL</span>)</span>
<span id="cb57-19"><a href="#cb57-19" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb57-20"><a href="#cb57-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-21"><a href="#cb57-21" aria-hidden="true" tabindex="-1"></a>geneGS <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, gs_list)</span>
<span id="cb57-22"><a href="#cb57-22" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(geneGS) <span class="ot">&lt;-</span> geneGS<span class="sc">$</span>miRNA</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Module membership (MM) ----</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Drop the "ME" prefix so we get clean module names</span></span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>modNames <span class="ot">&lt;-</span> <span class="fu">substring</span>(<span class="fu">colnames</span>(MEs), <span class="dv">3</span>)  <span class="co"># "MEturquoise" -&gt; "turquoise"</span></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>geneModuleMembership <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">cor</span>(datExpr, MEs, <span class="at">use =</span> <span class="st">"p"</span>))</span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(geneModuleMembership) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"MM"</span>, modNames)</span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(geneModuleMembership) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(datExpr)  <span class="co"># miRNA names</span></span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Put everything together</span></span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a>hubData <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">miRNA  =</span> <span class="fu">colnames</span>(datExpr),</span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">module =</span> moduleColors,</span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a>  geneModuleMembership,</span>
<span id="cb58-15"><a href="#cb58-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">GS     =</span> geneGS[<span class="fu">colnames</span>(datExpr), <span class="st">"GS"</span>],</span>
<span id="cb58-16"><a href="#cb58-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">row.names =</span> <span class="cn">NULL</span></span>
<span id="cb58-17"><a href="#cb58-17" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Intramodular connectivity &amp; hub miRNAs ----</span></span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>IMConn <span class="ot">&lt;-</span> <span class="fu">intramodularConnectivity</span>(adjacency, moduleColors)</span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>IMConn_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(IMConn)</span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a>IMConn_df<span class="sc">$</span>miRNA <span class="ot">&lt;-</span> <span class="fu">rownames</span>(IMConn_df)</span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Add kWithin into hubData</span></span>
<span id="cb59-7"><a href="#cb59-7" aria-hidden="true" tabindex="-1"></a>hubData <span class="ot">&lt;-</span> hubData <span class="sc">%&gt;%</span></span>
<span id="cb59-8"><a href="#cb59-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(IMConn_df[, <span class="fu">c</span>(<span class="st">"miRNA"</span>, <span class="st">"kWithin"</span>)], <span class="at">by =</span> <span class="st">"miRNA"</span>)</span>
<span id="cb59-9"><a href="#cb59-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-10"><a href="#cb59-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Top 10 hubs per module (by kWithin)</span></span>
<span id="cb59-11"><a href="#cb59-11" aria-hidden="true" tabindex="-1"></a>hubList <span class="ot">&lt;-</span> hubData <span class="sc">%&gt;%</span></span>
<span id="cb59-12"><a href="#cb59-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(module) <span class="sc">%&gt;%</span></span>
<span id="cb59-13"><a href="#cb59-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(kWithin)) <span class="sc">%&gt;%</span></span>
<span id="cb59-14"><a href="#cb59-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb59-15"><a href="#cb59-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb59-16"><a href="#cb59-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-17"><a href="#cb59-17" aria-hidden="true" tabindex="-1"></a>hubList</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 40 × 8
   miRNA        module MMturquoise MMblue MMbrown  MMgrey    GS kWithin
   &lt;chr&gt;        &lt;chr&gt;        &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt;
 1 ILMN_3168603 blue       0.0672   0.930   0.302 -0.0774 1.72     15.0
 2 ILMN_3167275 blue      -0.0461   0.911   0.261  0.0774 0.621    14.6
 3 ILMN_3168541 blue      -0.199    0.935   0.409  0.117  0.554    14.5
 4 ILMN_3167184 blue      -0.0164   0.916   0.215  0.0661 0.805    14.2
 5 ILMN_3168633 blue       0.0325   0.908   0.227  0.0416 1.28     13.7
 6 ILMN_3168736 blue      -0.145    0.910   0.308  0.0969 0.265    13.4
 7 ILMN_3168183 blue      -0.0273   0.925   0.353 -0.0624 0.878    13.4
 8 ILMN_3168848 blue      -0.0341   0.905   0.257  0.0677 0.740    13.4
 9 ILMN_3166996 blue      -0.0278   0.895   0.292  0.0182 0.861    12.5
10 ILMN_3167242 blue       0.00604  0.885   0.120  0.0544 1.24     11.8
# ℹ 30 more rows</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Example: GS vs MM for the turquoise module</span></span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>targetModule <span class="ot">&lt;-</span> <span class="st">"turquoise"</span></span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>inMod <span class="ot">&lt;-</span> hubData<span class="sc">$</span>module <span class="sc">==</span> targetModule</span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(hubData<span class="sc">$</span>MMturquoise[inMod],</span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a>     hubData<span class="sc">$</span>GS[inMod],</span>
<span id="cb61-6"><a href="#cb61-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"Module membership (turquoise)"</span>,</span>
<span id="cb61-7"><a href="#cb61-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">"Gene significance (|t| for tumour)"</span>,</span>
<span id="cb61-8"><a href="#cb61-8" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">"GS vs MM in"</span>, targetModule, <span class="st">"module"</span>))</span>
<span id="cb61-9"><a href="#cb61-9" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="fu">lm</span>(GS <span class="sc">~</span> MMturquoise, <span class="at">data =</span> hubData[inMod, ]), <span class="at">lty =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_Limma_DiffCoExp_Glasso_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>The closer the absolute value of the x-coordinate (MMturquoise) is to 1, the more strongly that miRNA belongs to the turquoise module. -&gt; looks great used an unsigned network, so it’s totally normal to have lots of negative MM values.</p>
<p>Unsigned used, thus: -&gt; Adjacency uses |correlation| between miRNAs. -&gt; That means miRNAs that are strongly negatively correlated can still be tightly connected and end up in the same module.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Export module miRNA lists ----</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>moduleGenes <span class="ot">&lt;-</span> <span class="fu">split</span>(hubData<span class="sc">$</span>miRNA, hubData<span class="sc">$</span>module)</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (mod <span class="cf">in</span> <span class="fu">names</span>(moduleGenes)) {</span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(moduleGenes[[mod]],</span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"module_"</span>, mod, <span class="st">"_miRNAs.txt"</span>),</span>
<span id="cb62-7"><a href="#cb62-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb62-8"><a href="#cb62-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">col.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb62-9"><a href="#cb62-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb62-10"><a href="#cb62-10" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb62-11"><a href="#cb62-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-12"><a href="#cb62-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Export top 10 hubs per module ----</span></span>
<span id="cb62-13"><a href="#cb62-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (mod <span class="cf">in</span> <span class="fu">unique</span>(hubList<span class="sc">$</span>module)) {</span>
<span id="cb62-14"><a href="#cb62-14" aria-hidden="true" tabindex="-1"></a>  sub <span class="ot">&lt;-</span> hubList[hubList<span class="sc">$</span>module <span class="sc">==</span> mod, ]</span>
<span id="cb62-15"><a href="#cb62-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(sub,</span>
<span id="cb62-16"><a href="#cb62-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"module_"</span>, mod, <span class="st">"_top10_hubs.txt"</span>),</span>
<span id="cb62-17"><a href="#cb62-17" aria-hidden="true" tabindex="-1"></a>              <span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb62-18"><a href="#cb62-18" aria-hidden="true" tabindex="-1"></a>              <span class="at">quote =</span> <span class="cn">FALSE</span>,</span>
<span id="cb62-19"><a href="#cb62-19" aria-hidden="true" tabindex="-1"></a>              <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb62-20"><a href="#cb62-20" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>table(moduleColors) moduleColors blue brown grey turquoise 201 85 250 294</p>
<p>NEW METHOD 1: LIMMA ————————————————————</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Define condition from title suffix</span></span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>condition <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"T$"</span>, pheno<span class="sc">$</span>title), <span class="st">"tumor"</span>, <span class="st">"normal"</span>)</span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Check that each remaining patient has both N &amp; T</span></span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(pheno<span class="sc">$</span>patient_id, pheno<span class="sc">$</span>condition)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>    
     normal tumor
  01      1     1
  02      1     1
  03      1     1
  04      1     1
  05      1     1
  06      1     1
  07      1     1
  08      1     1
  09      1     1
  10      1     1
  11      1     1
  12      1     1
  13      1     1
  14      1     1
  15      1     1
  16      1     1
  17      1     1
  18      1     1
  19      1     1
  20      1     1
  21      1     1
  23      1     1
  24      1     1
  25      1     1
  26      1     1
  27      1     1
  28      1     1
  29      1     1
  30      1     1
  31      1     1
  32      1     1
  33      1     1
  34      1     1
  35      1     1
  37      1     1
  38      1     1
  39      1     1
  40      1     1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>patient_ids <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">unique</span>(pheno<span class="sc">$</span>patient_id))</span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a><span class="co">#set up factors</span></span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>condition  <span class="ot">&lt;-</span> <span class="fu">factor</span>(pheno<span class="sc">$</span>condition, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"normal"</span>, <span class="st">"tumor"</span>))</span>
<span id="cb65-5"><a href="#cb65-5" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>patient_id <span class="ot">&lt;-</span> <span class="fu">factor</span>(pheno<span class="sc">$</span>patient_id)</span>
<span id="cb65-6"><a href="#cb65-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-7"><a href="#cb65-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-8"><a href="#cb65-8" aria-hidden="true" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> patient_id <span class="sc">+</span> condition, <span class="at">data =</span> pheno)</span>
<span id="cb65-9"><a href="#cb65-9" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(design)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "(Intercept)"    "patient_id02"   "patient_id03"   "patient_id04"  
 [5] "patient_id05"   "patient_id06"   "patient_id07"   "patient_id08"  
 [9] "patient_id09"   "patient_id10"   "patient_id11"   "patient_id12"  
[13] "patient_id13"   "patient_id14"   "patient_id15"   "patient_id16"  
[17] "patient_id17"   "patient_id18"   "patient_id19"   "patient_id20"  
[21] "patient_id21"   "patient_id23"   "patient_id24"   "patient_id25"  
[25] "patient_id26"   "patient_id27"   "patient_id28"   "patient_id29"  
[29] "patient_id30"   "patient_id31"   "patient_id32"   "patient_id33"  
[33] "patient_id34"   "patient_id35"   "patient_id37"   "patient_id38"  
[37] "patient_id39"   "patient_id40"   "conditiontumor"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="co">#run Limma</span></span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>fit  <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(expr_norm, design)</span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>fit  <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit)</span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-5"><a href="#cb67-5" aria-hidden="true" tabindex="-1"></a>tt <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit, <span class="at">coef =</span> <span class="st">"conditiontumor"</span>, <span class="at">number =</span> <span class="cn">Inf</span>)</span>
<span id="cb67-6"><a href="#cb67-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-7"><a href="#cb67-7" aria-hidden="true" tabindex="-1"></a><span class="co">#statistically significant miRNAs</span></span>
<span id="cb67-8"><a href="#cb67-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Method 1: choose the k most significant genes </span></span>
<span id="cb67-9"><a href="#cb67-9" aria-hidden="true" tabindex="-1"></a>ord_sig <span class="ot">&lt;-</span> tt[<span class="fu">order</span>(tt<span class="sc">$</span>adj.P.Val),]</span>
<span id="cb67-10"><a href="#cb67-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-11"><a href="#cb67-11" aria-hidden="true" tabindex="-1"></a>ord_sig[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,] <span class="co">#gives the 10 most significant miRNAs</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                  logFC   AveExpr         t      P.Value    adj.P.Val        B
ILMN_3168273  2.4260804  9.713189  16.73042 2.097688e-19 1.741081e-16 34.12231
ILMN_3168513 -1.5136403 11.774535 -14.80969 1.301994e-17 5.403275e-15 30.04113
ILMN_3168646  2.6285182  9.124647  14.10147 6.574290e-17 1.818887e-14 28.43427
ILMN_3168388  1.8846762 10.863465  13.77601 1.409816e-16 2.925367e-14 27.67623
ILMN_3167455 -1.6018103 11.937361 -12.94061 1.056713e-15 1.754143e-13 25.67197
ILMN_3167729 -0.4872745 13.487849 -12.48619 3.273583e-15 4.528456e-13 24.54534
ILMN_3168707 -0.8163931  6.368681 -12.18114 7.093300e-15 7.898140e-13 23.77431
ILMN_3167714  1.3804658  9.884744  12.15349 7.612665e-15 7.898140e-13 23.70383
ILMN_3168692  1.5141958  9.669786  11.62623 2.983492e-14 2.751442e-12 22.34087
ILMN_3168765  1.5954947  9.275411  11.54080 3.734949e-14 3.100007e-12 22.11661</code></pre>
</div>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Method 2: choose based on threshold</span></span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a><span class="co">#co_FC &lt;- 1</span></span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a><span class="co">#sig_miRNA &lt;- tt[tt$adj.P.Val&lt; 0.05 &amp; abs(tt$logFC) &gt; co_FC,]</span></span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a><span class="co">#n_sig &lt;- dim(sig_miRNA)[1]</span></span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a><span class="co">#n_tot &lt;- dim(tt)[1]</span></span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-8"><a href="#cb69-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-9"><a href="#cb69-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-10"><a href="#cb69-10" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(tt[tt<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>,])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 366   6</code></pre>
</div>
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>tt_rel <span class="ot">&lt;-</span> (tt[tt<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(tt<span class="sc">$</span>logFC)  <span class="sc">&gt;=</span> <span class="dv">1</span>,])</span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(tt_rel)[<span class="dv">1</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 80</code></pre>
</div>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(tt_rel[tt_rel<span class="sc">$</span>logFC<span class="sc">&lt;</span><span class="dv">0</span>,])[<span class="dv">1</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 46</code></pre>
</div>
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(tt_rel[tt_rel<span class="sc">$</span>logFC<span class="sc">&gt;</span><span class="dv">0</span>,])[<span class="dv">1</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 34</code></pre>
</div>
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(tt_rel)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                 logFC   AveExpr         t      P.Value    adj.P.Val        B
ILMN_3168273  2.426080  9.713189  16.73042 2.097688e-19 1.741081e-16 34.12231
ILMN_3168513 -1.513640 11.774535 -14.80969 1.301994e-17 5.403275e-15 30.04113
ILMN_3168646  2.628518  9.124647  14.10147 6.574290e-17 1.818887e-14 28.43427
ILMN_3168388  1.884676 10.863465  13.77601 1.409816e-16 2.925367e-14 27.67623
ILMN_3167455 -1.601810 11.937361 -12.94061 1.056713e-15 1.754143e-13 25.67197
ILMN_3167714  1.380466  9.884744  12.15349 7.612665e-15 7.898140e-13 23.70383</code></pre>
</div>
</div>
<p>Plotting Limma</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> pheno[<span class="fu">colnames</span>(expr_norm), ]</span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">rownames</span>(pheno), <span class="fu">colnames</span>(expr_norm)))</span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>cols <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(pheno<span class="sc">$</span>condition <span class="sc">==</span> <span class="st">"tumor"</span>, <span class="st">"purple"</span>, <span class="st">"black"</span>)</span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMDS</span>(expr_norm, </span>
<span id="cb79-8"><a href="#cb79-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">top =</span> <span class="dv">500</span>, </span>
<span id="cb79-9"><a href="#cb79-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> pheno<span class="sc">$</span>patient_id, </span>
<span id="cb79-10"><a href="#cb79-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">col =</span> cols)</span>
<span id="cb79-11"><a href="#cb79-11" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"topright"</span>, </span>
<span id="cb79-12"><a href="#cb79-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">legend=</span><span class="fu">c</span>(<span class="st">"normal"</span>,<span class="st">"tumor"</span>), </span>
<span id="cb79-13"><a href="#cb79-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">col=</span><span class="fu">c</span>(<span class="st">"black"</span>,<span class="st">"purple"</span>), </span>
<span id="cb79-14"><a href="#cb79-14" aria-hidden="true" tabindex="-1"></a>       <span class="at">pch=</span><span class="dv">16</span>, <span class="at">bty=</span><span class="st">"n"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_Limma_DiffCoExp_Glasso_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(fit, </span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">coef =</span> <span class="st">"conditiontumor"</span>, </span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">main=</span><span class="st">"Average Expression vs LogFC"</span>)</span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">0</span>, <span class="at">col =</span> <span class="st">"violet"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_Limma_DiffCoExp_Glasso_files/figure-html/unnamed-chunk-20-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="co">#gives two distinct clouds, separation along the the first dimension</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Volcano Plot</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>tt_volc <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit, <span class="at">coef=</span><span class="st">"conditiontumor"</span>, <span class="at">number=</span><span class="cn">Inf</span>)</span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a>tt_volc<span class="sc">$</span>negLog10FDR <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(tt<span class="sc">$</span>adj.P.Val)</span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a><span class="co">#these parameters were chosen</span></span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a>p_sig <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a>co_FC <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb82-9"><a href="#cb82-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-10"><a href="#cb82-10" aria-hidden="true" tabindex="-1"></a>sig <span class="ot">&lt;-</span> tt_volc<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> p_sig <span class="sc">&amp;</span> <span class="fu">abs</span>(tt_volc<span class="sc">$</span>logFC) <span class="sc">&gt;</span> co_FC</span>
<span id="cb82-11"><a href="#cb82-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-12"><a href="#cb82-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-13"><a href="#cb82-13" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tt_volc<span class="sc">$</span>logFC, tt_volc<span class="sc">$</span>negLog10FDR,</span>
<span id="cb82-14"><a href="#cb82-14" aria-hidden="true" tabindex="-1"></a>     <span class="at">pch =</span> <span class="dv">20</span>,</span>
<span id="cb82-15"><a href="#cb82-15" aria-hidden="true" tabindex="-1"></a>     <span class="at">col =</span> <span class="st">"grey70"</span>,</span>
<span id="cb82-16"><a href="#cb82-16" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"log2 fold change (tumour / normal)"</span>,</span>
<span id="cb82-17"><a href="#cb82-17" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">"-log10 FDR"</span>,</span>
<span id="cb82-18"><a href="#cb82-18" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"Volcano: tumour vs normal"</span>)</span>
<span id="cb82-19"><a href="#cb82-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-20"><a href="#cb82-20" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>), <span class="at">col =</span> <span class="st">"orange"</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb82-21"><a href="#cb82-21" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">col =</span> <span class="st">"orange"</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb82-22"><a href="#cb82-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-23"><a href="#cb82-23" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(tt_volc<span class="sc">$</span>logFC[sig], tt_volc<span class="sc">$</span>negLog10FDR[sig],</span>
<span id="cb82-24"><a href="#cb82-24" aria-hidden="true" tabindex="-1"></a>       <span class="at">pch =</span> <span class="dv">20</span>, <span class="at">col =</span> <span class="st">"purple"</span>)</span>
<span id="cb82-25"><a href="#cb82-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-26"><a href="#cb82-26" aria-hidden="true" tabindex="-1"></a><span class="co">#label top 10 by FDR</span></span>
<span id="cb82-27"><a href="#cb82-27" aria-hidden="true" tabindex="-1"></a>top <span class="ot">&lt;-</span> <span class="fu">head</span>(<span class="fu">order</span>(tt_volc<span class="sc">$</span>adj.P.Val), <span class="dv">10</span>)</span>
<span id="cb82-28"><a href="#cb82-28" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(tt_volc<span class="sc">$</span>logFC[top], tt_volc<span class="sc">$</span>negLog10FDR[top], <span class="at">labels=</span><span class="fu">rownames</span>(tt_volc)[top], <span class="at">cex=</span><span class="fl">0.7</span>, <span class="at">pos=</span><span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_Limma_DiffCoExp_Glasso_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Shows us the logFC plotted against statistical significance</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Heatmap for top DE miRNAs</span></span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>topN <span class="ot">&lt;-</span> <span class="dv">30</span></span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a>top_ids <span class="ot">&lt;-</span> <span class="fu">rownames</span>(tt)[<span class="fu">order</span>(tt<span class="sc">$</span>adj.P.Val)][<span class="dv">1</span><span class="sc">:</span>topN]</span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a>mat <span class="ot">&lt;-</span> expr_norm[top_ids, ]</span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-9"><a href="#cb84-9" aria-hidden="true" tabindex="-1"></a>ann <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb84-10"><a href="#cb84-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">condition  =</span> pheno<span class="sc">$</span>condition,</span>
<span id="cb84-11"><a href="#cb84-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">patient_id =</span> pheno<span class="sc">$</span>patient_id</span>
<span id="cb84-12"><a href="#cb84-12" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb84-13"><a href="#cb84-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ann) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(pheno)</span>
<span id="cb84-14"><a href="#cb84-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-15"><a href="#cb84-15" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(mat,</span>
<span id="cb84-16"><a href="#cb84-16" aria-hidden="true" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">"row"</span>,</span>
<span id="cb84-17"><a href="#cb84-17" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> ann,</span>
<span id="cb84-18"><a href="#cb84-18" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb84-19"><a href="#cb84-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">"Top"</span>, topN, <span class="st">"DE miRNAs (limma)"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_Limma_DiffCoExp_Glasso_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>COMPARING LIMMA AND PAIRED WGCNA (lw) ——————————————</p>
<p>Do the miRNAs deemed important by paired WGCNA and Limma overlap?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>N_comp_lw <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">200</span>,<span class="dv">150</span>,<span class="dv">100</span>,<span class="dv">50</span>)</span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(N_comp_lw)){</span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a>  limma_top <span class="ot">&lt;-</span> <span class="fu">rownames</span>(ord_sig)[<span class="dv">1</span><span class="sc">:</span>N_comp_lw[i]]</span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-5"><a href="#cb85-5" aria-hidden="true" tabindex="-1"></a>  <span class="co">#WGCNA "importance" score = GS_i * |MM_i|</span></span>
<span id="cb85-6"><a href="#cb85-6" aria-hidden="true" tabindex="-1"></a>  <span class="co">#GS = gene significance: how "phenotype associated" a miRNA is</span></span>
<span id="cb85-7"><a href="#cb85-7" aria-hidden="true" tabindex="-1"></a>  <span class="co">#MM = module membership: how much a miRNA "belongs" to a module</span></span>
<span id="cb85-8"><a href="#cb85-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb85-9"><a href="#cb85-9" aria-hidden="true" tabindex="-1"></a>  mm_col <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"MM"</span>, hubData<span class="sc">$</span>module)</span>
<span id="cb85-10"><a href="#cb85-10" aria-hidden="true" tabindex="-1"></a>  hubData<span class="sc">$</span>MM_self <span class="ot">&lt;-</span> <span class="fu">mapply</span>(<span class="cf">function</span>(i, col) hubData[i, col], <span class="fu">seq_len</span>(<span class="fu">nrow</span>(hubData)), mm_col)</span>
<span id="cb85-11"><a href="#cb85-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb85-12"><a href="#cb85-12" aria-hidden="true" tabindex="-1"></a>  hubData<span class="sc">$</span>wgcna_score <span class="ot">&lt;-</span> hubData<span class="sc">$</span>GS <span class="sc">*</span> <span class="fu">abs</span>(hubData<span class="sc">$</span>MM_self)</span>
<span id="cb85-13"><a href="#cb85-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb85-14"><a href="#cb85-14" aria-hidden="true" tabindex="-1"></a>  wgcna_top <span class="ot">&lt;-</span> hubData<span class="sc">$</span>miRNA[<span class="fu">order</span>(<span class="sc">-</span>hubData<span class="sc">$</span>wgcna_score)][<span class="dv">1</span><span class="sc">:</span>N_comp_lw[i]]</span>
<span id="cb85-15"><a href="#cb85-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb85-16"><a href="#cb85-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"N ="</span>, N_comp_lw[i],<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb85-17"><a href="#cb85-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"Numer of overlapping miRNAs:"</span>,<span class="fu">length</span>(<span class="fu">intersect</span>(limma_top, wgcna_top)))</span>
<span id="cb85-18"><a href="#cb85-18" aria-hidden="true" tabindex="-1"></a>  <span class="co">#head(intersect(limma_top, wgcna_top), 20)</span></span>
<span id="cb85-19"><a href="#cb85-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb85-20"><a href="#cb85-20" aria-hidden="true" tabindex="-1"></a>  <span class="co">#failsafe in case the two lists were different lengths</span></span>
<span id="cb85-21"><a href="#cb85-21" aria-hidden="true" tabindex="-1"></a>  universe <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(ord_sig), hubData<span class="sc">$</span>miRNA)</span>
<span id="cb85-22"><a href="#cb85-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb85-23"><a href="#cb85-23" aria-hidden="true" tabindex="-1"></a>  A <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">intersect</span>(limma_top, wgcna_top))</span>
<span id="cb85-24"><a href="#cb85-24" aria-hidden="true" tabindex="-1"></a>  B <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">setdiff</span>(wgcna_top, limma_top))</span>
<span id="cb85-25"><a href="#cb85-25" aria-hidden="true" tabindex="-1"></a>  C <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">setdiff</span>(limma_top, wgcna_top))</span>
<span id="cb85-26"><a href="#cb85-26" aria-hidden="true" tabindex="-1"></a>  D <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">setdiff</span>(universe, <span class="fu">union</span>(limma_top, wgcna_top)))</span>
<span id="cb85-27"><a href="#cb85-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb85-28"><a href="#cb85-28" aria-hidden="true" tabindex="-1"></a>  fish_comp_lw <span class="ot">&lt;-</span> <span class="fu">fisher.test</span>(<span class="fu">matrix</span>(<span class="fu">c</span>(A,B,C,D), <span class="at">nrow=</span><span class="dv">2</span>), <span class="at">alternative=</span><span class="st">"greater"</span>)</span>
<span id="cb85-29"><a href="#cb85-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb85-30"><a href="#cb85-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(fish_comp_lw)</span>
<span id="cb85-31"><a href="#cb85-31" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>N = 200 
Numer of overlapping miRNAs: 184
    Fisher's Exact Test for Count Data

data:  matrix(c(A, B, C, D), nrow = 2)
p-value &lt; 2.2e-16
alternative hypothesis: true odds ratio is greater than 1
95 percent confidence interval:
 229.3031      Inf
sample estimates:
odds ratio 
  433.8704 

N = 150 
Numer of overlapping miRNAs: 137
    Fisher's Exact Test for Count Data

data:  matrix(c(A, B, C, D), nrow = 2)
p-value &lt; 2.2e-16
alternative hypothesis: true odds ratio is greater than 1
95 percent confidence interval:
 257.825     Inf
sample estimates:
odds ratio 
  517.6788 

N = 100 
Numer of overlapping miRNAs: 89
    Fisher's Exact Test for Count Data

data:  matrix(c(A, B, C, D), nrow = 2)
p-value &lt; 2.2e-16
alternative hypothesis: true odds ratio is greater than 1
95 percent confidence interval:
 233.7978      Inf
sample estimates:
odds ratio 
   507.699 

N = 50 
Numer of overlapping miRNAs: 44
    Fisher's Exact Test for Count Data

data:  matrix(c(A, B, C, D), nrow = 2)
p-value &lt; 2.2e-16
alternative hypothesis: true odds ratio is greater than 1
95 percent confidence interval:
 304.9829      Inf
sample estimates:
odds ratio 
   894.164 </code></pre>
</div>
</div>
<p>Which modules contain the most miRNAs flagged by Limma? How likely is this result?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a>limma_map <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">miRNA      =</span> <span class="fu">rownames</span>(ord_sig),</span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">limma_FDR  =</span> ord_sig<span class="sc">$</span>adj.P.Val,</span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">limma_P    =</span> ord_sig<span class="sc">$</span>P.Value,</span>
<span id="cb87-5"><a href="#cb87-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">limma_t    =</span> ord_sig<span class="sc">$</span>t,</span>
<span id="cb87-6"><a href="#cb87-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">limma_logFC=</span> ord_sig<span class="sc">$</span>logFC,</span>
<span id="cb87-7"><a href="#cb87-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">row.names  =</span> <span class="cn">NULL</span></span>
<span id="cb87-8"><a href="#cb87-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb87-9"><a href="#cb87-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-10"><a href="#cb87-10" aria-hidden="true" tabindex="-1"></a>comp <span class="ot">&lt;-</span> hubData <span class="sc">%&gt;%</span></span>
<span id="cb87-11"><a href="#cb87-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(limma_map, <span class="at">by =</span> <span class="st">"miRNA"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb87-12"><a href="#cb87-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(miRNA <span class="sc">%in%</span> universe) <span class="sc">%&gt;%</span></span>
<span id="cb87-13"><a href="#cb87-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">limma_sig =</span> <span class="sc">-</span><span class="fu">log10</span>(limma_FDR))</span>
<span id="cb87-14"><a href="#cb87-14" aria-hidden="true" tabindex="-1"></a>DE <span class="ot">&lt;-</span> <span class="fu">rownames</span>(ord_sig)[ord_sig<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>]</span>
<span id="cb87-15"><a href="#cb87-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-16"><a href="#cb87-16" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"DE miRNAs (FDR&lt;0.05):"</span>, <span class="fu">length</span>(DE), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DE miRNAs (FDR&lt;0.05): 366 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a>moduleMembers <span class="ot">&lt;-</span> <span class="fu">split</span>(comp<span class="sc">$</span>miRNA, comp<span class="sc">$</span>module)</span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>mod_enrich <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">lapply</span>(<span class="fu">names</span>(moduleMembers), <span class="cf">function</span>(m) {</span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a>  genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(moduleMembers[[m]], universe)</span>
<span id="cb89-5"><a href="#cb89-5" aria-hidden="true" tabindex="-1"></a>  k <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">intersect</span>(genes, DE))      <span class="co"># DE inside module</span></span>
<span id="cb89-6"><a href="#cb89-6" aria-hidden="true" tabindex="-1"></a>  M <span class="ot">&lt;-</span> <span class="fu">length</span>(genes)                     <span class="co"># module size</span></span>
<span id="cb89-7"><a href="#cb89-7" aria-hidden="true" tabindex="-1"></a>  U <span class="ot">&lt;-</span> <span class="fu">length</span>(universe)                  <span class="co"># total tested</span></span>
<span id="cb89-8"><a href="#cb89-8" aria-hidden="true" tabindex="-1"></a>  n <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">intersect</span>(DE, universe))   <span class="co"># total DE</span></span>
<span id="cb89-9"><a href="#cb89-9" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">phyper</span>(k <span class="sc">-</span> <span class="dv">1</span>, M, U <span class="sc">-</span> M, n, <span class="at">lower.tail =</span> <span class="cn">FALSE</span>)</span>
<span id="cb89-10"><a href="#cb89-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-11"><a href="#cb89-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">module=</span>m, <span class="at">module_size=</span>M, <span class="at">DE_in_module=</span>k, <span class="at">frac_DE=</span>k<span class="sc">/</span>M, <span class="at">p_value=</span>p)</span>
<span id="cb89-12"><a href="#cb89-12" aria-hidden="true" tabindex="-1"></a>})) <span class="sc">%&gt;%</span></span>
<span id="cb89-13"><a href="#cb89-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FDR =</span> <span class="fu">p.adjust</span>(p_value, <span class="st">"BH"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb89-14"><a href="#cb89-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(FDR, <span class="fu">desc</span>(frac_DE))</span>
<span id="cb89-15"><a href="#cb89-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-16"><a href="#cb89-16" aria-hidden="true" tabindex="-1"></a>mod_enrich</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     module module_size DE_in_module   frac_DE       p_value           FDR
1 turquoise         294          274 0.9319728 2.013281e-110 8.053123e-110
2     brown          85           22 0.2588235  9.999202e-01  1.000000e+00
3      blue         201           36 0.1791045  1.000000e+00  1.000000e+00
4      grey         250           34 0.1360000  1.000000e+00  1.000000e+00</code></pre>
</div>
</div>
<p>Are the hubgenes more DE?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="co">#kWithin vs limma significance</span></span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a>cor_kWithin <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(comp<span class="sc">$</span>kWithin, comp<span class="sc">$</span>limma_sig, <span class="at">method=</span><span class="st">"spearman"</span>, <span class="at">use=</span><span class="st">"complete.obs"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in cor.test.default(comp$kWithin, comp$limma_sig, method = "spearman",
: Cannot compute exact p-value with ties</code></pre>
</div>
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="co"># |MM_self| vs limma significance</span></span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a>cor_MM <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(<span class="fu">abs</span>(comp<span class="sc">$</span>MM_self), comp<span class="sc">$</span>limma_sig, <span class="at">method=</span><span class="st">"spearman"</span>, <span class="at">use=</span><span class="st">"complete.obs"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in cor.test.default(abs(comp$MM_self), comp$limma_sig, method =
"spearman", : Cannot compute exact p-value with ties</code></pre>
</div>
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="co">#combined score vs limma significance</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a>cor_score <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(comp<span class="sc">$</span>wgcna_score, comp<span class="sc">$</span>limma_sig, <span class="at">method=</span><span class="st">"spearman"</span>, <span class="at">use=</span><span class="st">"complete.obs"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in cor.test.default(comp$wgcna_score, comp$limma_sig, method =
"spearman", : Cannot compute exact p-value with ties</code></pre>
</div>
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>cor_kWithin</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Spearman's rank correlation rho

data:  comp$kWithin and comp$limma_sig
S = 55909986, p-value &lt; 2.2e-16
alternative hypothesis: true rho is not equal to 0
sample estimates:
      rho 
0.4133123 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>cor_MM</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Spearman's rank correlation rho

data:  abs(comp$MM_self) and comp$limma_sig
S = 58902481, p-value &lt; 2.2e-16
alternative hypothesis: true rho is not equal to 0
sample estimates:
      rho 
0.3819107 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a>cor_score</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Spearman's rank correlation rho

data:  comp$wgcna_score and comp$limma_sig
S = 8115486, p-value &lt; 2.2e-16
alternative hypothesis: true rho is not equal to 0
sample estimates:
      rho 
0.9148407 </code></pre>
</div>
</div>
<p>NEW METHOD 3: DIFFCOEXP ——————————————————–</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>condition <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"T$"</span>, pheno<span class="sc">$</span>title), <span class="st">"tumor"</span>, <span class="st">"normal"</span>)</span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>condition <span class="ot">&lt;-</span> <span class="fu">factor</span>(pheno<span class="sc">$</span>condition, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"normal"</span>,<span class="st">"tumor"</span>))</span>
<span id="cb103-3"><a href="#cb103-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-4"><a href="#cb103-4" aria-hidden="true" tabindex="-1"></a>datExpr <span class="ot">&lt;-</span> <span class="fu">t</span>(expr_norm) </span>
<span id="cb103-5"><a href="#cb103-5" aria-hidden="true" tabindex="-1"></a>E_N_exp <span class="ot">&lt;-</span> datExpr[pheno<span class="sc">$</span>condition <span class="sc">==</span> <span class="st">"normal"</span>, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb103-6"><a href="#cb103-6" aria-hidden="true" tabindex="-1"></a>E_T_exp <span class="ot">&lt;-</span> datExpr[pheno<span class="sc">$</span>condition <span class="sc">==</span> <span class="st">"tumor"</span>,  , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb103-7"><a href="#cb103-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-8"><a href="#cb103-8" aria-hidden="true" tabindex="-1"></a><span class="co">#for diffcoexp we need the two expression matrices to have the same row names, we will name them by their patient_ID</span></span>
<span id="cb103-9"><a href="#cb103-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-10"><a href="#cb103-10" aria-hidden="true" tabindex="-1"></a>E_N_exp <span class="ot">&lt;-</span> E_N_exp[<span class="fu">order</span>(pheno<span class="sc">$</span>patient_id[pheno<span class="sc">$</span>condition<span class="sc">==</span><span class="st">"normal"</span>]), , drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb103-11"><a href="#cb103-11" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(E_N_exp) <span class="ot">&lt;-</span> pheno<span class="sc">$</span>patient_id[pheno<span class="sc">$</span>condition<span class="sc">==</span><span class="st">"normal"</span>][<span class="fu">order</span>(pheno<span class="sc">$</span>patient_id[pheno<span class="sc">$</span>condition<span class="sc">==</span><span class="st">"normal"</span>])]</span>
<span id="cb103-12"><a href="#cb103-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-13"><a href="#cb103-13" aria-hidden="true" tabindex="-1"></a>E_T_exp <span class="ot">&lt;-</span> E_T_exp[<span class="fu">order</span>(pheno<span class="sc">$</span>patient_id[pheno<span class="sc">$</span>condition<span class="sc">==</span><span class="st">"tumor"</span>]), , drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb103-14"><a href="#cb103-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(E_T_exp) <span class="ot">&lt;-</span> pheno<span class="sc">$</span>patient_id[pheno<span class="sc">$</span>condition<span class="sc">==</span><span class="st">"tumor"</span>][<span class="fu">order</span>(pheno<span class="sc">$</span>patient_id[pheno<span class="sc">$</span>condition<span class="sc">==</span><span class="st">"tumor"</span>])]</span>
<span id="cb103-15"><a href="#cb103-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-16"><a href="#cb103-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(diffcoexp)</span>
<span id="cb103-17"><a href="#cb103-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-18"><a href="#cb103-18" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">diffcoexp</span>(<span class="fu">t</span>(E_N_exp), <span class="fu">t</span>(E_T_exp), <span class="at">r.method =</span> <span class="st">"spearman"</span> )</span>
<span id="cb103-19"><a href="#cb103-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-20"><a href="#cb103-20" aria-hidden="true" tabindex="-1"></a><span class="co">#res calculates differentially coexpressed links (DCLs): gene pairs with significantly different</span></span>
<span id="cb103-21"><a href="#cb103-21" aria-hidden="true" tabindex="-1"></a><span class="co">#differentially: DCGs are genes with significantly more DCLs than by chance</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Get parameters for paper</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Default parameters for diffcoexp:"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Default parameters for diffcoexp:"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">args</span>(diffcoexp<span class="sc">::</span>diffcoexp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>function (exprs.1, exprs.2, r.method = c("pearson", "kendall", 
    "spearman")[1], q.method = c("BH", "holm", "hochberg", "hommel", 
    "bonferroni", "BY", "fdr", "none")[1], rth = 0.5, qth = 0.1, 
    r.diffth = 0.5, q.diffth = 0.1, q.dcgth = 0.1) 
NULL</code></pre>
</div>
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"number of DCLs:"</span>, <span class="fu">length</span>(<span class="fu">rownames</span>(res<span class="sc">$</span>DCLs)),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>number of DCLs: 440 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"number of DCGs:"</span>, <span class="fu">length</span>(<span class="fu">rownames</span>(res<span class="sc">$</span>DCGs)),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>number of DCGs: 28 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"number of same signed DCLs:"</span>,<span class="fu">length</span>(res<span class="sc">$</span>DCL<span class="sc">$</span>type[res<span class="sc">$</span>DCL<span class="sc">$</span>type <span class="sc">==</span> <span class="st">"same signed"</span>]), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>number of same signed DCLs: 101 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"number of diff signed DCLs:"</span>,<span class="fu">length</span>(res<span class="sc">$</span>DCL<span class="sc">$</span>type[res<span class="sc">$</span>DCL<span class="sc">$</span>type <span class="sc">==</span> <span class="st">"diff signed"</span>]),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>number of diff signed DCLs: 335 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"number of switched opposites DCLs:"</span>,<span class="fu">length</span>(res<span class="sc">$</span>DCL<span class="sc">$</span>type[res<span class="sc">$</span>DCL<span class="sc">$</span>type <span class="sc">==</span> <span class="st">"switched opposites"</span>]),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>number of switched opposites DCLs: 4 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="fu">max</span>(<span class="fu">abs</span>(res<span class="sc">$</span>DCLs<span class="sc">$</span>cor.diff))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1.130299</code></pre>
</div>
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="fu">min</span>(res<span class="sc">$</span>DCLs<span class="sc">$</span>q.diffcor)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 9.182365e-06</code></pre>
</div>
</div>
<p>COMPARING DIFFCOEXP TO PAIRED WGCNA (pw) —————————————</p>
<p>How do the top candidates overlap?</p>
<p>“DCG-ness” (DCL degree) with WGCNA hubness (kWithin)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a>dcg_degree <span class="ot">&lt;-</span> <span class="fu">table</span>(<span class="fu">c</span>(res<span class="sc">$</span>DCLs<span class="sc">$</span>Gene<span class="fl">.1</span>, res<span class="sc">$</span>DCLs<span class="sc">$</span>Gene<span class="fl">.2</span>))</span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a>dcg_degree <span class="ot">&lt;-</span> dcg_degree[<span class="fu">names</span>(dcg_degree) <span class="sc">%in%</span> hubData<span class="sc">$</span>miRNA]</span>
<span id="cb122-3"><a href="#cb122-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-4"><a href="#cb122-4" aria-hidden="true" tabindex="-1"></a>deg_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">miRNA =</span> <span class="fu">names</span>(dcg_degree),</span>
<span id="cb122-5"><a href="#cb122-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">dcl_degree =</span> <span class="fu">as.numeric</span>(dcg_degree),</span>
<span id="cb122-6"><a href="#cb122-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">row.names=</span><span class="cn">NULL</span>)</span>
<span id="cb122-7"><a href="#cb122-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-8"><a href="#cb122-8" aria-hidden="true" tabindex="-1"></a>comp_dcg <span class="ot">&lt;-</span> <span class="fu">merge</span>(hubData[, <span class="fu">c</span>(<span class="st">"miRNA"</span>,<span class="st">"kWithin"</span>,<span class="st">"module"</span>)], deg_df, <span class="at">by=</span><span class="st">"miRNA"</span>)</span>
<span id="cb122-9"><a href="#cb122-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-10"><a href="#cb122-10" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(comp_dcg<span class="sc">$</span>kWithin, comp_dcg<span class="sc">$</span>dcl_degree, <span class="at">method=</span><span class="st">"spearman"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in cor.test.default(comp_dcg$kWithin, comp_dcg$dcl_degree, method =
"spearman"): Cannot compute exact p-value with ties</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Spearman's rank correlation rho

data:  comp_dcg$kWithin and comp_dcg$dcl_degree
S = 5509448, p-value = 0.1082
alternative hypothesis: true rho is not equal to 0
sample estimates:
       rho 
0.08845251 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb125"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(comp_dcg<span class="sc">$</span>kWithin, comp_dcg<span class="sc">$</span>dcl_degree, <span class="at">pch=</span><span class="dv">20</span>,</span>
<span id="cb125-2"><a href="#cb125-2" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">"WGCNA kWithin"</span>, <span class="at">ylab=</span><span class="st">"DiffCoExp DCL degree"</span>,</span>
<span id="cb125-3"><a href="#cb125-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">main=</span><span class="st">"Are WGCNA hubs also DiffCoExp hubs?"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Dataset1_Limma_DiffCoExp_Glasso_files/figure-html/unnamed-chunk-28-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Overlap of top hubs</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb126"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a>topK <span class="ot">&lt;-</span> <span class="dv">100</span></span>
<span id="cb126-2"><a href="#cb126-2" aria-hidden="true" tabindex="-1"></a>hubs_wgcna <span class="ot">&lt;-</span> hubData<span class="sc">$</span>miRNA[<span class="fu">order</span>(<span class="sc">-</span>hubData<span class="sc">$</span>kWithin)][<span class="dv">1</span><span class="sc">:</span>topK]</span>
<span id="cb126-3"><a href="#cb126-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-4"><a href="#cb126-4" aria-hidden="true" tabindex="-1"></a><span class="co"># DiffCoExp hubs = highest DCL degree</span></span>
<span id="cb126-5"><a href="#cb126-5" aria-hidden="true" tabindex="-1"></a>hubs_diff <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">sort</span>(dcg_degree, <span class="at">decreasing=</span><span class="cn">TRUE</span>))[<span class="dv">1</span><span class="sc">:</span>topK]</span>
<span id="cb126-6"><a href="#cb126-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-7"><a href="#cb126-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-8"><a href="#cb126-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-9"><a href="#cb126-9" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">intersect</span>(hubs_wgcna, hubs_diff))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 23</code></pre>
</div>
<div class="sourceCode cell-code" id="cb128"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1" aria-hidden="true" tabindex="-1"></a><span class="fu">intersect</span>(hubs_wgcna, hubs_diff)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "ILMN_3168183" "ILMN_3168441" "ILMN_3168663" "ILMN_3168711" "ILMN_3168632"
 [6] "ILMN_3168864" "ILMN_3167455" "ILMN_3167609" "ILMN_3168605" "ILMN_3167988"
[11] "ILMN_3166974" "ILMN_3168773" "ILMN_3168009" "ILMN_3168535" "ILMN_3167729"
[16] "ILMN_3168388" "ILMN_3168840" "ILMN_3168642" "ILMN_3168676" "ILMN_3167913"
[21] "ILMN_3168718" "ILMN_3168314" "ILMN_3167437"</code></pre>
</div>
</div>
<p>Are DCLs mostly within the same WGCNA module (or between modules)?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb130"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a>dcl <span class="ot">&lt;-</span> res<span class="sc">$</span>DCLs</span>
<span id="cb130-2"><a href="#cb130-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-3"><a href="#cb130-3" aria-hidden="true" tabindex="-1"></a><span class="co">#module lookup: miRNA -&gt; moduleColor</span></span>
<span id="cb130-4"><a href="#cb130-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-5"><a href="#cb130-5" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">setNames</span>(moduleColors_wgcna<span class="sc">$</span>moduleColor, moduleColors_wgcna<span class="sc">$</span>miRNA)</span>
<span id="cb130-6"><a href="#cb130-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-7"><a href="#cb130-7" aria-hidden="true" tabindex="-1"></a>dcl<span class="sc">$</span>mod1 <span class="ot">&lt;-</span> mod[dcl<span class="sc">$</span>Gene<span class="fl">.1</span>]</span>
<span id="cb130-8"><a href="#cb130-8" aria-hidden="true" tabindex="-1"></a>dcl<span class="sc">$</span>mod2 <span class="ot">&lt;-</span> mod[dcl<span class="sc">$</span>Gene<span class="fl">.2</span>]</span>
<span id="cb130-9"><a href="#cb130-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-10"><a href="#cb130-10" aria-hidden="true" tabindex="-1"></a><span class="co">#drop pairs where a gene/miRNA wasnt found</span></span>
<span id="cb130-11"><a href="#cb130-11" aria-hidden="true" tabindex="-1"></a>dcls2 <span class="ot">&lt;-</span> <span class="fu">subset</span>(dcl, <span class="sc">!</span><span class="fu">is.na</span>(mod1) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(mod2))</span>
<span id="cb130-12"><a href="#cb130-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-13"><a href="#cb130-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-14"><a href="#cb130-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-15"><a href="#cb130-15" aria-hidden="true" tabindex="-1"></a><span class="co">#within-module vs between-module</span></span>
<span id="cb130-16"><a href="#cb130-16" aria-hidden="true" tabindex="-1"></a>within <span class="ot">&lt;-</span> <span class="fu">mean</span>(dcls2<span class="sc">$</span>mod1 <span class="sc">==</span> dcls2<span class="sc">$</span>mod2)</span>
<span id="cb130-17"><a href="#cb130-17" aria-hidden="true" tabindex="-1"></a>within</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.3977273</code></pre>
</div>
<div class="sourceCode cell-code" id="cb132"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb132-1"><a href="#cb132-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(dcls2<span class="sc">$</span>mod1 <span class="sc">==</span> dcls2<span class="sc">$</span>mod2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
  265   175 </code></pre>
</div>
</div>
<p>Are DCLs “focused” on the “tumour” module?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb134"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb134-1"><a href="#cb134-1" aria-hidden="true" tabindex="-1"></a>tumMod <span class="ot">&lt;-</span> <span class="st">"turquoise"</span></span>
<span id="cb134-2"><a href="#cb134-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-3"><a href="#cb134-3" aria-hidden="true" tabindex="-1"></a><span class="co"># fraction of DCL edges that touch turquoise</span></span>
<span id="cb134-4"><a href="#cb134-4" aria-hidden="true" tabindex="-1"></a>touch_turq <span class="ot">&lt;-</span> <span class="fu">mean</span>(dcls2<span class="sc">$</span>mod1 <span class="sc">==</span> tumMod <span class="sc">|</span> dcls2<span class="sc">$</span>mod2 <span class="sc">==</span> tumMod)</span>
<span id="cb134-5"><a href="#cb134-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-6"><a href="#cb134-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-7"><a href="#cb134-7" aria-hidden="true" tabindex="-1"></a><span class="co"># fraction of DCL edges that are internal to turquoise</span></span>
<span id="cb134-8"><a href="#cb134-8" aria-hidden="true" tabindex="-1"></a>internal_turq <span class="ot">&lt;-</span> <span class="fu">mean</span>(dcls2<span class="sc">$</span>mod1 <span class="sc">==</span> tumMod <span class="sc">&amp;</span> dcls2<span class="sc">$</span>mod2 <span class="sc">==</span> tumMod)</span>
<span id="cb134-9"><a href="#cb134-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-10"><a href="#cb134-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-11"><a href="#cb134-11" aria-hidden="true" tabindex="-1"></a><span class="co">#DCL edges that dont involve turquoise</span></span>
<span id="cb134-12"><a href="#cb134-12" aria-hidden="true" tabindex="-1"></a>not_touch_turq_logical <span class="ot">&lt;-</span> <span class="sc">!</span>(dcls2<span class="sc">$</span>mod1 <span class="sc">==</span> tumMod <span class="sc">|</span> dcls2<span class="sc">$</span>mod2 <span class="sc">==</span> tumMod)</span>
<span id="cb134-13"><a href="#cb134-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-14"><a href="#cb134-14" aria-hidden="true" tabindex="-1"></a>n_not_touch_turq <span class="ot">&lt;-</span> <span class="fu">sum</span>(not_touch_turq_logical)</span>
<span id="cb134-15"><a href="#cb134-15" aria-hidden="true" tabindex="-1"></a>frac_not_touch_turq <span class="ot">&lt;-</span> <span class="fu">mean</span>(not_touch_turq_logical)</span>
<span id="cb134-16"><a href="#cb134-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-17"><a href="#cb134-17" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"DCLs not involving turquoise:"</span>, n_not_touch_turq, <span class="st">"out of"</span>, <span class="fu">nrow</span>(dcls2),</span>
<span id="cb134-18"><a href="#cb134-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sprintf</span>(<span class="st">"(%.1f%%)</span><span class="sc">\n</span><span class="st">"</span>, <span class="dv">100</span><span class="sc">*</span>frac_not_touch_turq))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DCLs not involving turquoise: 140 out of 440 (31.8%)</code></pre>
</div>
</div>
<p>Are the pairs whose correlation changes also strongly connected in the overall WGCNA network?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb136"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb136-1"><a href="#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="co">#prepare for comparison</span></span>
<span id="cb136-2"><a href="#cb136-2" aria-hidden="true" tabindex="-1"></a>TOM_wgcna <span class="ot">&lt;-</span> TOM</span>
<span id="cb136-3"><a href="#cb136-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-4"><a href="#cb136-4" aria-hidden="true" tabindex="-1"></a><span class="co"># add dimnames so you can index by miRNA name</span></span>
<span id="cb136-5"><a href="#cb136-5" aria-hidden="true" tabindex="-1"></a>miRNA_ids <span class="ot">&lt;-</span> <span class="fu">colnames</span>(datExpr)</span>
<span id="cb136-6"><a href="#cb136-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-7"><a href="#cb136-7" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(TOM_wgcna) <span class="ot">&lt;-</span> <span class="fu">list</span>(miRNA_ids, miRNA_ids)</span>
<span id="cb136-8"><a href="#cb136-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-9"><a href="#cb136-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-10"><a href="#cb136-10" aria-hidden="true" tabindex="-1"></a><span class="co">#TOM weight for each DCL edge</span></span>
<span id="cb136-11"><a href="#cb136-11" aria-hidden="true" tabindex="-1"></a>tom_dcl <span class="ot">&lt;-</span> <span class="fu">mapply</span>(<span class="cf">function</span>(a, b) TOM_wgcna[a, b],</span>
<span id="cb136-12"><a href="#cb136-12" aria-hidden="true" tabindex="-1"></a>                  dcls2<span class="sc">$</span>Gene<span class="fl">.1</span>, dcls2<span class="sc">$</span>Gene<span class="fl">.2</span>)</span>
<span id="cb136-13"><a href="#cb136-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-14"><a href="#cb136-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-15"><a href="#cb136-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-16"><a href="#cb136-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare to background distribution of TOM edges (all pairs)</span></span>
<span id="cb136-17"><a href="#cb136-17" aria-hidden="true" tabindex="-1"></a>tom_all <span class="ot">&lt;-</span> TOM_wgcna[<span class="fu">upper.tri</span>(TOM_wgcna)]</span>
<span id="cb136-18"><a href="#cb136-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-19"><a href="#cb136-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-20"><a href="#cb136-20" aria-hidden="true" tabindex="-1"></a><span class="co">#are DCL edges generally higher TOM than average?</span></span>
<span id="cb136-21"><a href="#cb136-21" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(tom_dcl, tom_all)  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  tom_dcl and tom_all
W = 107915153, p-value &lt; 2.2e-16
alternative hypothesis: true location shift is not equal to 0</code></pre>
</div>
<div class="sourceCode cell-code" id="cb138"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb138-1"><a href="#cb138-1" aria-hidden="true" tabindex="-1"></a><span class="fu">median</span>(tom_dcl)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.0006143351</code></pre>
</div>
<div class="sourceCode cell-code" id="cb140"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb140-1"><a href="#cb140-1" aria-hidden="true" tabindex="-1"></a><span class="fu">median</span>(tom_all)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 5.446753e-05</code></pre>
</div>
</div>
<p>NEW METHOD 4: GRAPHICAL LASSO ————————————————–</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb142"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(huge)</span>
<span id="cb142-2"><a href="#cb142-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb142-3"><a href="#cb142-3" aria-hidden="true" tabindex="-1"></a>pheno_graph <span class="ot">&lt;-</span> pheno</span>
<span id="cb142-4"><a href="#cb142-4" aria-hidden="true" tabindex="-1"></a>pheno_graph <span class="ot">&lt;-</span> pheno_graph[<span class="fu">colnames</span>(expr_norm), , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb142-5"><a href="#cb142-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb142-6"><a href="#cb142-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Split samples by condition</span></span>
<span id="cb142-7"><a href="#cb142-7" aria-hidden="true" tabindex="-1"></a>E_N_graph <span class="ot">&lt;-</span> <span class="fu">rownames</span>(pheno)[pheno<span class="sc">$</span>condition <span class="sc">==</span> <span class="st">"normal"</span>]</span>
<span id="cb142-8"><a href="#cb142-8" aria-hidden="true" tabindex="-1"></a>E_T_graph <span class="ot">&lt;-</span> <span class="fu">rownames</span>(pheno)[pheno<span class="sc">$</span>condition <span class="sc">==</span> <span class="st">"tumor"</span>]</span>
<span id="cb142-9"><a href="#cb142-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb142-10"><a href="#cb142-10" aria-hidden="true" tabindex="-1"></a>X_N_graph <span class="ot">&lt;-</span> <span class="fu">t</span>(expr_norm[, E_N_graph, <span class="at">drop =</span> <span class="cn">FALSE</span>])  </span>
<span id="cb142-11"><a href="#cb142-11" aria-hidden="true" tabindex="-1"></a>X_T_graph <span class="ot">&lt;-</span> <span class="fu">t</span>(expr_norm[, E_T_graph, <span class="at">drop =</span> <span class="cn">FALSE</span>])</span>
<span id="cb142-12"><a href="#cb142-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb142-13"><a href="#cb142-13" aria-hidden="true" tabindex="-1"></a><span class="co">#standardize features </span></span>
<span id="cb142-14"><a href="#cb142-14" aria-hidden="true" tabindex="-1"></a>X_N_graph <span class="ot">&lt;-</span> <span class="fu">scale</span>(X_N_graph)</span>
<span id="cb142-15"><a href="#cb142-15" aria-hidden="true" tabindex="-1"></a>X_T_graph <span class="ot">&lt;-</span> <span class="fu">scale</span>(X_T_graph)</span>
<span id="cb142-16"><a href="#cb142-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb142-17"><a href="#cb142-17" aria-hidden="true" tabindex="-1"></a><span class="co">#fit sparse networks with StARS to choose lambda</span></span>
<span id="cb142-18"><a href="#cb142-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb142-19"><a href="#cb142-19" aria-hidden="true" tabindex="-1"></a>fitN <span class="ot">&lt;-</span> <span class="fu">huge</span>(X_N_graph, <span class="at">method =</span> <span class="st">"glasso"</span>, <span class="at">nlambda =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 9%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 19%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 30%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 40%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 50%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 60%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 70%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 80%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 90%
Conducting the graphical lasso (glasso)....done.                                          </code></pre>
</div>
<div class="sourceCode cell-code" id="cb144"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb144-1"><a href="#cb144-1" aria-hidden="true" tabindex="-1"></a>selN <span class="ot">&lt;-</span> <span class="fu">huge.select</span>(fitN, <span class="at">criterion =</span> <span class="st">"stars"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Conducting Subsampling....in progress:5% 
Conducting Subsampling....in progress:10% 
Conducting Subsampling....in progress:15% 
Conducting Subsampling....in progress:20% 
Conducting Subsampling....in progress:25% 
Conducting Subsampling....in progress:30% 
Conducting Subsampling....in progress:35% 
Conducting Subsampling....in progress:40% 
Conducting Subsampling....in progress:45% 
Conducting Subsampling....in progress:50% 
Conducting Subsampling....in progress:55% 
Conducting Subsampling....in progress:60% 
Conducting Subsampling....in progress:65% 
Conducting Subsampling....in progress:70% 
Conducting Subsampling....in progress:75% 
Conducting Subsampling....in progress:80% 
Conducting Subsampling....in progress:85% 
Conducting Subsampling....in progress:90% 
Conducting Subsampling....in progress:95% 
Conducting Subsampling....in progress:100% 
Conducting Subsampling....done.                  </code></pre>
</div>
<div class="sourceCode cell-code" id="cb146"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb146-1"><a href="#cb146-1" aria-hidden="true" tabindex="-1"></a>ThetaN <span class="ot">&lt;-</span> selN<span class="sc">$</span>opt.icov  </span>
<span id="cb146-2"><a href="#cb146-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-3"><a href="#cb146-3" aria-hidden="true" tabindex="-1"></a>fitT <span class="ot">&lt;-</span> <span class="fu">huge</span>(X_T_graph, <span class="at">method =</span> <span class="st">"glasso"</span>, <span class="at">nlambda =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 9%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 19%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 30%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 40%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 50%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 60%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 70%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 80%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 90%
Conducting the graphical lasso (glasso)....done.                                          </code></pre>
</div>
<div class="sourceCode cell-code" id="cb148"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb148-1"><a href="#cb148-1" aria-hidden="true" tabindex="-1"></a>selT <span class="ot">&lt;-</span> <span class="fu">huge.select</span>(fitT, <span class="at">criterion =</span> <span class="st">"stars"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Conducting Subsampling....in progress:5% 
Conducting Subsampling....in progress:10% 
Conducting Subsampling....in progress:15% 
Conducting Subsampling....in progress:20% 
Conducting Subsampling....in progress:25% 
Conducting Subsampling....in progress:30% 
Conducting Subsampling....in progress:35% 
Conducting Subsampling....in progress:40% 
Conducting Subsampling....in progress:45% 
Conducting Subsampling....in progress:50% 
Conducting Subsampling....in progress:55% 
Conducting Subsampling....in progress:60% 
Conducting Subsampling....in progress:65% 
Conducting Subsampling....in progress:70% 
Conducting Subsampling....in progress:75% 
Conducting Subsampling....in progress:80% 
Conducting Subsampling....in progress:85% 
Conducting Subsampling....in progress:90% 
Conducting Subsampling....in progress:95% 
Conducting Subsampling....in progress:100% 
Conducting Subsampling....done.                  </code></pre>
</div>
<div class="sourceCode cell-code" id="cb150"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb150-1"><a href="#cb150-1" aria-hidden="true" tabindex="-1"></a>ThetaT <span class="ot">&lt;-</span> selT<span class="sc">$</span>opt.icov</span>
<span id="cb150-2"><a href="#cb150-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb150-3"><a href="#cb150-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb150-4"><a href="#cb150-4" aria-hidden="true" tabindex="-1"></a>eps <span class="ot">&lt;-</span> <span class="fl">1e-8</span></span>
<span id="cb150-5"><a href="#cb150-5" aria-hidden="true" tabindex="-1"></a>ThetaN <span class="ot">&lt;-</span> (ThetaN <span class="sc">+</span> <span class="fu">t</span>(ThetaN))<span class="sc">/</span><span class="dv">2</span>; ThetaT <span class="ot">&lt;-</span> (ThetaT <span class="sc">+</span> <span class="fu">t</span>(ThetaT))<span class="sc">/</span><span class="dv">2</span></span>
<span id="cb150-6"><a href="#cb150-6" aria-hidden="true" tabindex="-1"></a>AdjN <span class="ot">&lt;-</span> ((<span class="fu">abs</span>(ThetaN) <span class="sc">&gt;</span> eps) <span class="sc">+</span> <span class="fu">t</span>(<span class="fu">abs</span>(ThetaN) <span class="sc">&gt;</span> eps) <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">*</span> <span class="dv">1</span>; <span class="fu">diag</span>(AdjN) <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb150-7"><a href="#cb150-7" aria-hidden="true" tabindex="-1"></a>AdjT <span class="ot">&lt;-</span> ((<span class="fu">abs</span>(ThetaT) <span class="sc">&gt;</span> eps) <span class="sc">+</span> <span class="fu">t</span>(<span class="fu">abs</span>(ThetaT) <span class="sc">&gt;</span> eps) <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">*</span> <span class="dv">1</span>; <span class="fu">diag</span>(AdjT) <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb150-8"><a href="#cb150-8" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ncol</span>(AdjN); EN <span class="ot">&lt;-</span> <span class="fu">sum</span>(AdjN[<span class="fu">upper.tri</span>(AdjN)]); ET <span class="ot">&lt;-</span> <span class="fu">sum</span>(AdjT[<span class="fu">upper.tri</span>(AdjT)])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Parameters for paper</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb151"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb151-1"><a href="#cb151-1" aria-hidden="true" tabindex="-1"></a>prec_to_pcor <span class="ot">&lt;-</span> <span class="cf">function</span>(Theta) {</span>
<span id="cb151-2"><a href="#cb151-2" aria-hidden="true" tabindex="-1"></a>  D <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="fu">diag</span>(Theta))</span>
<span id="cb151-3"><a href="#cb151-3" aria-hidden="true" tabindex="-1"></a>  P <span class="ot">&lt;-</span> <span class="sc">-</span>Theta <span class="sc">/</span> (D <span class="sc">%o%</span> D)</span>
<span id="cb151-4"><a href="#cb151-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">diag</span>(P) <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb151-5"><a href="#cb151-5" aria-hidden="true" tabindex="-1"></a>  P</span>
<span id="cb151-6"><a href="#cb151-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb151-7"><a href="#cb151-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-8"><a href="#cb151-8" aria-hidden="true" tabindex="-1"></a>network_summary <span class="ot">&lt;-</span> <span class="cf">function</span>(Theta, <span class="at">eps =</span> <span class="fl">1e-8</span>) {</span>
<span id="cb151-9"><a href="#cb151-9" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ncol</span>(Theta)</span>
<span id="cb151-10"><a href="#cb151-10" aria-hidden="true" tabindex="-1"></a>  Adj <span class="ot">&lt;-</span> (<span class="fu">abs</span>(Theta) <span class="sc">&gt;</span> eps) <span class="sc">*</span> <span class="dv">1</span></span>
<span id="cb151-11"><a href="#cb151-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">diag</span>(Adj) <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb151-12"><a href="#cb151-12" aria-hidden="true" tabindex="-1"></a>  E <span class="ot">&lt;-</span> <span class="fu">sum</span>(Adj) <span class="sc">/</span> <span class="dv">2</span></span>
<span id="cb151-13"><a href="#cb151-13" aria-hidden="true" tabindex="-1"></a>  density <span class="ot">&lt;-</span> E <span class="sc">/</span> (p <span class="sc">*</span> (p <span class="sc">-</span> <span class="dv">1</span>) <span class="sc">/</span> <span class="dv">2</span>)</span>
<span id="cb151-14"><a href="#cb151-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">p =</span> p, <span class="at">E =</span> E, <span class="at">density =</span> density, <span class="at">Adj =</span> Adj)</span>
<span id="cb151-15"><a href="#cb151-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb151-16"><a href="#cb151-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-17"><a href="#cb151-17" aria-hidden="true" tabindex="-1"></a>sumN <span class="ot">&lt;-</span> <span class="fu">network_summary</span>(ThetaN)</span>
<span id="cb151-18"><a href="#cb151-18" aria-hidden="true" tabindex="-1"></a>sumT <span class="ot">&lt;-</span> <span class="fu">network_summary</span>(ThetaT)</span>
<span id="cb151-19"><a href="#cb151-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-20"><a href="#cb151-20" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Normal edges:"</span>, sumN<span class="sc">$</span>E, <span class="st">" density:"</span>, <span class="fu">round</span>(sumN<span class="sc">$</span>density, <span class="dv">4</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Normal edges: 15460  density: 0.0449 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb153"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb153-1"><a href="#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Tumour edges:"</span>, sumT<span class="sc">$</span>E, <span class="st">" density:"</span>, <span class="fu">round</span>(sumT<span class="sc">$</span>density, <span class="dv">4</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Tumour edges: 15832  density: 0.046 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb155"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb155-1"><a href="#cb155-1" aria-hidden="true" tabindex="-1"></a><span class="co">#gained/lost edges (tumour vs normal)</span></span>
<span id="cb155-2"><a href="#cb155-2" aria-hidden="true" tabindex="-1"></a>AdjN <span class="ot">&lt;-</span> sumN<span class="sc">$</span>Adj</span>
<span id="cb155-3"><a href="#cb155-3" aria-hidden="true" tabindex="-1"></a>AdjT <span class="ot">&lt;-</span> sumT<span class="sc">$</span>Adj</span>
<span id="cb155-4"><a href="#cb155-4" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">colnames</span>(X_N_graph)   </span>
<span id="cb155-5"><a href="#cb155-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-6"><a href="#cb155-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(AdjN) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(AdjN) <span class="ot">&lt;-</span> genes</span>
<span id="cb155-7"><a href="#cb155-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(AdjT) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(AdjT) <span class="ot">&lt;-</span> genes</span>
<span id="cb155-8"><a href="#cb155-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-9"><a href="#cb155-9" aria-hidden="true" tabindex="-1"></a>gain_idx <span class="ot">&lt;-</span> <span class="fu">which</span>(AdjT <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> AdjN <span class="sc">==</span> <span class="dv">0</span>, <span class="at">arr.ind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb155-10"><a href="#cb155-10" aria-hidden="true" tabindex="-1"></a>loss_idx <span class="ot">&lt;-</span> <span class="fu">which</span>(AdjT <span class="sc">==</span> <span class="dv">0</span> <span class="sc">&amp;</span> AdjN <span class="sc">==</span> <span class="dv">1</span>, <span class="at">arr.ind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb155-11"><a href="#cb155-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-12"><a href="#cb155-12" aria-hidden="true" tabindex="-1"></a><span class="co">#keep upper triangle only</span></span>
<span id="cb155-13"><a href="#cb155-13" aria-hidden="true" tabindex="-1"></a>gain_idx <span class="ot">&lt;-</span> gain_idx[gain_idx[,<span class="dv">1</span>] <span class="sc">&lt;</span> gain_idx[,<span class="dv">2</span>], , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb155-14"><a href="#cb155-14" aria-hidden="true" tabindex="-1"></a>loss_idx <span class="ot">&lt;-</span> loss_idx[loss_idx[,<span class="dv">1</span>] <span class="sc">&lt;</span> loss_idx[,<span class="dv">2</span>], , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb155-15"><a href="#cb155-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-16"><a href="#cb155-16" aria-hidden="true" tabindex="-1"></a>gain_edges <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb155-17"><a href="#cb155-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">miRNA1 =</span> <span class="fu">rownames</span>(AdjT)[gain_idx[,<span class="dv">1</span>]],</span>
<span id="cb155-18"><a href="#cb155-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">miRNA2 =</span> <span class="fu">colnames</span>(AdjT)[gain_idx[,<span class="dv">2</span>]],</span>
<span id="cb155-19"><a href="#cb155-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">change =</span> <span class="st">"gained_in_tumour"</span></span>
<span id="cb155-20"><a href="#cb155-20" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb155-21"><a href="#cb155-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-22"><a href="#cb155-22" aria-hidden="true" tabindex="-1"></a>loss_edges <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb155-23"><a href="#cb155-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">miRNA1 =</span> <span class="fu">rownames</span>(AdjN)[loss_idx[,<span class="dv">1</span>]],</span>
<span id="cb155-24"><a href="#cb155-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">miRNA2 =</span> <span class="fu">colnames</span>(AdjN)[loss_idx[,<span class="dv">2</span>]],</span>
<span id="cb155-25"><a href="#cb155-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">change =</span> <span class="st">"lost_in_tumour"</span></span>
<span id="cb155-26"><a href="#cb155-26" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb155-27"><a href="#cb155-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-28"><a href="#cb155-28" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Gained in tumour:"</span>, <span class="fu">nrow</span>(gain_edges), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Gained in tumour: 13996 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb157"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb157-1"><a href="#cb157-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Lost in tumour:"</span>, <span class="fu">nrow</span>(loss_edges), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Lost in tumour: 13624 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb159"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb159-1"><a href="#cb159-1" aria-hidden="true" tabindex="-1"></a><span class="co">#hubness: degree and strength </span></span>
<span id="cb159-2"><a href="#cb159-2" aria-hidden="true" tabindex="-1"></a><span class="co">#degree = number of non-zero neighbours</span></span>
<span id="cb159-3"><a href="#cb159-3" aria-hidden="true" tabindex="-1"></a>degN <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(AdjN)</span>
<span id="cb159-4"><a href="#cb159-4" aria-hidden="true" tabindex="-1"></a>degT <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(AdjT)</span>
<span id="cb159-5"><a href="#cb159-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb159-6"><a href="#cb159-6" aria-hidden="true" tabindex="-1"></a><span class="co">#strength = sum of abs(partial correlations) over neighbours</span></span>
<span id="cb159-7"><a href="#cb159-7" aria-hidden="true" tabindex="-1"></a>PcorN <span class="ot">&lt;-</span> <span class="fu">prec_to_pcor</span>(ThetaN)</span>
<span id="cb159-8"><a href="#cb159-8" aria-hidden="true" tabindex="-1"></a>PcorT <span class="ot">&lt;-</span> <span class="fu">prec_to_pcor</span>(ThetaT)</span>
<span id="cb159-9"><a href="#cb159-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(PcorN) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(PcorN) <span class="ot">&lt;-</span> genes</span>
<span id="cb159-10"><a href="#cb159-10" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(PcorT) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(PcorT) <span class="ot">&lt;-</span> genes</span>
<span id="cb159-11"><a href="#cb159-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb159-12"><a href="#cb159-12" aria-hidden="true" tabindex="-1"></a>strengthN <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">abs</span>(PcorN) <span class="sc">*</span> AdjN)   </span>
<span id="cb159-13"><a href="#cb159-13" aria-hidden="true" tabindex="-1"></a>strengthT <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">abs</span>(PcorT) <span class="sc">*</span> AdjT)</span>
<span id="cb159-14"><a href="#cb159-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb159-15"><a href="#cb159-15" aria-hidden="true" tabindex="-1"></a>hub_df <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb159-16"><a href="#cb159-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">miRNA =</span> genes,</span>
<span id="cb159-17"><a href="#cb159-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">degree_normal =</span> degN,</span>
<span id="cb159-18"><a href="#cb159-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">degree_tumour =</span> degT,</span>
<span id="cb159-19"><a href="#cb159-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">strength_normal =</span> strengthN,</span>
<span id="cb159-20"><a href="#cb159-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">strength_tumour =</span> strengthT,</span>
<span id="cb159-21"><a href="#cb159-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">degree_delta =</span> degT <span class="sc">-</span> degN,</span>
<span id="cb159-22"><a href="#cb159-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">strength_delta =</span> strengthT <span class="sc">-</span> strengthN</span>
<span id="cb159-23"><a href="#cb159-23" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb159-24"><a href="#cb159-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb159-25"><a href="#cb159-25" aria-hidden="true" tabindex="-1"></a><span class="co">#top hubs </span></span>
<span id="cb159-26"><a href="#cb159-26" aria-hidden="true" tabindex="-1"></a>top_n <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb159-27"><a href="#cb159-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb159-28"><a href="#cb159-28" aria-hidden="true" tabindex="-1"></a>top_deg_normal <span class="ot">&lt;-</span> hub_df <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(degree_normal)) <span class="sc">%&gt;%</span> <span class="fu">slice_head</span>(<span class="at">n =</span> top_n)</span>
<span id="cb159-29"><a href="#cb159-29" aria-hidden="true" tabindex="-1"></a>top_deg_tumour <span class="ot">&lt;-</span> hub_df <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(degree_tumour)) <span class="sc">%&gt;%</span> <span class="fu">slice_head</span>(<span class="at">n =</span> top_n)</span>
<span id="cb159-30"><a href="#cb159-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb159-31"><a href="#cb159-31" aria-hidden="true" tabindex="-1"></a>top_str_normal <span class="ot">&lt;-</span> hub_df <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(strength_normal)) <span class="sc">%&gt;%</span> <span class="fu">slice_head</span>(<span class="at">n =</span> top_n)</span>
<span id="cb159-32"><a href="#cb159-32" aria-hidden="true" tabindex="-1"></a>top_str_tumour <span class="ot">&lt;-</span> hub_df <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(strength_tumour)) <span class="sc">%&gt;%</span> <span class="fu">slice_head</span>(<span class="at">n =</span> top_n)</span>
<span id="cb159-33"><a href="#cb159-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb159-34"><a href="#cb159-34" aria-hidden="true" tabindex="-1"></a><span class="co">#hubs that increase most in tumour</span></span>
<span id="cb159-35"><a href="#cb159-35" aria-hidden="true" tabindex="-1"></a>top_degree_gain <span class="ot">&lt;-</span> hub_df <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(degree_delta)) <span class="sc">%&gt;%</span> <span class="fu">slice_head</span>(<span class="at">n =</span> top_n)</span>
<span id="cb159-36"><a href="#cb159-36" aria-hidden="true" tabindex="-1"></a>top_strength_gain <span class="ot">&lt;-</span> hub_df <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(strength_delta)) <span class="sc">%&gt;%</span> <span class="fu">slice_head</span>(<span class="at">n =</span> top_n)</span>
<span id="cb159-37"><a href="#cb159-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb159-38"><a href="#cb159-38" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(top_deg_normal)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 7
  miRNA degree_normal degree_tumour strength_normal strength_tumour degree_delta
  &lt;chr&gt;         &lt;dbl&gt;         &lt;dbl&gt;           &lt;dbl&gt;           &lt;dbl&gt;        &lt;dbl&gt;
1 ILMN…            57            24            1.74            1.04          -33
2 ILMN…            54            46            1.61            1.98           -8
3 ILMN…            54            43            1.58            1.53          -11
4 ILMN…            53            33            1.56            1.19          -20
5 ILMN…            53            41            1.77            1.61          -12
6 ILMN…            53            30            1.56            1.54          -23
# ℹ 1 more variable: strength_delta &lt;dbl&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb161"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb161-1"><a href="#cb161-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(top_deg_tumour)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 7
  miRNA degree_normal degree_tumour strength_normal strength_tumour degree_delta
  &lt;chr&gt;         &lt;dbl&gt;         &lt;dbl&gt;           &lt;dbl&gt;           &lt;dbl&gt;        &lt;dbl&gt;
1 ILMN…            41            60            1.23            1.77           19
2 ILMN…            47            59            1.48            1.71           12
3 ILMN…            43            55            1.30            1.51           12
4 ILMN…            32            55            1.29            1.63           23
5 ILMN…            35            55            1.13            1.47           20
6 ILMN…            41            54            1.52            1.76           13
# ℹ 1 more variable: strength_delta &lt;dbl&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb163"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb163-1"><a href="#cb163-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(top_degree_gain)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 7
  miRNA degree_normal degree_tumour strength_normal strength_tumour degree_delta
  &lt;chr&gt;         &lt;dbl&gt;         &lt;dbl&gt;           &lt;dbl&gt;           &lt;dbl&gt;        &lt;dbl&gt;
1 ILMN…            28            54           1.32             1.60           26
2 ILMN…            17            41           1.01             1.39           24
3 ILMN…            32            55           1.29             1.63           23
4 ILMN…            24            47           0.853            1.41           23
5 ILMN…            20            43           1.07             1.47           23
6 ILMN…            15            37           0.849            1.29           22
# ℹ 1 more variable: strength_delta &lt;dbl&gt;</code></pre>
</div>
</div>
<p>COMPARING GLASSO AND PAIRED WGCNA (gw) —————————————–</p>
<p>Do Glasso edges fall inside WGCNA modules?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb165"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb165-1"><a href="#cb165-1" aria-hidden="true" tabindex="-1"></a><span class="co">#such that there is no variable confusion</span></span>
<span id="cb165-2"><a href="#cb165-2" aria-hidden="true" tabindex="-1"></a>moduleColors_wgcna <span class="ot">&lt;-</span> moduleColors</span>
<span id="cb165-3"><a href="#cb165-3" aria-hidden="true" tabindex="-1"></a>TOM_wgcna <span class="ot">&lt;-</span> TOM</span>
<span id="cb165-4"><a href="#cb165-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb165-5"><a href="#cb165-5" aria-hidden="true" tabindex="-1"></a>miRNA_ids <span class="ot">&lt;-</span> <span class="fu">colnames</span>(datExpr)</span>
<span id="cb165-6"><a href="#cb165-6" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">length</span>(miRNA_ids) <span class="sc">==</span> <span class="fu">nrow</span>(TOM_wgcna))</span>
<span id="cb165-7"><a href="#cb165-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb165-8"><a href="#cb165-8" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(TOM_wgcna) <span class="ot">&lt;-</span> <span class="fu">list</span>(miRNA_ids, miRNA_ids)</span>
<span id="cb165-9"><a href="#cb165-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(moduleColors_wgcna) <span class="ot">&lt;-</span> miRNA_ids</span>
<span id="cb165-10"><a href="#cb165-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb165-11"><a href="#cb165-11" aria-hidden="true" tabindex="-1"></a><span class="co">#correct dimnames </span></span>
<span id="cb165-12"><a href="#cb165-12" aria-hidden="true" tabindex="-1"></a>featN <span class="ot">&lt;-</span> <span class="fu">colnames</span>(X_N_graph)</span>
<span id="cb165-13"><a href="#cb165-13" aria-hidden="true" tabindex="-1"></a>featT <span class="ot">&lt;-</span> <span class="fu">colnames</span>(X_T_graph)</span>
<span id="cb165-14"><a href="#cb165-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb165-15"><a href="#cb165-15" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(ThetaN) <span class="ot">&lt;-</span> <span class="fu">list</span>(featN, featN)</span>
<span id="cb165-16"><a href="#cb165-16" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(ThetaT) <span class="ot">&lt;-</span> <span class="fu">list</span>(featT, featT)</span>
<span id="cb165-17"><a href="#cb165-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb165-18"><a href="#cb165-18" aria-hidden="true" tabindex="-1"></a><span class="co">#common miRNAs between WGCNA and both glasso networks</span></span>
<span id="cb165-19"><a href="#cb165-19" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">names</span>(moduleColors_wgcna)</span>
<span id="cb165-20"><a href="#cb165-20" aria-hidden="true" tabindex="-1"></a>common_glasso <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(intersect, <span class="fu">list</span>(genes, <span class="fu">rownames</span>(ThetaN), <span class="fu">rownames</span>(ThetaT)))</span>
<span id="cb165-21"><a href="#cb165-21" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Common miRNAs used for comparison:"</span>, <span class="fu">length</span>(common_glasso), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Common miRNAs used for comparison: 830 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb167"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb167-1"><a href="#cb167-1" aria-hidden="true" tabindex="-1"></a>mod2 <span class="ot">&lt;-</span> moduleColors_wgcna[common_glasso]  </span>
<span id="cb167-2"><a href="#cb167-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb167-3"><a href="#cb167-3" aria-hidden="true" tabindex="-1"></a><span class="co">#count within-module edges and total edges in a glasso adjacency</span></span>
<span id="cb167-4"><a href="#cb167-4" aria-hidden="true" tabindex="-1"></a>edge_within_counts <span class="ot">&lt;-</span> <span class="cf">function</span>(Theta, mod, common) {</span>
<span id="cb167-5"><a href="#cb167-5" aria-hidden="true" tabindex="-1"></a>  Theta <span class="ot">&lt;-</span> Theta[common, common, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb167-6"><a href="#cb167-6" aria-hidden="true" tabindex="-1"></a>  A <span class="ot">&lt;-</span> <span class="fu">abs</span>(Theta) <span class="sc">&gt;</span> eps</span>
<span id="cb167-7"><a href="#cb167-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">diag</span>(A) <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb167-8"><a href="#cb167-8" aria-hidden="true" tabindex="-1"></a>  idx <span class="ot">&lt;-</span> <span class="fu">which</span>(A <span class="sc">&amp;</span> <span class="fu">upper.tri</span>(A), <span class="at">arr.ind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb167-9"><a href="#cb167-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">nrow</span>(idx) <span class="sc">==</span> <span class="dv">0</span>) <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">m=</span><span class="dv">0</span>L, <span class="at">x=</span><span class="dv">0</span>L, <span class="at">frac=</span><span class="cn">NA_real_</span>))</span>
<span id="cb167-10"><a href="#cb167-10" aria-hidden="true" tabindex="-1"></a>  within <span class="ot">&lt;-</span> mod[idx[,<span class="dv">1</span>]] <span class="sc">==</span> mod[idx[,<span class="dv">2</span>]]</span>
<span id="cb167-11"><a href="#cb167-11" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> <span class="fu">sum</span>(within)        </span>
<span id="cb167-12"><a href="#cb167-12" aria-hidden="true" tabindex="-1"></a>  m <span class="ot">&lt;-</span> <span class="fu">length</span>(within)     </span>
<span id="cb167-13"><a href="#cb167-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">m=</span>m, <span class="at">x=</span>x, <span class="at">frac=</span>x<span class="sc">/</span>m)</span>
<span id="cb167-14"><a href="#cb167-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb167-15"><a href="#cb167-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb167-16"><a href="#cb167-16" aria-hidden="true" tabindex="-1"></a><span class="co">#observed within fractions for normal/tumour glasso networks</span></span>
<span id="cb167-17"><a href="#cb167-17" aria-hidden="true" tabindex="-1"></a>N_counts <span class="ot">&lt;-</span> <span class="fu">edge_within_counts</span>(ThetaN, mod2, common_glasso)</span>
<span id="cb167-18"><a href="#cb167-18" aria-hidden="true" tabindex="-1"></a>T_counts <span class="ot">&lt;-</span> <span class="fu">edge_within_counts</span>(ThetaT, mod2, common_glasso)</span>
<span id="cb167-19"><a href="#cb167-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb167-20"><a href="#cb167-20" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Within-edge fraction (normal):"</span>, N_counts<span class="sc">$</span>frac, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Within-edge fraction (normal): 0.4313713 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb169"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb169-1"><a href="#cb169-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Within-edge fraction (tumour):"</span>, T_counts<span class="sc">$</span>frac, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Within-edge fraction (tumour): 0.4190248 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb171"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb171-1"><a href="#cb171-1" aria-hidden="true" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">length</span>(common_glasso)</span>
<span id="cb171-2"><a href="#cb171-2" aria-hidden="true" tabindex="-1"></a>Npairs <span class="ot">&lt;-</span> <span class="fu">choose</span>(n, <span class="dv">2</span>)</span>
<span id="cb171-3"><a href="#cb171-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb171-4"><a href="#cb171-4" aria-hidden="true" tabindex="-1"></a>W <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">sapply</span>(<span class="fu">split</span>(mod2, mod2), <span class="cf">function</span>(x) <span class="fu">choose</span>(<span class="fu">length</span>(x), <span class="dv">2</span>))) </span>
<span id="cb171-5"><a href="#cb171-5" aria-hidden="true" tabindex="-1"></a>expected_within <span class="ot">&lt;-</span> W <span class="sc">/</span> Npairs</span>
<span id="cb171-6"><a href="#cb171-6" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Expected within-edge fraction under random pairs:"</span>, expected_within, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Expected within-edge fraction under random pairs: 0.2844652 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb173"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb173-1"><a href="#cb173-1" aria-hidden="true" tabindex="-1"></a>p_enrich_N <span class="ot">&lt;-</span> <span class="fu">phyper</span>(N_counts<span class="sc">$</span>x <span class="sc">-</span> <span class="dv">1</span>, W, Npairs <span class="sc">-</span> W, N_counts<span class="sc">$</span>m, <span class="at">lower.tail =</span> <span class="cn">FALSE</span>)</span>
<span id="cb173-2"><a href="#cb173-2" aria-hidden="true" tabindex="-1"></a>p_enrich_T <span class="ot">&lt;-</span> <span class="fu">phyper</span>(T_counts<span class="sc">$</span>x <span class="sc">-</span> <span class="dv">1</span>, W, Npairs <span class="sc">-</span> W, T_counts<span class="sc">$</span>m, <span class="at">lower.tail =</span> <span class="cn">FALSE</span>)</span>
<span id="cb173-3"><a href="#cb173-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb173-4"><a href="#cb173-4" aria-hidden="true" tabindex="-1"></a><span class="co">#normal vs tumour difference in within/between proportions </span></span>
<span id="cb173-5"><a href="#cb173-5" aria-hidden="true" tabindex="-1"></a>mat <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">c</span>(N_counts<span class="sc">$</span>x, N_counts<span class="sc">$</span>m <span class="sc">-</span> N_counts<span class="sc">$</span>x,</span>
<span id="cb173-6"><a href="#cb173-6" aria-hidden="true" tabindex="-1"></a>                T_counts<span class="sc">$</span>x, T_counts<span class="sc">$</span>m <span class="sc">-</span> T_counts<span class="sc">$</span>x),</span>
<span id="cb173-7"><a href="#cb173-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">nrow=</span><span class="dv">2</span>, <span class="at">byrow=</span><span class="cn">TRUE</span>,</span>
<span id="cb173-8"><a href="#cb173-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">dimnames=</span><span class="fu">list</span>(<span class="fu">c</span>(<span class="st">"normal"</span>,<span class="st">"tumour"</span>), <span class="fu">c</span>(<span class="st">"within"</span>,<span class="st">"between"</span>)))</span>
<span id="cb173-9"><a href="#cb173-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb173-10"><a href="#cb173-10" aria-hidden="true" tabindex="-1"></a>test_diff <span class="ot">&lt;-</span> <span class="fu">fisher.test</span>(mat)</span>
<span id="cb173-11"><a href="#cb173-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb173-12"><a href="#cb173-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb173-13"><a href="#cb173-13" aria-hidden="true" tabindex="-1"></a>res_glasso_wgcna <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb173-14"><a href="#cb173-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">condition =</span> <span class="fu">c</span>(<span class="st">"normal"</span>,<span class="st">"tumour"</span>),</span>
<span id="cb173-15"><a href="#cb173-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">total_edges_m =</span> <span class="fu">c</span>(N_counts<span class="sc">$</span>m, T_counts<span class="sc">$</span>m),</span>
<span id="cb173-16"><a href="#cb173-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">within_edges_x =</span> <span class="fu">c</span>(N_counts<span class="sc">$</span>x, T_counts<span class="sc">$</span>x),</span>
<span id="cb173-17"><a href="#cb173-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">within_fraction =</span> <span class="fu">c</span>(N_counts<span class="sc">$</span>frac, T_counts<span class="sc">$</span>frac),</span>
<span id="cb173-18"><a href="#cb173-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">expected_fraction =</span> expected_within,</span>
<span id="cb173-19"><a href="#cb173-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">enrichment =</span> <span class="fu">c</span>(N_counts<span class="sc">$</span>frac, T_counts<span class="sc">$</span>frac) <span class="sc">/</span> expected_within,</span>
<span id="cb173-20"><a href="#cb173-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">p_value_enrichment =</span> <span class="fu">c</span>(p_enrich_N, p_enrich_T)</span>
<span id="cb173-21"><a href="#cb173-21" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb173-22"><a href="#cb173-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb173-23"><a href="#cb173-23" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(res_glasso_wgcna)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  condition total_edges_m within_edges_x within_fraction expected_fraction
1    normal         15460           6669       0.4313713         0.2844652
2    tumour         15832           6634       0.4190248         0.2844652
  enrichment p_value_enrichment
1   1.516429       0.000000e+00
2   1.473026      3.936861e-301</code></pre>
</div>
<div class="sourceCode cell-code" id="cb175"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb175-1"><a href="#cb175-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">2x2 Fisher test (normal vs tumour within/between proportions):</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
2x2 Fisher test (normal vs tumour within/between proportions):</code></pre>
</div>
<div class="sourceCode cell-code" id="cb177"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb177-1"><a href="#cb177-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(test_diff)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Fisher's Exact Test for Count Data

data:  mat
p-value = 0.02731
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 1.005437 1.100313
sample estimates:
odds ratio 
  1.051784 </code></pre>
</div>
</div>
<p>Which WGCNA modules gain/lose conditional dependence edges in tumour?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb179"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb179-1"><a href="#cb179-1" aria-hidden="true" tabindex="-1"></a><span class="co">#make sure we got right names</span></span>
<span id="cb179-2"><a href="#cb179-2" aria-hidden="true" tabindex="-1"></a>featN <span class="ot">&lt;-</span> <span class="fu">colnames</span>(X_N_graph)</span>
<span id="cb179-3"><a href="#cb179-3" aria-hidden="true" tabindex="-1"></a>featT <span class="ot">&lt;-</span> <span class="fu">colnames</span>(X_T_graph)</span>
<span id="cb179-4"><a href="#cb179-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(ThetaN) <span class="ot">&lt;-</span> <span class="fu">list</span>(featN, featN)</span>
<span id="cb179-5"><a href="#cb179-5" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(ThetaT) <span class="ot">&lt;-</span> <span class="fu">list</span>(featT, featT)</span>
<span id="cb179-6"><a href="#cb179-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-7"><a href="#cb179-7" aria-hidden="true" tabindex="-1"></a><span class="co">#define common miRNAs AFTER dimnames exist</span></span>
<span id="cb179-8"><a href="#cb179-8" aria-hidden="true" tabindex="-1"></a>common_glasso <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(intersect, <span class="fu">list</span>(</span>
<span id="cb179-9"><a href="#cb179-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(moduleColors_wgcna),</span>
<span id="cb179-10"><a href="#cb179-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(ThetaN),</span>
<span id="cb179-11"><a href="#cb179-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(ThetaT)</span>
<span id="cb179-12"><a href="#cb179-12" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb179-13"><a href="#cb179-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-14"><a href="#cb179-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-15"><a href="#cb179-15" aria-hidden="true" tabindex="-1"></a><span class="co">#module labels</span></span>
<span id="cb179-16"><a href="#cb179-16" aria-hidden="true" tabindex="-1"></a>mod2 <span class="ot">&lt;-</span> moduleColors_wgcna[common_glasso]</span>
<span id="cb179-17"><a href="#cb179-17" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(mod2) <span class="ot">&lt;-</span> common_glasso</span>
<span id="cb179-18"><a href="#cb179-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-19"><a href="#cb179-19" aria-hidden="true" tabindex="-1"></a>count_edges_between_modules <span class="ot">&lt;-</span> <span class="cf">function</span>(Theta, mod, common, <span class="at">eps =</span> <span class="fl">1e-8</span>) {</span>
<span id="cb179-20"><a href="#cb179-20" aria-hidden="true" tabindex="-1"></a>  Th <span class="ot">&lt;-</span> Theta[common, common, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb179-21"><a href="#cb179-21" aria-hidden="true" tabindex="-1"></a>  A <span class="ot">&lt;-</span> <span class="fu">abs</span>(Th) <span class="sc">&gt;</span> eps</span>
<span id="cb179-22"><a href="#cb179-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">diag</span>(A) <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb179-23"><a href="#cb179-23" aria-hidden="true" tabindex="-1"></a>  idx <span class="ot">&lt;-</span> <span class="fu">which</span>(A <span class="sc">&amp;</span> <span class="fu">upper.tri</span>(A), <span class="at">arr.ind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb179-24"><a href="#cb179-24" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">nrow</span>(idx) <span class="sc">==</span> <span class="dv">0</span>) <span class="fu">return</span>(<span class="fu">table</span>())  <span class="co"># empty</span></span>
<span id="cb179-25"><a href="#cb179-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-26"><a href="#cb179-26" aria-hidden="true" tabindex="-1"></a>  g1 <span class="ot">&lt;-</span> <span class="fu">rownames</span>(Th)[idx[,<span class="dv">1</span>]]</span>
<span id="cb179-27"><a href="#cb179-27" aria-hidden="true" tabindex="-1"></a>  g2 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(Th)[idx[,<span class="dv">2</span>]]</span>
<span id="cb179-28"><a href="#cb179-28" aria-hidden="true" tabindex="-1"></a>  m1 <span class="ot">&lt;-</span> mod[g1]</span>
<span id="cb179-29"><a href="#cb179-29" aria-hidden="true" tabindex="-1"></a>  m2 <span class="ot">&lt;-</span> mod[g2]</span>
<span id="cb179-30"><a href="#cb179-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">table</span>(<span class="fu">pmin</span>(m1, m2), <span class="fu">pmax</span>(m1, m2))</span>
<span id="cb179-31"><a href="#cb179-31" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb179-32"><a href="#cb179-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-33"><a href="#cb179-33" aria-hidden="true" tabindex="-1"></a>tabN <span class="ot">&lt;-</span> <span class="fu">count_edges_between_modules</span>(ThetaN, mod2, common_glasso, <span class="at">eps =</span> eps)</span>
<span id="cb179-34"><a href="#cb179-34" aria-hidden="true" tabindex="-1"></a>tabT <span class="ot">&lt;-</span> <span class="fu">count_edges_between_modules</span>(ThetaT, mod2, common_glasso, <span class="at">eps =</span> eps)</span>
<span id="cb179-35"><a href="#cb179-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-36"><a href="#cb179-36" aria-hidden="true" tabindex="-1"></a>tabN</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           
            blue brown grey turquoise
  blue      1760   436 1389      2149
  brown        0   655  711       857
  grey         0     0 1675      3249
  turquoise    0     0    0      2579</code></pre>
</div>
<div class="sourceCode cell-code" id="cb181"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb181-1"><a href="#cb181-1" aria-hidden="true" tabindex="-1"></a>tabT</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           
            blue brown grey turquoise
  blue      1680   538 1595      2131
  brown        0   609  709       876
  grey         0     0 1655      3349
  turquoise    0     0    0      2690</code></pre>
</div>
</div>
<p>WGCNA kWithin vs glasso degree</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb183"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb183-1"><a href="#cb183-1" aria-hidden="true" tabindex="-1"></a>hubData2 <span class="ot">&lt;-</span> hubData[hubData<span class="sc">$</span>module <span class="sc">!=</span> <span class="st">"grey"</span>,]</span>
<span id="cb183-2"><a href="#cb183-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-3"><a href="#cb183-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-4"><a href="#cb183-4" aria-hidden="true" tabindex="-1"></a>kW <span class="ot">&lt;-</span> hubData2<span class="sc">$</span>kWithin</span>
<span id="cb183-5"><a href="#cb183-5" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(kW) <span class="ot">&lt;-</span> hubData2<span class="sc">$</span>miRNA</span>
<span id="cb183-6"><a href="#cb183-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-7"><a href="#cb183-7" aria-hidden="true" tabindex="-1"></a>common_glasso <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">names</span>(degN), <span class="fu">names</span>(kW))</span>
<span id="cb183-8"><a href="#cb183-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-9"><a href="#cb183-9" aria-hidden="true" tabindex="-1"></a><span class="co"># adjacency with proper dimnames</span></span>
<span id="cb183-10"><a href="#cb183-10" aria-hidden="true" tabindex="-1"></a>A_N <span class="ot">&lt;-</span> <span class="fu">abs</span>(ThetaN[common_glasso, common_glasso]) <span class="sc">&gt;</span> eps; <span class="fu">diag</span>(A_N) <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb183-11"><a href="#cb183-11" aria-hidden="true" tabindex="-1"></a>A_T <span class="ot">&lt;-</span> <span class="fu">abs</span>(ThetaT[common_glasso, common_glasso]) <span class="sc">&gt;</span> eps; <span class="fu">diag</span>(A_T) <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb183-12"><a href="#cb183-12" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(A_N) <span class="ot">&lt;-</span> <span class="fu">list</span>(common_glasso, common_glasso)</span>
<span id="cb183-13"><a href="#cb183-13" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(A_T) <span class="ot">&lt;-</span> <span class="fu">list</span>(common_glasso, common_glasso)</span>
<span id="cb183-14"><a href="#cb183-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-15"><a href="#cb183-15" aria-hidden="true" tabindex="-1"></a>degN <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">degree</span>(igraph<span class="sc">::</span><span class="fu">graph_from_adjacency_matrix</span>(A_N, <span class="at">mode=</span><span class="st">"undirected"</span>, <span class="at">diag=</span><span class="cn">FALSE</span>))</span>
<span id="cb183-16"><a href="#cb183-16" aria-hidden="true" tabindex="-1"></a>degT <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">degree</span>(igraph<span class="sc">::</span><span class="fu">graph_from_adjacency_matrix</span>(A_T, <span class="at">mode=</span><span class="st">"undirected"</span>, <span class="at">diag=</span><span class="cn">FALSE</span>))</span>
<span id="cb183-17"><a href="#cb183-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-18"><a href="#cb183-18" aria-hidden="true" tabindex="-1"></a><span class="co">#head(degN)</span></span>
<span id="cb183-19"><a href="#cb183-19" aria-hidden="true" tabindex="-1"></a><span class="co">#head(kW)</span></span>
<span id="cb183-20"><a href="#cb183-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-21"><a href="#cb183-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-22"><a href="#cb183-22" aria-hidden="true" tabindex="-1"></a><span class="co">#compute Spearman on aligned vectors</span></span>
<span id="cb183-23"><a href="#cb183-23" aria-hidden="true" tabindex="-1"></a>cor_glasso_normal <span class="ot">&lt;-</span> <span class="fu">cor</span>(degN[common_glasso], kW[common_glasso], <span class="at">method =</span> <span class="st">"spearman"</span>, <span class="at">use =</span> <span class="st">"complete.obs"</span>)</span>
<span id="cb183-24"><a href="#cb183-24" aria-hidden="true" tabindex="-1"></a>cor_glasso_tumour <span class="ot">&lt;-</span> <span class="fu">cor</span>(degT[common_glasso], kW[common_glasso], <span class="at">method =</span> <span class="st">"spearman"</span>, <span class="at">use =</span> <span class="st">"complete.obs"</span>)</span>
<span id="cb183-25"><a href="#cb183-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-26"><a href="#cb183-26" aria-hidden="true" tabindex="-1"></a>cor_glasso_normal</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.4207676</code></pre>
</div>
<div class="sourceCode cell-code" id="cb185"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb185-1"><a href="#cb185-1" aria-hidden="true" tabindex="-1"></a>cor_glasso_tumour</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.4150968</code></pre>
</div>
<div class="sourceCode cell-code" id="cb187"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb187-1"><a href="#cb187-1" aria-hidden="true" tabindex="-1"></a>topk <span class="ot">&lt;-</span> <span class="dv">100</span></span>
<span id="cb187-2"><a href="#cb187-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-3"><a href="#cb187-3" aria-hidden="true" tabindex="-1"></a>top_gl_N <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">sort</span>(degN, <span class="at">decreasing =</span> <span class="cn">TRUE</span>))[<span class="dv">1</span><span class="sc">:</span>topk]</span>
<span id="cb187-4"><a href="#cb187-4" aria-hidden="true" tabindex="-1"></a>top_gl_T <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">sort</span>(degT, <span class="at">decreasing =</span> <span class="cn">TRUE</span>))[<span class="dv">1</span><span class="sc">:</span>topk]</span>
<span id="cb187-5"><a href="#cb187-5" aria-hidden="true" tabindex="-1"></a>top_wgc  <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">sort</span>(kW,  <span class="at">decreasing =</span> <span class="cn">TRUE</span>))[<span class="dv">1</span><span class="sc">:</span>topk]</span>
<span id="cb187-6"><a href="#cb187-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-7"><a href="#cb187-7" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="at">overlap_normal =</span> <span class="fu">length</span>(<span class="fu">intersect</span>(top_gl_N, top_wgc)), <span class="at">overlap_tumour =</span> <span class="fu">length</span>(<span class="fu">intersect</span>(top_gl_T, top_wgc)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>overlap_normal overlap_tumour 
            38             38 </code></pre>
</div>
</div>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>