<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2025-12-23">

<title>SD_WGCNA_exactly</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="SD_WGCNA_comp_all_files/libs/clipboard/clipboard.min.js"></script>
<script src="SD_WGCNA_comp_all_files/libs/quarto-html/quarto.js" type="module"></script>
<script src="SD_WGCNA_comp_all_files/libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="SD_WGCNA_comp_all_files/libs/quarto-html/popper.min.js"></script>
<script src="SD_WGCNA_comp_all_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="SD_WGCNA_comp_all_files/libs/quarto-html/anchor.min.js"></script>
<link href="SD_WGCNA_comp_all_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="SD_WGCNA_comp_all_files/libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="SD_WGCNA_comp_all_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="SD_WGCNA_comp_all_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="SD_WGCNA_comp_all_files/libs/bootstrap/bootstrap-81267100e462c21b3d6c0d5bf76a3417.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent quarto-light">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">SD_WGCNA_exactly</h1>
</div>



<div class="quarto-title-meta">

    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">2025-12-23</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Load cleaned dataset + safety stuff</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(WGCNA)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(nlme)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">disableWGCNAThreads</span>()  <span class="co"># reproducibility</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Point to your saved file</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>infile <span class="ot">&lt;-</span> <span class="st">"GSE62043_clean_noOutliers_20251223.RData"</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(infile)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Expect: expr_gene (genes x samples), sample_info, datExpr0 (samples x genes) optional,</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="co"># and sample_traits already saved.</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">exists</span>(<span class="st">"expr_gene"</span>), <span class="fu">exists</span>(<span class="st">"sample_info"</span>))</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">exists</span>(<span class="st">"datExpr0"</span>)) datExpr0 <span class="ot">&lt;-</span> <span class="fu">t</span>(expr_gene)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">rownames</span>(datExpr0), <span class="fu">rownames</span>(sample_info)))</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Build/confirm traits: patient_id, tumour, Age, Gender (no Stage_num)</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">exists</span>(<span class="st">"sample_traits"</span>)) {</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>  sample_traits <span class="ot">&lt;-</span> sample_info[, <span class="fu">intersect</span>(<span class="fu">c</span>(<span class="st">"patient_id"</span>,<span class="st">"tumour"</span>,<span class="st">"Age"</span>,<span class="st">"Gender"</span>),</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>                                           <span class="fu">colnames</span>(sample_info)), drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Coerce types safely</span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>sample_traits<span class="sc">$</span>patient_id <span class="ot">&lt;-</span> <span class="fu">factor</span>(sample_traits<span class="sc">$</span>patient_id)</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>sample_traits<span class="sc">$</span>tumour     <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(sample_traits<span class="sc">$</span>tumour)</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"Age"</span> <span class="sc">%in%</span> <span class="fu">names</span>(sample_traits))    sample_traits<span class="sc">$</span>Age <span class="ot">&lt;-</span> <span class="fu">suppressWarnings</span>(<span class="fu">as.numeric</span>(sample_traits<span class="sc">$</span>Age))</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"Gender"</span> <span class="sc">%in%</span> <span class="fu">names</span>(sample_traits)) {</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>  g <span class="ot">&lt;-</span> <span class="fu">toupper</span>(<span class="fu">trimws</span>(<span class="fu">as.character</span>(sample_traits<span class="sc">$</span>Gender)))</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>  g[<span class="sc">!</span>g <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"F"</span>,<span class="st">"M"</span>)] <span class="ot">&lt;-</span> <span class="cn">NA_character_</span></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>  sample_traits<span class="sc">$</span>Gender <span class="ot">&lt;-</span> <span class="fu">factor</span>(g, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"F"</span>,<span class="st">"M"</span>))</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sample_traits) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(sample_info)</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">rownames</span>(datExpr0), <span class="fu">rownames</span>(sample_traits)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># WGCNA: Choosing soft-threshold (unsigned)</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>gsg <span class="ot">&lt;-</span> <span class="fu">goodSamplesGenes</span>(datExpr0, <span class="at">verbose =</span> <span class="dv">3</span>); <span class="fu">stopifnot</span>(gsg<span class="sc">$</span>allOK)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Flagging genes and samples with too many missing values...
  ..step 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>nGenesForSFT <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="dv">5000</span>, <span class="fu">ncol</span>(datExpr0))</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>vars <span class="ot">&lt;-</span> <span class="fu">apply</span>(datExpr0, <span class="dv">2</span>, var)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>topGenesSFT <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">sort</span>(vars, <span class="at">decreasing =</span> <span class="cn">TRUE</span>))[<span class="dv">1</span><span class="sc">:</span>nGenesForSFT]</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>datExpr_sft <span class="ot">&lt;-</span> datExpr0[, topGenesSFT, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>powers <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>sft <span class="ot">&lt;-</span> <span class="fu">pickSoftThreshold</span>(datExpr_sft, <span class="at">powerVector =</span> powers,</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">networkType =</span> <span class="st">"unsigned"</span>, <span class="at">verbose =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>pickSoftThreshold: will use block size 5000.
 pickSoftThreshold: calculating connectivity for given powers...
   ..working on genes 1 through 5000 of 5000
   Power SFT.R.sq  slope truncated.R.sq  mean.k. median.k. max.k.
1      1    0.132  0.682          0.944 1050.000  1.04e+03 1910.0
2      2    0.425 -0.858          0.912  352.000  3.10e+02  976.0
3      3    0.706 -1.270          0.938  148.000  1.11e+02  569.0
4      4    0.775 -1.450          0.944   72.200  4.57e+01  359.0
5      5    0.800 -1.500          0.946   39.100  2.08e+01  240.0
6      6    0.836 -1.490          0.957   22.900  1.01e+01  167.0
7      7    0.831 -1.520          0.938   14.300  5.07e+00  120.0
8      8    0.894 -1.450          0.975    9.340  2.73e+00   88.7
9      9    0.928 -1.390          0.990    6.380  1.54e+00   67.1
10    10    0.952 -1.340          0.996    4.520  9.07e-01   51.7
11    11    0.960 -1.340          0.992    3.300  5.47e-01   41.6
12    12    0.956 -1.350          0.994    2.470  3.27e-01   34.7
13    13    0.952 -1.360          0.993    1.900  2.07e-01   29.5
14    14    0.920 -1.400          0.961    1.490  1.31e-01   25.4
15    15    0.937 -1.390          0.984    1.190  8.40e-02   22.1
16    16    0.935 -1.400          0.989    0.961  5.38e-02   19.4
17    17    0.921 -1.420          0.979    0.790  3.47e-02   17.2
18    18    0.921 -1.420          0.983    0.658  2.32e-02   15.3
19    19    0.909 -1.430          0.978    0.554  1.51e-02   13.7
20    20    0.908 -1.410          0.978    0.471  9.89e-03   12.3</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>     <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">3</span>]) <span class="sc">*</span> sft<span class="sc">$</span>fitIndices[,<span class="dv">2</span>],</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"Soft Threshold (power)"</span>,</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">"Scale Free Topology Model Fit, signed R^2"</span>,</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">type =</span> <span class="st">"n"</span>, <span class="at">main =</span> <span class="st">"Scale independence"</span>)</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>     <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">3</span>]) <span class="sc">*</span> sft<span class="sc">$</span>fitIndices[,<span class="dv">2</span>],</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels =</span> powers, <span class="at">cex =</span> <span class="fl">0.8</span>, <span class="at">col =</span> <span class="st">"red"</span>)</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="fl">0.90</span>, <span class="at">col =</span> <span class="st">"red"</span>)</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], sft<span class="sc">$</span>fitIndices[,<span class="dv">5</span>],</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"Soft Threshold (power)"</span>,</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">"Mean Connectivity"</span>,</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>     <span class="at">type =</span> <span class="st">"n"</span>, <span class="at">main =</span> <span class="st">"Mean connectivity"</span>)</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], sft<span class="sc">$</span>fitIndices[,<span class="dv">5</span>], <span class="at">labels =</span> powers, <span class="at">cex =</span> <span class="fl">0.8</span>, <span class="at">col =</span> <span class="st">"red"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SD_WGCNA_comp_all_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">3</span>]) <span class="sc">*</span> sft<span class="sc">$</span>fitIndices[,<span class="dv">2</span>]</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>softPower <span class="ot">&lt;-</span> <span class="cf">if</span> (<span class="fu">any</span>(fit <span class="sc">&gt;=</span> <span class="fl">0.90</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) powers[<span class="fu">which</span>(fit <span class="sc">&gt;=</span> <span class="fl">0.90</span>)[<span class="dv">1</span>]] <span class="cf">else</span> <span class="dv">7</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>softPower <span class="co"># 9</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 9</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Network construction (unsigned Pearson)</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>maxBlockSize <span class="ot">&lt;-</span> <span class="dv">12000</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>net <span class="ot">&lt;-</span> <span class="fu">blockwiseModules</span>(</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  datExpr0,</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">power              =</span> softPower,</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">networkType        =</span> <span class="st">"unsigned"</span>,</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">TOMType            =</span> <span class="st">"unsigned"</span>,</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">corType            =</span> <span class="st">"pearson"</span>,</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">maxBlockSize       =</span> maxBlockSize,</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">minModuleSize      =</span> <span class="dv">70</span>,</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">reassignThreshold  =</span> <span class="dv">0</span>,</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">mergeCutHeight     =</span> <span class="fl">0.25</span>,</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">deepSplit          =</span> <span class="dv">2</span>,</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">pamRespectsDendro  =</span> <span class="cn">FALSE</span>,</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">numericLabels      =</span> <span class="cn">FALSE</span>,</span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">saveTOMs           =</span> <span class="cn">TRUE</span>,</span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">saveTOMFileBase    =</span> <span class="fu">sprintf</span>(<span class="st">"GSE62043_TOM_unsigned_power%i"</span>, softPower),</span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose            =</span> <span class="dv">3</span></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Calculating module eigengenes block-wise from all genes
   Flagging genes and samples with too many missing values...
    ..step 1
 ....pre-clustering genes to determine blocks..
   Projective K-means:
   ..k-means clustering..
   ..merging smaller clusters...
Block sizes:
gBlocks
    1     2 
11988  7607 
 ..Working on block 1 .
    TOM calculation: adjacency..
    ..will not use multithreading.
     Fraction of slow calculations: 0.000000
    ..connectivity..
    ..matrix multiplication (system BLAS)..
    ..normalization..
    ..done.
   ..saving TOM for block 1 into file GSE62043_TOM_unsigned_power9-block.1.RData
 ....clustering..
 ....detecting modules..
 ....calculating module eigengenes..
 ....checking kME in modules..
     ..removing 1 genes from module 1 because their KME is too low.
     ..removing 9 genes from module 4 because their KME is too low.
     ..removing 1 genes from module 8 because their KME is too low.
 ..Working on block 2 .
    TOM calculation: adjacency..
    ..will not use multithreading.
     Fraction of slow calculations: 0.000000
    ..connectivity..
    ..matrix multiplication (system BLAS)..
    ..normalization..
    ..done.
   ..saving TOM for block 2 into file GSE62043_TOM_unsigned_power9-block.2.RData
 ....clustering..
 ....detecting modules..
 ....calculating module eigengenes..
 ....checking kME in modules..
     ..removing 21 genes from module 3 because their KME is too low.
 ..merging modules that are too close..
     mergeCloseModules: Merging modules whose distance is less than 0.25
       Calculating new MEs...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>moduleColors <span class="ot">&lt;-</span> net<span class="sc">$</span>colors</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(moduleColors) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(datExpr0)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>MEs <span class="ot">&lt;-</span> <span class="fu">orderMEs</span>(net<span class="sc">$</span>MEs)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>MEs <span class="ot">&lt;-</span> MEs[, <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"^MEgrey$"</span>, <span class="fu">names</span>(MEs)), drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(moduleColors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>moduleColors
      black        blue       brown        cyan       green greenyellow 
        323        1648        1312          95        1029         131 
       grey     magenta        pink      purple         red      salmon 
       9247         172         198         160         493          97 
        tan   turquoise      yellow 
        104        3371        1215 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Li-style LMMs for module eigengenes:</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="co">#   ME ~ tumour + Age + Gender + (1|patient_id)</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   (paired subjects only; drop if missing covars)</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>ctrl_lme <span class="ot">&lt;-</span> nlme<span class="sc">::</span><span class="fu">lmeControl</span>(</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">singular.ok =</span> <span class="cn">TRUE</span>, <span class="at">returnObject =</span> <span class="cn">TRUE</span>,</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">opt =</span> <span class="st">"optim"</span>, <span class="at">msMaxIter =</span> <span class="dv">200</span>, <span class="at">maxIter =</span> <span class="dv">200</span>, <span class="at">niterEM =</span> <span class="dv">50</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">rownames</span>(MEs), <span class="fu">rownames</span>(sample_traits)))</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>datLMM <span class="ot">&lt;-</span> <span class="fu">cbind</span>(sample_traits, <span class="fu">as.data.frame</span>(MEs))</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a><span class="co"># covariates are fixed: Age + Gender (drop if all-NA)</span></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>covars <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Age"</span>,<span class="st">"Gender"</span>)</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>covars <span class="ot">&lt;-</span> covars[<span class="sc">!</span><span class="fu">sapply</span>(datLMM[covars], <span class="cf">function</span>(v) <span class="fu">all</span>(<span class="fu">is.na</span>(v)))]</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>rhs <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">c</span>(<span class="st">"tumour"</span>, covars), <span class="at">collapse =</span> <span class="st">" + "</span>)</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>modNames <span class="ot">&lt;-</span> <span class="fu">colnames</span>(MEs)</span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>lmm_results <span class="ot">&lt;-</span> <span class="fu">lapply</span>(modNames, <span class="cf">function</span>(mn) {</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Keep only rows with complete data for this module + covars</span></span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>  cols_needed <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"patient_id"</span>, <span class="st">"tumour"</span>, covars, mn)</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>  d <span class="ot">&lt;-</span> datLMM[, cols_needed, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>  d <span class="ot">&lt;-</span> d[<span class="fu">complete.cases</span>(d), , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Now enforce true pairing on the rows that will actually be modeled</span></span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a>  tabTN <span class="ot">&lt;-</span> <span class="fu">with</span>(d, <span class="fu">table</span>(patient_id, tumour))</span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>  keep_pat <span class="ot">&lt;-</span> <span class="fu">rownames</span>(tabTN)[<span class="fu">rowSums</span>(tabTN <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">==</span> <span class="dv">2</span>]</span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a>  d <span class="ot">&lt;-</span> d[d<span class="sc">$</span>patient_id <span class="sc">%in%</span> keep_pat, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># If nothing left, return NAs</span></span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">nrow</span>(d) <span class="sc">&lt;</span> <span class="dv">4</span>) {</span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">data.frame</span>(<span class="at">module =</span> mn, <span class="at">beta_tumour =</span> <span class="cn">NA</span>, <span class="at">t_tumour =</span> <span class="cn">NA</span>, <span class="at">p_tumour =</span> <span class="cn">NA</span>))</span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a>  form <span class="ot">&lt;-</span> <span class="fu">as.formula</span>(<span class="fu">paste</span>(mn, <span class="st">"~"</span>, rhs))</span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a>  fit <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>(</span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a>    nlme<span class="sc">::</span><span class="fu">lme</span>(form, <span class="at">random =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> patient_id,</span>
<span id="cb13-41"><a href="#cb13-41" aria-hidden="true" tabindex="-1"></a>              <span class="at">data =</span> d, <span class="at">method =</span> <span class="st">"REML"</span>, <span class="at">control =</span> ctrl_lme),</span>
<span id="cb13-42"><a href="#cb13-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">error =</span> <span class="cf">function</span>(e) <span class="cn">NULL</span></span>
<span id="cb13-43"><a href="#cb13-43" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb13-44"><a href="#cb13-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-45"><a href="#cb13-45" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(fit)) {</span>
<span id="cb13-46"><a href="#cb13-46" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">data.frame</span>(<span class="at">module =</span> mn, <span class="at">beta_tumour =</span> <span class="cn">NA</span>, <span class="at">t_tumour =</span> <span class="cn">NA</span>, <span class="at">p_tumour =</span> <span class="cn">NA</span>))</span>
<span id="cb13-47"><a href="#cb13-47" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb13-48"><a href="#cb13-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-49"><a href="#cb13-49" aria-hidden="true" tabindex="-1"></a>  sm <span class="ot">&lt;-</span> <span class="fu">summary</span>(fit)</span>
<span id="cb13-50"><a href="#cb13-50" aria-hidden="true" tabindex="-1"></a>  coef_name <span class="ot">&lt;-</span> <span class="cf">if</span> (<span class="st">"tumour"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(sm<span class="sc">$</span>tTable)) <span class="st">"tumour"</span> <span class="cf">else</span></span>
<span id="cb13-51"><a href="#cb13-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">grep</span>(<span class="st">"^tumour"</span>, <span class="fu">rownames</span>(sm<span class="sc">$</span>tTable), <span class="at">value =</span> <span class="cn">TRUE</span>)[<span class="dv">1</span>]</span>
<span id="cb13-52"><a href="#cb13-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-53"><a href="#cb13-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb13-54"><a href="#cb13-54" aria-hidden="true" tabindex="-1"></a>    <span class="at">module      =</span> mn,</span>
<span id="cb13-55"><a href="#cb13-55" aria-hidden="true" tabindex="-1"></a>    <span class="at">beta_tumour =</span> sm<span class="sc">$</span>tTable[coef_name, <span class="st">"Value"</span>],</span>
<span id="cb13-56"><a href="#cb13-56" aria-hidden="true" tabindex="-1"></a>    <span class="at">t_tumour    =</span> sm<span class="sc">$</span>tTable[coef_name, <span class="st">"t-value"</span>],</span>
<span id="cb13-57"><a href="#cb13-57" aria-hidden="true" tabindex="-1"></a>    <span class="at">p_tumour    =</span> sm<span class="sc">$</span>tTable[coef_name, <span class="st">"p-value"</span>],</span>
<span id="cb13-58"><a href="#cb13-58" aria-hidden="true" tabindex="-1"></a>    <span class="at">row.names   =</span> <span class="cn">NULL</span></span>
<span id="cb13-59"><a href="#cb13-59" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb13-60"><a href="#cb13-60" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb13-61"><a href="#cb13-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-62"><a href="#cb13-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-63"><a href="#cb13-63" aria-hidden="true" tabindex="-1"></a>lmm_results_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, lmm_results)</span>
<span id="cb13-64"><a href="#cb13-64" aria-hidden="true" tabindex="-1"></a>lmm_results_df<span class="sc">$</span>FDR_tumour <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(lmm_results_df<span class="sc">$</span>p_tumour, <span class="at">method =</span> <span class="st">"BH"</span>)</span>
<span id="cb13-65"><a href="#cb13-65" aria-hidden="true" tabindex="-1"></a>lmm_results_df <span class="ot">&lt;-</span> lmm_results_df[<span class="fu">order</span>(lmm_results_df<span class="sc">$</span>p_tumour), ]</span>
<span id="cb13-66"><a href="#cb13-66" aria-hidden="true" tabindex="-1"></a>lmm_results_df</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          module  beta_tumour   t_tumour     p_tumour   FDR_tumour
13       MEgreen  0.113865010  17.931365 8.849655e-32 1.238952e-30
4    MEturquoise -0.113148008 -16.703370 1.324776e-29 9.273432e-29
14      MEsalmon  0.099423074  15.277010 5.663103e-27 2.642782e-26
6        MEbrown -0.098708731 -12.482956 1.661738e-21 5.816082e-21
7          MEtan -0.082873705 -12.011999 1.510032e-20 4.228091e-20
9         MEcyan  0.061724581   8.259288 1.037621e-12 2.421115e-12
12        MEpink  0.069675009   7.812670 8.815625e-12 1.763125e-11
1      MEmagenta  0.017756193   5.788195 9.769685e-08 1.709695e-07
5          MEred -0.041631923  -5.155986 1.438200e-06 2.237201e-06
8       MEpurple  0.042942543   4.850331 4.999605e-06 6.999447e-06
10 MEgreenyellow  0.038229392   4.245908 5.196763e-05 6.614061e-05
2        MEblack  0.005441032   1.556064 1.231263e-01 1.436473e-01
11      MEyellow  0.013311006   1.392200 1.672179e-01 1.800808e-01
3         MEblue  0.003920094   1.030574 3.054423e-01 3.054423e-01</code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Heatmap</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">requireNamespace</span>(<span class="st">"pheatmap"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  lmm_results_df<span class="sc">$</span>module_clean <span class="ot">&lt;-</span> <span class="fu">substring</span>(lmm_results_df<span class="sc">$</span>module, <span class="dv">3</span>)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  tmat <span class="ot">&lt;-</span> <span class="fu">matrix</span>(lmm_results_df<span class="sc">$</span>t_tumour,</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>                 <span class="at">nrow =</span> <span class="fu">nrow</span>(lmm_results_df),</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">dimnames =</span> <span class="fu">list</span>(lmm_results_df<span class="sc">$</span>module_clean, <span class="st">"Tumour (t)"</span>))</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>  pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(tmat, <span class="at">main =</span> <span class="st">"Moduleâ€“tumour associations (LMM t-values)"</span>,</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>, <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>                     <span class="at">display_numbers =</span> <span class="fu">round</span>(tmat, <span class="dv">2</span>))</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SD_WGCNA_comp_all_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># MM (module membership) + GS (per-gene LMM |t| for tumour)</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>datExpr <span class="ot">&lt;-</span> datExpr0  <span class="co"># samples x genes</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="co"># MM</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>modNames_clean <span class="ot">&lt;-</span> <span class="fu">substring</span>(<span class="fu">colnames</span>(MEs), <span class="dv">3</span>)</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>geneModuleMembership <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(WGCNA<span class="sc">::</span><span class="fu">cor</span>(datExpr, MEs, <span class="at">use =</span> <span class="st">"p"</span>))</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(geneModuleMembership) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"MM"</span>, modNames_clean)</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>geneModuleMembership<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">colnames</span>(datExpr)</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Traits aligned to datExpr</span></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>traits_full <span class="ot">&lt;-</span> sample_traits[<span class="fu">rownames</span>(datExpr), , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>need_vars <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">c</span>(<span class="st">"patient_id"</span>,<span class="st">"tumour"</span>, covars), <span class="fu">names</span>(traits_full))</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>keep1 <span class="ot">&lt;-</span> <span class="fu">complete.cases</span>(traits_full[, need_vars, <span class="at">drop =</span> <span class="cn">FALSE</span>])</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>datExpr_ok <span class="ot">&lt;-</span> datExpr[keep1, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>traits_ok  <span class="ot">&lt;-</span> traits_full[keep1, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only subjects with both tumour states (0 and 1)</span></span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>tab <span class="ot">&lt;-</span> <span class="fu">with</span>(traits_ok, <span class="fu">table</span>(patient_id, tumour))</span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>keep_patients2 <span class="ot">&lt;-</span> <span class="fu">rownames</span>(tab)[<span class="fu">rowSums</span>(tab <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">==</span> <span class="dv">2</span>]</span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a>keep_rows2 <span class="ot">&lt;-</span> traits_ok<span class="sc">$</span>patient_id <span class="sc">%in%</span> keep_patients2</span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a>datExpr_ok <span class="ot">&lt;-</span> datExpr_ok[keep_rows2, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>traits_ok  <span class="ot">&lt;-</span> traits_ok [keep_rows2, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">rownames</span>(datExpr_ok), <span class="fu">rownames</span>(traits_ok)))</span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a><span class="co"># pick genes for GS</span></span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a>nGS <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="dv">8000</span>, <span class="fu">ncol</span>(datExpr_ok))</span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a>vars_ok <span class="ot">&lt;-</span> <span class="fu">apply</span>(datExpr_ok, <span class="dv">2</span>, var, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a>genes_for_GS <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">sort</span>(vars_ok, <span class="at">decreasing =</span> <span class="cn">TRUE</span>))[<span class="dv">1</span><span class="sc">:</span>nGS]</span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-35"><a href="#cb16-35" aria-hidden="true" tabindex="-1"></a><span class="co"># safe column names</span></span>
<span id="cb16-36"><a href="#cb16-36" aria-hidden="true" tabindex="-1"></a>X <span class="ot">&lt;-</span> datExpr_ok[, genes_for_GS, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb16-37"><a href="#cb16-37" aria-hidden="true" tabindex="-1"></a>safe_names   <span class="ot">&lt;-</span> <span class="fu">make.names</span>(<span class="fu">colnames</span>(X), <span class="at">unique =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-38"><a href="#cb16-38" aria-hidden="true" tabindex="-1"></a>gene_to_safe <span class="ot">&lt;-</span> <span class="fu">setNames</span>(safe_names, <span class="fu">colnames</span>(X))</span>
<span id="cb16-39"><a href="#cb16-39" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(X)  <span class="ot">&lt;-</span> safe_names</span>
<span id="cb16-40"><a href="#cb16-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-41"><a href="#cb16-41" aria-hidden="true" tabindex="-1"></a>datGS <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(traits_ok, X, <span class="at">check.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb16-42"><a href="#cb16-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-43"><a href="#cb16-43" aria-hidden="true" tabindex="-1"></a>gs_rhs <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">c</span>(<span class="st">"tumour"</span>, covars), <span class="at">collapse =</span> <span class="st">" + "</span>)  <span class="co"># same as ME models</span></span>
<span id="cb16-44"><a href="#cb16-44" aria-hidden="true" tabindex="-1"></a>gs_list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(genes_for_GS, <span class="cf">function</span>(g) {</span>
<span id="cb16-45"><a href="#cb16-45" aria-hidden="true" tabindex="-1"></a>  g_safe <span class="ot">&lt;-</span> gene_to_safe[[g]]</span>
<span id="cb16-46"><a href="#cb16-46" aria-hidden="true" tabindex="-1"></a>  form <span class="ot">&lt;-</span> <span class="fu">as.formula</span>(<span class="fu">paste0</span>(g_safe, <span class="st">" ~ "</span>, gs_rhs))</span>
<span id="cb16-47"><a href="#cb16-47" aria-hidden="true" tabindex="-1"></a>  fit <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>(</span>
<span id="cb16-48"><a href="#cb16-48" aria-hidden="true" tabindex="-1"></a>    nlme<span class="sc">::</span><span class="fu">lme</span>(form, <span class="at">random =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> patient_id,</span>
<span id="cb16-49"><a href="#cb16-49" aria-hidden="true" tabindex="-1"></a>              <span class="at">data =</span> datGS, <span class="at">na.action =</span> na.omit,</span>
<span id="cb16-50"><a href="#cb16-50" aria-hidden="true" tabindex="-1"></a>              <span class="at">method =</span> <span class="st">"REML"</span>, <span class="at">control =</span> ctrl_lme),</span>
<span id="cb16-51"><a href="#cb16-51" aria-hidden="true" tabindex="-1"></a>    <span class="at">error =</span> <span class="cf">function</span>(e) <span class="cn">NULL</span></span>
<span id="cb16-52"><a href="#cb16-52" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb16-53"><a href="#cb16-53" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(fit)) <span class="fu">return</span>(<span class="fu">data.frame</span>(<span class="at">gene =</span> g, <span class="at">GS =</span> <span class="cn">NA_real_</span>))</span>
<span id="cb16-54"><a href="#cb16-54" aria-hidden="true" tabindex="-1"></a>  sm <span class="ot">&lt;-</span> <span class="fu">summary</span>(fit)</span>
<span id="cb16-55"><a href="#cb16-55" aria-hidden="true" tabindex="-1"></a>  coef_name <span class="ot">&lt;-</span> <span class="cf">if</span> (<span class="st">"tumour"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(sm<span class="sc">$</span>tTable)) <span class="st">"tumour"</span> <span class="cf">else</span></span>
<span id="cb16-56"><a href="#cb16-56" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">grep</span>(<span class="st">"^tumour"</span>, <span class="fu">rownames</span>(sm<span class="sc">$</span>tTable), <span class="at">value =</span> <span class="cn">TRUE</span>)[<span class="dv">1</span>]</span>
<span id="cb16-57"><a href="#cb16-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">gene =</span> g, <span class="at">GS =</span> <span class="fu">abs</span>(sm<span class="sc">$</span>tTable[coef_name, <span class="st">"t-value"</span>]))</span>
<span id="cb16-58"><a href="#cb16-58" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb16-59"><a href="#cb16-59" aria-hidden="true" tabindex="-1"></a>geneGS <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, gs_list)</span>
<span id="cb16-60"><a href="#cb16-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-61"><a href="#cb16-61" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine hub metrics</span></span>
<span id="cb16-62"><a href="#cb16-62" aria-hidden="true" tabindex="-1"></a>hubData <span class="ot">&lt;-</span> geneModuleMembership <span class="sc">%&gt;%</span></span>
<span id="cb16-63"><a href="#cb16-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(geneGS, <span class="at">by =</span> <span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb16-64"><a href="#cb16-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">module =</span> moduleColors[<span class="fu">match</span>(gene, <span class="fu">names</span>(moduleColors))])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># kWithin (chunked within-module adjacency; memory-safe, faster)</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Make sure datExpr exists and is aligned</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>datExpr <span class="ot">&lt;-</span> datExpr0</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>datExpr <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(datExpr)</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="fu">storage.mode</span>(datExpr) <span class="ot">&lt;-</span> <span class="st">"double"</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">colnames</span>(datExpr), <span class="fu">names</span>(moduleColors)))</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>compute_kWithin_chunked <span class="ot">&lt;-</span> <span class="cf">function</span>(datExpr, moduleColors, softPower,</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">chunk_size =</span> <span class="dv">500</span>,</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">exclude_modules =</span> <span class="st">"grey"</span>,</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">cor_use =</span> <span class="st">"p"</span>) {</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>  genes <span class="ot">&lt;-</span> <span class="fu">colnames</span>(datExpr)</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>  moduleColors <span class="ot">&lt;-</span> moduleColors[genes]  <span class="co"># align defensively</span></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>  mods <span class="ot">&lt;-</span> <span class="fu">unique</span>(moduleColors)</span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>  mods <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(mods, exclude_modules)</span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>  kWithin <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">rep</span>(<span class="cn">NA_real_</span>, <span class="fu">length</span>(genes)), genes)</span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (mod <span class="cf">in</span> mods) {</span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a>    genes_in_mod <span class="ot">&lt;-</span> genes[moduleColors <span class="sc">==</span> mod]</span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a>    m0 <span class="ot">&lt;-</span> <span class="fu">length</span>(genes_in_mod)</span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (m0 <span class="sc">&lt;</span> <span class="dv">5</span>) <span class="cf">next</span></span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a>    X_all <span class="ot">&lt;-</span> datExpr[, genes_in_mod, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Drop zero-variance genes (they cause NA correlations)</span></span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a>    v <span class="ot">&lt;-</span> <span class="fu">apply</span>(X_all, <span class="dv">2</span>, var, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a>    keep <span class="ot">&lt;-</span> <span class="fu">is.finite</span>(v) <span class="sc">&amp;</span> v <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb17-32"><a href="#cb17-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-33"><a href="#cb17-33" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(keep)) {</span>
<span id="cb17-34"><a href="#cb17-34" aria-hidden="true" tabindex="-1"></a>      genes_in_mod <span class="ot">&lt;-</span> genes_in_mod[keep]</span>
<span id="cb17-35"><a href="#cb17-35" aria-hidden="true" tabindex="-1"></a>      X_all <span class="ot">&lt;-</span> X_all[, keep, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb17-36"><a href="#cb17-36" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb17-37"><a href="#cb17-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-38"><a href="#cb17-38" aria-hidden="true" tabindex="-1"></a>    m <span class="ot">&lt;-</span> <span class="fu">ncol</span>(X_all)</span>
<span id="cb17-39"><a href="#cb17-39" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (m <span class="sc">&lt;</span> <span class="dv">5</span>) <span class="cf">next</span></span>
<span id="cb17-40"><a href="#cb17-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-41"><a href="#cb17-41" aria-hidden="true" tabindex="-1"></a>    idx_in_mod <span class="ot">&lt;-</span> <span class="fu">seq_len</span>(m); <span class="fu">names</span>(idx_in_mod) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(X_all)</span>
<span id="cb17-42"><a href="#cb17-42" aria-hidden="true" tabindex="-1"></a>    kW <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">rep</span>(<span class="cn">NA_real_</span>, m), <span class="fu">colnames</span>(X_all))</span>
<span id="cb17-43"><a href="#cb17-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-44"><a href="#cb17-44" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (s <span class="cf">in</span> <span class="fu">seq</span>(<span class="dv">1</span>, m, <span class="at">by =</span> chunk_size)) {</span>
<span id="cb17-45"><a href="#cb17-45" aria-hidden="true" tabindex="-1"></a>      e <span class="ot">&lt;-</span> <span class="fu">min</span>(m, s <span class="sc">+</span> chunk_size <span class="sc">-</span> <span class="dv">1</span>)</span>
<span id="cb17-46"><a href="#cb17-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-47"><a href="#cb17-47" aria-hidden="true" tabindex="-1"></a>      X_chunk <span class="ot">&lt;-</span> X_all[, s<span class="sc">:</span>e, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb17-48"><a href="#cb17-48" aria-hidden="true" tabindex="-1"></a>      chunk_genes <span class="ot">&lt;-</span> <span class="fu">colnames</span>(X_chunk)</span>
<span id="cb17-49"><a href="#cb17-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-50"><a href="#cb17-50" aria-hidden="true" tabindex="-1"></a>      C <span class="ot">&lt;-</span> WGCNA<span class="sc">::</span><span class="fu">cor</span>(X_chunk, X_all, <span class="at">use =</span> cor_use)</span>
<span id="cb17-51"><a href="#cb17-51" aria-hidden="true" tabindex="-1"></a>      A <span class="ot">&lt;-</span> <span class="fu">abs</span>(C)<span class="sc">^</span>softPower</span>
<span id="cb17-52"><a href="#cb17-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-53"><a href="#cb17-53" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Zero out self-adjacency for the chunk rows (vectorized)</span></span>
<span id="cb17-54"><a href="#cb17-54" aria-hidden="true" tabindex="-1"></a>      diag_idx <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">seq_len</span>(<span class="fu">nrow</span>(A)), idx_in_mod[chunk_genes])</span>
<span id="cb17-55"><a href="#cb17-55" aria-hidden="true" tabindex="-1"></a>      A[diag_idx] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb17-56"><a href="#cb17-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-57"><a href="#cb17-57" aria-hidden="true" tabindex="-1"></a>      kW[chunk_genes] <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(A, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-58"><a href="#cb17-58" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb17-59"><a href="#cb17-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-60"><a href="#cb17-60" aria-hidden="true" tabindex="-1"></a>    kWithin[<span class="fu">names</span>(kW)] <span class="ot">&lt;-</span> kW</span>
<span id="cb17-61"><a href="#cb17-61" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"Done kWithin for module %s (%d genes)</span><span class="sc">\n</span><span class="st">"</span>, mod, <span class="fu">length</span>(kW)))</span>
<span id="cb17-62"><a href="#cb17-62" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb17-63"><a href="#cb17-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-64"><a href="#cb17-64" aria-hidden="true" tabindex="-1"></a>  kWithin</span>
<span id="cb17-65"><a href="#cb17-65" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-66"><a href="#cb17-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-67"><a href="#cb17-67" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute + attach</span></span>
<span id="cb17-68"><a href="#cb17-68" aria-hidden="true" tabindex="-1"></a>kWithin_vec <span class="ot">&lt;-</span> <span class="fu">compute_kWithin_chunked</span>(datExpr, moduleColors, softPower, <span class="at">chunk_size =</span> <span class="dv">500</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Done kWithin for module yellow (1215 genes)
Done kWithin for module green (1029 genes)
Done kWithin for module turquoise (3371 genes)
Done kWithin for module blue (1648 genes)
Done kWithin for module brown (1312 genes)
Done kWithin for module salmon (97 genes)
Done kWithin for module pink (198 genes)
Done kWithin for module red (493 genes)
Done kWithin for module purple (160 genes)
Done kWithin for module cyan (95 genes)
Done kWithin for module greenyellow (131 genes)
Done kWithin for module tan (104 genes)
Done kWithin for module black (323 genes)
Done kWithin for module magenta (172 genes)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>hubData<span class="sc">$</span>kWithin <span class="ot">&lt;-</span> <span class="fu">unname</span>(kWithin_vec[hubData<span class="sc">$</span>gene])</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Top hubs per module (same as before, but now faster upstream)</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>hubList <span class="ot">&lt;-</span> hubData <span class="sc">%&gt;%</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(kWithin)) <span class="sc">%&gt;%</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(module) <span class="sc">%&gt;%</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(dplyr<span class="sc">::</span><span class="fu">desc</span>(kWithin), <span class="at">.by_group =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>()</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>hubList</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 140 Ã— 18
   MMmagenta MMblack MMblue MMturquoise   MMred MMbrown MMtan MMpurple  MMcyan
       &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt;       &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt;
 1     0.425   0.920  0.558     -0.183  -0.0180 -0.0447 0.324   0.131   0.0664
 2     0.421   0.890  0.661     -0.0118  0.194   0.200  0.511   0.116  -0.0597
 3     0.377   0.876  0.674     -0.116   0.0737 -0.0396 0.387   0.104   0.0886
 4     0.542   0.910  0.599      0.0198  0.201   0.151  0.491  -0.0204 -0.0572
 5     0.424   0.888  0.622     -0.0550  0.139   0.249  0.488   0.190  -0.0269
 6     0.543   0.885  0.490      0.0802  0.0507  0.186  0.503  -0.0349 -0.0346
 7     0.401   0.889  0.612     -0.124   0.158   0.0199 0.351   0.130  -0.0185
 8     0.381   0.897  0.523     -0.0747  0.0777  0.0346 0.357   0.0289  0.0530
 9     0.412   0.874  0.548     -0.195  -0.0720 -0.0661 0.287   0.152   0.0818
10     0.356   0.861  0.621     -0.107   0.0112 -0.0356 0.366   0.0891  0.0233
# â„¹ 130 more rows
# â„¹ 9 more variables: MMgreenyellow &lt;dbl&gt;, MMyellow &lt;dbl&gt;, MMpink &lt;dbl&gt;,
#   MMgreen &lt;dbl&gt;, MMsalmon &lt;dbl&gt;, gene &lt;chr&gt;, GS &lt;dbl&gt;, module &lt;chr&gt;,
#   kWithin &lt;dbl&gt;</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Table: Top 5 genes by kWithin in green/turquoise/salmon</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>target_modules <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"green"</span>, <span class="st">"turquoise"</span>, <span class="st">"salmon"</span>)</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>pool_n <span class="ot">&lt;-</span> <span class="dv">80</span>  <span class="co"># take more than 5 so we can skip any NA/failed GS fits safely</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="co"># --- Helper: compute GS for a given gene set (paired subset, same covars, same LMM spec) ---</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>compute_GS_for_genes <span class="ot">&lt;-</span> <span class="cf">function</span>(genes, datExpr_ok, traits_ok, covars, ctrl_lme) {</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>  genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(genes, <span class="fu">colnames</span>(datExpr_ok))</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(genes) <span class="sc">==</span> <span class="dv">0</span>) <span class="fu">return</span>(<span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">character</span>(), <span class="at">GS =</span> <span class="fu">numeric</span>()))</span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>  X <span class="ot">&lt;-</span> datExpr_ok[, genes, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>  safe <span class="ot">&lt;-</span> <span class="fu">make.names</span>(<span class="fu">colnames</span>(X), <span class="at">unique =</span> <span class="cn">TRUE</span>)</span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>  map  <span class="ot">&lt;-</span> <span class="fu">setNames</span>(safe, <span class="fu">colnames</span>(X))</span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(X) <span class="ot">&lt;-</span> safe</span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>  datGS_small <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(traits_ok, X, <span class="at">check.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>  rhs <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">c</span>(<span class="st">"tumour"</span>, covars), <span class="at">collapse =</span> <span class="st">" + "</span>)</span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">lapply</span>(genes, <span class="cf">function</span>(g) {</span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>    g_safe <span class="ot">&lt;-</span> map[[g]]</span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>    form <span class="ot">&lt;-</span> <span class="fu">as.formula</span>(<span class="fu">paste0</span>(g_safe, <span class="st">" ~ "</span>, rhs))</span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a>    fit <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>(</span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a>      nlme<span class="sc">::</span><span class="fu">lme</span>(form, <span class="at">random =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> patient_id,</span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> datGS_small, <span class="at">na.action =</span> na.omit,</span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">"REML"</span>, <span class="at">control =</span> ctrl_lme),</span>
<span id="cb21-28"><a href="#cb21-28" aria-hidden="true" tabindex="-1"></a>      <span class="at">error =</span> <span class="cf">function</span>(e) <span class="cn">NULL</span></span>
<span id="cb21-29"><a href="#cb21-29" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb21-30"><a href="#cb21-30" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.null</span>(fit)) <span class="fu">return</span>(<span class="fu">data.frame</span>(<span class="at">gene =</span> g, <span class="at">GS =</span> <span class="cn">NA_real_</span>))</span>
<span id="cb21-31"><a href="#cb21-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-32"><a href="#cb21-32" aria-hidden="true" tabindex="-1"></a>    sm <span class="ot">&lt;-</span> <span class="fu">summary</span>(fit)</span>
<span id="cb21-33"><a href="#cb21-33" aria-hidden="true" tabindex="-1"></a>    rn <span class="ot">&lt;-</span> <span class="fu">rownames</span>(sm<span class="sc">$</span>tTable)</span>
<span id="cb21-34"><a href="#cb21-34" aria-hidden="true" tabindex="-1"></a>    coef_name <span class="ot">&lt;-</span> <span class="cf">if</span> (<span class="st">"tumour"</span> <span class="sc">%in%</span> rn) <span class="st">"tumour"</span> <span class="cf">else</span> <span class="fu">grep</span>(<span class="st">"^tumour"</span>, rn, <span class="at">value =</span> <span class="cn">TRUE</span>)[<span class="dv">1</span>]</span>
<span id="cb21-35"><a href="#cb21-35" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.na</span>(coef_name)) <span class="fu">return</span>(<span class="fu">data.frame</span>(<span class="at">gene =</span> g, <span class="at">GS =</span> <span class="cn">NA_real_</span>))</span>
<span id="cb21-36"><a href="#cb21-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-37"><a href="#cb21-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">gene =</span> g, <span class="at">GS =</span> <span class="fu">abs</span>(sm<span class="sc">$</span>tTable[coef_name, <span class="st">"t-value"</span>]))</span>
<span id="cb21-38"><a href="#cb21-38" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb21-39"><a href="#cb21-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-40"><a href="#cb21-40" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(out)</span>
<span id="cb21-41"><a href="#cb21-41" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb21-42"><a href="#cb21-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-43"><a href="#cb21-43" aria-hidden="true" tabindex="-1"></a><span class="co"># --- Build a candidate pool: top kWithin genes per module ---</span></span>
<span id="cb21-44"><a href="#cb21-44" aria-hidden="true" tabindex="-1"></a>pool <span class="ot">&lt;-</span> hubData <span class="sc">%&gt;%</span></span>
<span id="cb21-45"><a href="#cb21-45" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(module <span class="sc">%in%</span> target_modules, <span class="sc">!</span><span class="fu">is.na</span>(kWithin)) <span class="sc">%&gt;%</span></span>
<span id="cb21-46"><a href="#cb21-46" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(module) <span class="sc">%&gt;%</span></span>
<span id="cb21-47"><a href="#cb21-47" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(dplyr<span class="sc">::</span><span class="fu">desc</span>(kWithin), <span class="at">.by_group =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb21-48"><a href="#cb21-48" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">slice_head</span>(<span class="at">n =</span> pool_n) <span class="sc">%&gt;%</span></span>
<span id="cb21-49"><a href="#cb21-49" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb21-50"><a href="#cb21-50" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene, module, kWithin, GS)</span>
<span id="cb21-51"><a href="#cb21-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-52"><a href="#cb21-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-53"><a href="#cb21-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-54"><a href="#cb21-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-55"><a href="#cb21-55" aria-hidden="true" tabindex="-1"></a><span class="co"># --- Add module membership (MM for that gene's own module) ---</span></span>
<span id="cb21-56"><a href="#cb21-56" aria-hidden="true" tabindex="-1"></a>mm_df <span class="ot">&lt;-</span> geneModuleMembership</span>
<span id="cb21-57"><a href="#cb21-57" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(mm_df) <span class="ot">&lt;-</span> mm_df<span class="sc">$</span>gene</span>
<span id="cb21-58"><a href="#cb21-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-59"><a href="#cb21-59" aria-hidden="true" tabindex="-1"></a>pool<span class="sc">$</span>MM <span class="ot">&lt;-</span> <span class="fu">mapply</span>(<span class="cf">function</span>(g, mod) {</span>
<span id="cb21-60"><a href="#cb21-60" aria-hidden="true" tabindex="-1"></a>  col <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"MM"</span>, mod)</span>
<span id="cb21-61"><a href="#cb21-61" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span>col <span class="sc">%in%</span> <span class="fu">colnames</span>(mm_df)) <span class="fu">return</span>(<span class="cn">NA_real_</span>)</span>
<span id="cb21-62"><a href="#cb21-62" aria-hidden="true" tabindex="-1"></a>  i <span class="ot">&lt;-</span> <span class="fu">match</span>(g, mm_df<span class="sc">$</span>gene)</span>
<span id="cb21-63"><a href="#cb21-63" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.na</span>(i)) <span class="fu">return</span>(<span class="cn">NA_real_</span>)</span>
<span id="cb21-64"><a href="#cb21-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.numeric</span>(mm_df[i, col, <span class="at">drop =</span> <span class="cn">TRUE</span>])</span>
<span id="cb21-65"><a href="#cb21-65" aria-hidden="true" tabindex="-1"></a>}, pool<span class="sc">$</span>gene, pool<span class="sc">$</span>module)</span>
<span id="cb21-66"><a href="#cb21-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-67"><a href="#cb21-67" aria-hidden="true" tabindex="-1"></a><span class="co"># --- Ensure GS exists for all genes in pool (compute missing ones only) ---</span></span>
<span id="cb21-68"><a href="#cb21-68" aria-hidden="true" tabindex="-1"></a>missing_gs <span class="ot">&lt;-</span> pool<span class="sc">$</span>gene[<span class="fu">is.na</span>(pool<span class="sc">$</span>GS)]</span>
<span id="cb21-69"><a href="#cb21-69" aria-hidden="true" tabindex="-1"></a>missing_gs <span class="ot">&lt;-</span> <span class="fu">unique</span>(missing_gs)</span>
<span id="cb21-70"><a href="#cb21-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-71"><a href="#cb21-71" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">length</span>(missing_gs) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb21-72"><a href="#cb21-72" aria-hidden="true" tabindex="-1"></a>  gs_new <span class="ot">&lt;-</span> <span class="fu">compute_GS_for_genes</span>(missing_gs, datExpr_ok, traits_ok, covars, ctrl_lme)</span>
<span id="cb21-73"><a href="#cb21-73" aria-hidden="true" tabindex="-1"></a>  pool <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(pool, gs_new, <span class="at">by =</span> <span class="st">"gene"</span>, <span class="at">suffix =</span> <span class="fu">c</span>(<span class="st">""</span>, <span class="st">".new"</span>))</span>
<span id="cb21-74"><a href="#cb21-74" aria-hidden="true" tabindex="-1"></a>  pool<span class="sc">$</span>GS <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(pool<span class="sc">$</span>GS), pool<span class="sc">$</span>GS.new, pool<span class="sc">$</span>GS)</span>
<span id="cb21-75"><a href="#cb21-75" aria-hidden="true" tabindex="-1"></a>  pool<span class="sc">$</span>GS.new <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb21-76"><a href="#cb21-76" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb21-77"><a href="#cb21-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-78"><a href="#cb21-78" aria-hidden="true" tabindex="-1"></a><span class="co"># --- Final: keep complete rows, take top 5 per module, order by kWithin ---</span></span>
<span id="cb21-79"><a href="#cb21-79" aria-hidden="true" tabindex="-1"></a>top15 <span class="ot">&lt;-</span> pool <span class="sc">%&gt;%</span></span>
<span id="cb21-80"><a href="#cb21-80" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">is.finite</span>(kWithin), <span class="fu">is.finite</span>(MM), <span class="fu">is.finite</span>(GS)) <span class="sc">%&gt;%</span></span>
<span id="cb21-81"><a href="#cb21-81" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(module) <span class="sc">%&gt;%</span></span>
<span id="cb21-82"><a href="#cb21-82" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(dplyr<span class="sc">::</span><span class="fu">desc</span>(kWithin), <span class="at">.by_group =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb21-83"><a href="#cb21-83" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb21-84"><a href="#cb21-84" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb21-85"><a href="#cb21-85" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(dplyr<span class="sc">::</span><span class="fu">desc</span>(kWithin)) <span class="sc">%&gt;%</span></span>
<span id="cb21-86"><a href="#cb21-86" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene, module, kWithin, MM, GS)</span>
<span id="cb21-87"><a href="#cb21-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-88"><a href="#cb21-88" aria-hidden="true" tabindex="-1"></a><span class="co"># If you ended up with &lt; 15 rows, increase pool_n (or allow more genes per module)</span></span>
<span id="cb21-89"><a href="#cb21-89" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">nrow</span>(top15) <span class="sc">&lt;</span> <span class="dv">15</span>) {</span>
<span id="cb21-90"><a href="#cb21-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">warning</span>(<span class="st">"Got only "</span>, <span class="fu">nrow</span>(top15), <span class="st">" rows without NA. Try increasing pool_n (e.g., 150)."</span>)</span>
<span id="cb21-91"><a href="#cb21-91" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb21-92"><a href="#cb21-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-93"><a href="#cb21-93" aria-hidden="true" tabindex="-1"></a><span class="co"># Render as a table</span></span>
<span id="cb21-94"><a href="#cb21-94" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(top15, <span class="at">digits =</span> <span class="dv">3</span>,</span>
<span id="cb21-95"><a href="#cb21-95" aria-hidden="true" tabindex="-1"></a>             <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">"Gene"</span>, <span class="st">"Module"</span>, <span class="st">"kWithin"</span>, <span class="st">"Module membership (MM)"</span>, <span class="st">"Gene significance (GS)"</span>),</span>
<span id="cb21-96"><a href="#cb21-96" aria-hidden="true" tabindex="-1"></a>             <span class="at">caption =</span> <span class="st">"Top 5 genes (by kWithin) in green, turquoise, salmon modules (no NA; ordered by kWithin)."</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<caption>Top 5 genes (by kWithin) in green, turquoise, salmon modules (no NA; ordered by kWithin).</caption>
<colgroup>
<col style="width: 11%">
<col style="width: 13%">
<col style="width: 11%">
<col style="width: 31%">
<col style="width: 31%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Gene</th>
<th style="text-align: left;">Module</th>
<th style="text-align: right;">kWithin</th>
<th style="text-align: right;">Module membership (MM)</th>
<th style="text-align: right;">Gene significance (GS)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">GMFG</td>
<td style="text-align: left;">turquoise</td>
<td style="text-align: right;">82.849</td>
<td style="text-align: right;">0.930</td>
<td style="text-align: right;">14.057</td>
</tr>
<tr class="even">
<td style="text-align: left;">LCP2</td>
<td style="text-align: left;">turquoise</td>
<td style="text-align: right;">82.602</td>
<td style="text-align: right;">0.904</td>
<td style="text-align: right;">13.561</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ARHGDIB</td>
<td style="text-align: left;">turquoise</td>
<td style="text-align: right;">82.082</td>
<td style="text-align: right;">0.916</td>
<td style="text-align: right;">11.556</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD53</td>
<td style="text-align: left;">turquoise</td>
<td style="text-align: right;">79.640</td>
<td style="text-align: right;">0.865</td>
<td style="text-align: right;">9.196</td>
</tr>
<tr class="odd">
<td style="text-align: left;">RCSD1</td>
<td style="text-align: left;">turquoise</td>
<td style="text-align: right;">79.465</td>
<td style="text-align: right;">0.906</td>
<td style="text-align: right;">11.444</td>
</tr>
<tr class="even">
<td style="text-align: left;">SPC25</td>
<td style="text-align: left;">green</td>
<td style="text-align: right;">55.214</td>
<td style="text-align: right;">0.948</td>
<td style="text-align: right;">16.699</td>
</tr>
<tr class="odd">
<td style="text-align: left;">BUB1</td>
<td style="text-align: left;">green</td>
<td style="text-align: right;">54.136</td>
<td style="text-align: right;">0.948</td>
<td style="text-align: right;">14.231</td>
</tr>
<tr class="even">
<td style="text-align: left;">MELK</td>
<td style="text-align: left;">green</td>
<td style="text-align: right;">53.957</td>
<td style="text-align: right;">0.941</td>
<td style="text-align: right;">15.936</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CDCA5</td>
<td style="text-align: left;">green</td>
<td style="text-align: right;">52.737</td>
<td style="text-align: right;">0.943</td>
<td style="text-align: right;">17.437</td>
</tr>
<tr class="even">
<td style="text-align: left;">NUF2</td>
<td style="text-align: left;">green</td>
<td style="text-align: right;">52.158</td>
<td style="text-align: right;">0.932</td>
<td style="text-align: right;">15.543</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SCAMP3</td>
<td style="text-align: left;">salmon</td>
<td style="text-align: right;">7.472</td>
<td style="text-align: right;">0.936</td>
<td style="text-align: right;">16.279</td>
</tr>
<tr class="even">
<td style="text-align: left;">SNX27</td>
<td style="text-align: left;">salmon</td>
<td style="text-align: right;">5.832</td>
<td style="text-align: right;">0.920</td>
<td style="text-align: right;">14.665</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SCNM1</td>
<td style="text-align: left;">salmon</td>
<td style="text-align: right;">5.430</td>
<td style="text-align: right;">0.910</td>
<td style="text-align: right;">13.348</td>
</tr>
<tr class="even">
<td style="text-align: left;">PRCC</td>
<td style="text-align: left;">salmon</td>
<td style="text-align: right;">4.748</td>
<td style="text-align: right;">0.897</td>
<td style="text-align: right;">11.012</td>
</tr>
<tr class="odd">
<td style="text-align: left;">VPS72</td>
<td style="text-align: left;">salmon</td>
<td style="text-align: right;">4.570</td>
<td style="text-align: right;">0.895</td>
<td style="text-align: right;">15.120</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>raw_top5 <span class="ot">&lt;-</span> hubData <span class="sc">%&gt;%</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(module <span class="sc">%in%</span> target_modules, <span class="sc">!</span><span class="fu">is.na</span>(kWithin)) <span class="sc">%&gt;%</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(module) <span class="sc">%&gt;%</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(dplyr<span class="sc">::</span><span class="fu">desc</span>(kWithin), <span class="at">.by_group =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene, module, kWithin)</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>final_top5 <span class="ot">&lt;-</span> top15 <span class="sc">%&gt;%</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(module) <span class="sc">%&gt;%</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene, module, kWithin)</span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">anti_join</span>(raw_top5, final_top5, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene"</span>,<span class="st">"module"</span>)) <span class="co"># returns 0 rows</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 0 Ã— 3
# â„¹ 3 variables: gene &lt;chr&gt;, module &lt;chr&gt;, kWithin &lt;dbl&gt;</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Example GS vs MM plot (largest module that's not grey)</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>tab <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">table</span>(moduleColors), <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>targetModule <span class="ot">&lt;-</span> <span class="fu">names</span>(tab)[<span class="fu">names</span>(tab) <span class="sc">!=</span> <span class="st">"grey"</span>][<span class="dv">1</span>]</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>mm_col <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"MM"</span>, targetModule)</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>inMod  <span class="ot">&lt;-</span> hubData<span class="sc">$</span>module <span class="sc">==</span> targetModule</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span>mm_col <span class="sc">%in%</span> <span class="fu">names</span>(hubData)) <span class="fu">stop</span>(<span class="st">"Missing column "</span>, mm_col, <span class="st">" (did you remove MEgrey but picked grey?)"</span>)</span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> hubData[inMod, <span class="fu">c</span>(<span class="st">"GS"</span>, mm_col), drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> df[<span class="fu">complete.cases</span>(df), , drop <span class="ot">=</span> <span class="cn">FALSE</span>]  <span class="co"># drop NA GS/MM rows</span></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">nrow</span>(df) <span class="sc">&lt;</span> <span class="dv">2</span>) <span class="fu">stop</span>(<span class="st">"No non-NA points to plot for module "</span>, targetModule)</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"GS"</span>,<span class="st">"MM"</span>)</span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(df<span class="sc">$</span>MM, df<span class="sc">$</span>GS,</span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="fu">paste0</span>(<span class="st">"Module membership ("</span>, targetModule, <span class="st">")"</span>),</span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">"Gene significance (|t| for tumour)"</span>,</span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">"GS vs MM in"</span>, targetModule, <span class="st">"module"</span>))</span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="fu">lm</span>(GS <span class="sc">~</span> MM, <span class="at">data =</span> df), <span class="at">lty =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SD_WGCNA_comp_all_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Exports</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>stamp <span class="ot">&lt;-</span> <span class="fu">format</span>(<span class="fu">Sys.Date</span>(), <span class="st">"%Y%m%d"</span>)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(lmm_results_df, <span class="at">file =</span> <span class="fu">sprintf</span>(<span class="st">"WGCNA_ME_LMM_AgeGender_%s.csv"</span>, stamp), <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(hubList,       <span class="at">file =</span> <span class="fu">sprintf</span>(<span class="st">"WGCNA_top10_hubs_%s.csv"</span>, stamp), <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>moduleGenes <span class="ot">&lt;-</span> <span class="fu">split</span>(hubData<span class="sc">$</span>gene, hubData<span class="sc">$</span>module)</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (mod <span class="cf">in</span> <span class="fu">names</span>(moduleGenes)) {</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(moduleGenes[[mod]], <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"module_"</span>, mod, <span class="st">"_genes.txt"</span>),</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>              <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">FALSE</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Save analysis state (optional)</span></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(net, moduleColors, MEs, softPower,</span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>     lmm_results_df, geneModuleMembership, geneGS, hubData, hubList,</span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>     <span class="at">file =</span> <span class="fu">sprintf</span>(<span class="st">"WGCNA_outputs_AgeGender_%s.RData"</span>, stamp), <span class="at">compress =</span> <span class="st">"xz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ---- Save paired WGCNA outputs for reuse (2nd dataset) ----</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Make sure gene IDs are attached everywhere</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>moduleColors_paired <span class="ot">&lt;-</span> moduleColors</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(moduleColors_paired) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(datExpr0)</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>MEs_paired <span class="ot">&lt;-</span> MEs</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a><span class="co"># (MEs already have rownames = samples; keep them)</span></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">rownames</span>(MEs_paired), <span class="fu">rownames</span>(sample_traits)))</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>hubData_paired <span class="ot">&lt;-</span> hubData</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Give hubData stable rownames for easy joins later</span></span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"gene"</span> <span class="sc">%in%</span> <span class="fu">names</span>(hubData_paired)) <span class="fu">rownames</span>(hubData_paired) <span class="ot">&lt;-</span> hubData_paired<span class="sc">$</span>gene</span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>lmm_results_paired <span class="ot">&lt;-</span> lmm_results_df</span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a><span class="co"># TOM note:</span></span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a><span class="co"># In this pipeline you used blockwiseModules(..., saveTOMs = FALSE),</span></span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a><span class="co"># and computed kWithin chunk-wise (no full TOM/adjacency stored).</span></span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a><span class="co"># So we export TOM as NULL (but keep TOM file info if it exists).</span></span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a>TOM_paired <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>TOM_files  <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">exists</span>(<span class="st">"net"</span>) <span class="sc">&amp;&amp;</span> <span class="fu">is.list</span>(net)) {</span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(net<span class="sc">$</span>TOMFiles)) TOM_files <span class="ot">&lt;-</span> net<span class="sc">$</span>TOMFiles</span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(net<span class="sc">$</span>TOMFile))  TOM_files <span class="ot">&lt;-</span> net<span class="sc">$</span>TOMFile</span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-28"><a href="#cb26-28" aria-hidden="true" tabindex="-1"></a>paired_wgcna_export <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb26-29"><a href="#cb26-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">method      =</span> <span class="st">"paired_WGCNA_blockwise_unsigned_AgeGender"</span>,</span>
<span id="cb26-30"><a href="#cb26-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">dataset     =</span> <span class="st">"GSE62043"</span>,</span>
<span id="cb26-31"><a href="#cb26-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">date        =</span> <span class="fu">Sys.Date</span>(),</span>
<span id="cb26-32"><a href="#cb26-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-33"><a href="#cb26-33" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Network settings</span></span>
<span id="cb26-34"><a href="#cb26-34" aria-hidden="true" tabindex="-1"></a>  <span class="at">softPower   =</span> softPower,</span>
<span id="cb26-35"><a href="#cb26-35" aria-hidden="true" tabindex="-1"></a>  <span class="at">networkType =</span> <span class="st">"unsigned"</span>,</span>
<span id="cb26-36"><a href="#cb26-36" aria-hidden="true" tabindex="-1"></a>  <span class="at">corType     =</span> <span class="st">"pearson"</span>,</span>
<span id="cb26-37"><a href="#cb26-37" aria-hidden="true" tabindex="-1"></a>  <span class="at">TOMType     =</span> <span class="st">"unsigned"</span>,</span>
<span id="cb26-38"><a href="#cb26-38" aria-hidden="true" tabindex="-1"></a>  <span class="at">params      =</span> <span class="fu">list</span>(</span>
<span id="cb26-39"><a href="#cb26-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">minModuleSize  =</span> <span class="dv">70</span>,</span>
<span id="cb26-40"><a href="#cb26-40" aria-hidden="true" tabindex="-1"></a>    <span class="at">mergeCutHeight =</span> <span class="fl">0.25</span>,</span>
<span id="cb26-41"><a href="#cb26-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">deepSplit      =</span> <span class="dv">2</span>,</span>
<span id="cb26-42"><a href="#cb26-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">maxBlockSize   =</span> <span class="cf">if</span> (<span class="fu">exists</span>(<span class="st">"maxBlockSize"</span>)) maxBlockSize <span class="cf">else</span> <span class="cn">NA_integer_</span></span>
<span id="cb26-43"><a href="#cb26-43" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb26-44"><a href="#cb26-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-45"><a href="#cb26-45" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Covariates used in LMMs</span></span>
<span id="cb26-46"><a href="#cb26-46" aria-hidden="true" tabindex="-1"></a>  <span class="at">covariates  =</span> covars,</span>
<span id="cb26-47"><a href="#cb26-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-48"><a href="#cb26-48" aria-hidden="true" tabindex="-1"></a>  <span class="co"># IDs</span></span>
<span id="cb26-49"><a href="#cb26-49" aria-hidden="true" tabindex="-1"></a>  <span class="at">samples     =</span> <span class="fu">rownames</span>(datExpr0),</span>
<span id="cb26-50"><a href="#cb26-50" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes       =</span> <span class="fu">colnames</span>(datExpr0),</span>
<span id="cb26-51"><a href="#cb26-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-52"><a href="#cb26-52" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Core WGCNA outputs</span></span>
<span id="cb26-53"><a href="#cb26-53" aria-hidden="true" tabindex="-1"></a>  <span class="at">moduleColors =</span> moduleColors_paired,</span>
<span id="cb26-54"><a href="#cb26-54" aria-hidden="true" tabindex="-1"></a>  <span class="at">TOM          =</span> TOM_paired,</span>
<span id="cb26-55"><a href="#cb26-55" aria-hidden="true" tabindex="-1"></a>  <span class="at">TOM_files    =</span> TOM_files,</span>
<span id="cb26-56"><a href="#cb26-56" aria-hidden="true" tabindex="-1"></a>  <span class="at">MEs          =</span> MEs_paired,</span>
<span id="cb26-57"><a href="#cb26-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-58"><a href="#cb26-58" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Downstream association + hub summaries</span></span>
<span id="cb26-59"><a href="#cb26-59" aria-hidden="true" tabindex="-1"></a>  <span class="at">lmm_results  =</span> lmm_results_paired,</span>
<span id="cb26-60"><a href="#cb26-60" aria-hidden="true" tabindex="-1"></a>  <span class="at">hubData      =</span> hubData_paired,</span>
<span id="cb26-61"><a href="#cb26-61" aria-hidden="true" tabindex="-1"></a>  <span class="at">hubList      =</span> hubList,</span>
<span id="cb26-62"><a href="#cb26-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-63"><a href="#cb26-63" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Helpful extras for re-use / comparison</span></span>
<span id="cb26-64"><a href="#cb26-64" aria-hidden="true" tabindex="-1"></a>  <span class="at">geneModuleMembership =</span> geneModuleMembership,</span>
<span id="cb26-65"><a href="#cb26-65" aria-hidden="true" tabindex="-1"></a>  <span class="at">geneGS               =</span> geneGS,</span>
<span id="cb26-66"><a href="#cb26-66" aria-hidden="true" tabindex="-1"></a>  <span class="at">sample_traits        =</span> sample_traits</span>
<span id="cb26-67"><a href="#cb26-67" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb26-68"><a href="#cb26-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-69"><a href="#cb26-69" aria-hidden="true" tabindex="-1"></a>paired_wgcna_export<span class="sc">$</span>TOM_files <span class="ot">&lt;-</span> TOM_files</span>
<span id="cb26-70"><a href="#cb26-70" aria-hidden="true" tabindex="-1"></a>paired_wgcna_export<span class="sc">$</span>TOM <span class="ot">&lt;-</span> <span class="cn">NULL</span>  <span class="co"># keep NULL (TOM itself is huge)</span></span>
<span id="cb26-71"><a href="#cb26-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-72"><a href="#cb26-72" aria-hidden="true" tabindex="-1"></a>stamp <span class="ot">&lt;-</span> <span class="fu">format</span>(<span class="fu">Sys.Date</span>(), <span class="st">"%Y%m%d"</span>)</span>
<span id="cb26-73"><a href="#cb26-73" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(paired_wgcna_export,</span>
<span id="cb26-74"><a href="#cb26-74" aria-hidden="true" tabindex="-1"></a>        <span class="at">file =</span> <span class="fu">sprintf</span>(<span class="st">"paired_wgcna_export_GSE62043_%s.rds"</span>, stamp))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>NEW METHOD 1: LIMMA â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="co">#set up factors</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>limma_data <span class="ot">&lt;-</span> datLMM</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a><span class="co">#datExpr_ok are complete cases</span></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>limma_expr <span class="ot">&lt;-</span> <span class="fu">t</span>(datExpr_ok)</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>limma_data<span class="sc">$</span>tumour  <span class="ot">&lt;-</span> <span class="fu">factor</span>(limma_data<span class="sc">$</span>tumour, </span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>                             <span class="at">levels =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),</span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>                             <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"normal"</span>, <span class="st">"tumour"</span>))</span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>limma_data<span class="sc">$</span>patient_id <span class="ot">&lt;-</span> <span class="fu">factor</span>(limma_data<span class="sc">$</span>patient_id)</span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a><span class="co">#same design as we are using for the first dataset</span></span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> patient_id <span class="sc">+</span> tumour, <span class="at">data =</span> limma_data)</span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a><span class="co">#colnames(design)</span></span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a><span class="co">#run Limma</span></span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a>fit_limma2  <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(limma_expr, design)</span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a>fit_limma2  <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit_limma2)</span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(fit_limma2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                 Length  Class  Mode     
coefficients     1841930 -none- numeric  
rank                   1 -none- numeric  
assign                94 -none- numeric  
qr                     5 qr     list     
df.residual        19595 -none- numeric  
sigma              19595 -none- numeric  
cov.coefficients    8836 -none- numeric  
stdev.unscaled   1841930 -none- numeric  
pivot                 94 -none- numeric  
Amean              19595 -none- numeric  
method                 1 -none- character
design             17484 -none- numeric  
df.prior               1 -none- numeric  
s2.prior               1 -none- numeric  
var.prior             94 -none- numeric  
proportion             1 -none- numeric  
s2.post            19595 -none- numeric  
t                1841930 -none- numeric  
df.total           19595 -none- numeric  
p.value          1841930 -none- numeric  
lods             1841930 -none- numeric  
F                  19595 -none- numeric  
F.p.value          19595 -none- numeric  </code></pre>
</div>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>tt_limma2 <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit_limma2, <span class="at">coef =</span> <span class="st">"tumourtumour"</span>,<span class="at">number =</span> <span class="cn">Inf</span>)</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="co">#tt_limma2</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="co">#statistically significant miRNAs</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Method 1: choose the k most significant genes </span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>ord_sig_limma2 <span class="ot">&lt;-</span> tt_limma2[<span class="fu">order</span>(tt_limma2<span class="sc">$</span>adj.P.Val),]</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a><span class="co">#gives the 10 most significant miRNAs</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>ord_sig_limma2[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           logFC   AveExpr         t      P.Value    adj.P.Val        B
CLEC4G -3.801440 11.637685 -38.09557 6.640119e-59 1.255166e-54 123.3116
MARCO  -3.307660 12.670715 -37.81282 1.281109e-58 1.255166e-54 122.6776
FCN3   -3.742258 11.885255 -35.61374 2.473461e-56 1.615582e-52 117.5870
CRHBP  -3.885106 10.392471 -34.65449 2.684458e-55 1.315049e-51 115.2732
STAB2  -2.292635 10.547746 -33.60708 3.872573e-54 1.517661e-50 112.6780
CXCL14 -4.107626 10.957329 -31.51184 1.003349e-51 3.276771e-48 107.2582
CLEC1B -3.729041  9.510247 -31.13740 2.797636e-51 7.831381e-48 106.2559
OIT3   -4.917527 10.068017 -31.07033 3.365367e-51 8.243046e-48 106.0752
CFP    -2.616591 11.671267 -30.82960 6.549277e-51 1.425923e-47 105.4239
FCN2   -3.581026 10.407283 -30.08109 5.337062e-50 1.045797e-46 103.3701</code></pre>
</div>
</div>
<p>Plotting Limma</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>cols <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(limma_data<span class="sc">$</span>tumour <span class="sc">==</span> <span class="st">"tumour"</span>, <span class="st">"purple"</span>, <span class="st">"black"</span>)</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="co">#MDS plot</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMDS</span>(limma_expr,</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">top =</span> <span class="dv">500</span>,</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> limma_data<span class="sc">$</span>patient_id,</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">col =</span> cols)</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"topright"</span>,</span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">"normal"</span>, <span class="st">"tumour"</span>),</span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">col    =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"purple"</span>),</span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">pch =</span> <span class="dv">16</span>, <span class="at">bty =</span> <span class="st">"n"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SD_WGCNA_comp_all_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co">#MA plot for tumour effect (tumour - normal)</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>limma<span class="sc">::</span><span class="fu">plotMA</span>(fit_limma2,</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">coef =</span> <span class="st">"tumourtumour"</span>,</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">main =</span> <span class="st">"Average Expression vs LogFC (tumour - normal)"</span>)</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">0</span>, <span class="at">col =</span> <span class="st">"violet"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SD_WGCNA_comp_all_files/figure-html/unnamed-chunk-12-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>topN <span class="ot">&lt;-</span> <span class="dv">30</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>top_ids <span class="ot">&lt;-</span> <span class="fu">rownames</span>(tt_limma2)[<span class="fu">order</span>(tt_limma2<span class="sc">$</span>adj.P.Val)][<span class="dv">1</span><span class="sc">:</span>topN]</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>top_ids</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "CLEC4G"   "MARCO"    "FCN3"     "CRHBP"    "STAB2"    "CXCL14"  
 [7] "CLEC1B"   "OIT3"     "CFP"      "FCN2"     "DNASE1L3" "ECM1"    
[13] "ADAMTS13" "CLEC4M"   "CETP"     "IGFALS"   "PTTG1"    "CXCL12"  
[19] "PTH1R"    "LEPREL1"  "EFNB3"    "C7"       "PLAC8"    "GDF2"    
[25] "LIFR"     "DCN"      "HAMP"     "ANGPTL6"  "LILRB5"   "CYP26A1" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>mat <span class="ot">&lt;-</span> <span class="fu">t</span>(datExpr0[, top_ids, <span class="at">drop =</span> <span class="cn">FALSE</span>])  <span class="co"># genes x samples</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>ann_col <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">condition  =</span> limma_data<span class="sc">$</span>tumour,</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">patient_id =</span> limma_data<span class="sc">$</span>patient_id</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>limma_data<span class="sc">$</span>tumour</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  [1] tumour tumour tumour tumour tumour tumour tumour tumour tumour tumour
 [11] tumour tumour tumour tumour tumour tumour tumour tumour tumour tumour
 [21] tumour tumour tumour tumour tumour tumour tumour tumour tumour tumour
 [31] tumour tumour tumour tumour tumour tumour tumour tumour tumour tumour
 [41] tumour tumour tumour tumour tumour tumour tumour tumour tumour tumour
 [51] tumour tumour tumour tumour tumour tumour tumour tumour tumour tumour
 [61] tumour tumour tumour tumour tumour tumour tumour tumour tumour tumour
 [71] tumour tumour tumour tumour tumour tumour tumour tumour tumour tumour
 [81] tumour tumour tumour tumour tumour tumour tumour tumour tumour tumour
 [91] tumour tumour tumour normal normal normal normal normal normal normal
[101] normal normal normal normal normal normal normal normal normal normal
[111] normal normal normal normal normal normal normal normal normal normal
[121] normal normal normal normal normal normal normal normal normal normal
[131] normal normal normal normal normal normal normal normal normal normal
[141] normal normal normal normal normal normal normal normal normal normal
[151] normal normal normal normal normal normal normal normal normal normal
[161] normal normal normal normal normal normal normal normal normal normal
[171] normal normal normal normal normal normal normal normal normal normal
[181] normal normal normal normal normal normal
Levels: normal tumour</code></pre>
</div>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ann_col) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(limma_data)</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="co">#align annotation rows to the SAMPLE columns of mat</span></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>ann_col <span class="ot">&lt;-</span> ann_col[<span class="fu">colnames</span>(mat), , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(mat,</span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">"row"</span>,</span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> ann_col,</span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">"Top"</span>, <span class="fu">length</span>(top_ids), <span class="st">"DE genes (limma)"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SD_WGCNA_comp_all_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Volcano Plot</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>tt_volc2 <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit_limma2, <span class="at">coef =</span> <span class="st">"tumourtumour"</span>,<span class="at">number =</span> <span class="cn">Inf</span>)</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>tt_volc2<span class="sc">$</span>negLog10FDR <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(tt_volc2<span class="sc">$</span>adj.P.Val)</span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a><span class="co">#these parameters were chosen</span></span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>p_sig <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>co_FC <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>sig <span class="ot">&lt;-</span> tt_volc2<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> p_sig <span class="sc">&amp;</span> <span class="fu">abs</span>(tt_volc2<span class="sc">$</span>logFC) <span class="sc">&gt;</span> co_FC</span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tt_volc2<span class="sc">$</span>logFC, tt_volc2<span class="sc">$</span>negLog10FDR,</span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a>     <span class="at">pch=</span><span class="dv">20</span>,</span>
<span id="cb38-14"><a href="#cb38-14" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">"log2 fold change (tumour / normal)"</span>,</span>
<span id="cb38-15"><a href="#cb38-15" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab=</span><span class="st">"-log10 FDR"</span>,</span>
<span id="cb38-16"><a href="#cb38-16" aria-hidden="true" tabindex="-1"></a>     <span class="at">main=</span><span class="st">"Volcano: tumour vs normal"</span>)</span>
<span id="cb38-17"><a href="#cb38-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-18"><a href="#cb38-18" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>), <span class="at">col=</span><span class="st">"orange"</span>, <span class="at">lty=</span><span class="dv">2</span>)</span>
<span id="cb38-19"><a href="#cb38-19" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">col=</span><span class="st">"orange"</span>, <span class="at">lty=</span><span class="dv">2</span>)</span>
<span id="cb38-20"><a href="#cb38-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-21"><a href="#cb38-21" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(tt_volc2<span class="sc">$</span>logFC[sig], tt_volc2<span class="sc">$</span>negLog10FDR[sig],</span>
<span id="cb38-22"><a href="#cb38-22" aria-hidden="true" tabindex="-1"></a>       <span class="at">pch =</span> <span class="dv">20</span>, <span class="at">col =</span> <span class="st">"purple"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SD_WGCNA_comp_all_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Get parameters for paper</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>tot_dim <span class="ot">&lt;-</span> <span class="fu">dim</span>(tt_limma2)[<span class="dv">1</span>]</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>tot_dim</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 19595</code></pre>
</div>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>dim_sig <span class="ot">&lt;-</span> <span class="fu">dim</span>(tt_limma2[tt_limma2<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>,])[<span class="dv">1</span>]</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>tt_rel2 <span class="ot">&lt;-</span> (tt_limma2[tt_limma2<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(tt_limma2<span class="sc">$</span>logFC) <span class="sc">&gt;=</span> <span class="dv">1</span>,])</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>dim_rel <span class="ot">&lt;-</span> <span class="fu">dim</span>(tt_rel2)[<span class="dv">1</span>]</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>dim_down <span class="ot">&lt;-</span> <span class="fu">dim</span>(tt_rel2[tt_rel2<span class="sc">$</span>logFC<span class="sc">&lt;</span><span class="dv">0</span>,])[<span class="dv">1</span>]</span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>dim_up <span class="ot">&lt;-</span> <span class="fu">dim</span>(tt_rel2[tt_rel2<span class="sc">$</span>logFC<span class="sc">&gt;</span><span class="dv">0</span>,])[<span class="dv">1</span>]</span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>frac <span class="ot">&lt;-</span> dim_rel<span class="sc">/</span>tot_dim</span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"fraction significant:"</span>, frac, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>fraction significant: 0.04031641 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"both conditions:"</span>, dim_rel, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>both conditions: 790 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"down:"</span>, dim_down, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>down: 656 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"up:"</span>, dim_up, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>up: 134 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>dim_down<span class="sc">/</span>dim_rel</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.8303797</code></pre>
</div>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>dim_up<span class="sc">/</span>dim_rel</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1696203</code></pre>
</div>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="co">#head(tt_rel2)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>COMPARING LIMMA AND PAIRED WGCNA (lw) â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”</p>
<p>Do the miRNAs deemed important by paired WGCNA and Limma overlap?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co">#we are filtering out genes from the grey module, since those do not have a kWithin score</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a><span class="co">#We can do this since the genes in the grey module are just the ones "left over"</span></span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>hubData2 <span class="ot">&lt;-</span> hubData[hubData<span class="sc">$</span>module<span class="sc">!=</span> <span class="st">"grey"</span>,]</span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a><span class="co">#add new column to make extracting easier</span></span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a>hubData2<span class="sc">$</span>MM_self <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(hubData2[<span class="fu">cbind</span>(<span class="fu">seq_len</span>(<span class="fu">nrow</span>(hubData2)), <span class="fu">match</span>(<span class="fu">paste0</span>(<span class="st">"MM"</span>, hubData2<span class="sc">$</span>module), <span class="fu">names</span>(hubData2)))])</span>
<span id="cb54-9"><a href="#cb54-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-10"><a href="#cb54-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-11"><a href="#cb54-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-12"><a href="#cb54-12" aria-hidden="true" tabindex="-1"></a>N_comp_lw <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">200</span>,<span class="dv">150</span>,<span class="dv">100</span>,<span class="dv">50</span>)</span>
<span id="cb54-13"><a href="#cb54-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(N_comp_lw)){</span>
<span id="cb54-14"><a href="#cb54-14" aria-hidden="true" tabindex="-1"></a>  limma2_top <span class="ot">&lt;-</span> <span class="fu">rownames</span>(ord_sig_limma2)[<span class="dv">1</span><span class="sc">:</span>N_comp_lw[i]]</span>
<span id="cb54-15"><a href="#cb54-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-16"><a href="#cb54-16" aria-hidden="true" tabindex="-1"></a>  <span class="co">#WGCNA "importance" score = GS_i * |MM_i|</span></span>
<span id="cb54-17"><a href="#cb54-17" aria-hidden="true" tabindex="-1"></a>  <span class="co">#GS = gene significance: how "phenotype associated" a miRNA is</span></span>
<span id="cb54-18"><a href="#cb54-18" aria-hidden="true" tabindex="-1"></a>  <span class="co">#MM = module membership: how much a miRNA "belongs" to a module</span></span>
<span id="cb54-19"><a href="#cb54-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-20"><a href="#cb54-20" aria-hidden="true" tabindex="-1"></a>  mm_col <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"MM"</span>, hubData2<span class="sc">$</span>module)</span>
<span id="cb54-21"><a href="#cb54-21" aria-hidden="true" tabindex="-1"></a>  <span class="co">#hubData$MM_self &lt;- mapply(function(i, col) hubData[i, col], seq_len(nrow(hubData2)), mm_col)</span></span>
<span id="cb54-22"><a href="#cb54-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-23"><a href="#cb54-23" aria-hidden="true" tabindex="-1"></a>  <span class="co">#hubData2$MM_self</span></span>
<span id="cb54-24"><a href="#cb54-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-25"><a href="#cb54-25" aria-hidden="true" tabindex="-1"></a>  hubData2<span class="sc">$</span>wgcna_score <span class="ot">&lt;-</span> hubData2<span class="sc">$</span>GS <span class="sc">*</span> <span class="fu">abs</span>(hubData2<span class="sc">$</span>MM_self)</span>
<span id="cb54-26"><a href="#cb54-26" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-27"><a href="#cb54-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-28"><a href="#cb54-28" aria-hidden="true" tabindex="-1"></a>  wgcna_top <span class="ot">&lt;-</span> hubData2<span class="sc">$</span>gene[<span class="fu">order</span>(<span class="sc">-</span>hubData2<span class="sc">$</span>wgcna_score)][<span class="dv">1</span><span class="sc">:</span>N_comp_lw[i]]</span>
<span id="cb54-29"><a href="#cb54-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-30"><a href="#cb54-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"N ="</span>, N_comp_lw[i],<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb54-31"><a href="#cb54-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"Numer of overlapping miRNAs:"</span>,<span class="fu">length</span>(<span class="fu">intersect</span>(limma2_top, wgcna_top)))</span>
<span id="cb54-32"><a href="#cb54-32" aria-hidden="true" tabindex="-1"></a>  <span class="co">#head(intersect(limma_top, wgcna_top), 20)</span></span>
<span id="cb54-33"><a href="#cb54-33" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-34"><a href="#cb54-34" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-35"><a href="#cb54-35" aria-hidden="true" tabindex="-1"></a>  universe <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(ord_sig_limma2), hubData2<span class="sc">$</span>gene)</span>
<span id="cb54-36"><a href="#cb54-36" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-37"><a href="#cb54-37" aria-hidden="true" tabindex="-1"></a>  A <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">intersect</span>(limma2_top, wgcna_top))</span>
<span id="cb54-38"><a href="#cb54-38" aria-hidden="true" tabindex="-1"></a>  B <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">setdiff</span>(wgcna_top, limma2_top))</span>
<span id="cb54-39"><a href="#cb54-39" aria-hidden="true" tabindex="-1"></a>  C <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">setdiff</span>(limma2_top, wgcna_top))</span>
<span id="cb54-40"><a href="#cb54-40" aria-hidden="true" tabindex="-1"></a>  D <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">setdiff</span>(universe, <span class="fu">union</span>(limma2_top, wgcna_top)))</span>
<span id="cb54-41"><a href="#cb54-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-42"><a href="#cb54-42" aria-hidden="true" tabindex="-1"></a>  fish_comp_lw <span class="ot">&lt;-</span> <span class="fu">fisher.test</span>(<span class="fu">matrix</span>(<span class="fu">c</span>(A,B,C,D), <span class="at">nrow=</span><span class="dv">2</span>), <span class="at">alternative=</span><span class="st">"greater"</span>)</span>
<span id="cb54-43"><a href="#cb54-43" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-44"><a href="#cb54-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(fish_comp_lw)</span>
<span id="cb54-45"><a href="#cb54-45" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>N = 200 
Numer of overlapping miRNAs: 154
    Fisher's Exact Test for Count Data

data:  matrix(c(A, B, C, D), nrow = 2)
p-value &lt; 2.2e-16
alternative hypothesis: true odds ratio is greater than 1
95 percent confidence interval:
 505.0739      Inf
sample estimates:
odds ratio 
  714.8289 

N = 150 
Numer of overlapping miRNAs: 115
    Fisher's Exact Test for Count Data

data:  matrix(c(A, B, C, D), nrow = 2)
p-value &lt; 2.2e-16
alternative hypothesis: true odds ratio is greater than 1
95 percent confidence interval:
 599.5175      Inf
sample estimates:
odds ratio 
  931.1746 

N = 100 
Numer of overlapping miRNAs: 76
    Fisher's Exact Test for Count Data

data:  matrix(c(A, B, C, D), nrow = 2)
p-value &lt; 2.2e-16
alternative hypothesis: true odds ratio is greater than 1
95 percent confidence interval:
 758.6419      Inf
sample estimates:
odds ratio 
  1277.573 

N = 50 
Numer of overlapping miRNAs: 36
    Fisher's Exact Test for Count Data

data:  matrix(c(A, B, C, D), nrow = 2)
p-value &lt; 2.2e-16
alternative hypothesis: true odds ratio is greater than 1
95 percent confidence interval:
 873.3253      Inf
sample estimates:
odds ratio 
  1760.158 </code></pre>
</div>
</div>
<p>Limma camera: gene-set test for each module</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>common_samp <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(limma_data), <span class="fu">rownames</span>(datExpr_ok))</span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>design_cam <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> patient_id <span class="sc">+</span> tumour, <span class="at">data =</span> limma_data[common_samp, , <span class="at">drop =</span> <span class="cn">FALSE</span>])</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>expr_cam <span class="ot">&lt;-</span> <span class="fu">t</span>(datExpr_ok[common_samp, , <span class="at">drop =</span> <span class="cn">FALSE</span>])</span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a><span class="co">#module sets</span></span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a>moduleColors_cam_limma <span class="ot">&lt;-</span> <span class="fu">setNames</span>(hubData2<span class="sc">$</span>module, hubData2<span class="sc">$</span>gene)</span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a>moduleColors_cam_limma <span class="ot">&lt;-</span> moduleColors_cam_limma[<span class="fu">names</span>(moduleColors_cam_limma) <span class="sc">%in%</span> <span class="fu">rownames</span>(expr_cam)]</span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a>moduleSets <span class="ot">&lt;-</span> <span class="fu">split</span>(<span class="fu">names</span>(moduleColors_cam_limma), moduleColors_cam_limma)</span>
<span id="cb56-11"><a href="#cb56-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-12"><a href="#cb56-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-13"><a href="#cb56-13" aria-hidden="true" tabindex="-1"></a>contrast_idx <span class="ot">&lt;-</span> <span class="fu">match</span>(<span class="st">"tumourtumour"</span>, <span class="fu">colnames</span>(design_cam))</span>
<span id="cb56-14"><a href="#cb56-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-15"><a href="#cb56-15" aria-hidden="true" tabindex="-1"></a>camera_res <span class="ot">&lt;-</span> <span class="fu">camera</span>(expr_cam, </span>
<span id="cb56-16"><a href="#cb56-16" aria-hidden="true" tabindex="-1"></a>                     <span class="at">index =</span> moduleSets,</span>
<span id="cb56-17"><a href="#cb56-17" aria-hidden="true" tabindex="-1"></a>                     <span class="at">design =</span> design_cam,</span>
<span id="cb56-18"><a href="#cb56-18" aria-hidden="true" tabindex="-1"></a>                     <span class="at">contrast =</span> contrast_idx)</span>
<span id="cb56-19"><a href="#cb56-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-20"><a href="#cb56-20" aria-hidden="true" tabindex="-1"></a>camera_res[<span class="fu">order</span>(camera_res<span class="sc">$</span>PValue), ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>            NGenes Direction       PValue          FDR
salmon          97        Up 6.750165e-31 9.450231e-30
green         1029        Up 1.804768e-22 1.263338e-21
tan            104      Down 6.393499e-20 2.983633e-19
brown         1312      Down 1.129718e-16 3.954013e-16
cyan            95        Up 1.809098e-12 5.065474e-12
turquoise     3371      Down 8.891444e-12 2.074670e-11
pink           198        Up 2.652478e-11 5.304956e-11
red            493      Down 1.969135e-08 3.445986e-08
greenyellow    131        Up 1.954594e-05 3.040479e-05
purple         160        Up 4.166509e-05 5.833113e-05
magenta        172        Up 2.824633e-03 3.594988e-03
yellow        1215        Up 1.359593e-01 1.586192e-01
black          323        Up 2.450343e-01 2.638831e-01
blue          1648        Up 2.864469e-01 2.864469e-01</code></pre>
</div>
</div>
<p>Which modules contain the most miRNAs flagged by Limma? How likely is this result?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a><span class="co">#map limma results onto gene names</span></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>limma_map <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene        =</span> <span class="fu">rownames</span>(ord_sig_limma2),</span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">limma_FDR   =</span> ord_sig_limma2<span class="sc">$</span>adj.P.Val,</span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">limma_P     =</span> ord_sig_limma2<span class="sc">$</span>P.Value,</span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">limma_t     =</span> ord_sig_limma2<span class="sc">$</span>t,</span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">limma_logFC =</span> ord_sig_limma2<span class="sc">$</span>logFC,</span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">row.names   =</span> <span class="cn">NULL</span></span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a><span class="co">#universe: genes present in both limma + WGCNA (hubData2 already non-grey)</span></span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a>universe <span class="ot">&lt;-</span> <span class="fu">intersect</span>(limma_map<span class="sc">$</span>gene, hubData2<span class="sc">$</span>gene)</span>
<span id="cb58-15"><a href="#cb58-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-16"><a href="#cb58-16" aria-hidden="true" tabindex="-1"></a><span class="co">#join and keep only universe genes</span></span>
<span id="cb58-17"><a href="#cb58-17" aria-hidden="true" tabindex="-1"></a>comp <span class="ot">&lt;-</span> hubData2 <span class="sc">%&gt;%</span></span>
<span id="cb58-18"><a href="#cb58-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(limma_map, <span class="at">by =</span> <span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb58-19"><a href="#cb58-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(gene <span class="sc">%in%</span> universe) <span class="sc">%&gt;%</span></span>
<span id="cb58-20"><a href="#cb58-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">limma_sig =</span> <span class="sc">-</span><span class="fu">log10</span>(limma_FDR))</span>
<span id="cb58-21"><a href="#cb58-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-22"><a href="#cb58-22" aria-hidden="true" tabindex="-1"></a><span class="co">#DE set from limma (FDR &lt; 0.05 and |logFC| &gt; 1)</span></span>
<span id="cb58-23"><a href="#cb58-23" aria-hidden="true" tabindex="-1"></a>DE <span class="ot">&lt;-</span> limma_map<span class="sc">$</span>gene[limma_map<span class="sc">$</span>limma_FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(limma_map<span class="sc">$</span>limma_logFC) <span class="sc">&gt;=</span><span class="dv">1</span>]</span>
<span id="cb58-24"><a href="#cb58-24" aria-hidden="true" tabindex="-1"></a>DE <span class="ot">&lt;-</span> <span class="fu">intersect</span>(DE, universe)</span>
<span id="cb58-25"><a href="#cb58-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-26"><a href="#cb58-26" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(DE)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 774</code></pre>
</div>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"DE genes (FDR&lt;0.05):"</span>, <span class="fu">length</span>(DE), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DE genes (FDR&lt;0.05): 774 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co">#enrichment per module (hypergeometric)</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>moduleMembers <span class="ot">&lt;-</span> <span class="fu">split</span>(comp<span class="sc">$</span>gene, comp<span class="sc">$</span>module)</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a>mod_enrich <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">lapply</span>(<span class="fu">names</span>(moduleMembers), <span class="cf">function</span>(m) {</span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a>  genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(moduleMembers[[m]], universe)</span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a>  k <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">intersect</span>(genes, DE))        <span class="co"># DE inside module</span></span>
<span id="cb62-7"><a href="#cb62-7" aria-hidden="true" tabindex="-1"></a>  M <span class="ot">&lt;-</span> <span class="fu">length</span>(genes)                       <span class="co"># module size</span></span>
<span id="cb62-8"><a href="#cb62-8" aria-hidden="true" tabindex="-1"></a>  U <span class="ot">&lt;-</span> <span class="fu">length</span>(universe)                    <span class="co"># total tested</span></span>
<span id="cb62-9"><a href="#cb62-9" aria-hidden="true" tabindex="-1"></a>  n <span class="ot">&lt;-</span> <span class="fu">length</span>(DE)                          <span class="co"># total DE in universe</span></span>
<span id="cb62-10"><a href="#cb62-10" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">phyper</span>(k <span class="sc">-</span> <span class="dv">1</span>, M, U <span class="sc">-</span> M, n, <span class="at">lower.tail =</span> <span class="cn">FALSE</span>)</span>
<span id="cb62-11"><a href="#cb62-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-12"><a href="#cb62-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">module =</span> m, <span class="at">module_size =</span> M, <span class="at">DE_in_module =</span> k,</span>
<span id="cb62-13"><a href="#cb62-13" aria-hidden="true" tabindex="-1"></a>             <span class="at">frac_DE =</span> k <span class="sc">/</span> M, <span class="at">p_value =</span> p)</span>
<span id="cb62-14"><a href="#cb62-14" aria-hidden="true" tabindex="-1"></a>})) <span class="sc">%&gt;%</span></span>
<span id="cb62-15"><a href="#cb62-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FDR =</span> <span class="fu">p.adjust</span>(p_value, <span class="st">"BH"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb62-16"><a href="#cb62-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(FDR, <span class="fu">desc</span>(frac_DE))</span>
<span id="cb62-17"><a href="#cb62-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-18"><a href="#cb62-18" aria-hidden="true" tabindex="-1"></a>mod_enrich</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>        module module_size DE_in_module     frac_DE      p_value          FDR
1        brown        1312          258 0.196646341 2.368804e-55 3.316326e-54
2    turquoise        3371          390 0.115692673 1.091235e-26 7.638648e-26
3        green        1029           98 0.095238095 6.316979e-03 2.947923e-02
4         cyan          95           14 0.147368421 1.071724e-02 3.751033e-02
5          tan         104            5 0.048076923 8.973396e-01 1.000000e+00
6       salmon          97            4 0.041237113 9.386437e-01 1.000000e+00
7       purple         160            4 0.025000000 9.983135e-01 1.000000e+00
8         pink         198            1 0.005050505 9.999998e-01 1.000000e+00
9        black         323            0 0.000000000 1.000000e+00 1.000000e+00
10        blue        1648            0 0.000000000 1.000000e+00 1.000000e+00
11 greenyellow         131            0 0.000000000 1.000000e+00 1.000000e+00
12     magenta         172            0 0.000000000 1.000000e+00 1.000000e+00
13         red         493            0 0.000000000 1.000000e+00 1.000000e+00
14      yellow        1215            0 0.000000000 1.000000e+00 1.000000e+00</code></pre>
</div>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Universe size U ="</span>, <span class="fu">length</span>(universe), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Universe size U = 10348 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"DE size n ="</span>, <span class="fu">length</span>(DE), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DE size n = 774 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"DE fraction n/U ="</span>, <span class="fu">length</span>(DE)<span class="sc">/</span><span class="fu">length</span>(universe), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DE fraction n/U = 0.07479706 </code></pre>
</div>
</div>
<p>Are the hubgenes more DE?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>cor_kWithin <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(comp<span class="sc">$</span>kWithin, comp<span class="sc">$</span>limma_sig,</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="st">"spearman"</span>, <span class="at">use =</span> <span class="st">"complete.obs"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in cor.test.default(comp$kWithin, comp$limma_sig, method = "spearman",
: Cannot compute exact p-value with ties</code></pre>
</div>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a>cor_MM <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(<span class="fu">abs</span>(comp<span class="sc">$</span>MM_self), comp<span class="sc">$</span>limma_sig,</span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">method =</span> <span class="st">"spearman"</span>, <span class="at">use =</span> <span class="st">"complete.obs"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in cor.test.default(abs(comp$MM_self), comp$limma_sig, method =
"spearman", : Cannot compute exact p-value with ties</code></pre>
</div>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>cor_score <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(comp<span class="sc">$</span>wgcna_score, comp<span class="sc">$</span>limma_sig,</span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>                      <span class="at">method =</span> <span class="st">"spearman"</span>, <span class="at">use =</span> <span class="st">"complete.obs"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in cor.test.default(comp$wgcna_score, comp$limma_sig, method =
"spearman", : Cannot compute exact p-value with ties</code></pre>
</div>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>cor_kWithin</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Spearman's rank correlation rho

data:  comp$kWithin and comp$limma_sig
S = 1.2957e+11, p-value &lt; 2.2e-16
alternative hypothesis: true rho is not equal to 0
sample estimates:
   rho 
0.2984 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>cor_MM</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Spearman's rank correlation rho

data:  abs(comp$MM_self) and comp$limma_sig
S = 1.449e+11, p-value &lt; 2.2e-16
alternative hypothesis: true rho is not equal to 0
sample estimates:
     rho 
0.215389 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>cor_score</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Spearman's rank correlation rho

data:  comp$wgcna_score and comp$limma_sig
S = 971442853, p-value &lt; 2.2e-16
alternative hypothesis: true rho is not equal to 0
sample estimates:
      rho 
0.9569777 </code></pre>
</div>
</div>
<p>NEW METHOD 3: DIFFCOEXP â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€“</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co">#library(dplyr)</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a><span class="co">#library(diffcoexp)</span></span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a>pheno_diff <span class="ot">&lt;-</span> sample_traits[<span class="fu">rownames</span>(datExpr_ok), , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a>pheno_diff<span class="sc">$</span>sample_id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(pheno_diff)   <span class="co"># &lt;-- add this</span></span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">rownames</span>(pheno_diff), <span class="fu">rownames</span>(datExpr_ok)))</span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-9"><a href="#cb82-9" aria-hidden="true" tabindex="-1"></a>pheno_diff<span class="sc">$</span>tumour <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.integer</span>(pheno_diff<span class="sc">$</span>tumour), <span class="at">levels =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"normal"</span>,<span class="st">"tumour"</span>))</span>
<span id="cb82-10"><a href="#cb82-10" aria-hidden="true" tabindex="-1"></a>pheno_diff<span class="sc">$</span>patient_id <span class="ot">&lt;-</span> <span class="fu">as.character</span>(pheno_diff<span class="sc">$</span>patient_id)</span>
<span id="cb82-11"><a href="#cb82-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-12"><a href="#cb82-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-13"><a href="#cb82-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-14"><a href="#cb82-14" aria-hidden="true" tabindex="-1"></a>expr2  <span class="ot">&lt;-</span> datExpr_ok</span>
<span id="cb82-15"><a href="#cb82-15" aria-hidden="true" tabindex="-1"></a>expr2 <span class="ot">&lt;-</span> expr2[pheno_diff<span class="sc">$</span>sample_id, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb82-16"><a href="#cb82-16" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">rownames</span>(expr2), pheno_diff<span class="sc">$</span>sample_id))</span>
<span id="cb82-17"><a href="#cb82-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-18"><a href="#cb82-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-19"><a href="#cb82-19" aria-hidden="true" tabindex="-1"></a><span class="co">#filter genes for speed to enable speed</span></span>
<span id="cb82-20"><a href="#cb82-20" aria-hidden="true" tabindex="-1"></a>nDiff <span class="ot">&lt;-</span> <span class="dv">5000</span>   <span class="co"># try 3000â€“8000 depending on machine</span></span>
<span id="cb82-21"><a href="#cb82-21" aria-hidden="true" tabindex="-1"></a>vars  <span class="ot">&lt;-</span> <span class="fu">apply</span>(expr2, <span class="dv">2</span>, var, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb82-22"><a href="#cb82-22" aria-hidden="true" tabindex="-1"></a>genes_keep <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">sort</span>(vars, <span class="at">decreasing =</span> <span class="cn">TRUE</span>))[<span class="dv">1</span><span class="sc">:</span><span class="fu">min</span>(nDiff, <span class="fu">ncol</span>(expr2))]</span>
<span id="cb82-23"><a href="#cb82-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-24"><a href="#cb82-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-25"><a href="#cb82-25" aria-hidden="true" tabindex="-1"></a>expr2 <span class="ot">&lt;-</span> expr2[, genes_keep, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb82-26"><a href="#cb82-26" aria-hidden="true" tabindex="-1"></a>tab <span class="ot">&lt;-</span> <span class="fu">with</span>(pheno_diff, <span class="fu">table</span>(patient_id, tumour))</span>
<span id="cb82-27"><a href="#cb82-27" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(tab)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>            tumour
patient_id   normal tumour
  GSM1518806      1      1
  GSM1518807      1      1
  GSM1518808      1      1
  GSM1518809      1      1
  GSM1518810      1      1
  GSM1518811      1      1
  GSM1518812      1      1
  GSM1518813      1      1
  GSM1518816      1      1
  GSM1518817      1      1
  GSM1518818      1      1
  GSM1518819      1      1
  GSM1518821      1      1
  GSM1518822      1      1
  GSM1518823      1      1
  GSM1518824      1      1
  GSM1518825      1      1
  GSM1518826      1      1
  GSM1518827      1      1
  GSM1518828      1      1
  GSM1518829      1      1
  GSM1518830      1      1
  GSM1518831      1      1
  GSM1518832      1      1
  GSM1518833      1      1
  GSM1518834      1      1
  GSM1518835      1      1
  GSM1518836      1      1
  GSM1518837      1      1
  GSM1518838      1      1
  GSM1518839      1      1
  GSM1518840      1      1
  GSM1518841      1      1
  GSM1518842      1      1
  GSM1518843      1      1
  GSM1518845      1      1
  GSM1518846      1      1
  GSM1518847      1      1
  GSM1518849      1      1
  GSM1518850      1      1
  GSM1518851      1      1
  GSM1518852      1      1
  GSM1518853      1      1
  GSM1518854      1      1
  GSM1518855      1      1
  GSM1518856      1      1
  GSM1518857      1      1
  GSM1518859      1      1
  GSM1518860      1      1
  GSM1518861      1      1
  GSM1518862      1      1
  GSM1518863      1      1
  GSM1518864      1      1
  GSM1518865      1      1
  GSM1518866      1      1
  GSM1518867      1      1
  GSM1518868      1      1
  GSM1518869      1      1
  GSM1518870      1      1
  GSM1518871      1      1
  GSM1518872      1      1
  GSM1518873      1      1
  GSM1518874      1      1
  GSM1518875      1      1
  GSM1518876      1      1
  GSM1518877      1      1
  GSM1518878      1      1
  GSM1518879      1      1
  GSM1518880      1      1
  GSM1518881      1      1
  GSM1518882      1      1
  GSM1518883      1      1
  GSM1518884      1      1
  GSM1518885      1      1
  GSM1518886      1      1
  GSM1518887      1      1
  GSM1518888      1      1
  GSM1518889      1      1
  GSM1518890      1      1
  GSM1518891      1      1
  GSM1518892      1      1
  GSM1518893      1      1
  GSM1518894      1      1
  GSM1518896      1      1
  GSM1518897      1      1
  GSM1518898      1      1
  GSM1518899      1      1
  GSM1518900      1      1
  GSM1518901      1      1
  GSM1518902      1      1
  GSM1518903      1      1
  GSM1518904      1      1
  GSM1518905      1      1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>paired_pat <span class="ot">&lt;-</span> <span class="fu">rownames</span>(tab)[<span class="fu">rowSums</span>(tab <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">==</span> <span class="dv">2</span>]</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(paired_pat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "GSM1518806" "GSM1518807" "GSM1518808" "GSM1518809" "GSM1518810"
 [6] "GSM1518811" "GSM1518812" "GSM1518813" "GSM1518816" "GSM1518817"
[11] "GSM1518818" "GSM1518819" "GSM1518821" "GSM1518822" "GSM1518823"
[16] "GSM1518824" "GSM1518825" "GSM1518826" "GSM1518827" "GSM1518828"
[21] "GSM1518829" "GSM1518830" "GSM1518831" "GSM1518832" "GSM1518833"
[26] "GSM1518834" "GSM1518835" "GSM1518836" "GSM1518837" "GSM1518838"
[31] "GSM1518839" "GSM1518840" "GSM1518841" "GSM1518842" "GSM1518843"
[36] "GSM1518845" "GSM1518846" "GSM1518847" "GSM1518849" "GSM1518850"
[41] "GSM1518851" "GSM1518852" "GSM1518853" "GSM1518854" "GSM1518855"
[46] "GSM1518856" "GSM1518857" "GSM1518859" "GSM1518860" "GSM1518861"
[51] "GSM1518862" "GSM1518863" "GSM1518864" "GSM1518865" "GSM1518866"
[56] "GSM1518867" "GSM1518868" "GSM1518869" "GSM1518870" "GSM1518871"
[61] "GSM1518872" "GSM1518873" "GSM1518874" "GSM1518875" "GSM1518876"
[66] "GSM1518877" "GSM1518878" "GSM1518879" "GSM1518880" "GSM1518881"
[71] "GSM1518882" "GSM1518883" "GSM1518884" "GSM1518885" "GSM1518886"
[76] "GSM1518887" "GSM1518888" "GSM1518889" "GSM1518890" "GSM1518891"
[81] "GSM1518892" "GSM1518893" "GSM1518894" "GSM1518896" "GSM1518897"
[86] "GSM1518898" "GSM1518899" "GSM1518900" "GSM1518901" "GSM1518902"
[91] "GSM1518903" "GSM1518904" "GSM1518905"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>ph <span class="ot">&lt;-</span> pheno_diff[pheno_diff<span class="sc">$</span>patient_id <span class="sc">%in%</span> paired_pat, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a><span class="co">#now build matched sample lists</span></span>
<span id="cb86-5"><a href="#cb86-5" aria-hidden="true" tabindex="-1"></a>pid <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">unique</span>(ph<span class="sc">$</span>patient_id))</span>
<span id="cb86-6"><a href="#cb86-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-7"><a href="#cb86-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-8"><a href="#cb86-8" aria-hidden="true" tabindex="-1"></a>sN <span class="ot">&lt;-</span> ph<span class="sc">$</span>sample_id[ph<span class="sc">$</span>tumour <span class="sc">==</span> <span class="st">"normal"</span>][<span class="fu">match</span>(pid, ph<span class="sc">$</span>patient_id[ph<span class="sc">$</span>tumour <span class="sc">==</span> <span class="st">"normal"</span>])]</span>
<span id="cb86-9"><a href="#cb86-9" aria-hidden="true" tabindex="-1"></a>sT <span class="ot">&lt;-</span> ph<span class="sc">$</span>sample_id[ph<span class="sc">$</span>tumour <span class="sc">==</span> <span class="st">"tumour"</span>][<span class="fu">match</span>(pid, ph<span class="sc">$</span>patient_id[ph<span class="sc">$</span>tumour <span class="sc">==</span> <span class="st">"tumour"</span>])]</span>
<span id="cb86-10"><a href="#cb86-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-11"><a href="#cb86-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-12"><a href="#cb86-12" aria-hidden="true" tabindex="-1"></a><span class="co">#genes Ã— patients (columns are paired patients in identical order)</span></span>
<span id="cb86-13"><a href="#cb86-13" aria-hidden="true" tabindex="-1"></a>E_N <span class="ot">&lt;-</span> <span class="fu">t</span>(expr2[sN, , <span class="at">drop =</span> <span class="cn">FALSE</span>])</span>
<span id="cb86-14"><a href="#cb86-14" aria-hidden="true" tabindex="-1"></a>E_T <span class="ot">&lt;-</span> <span class="fu">t</span>(expr2[sT, , <span class="at">drop =</span> <span class="cn">FALSE</span>])</span>
<span id="cb86-15"><a href="#cb86-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-16"><a href="#cb86-16" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(E_N) <span class="ot">&lt;-</span> pid</span>
<span id="cb86-17"><a href="#cb86-17" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(E_T) <span class="ot">&lt;-</span> pid</span>
<span id="cb86-18"><a href="#cb86-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-19"><a href="#cb86-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-20"><a href="#cb86-20" aria-hidden="true" tabindex="-1"></a><span class="co">#run DiffCoExp</span></span>
<span id="cb86-21"><a href="#cb86-21" aria-hidden="true" tabindex="-1"></a><span class="co">#Spearman is slower than Pearson -&gt; use pearson</span></span>
<span id="cb86-22"><a href="#cb86-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-23"><a href="#cb86-23" aria-hidden="true" tabindex="-1"></a>fn <span class="ot">&lt;-</span> diffcoexp<span class="sc">::</span>diffcoexp</span>
<span id="cb86-24"><a href="#cb86-24" aria-hidden="true" tabindex="-1"></a>avail <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">formals</span>(fn))</span>
<span id="cb86-25"><a href="#cb86-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-26"><a href="#cb86-26" aria-hidden="true" tabindex="-1"></a>params <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb86-27"><a href="#cb86-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">r.method  =</span> <span class="st">"pearson"</span>,</span>
<span id="cb86-28"><a href="#cb86-28" aria-hidden="true" tabindex="-1"></a>  <span class="co"># The following are common in diffcoexp; will be ignored if not supported:</span></span>
<span id="cb86-29"><a href="#cb86-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">rth       =</span> <span class="fl">0.4</span>,</span>
<span id="cb86-30"><a href="#cb86-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">r.diffth  =</span> <span class="fl">0.2</span>,</span>
<span id="cb86-31"><a href="#cb86-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">qth       =</span> <span class="fl">0.2</span>,</span>
<span id="cb86-32"><a href="#cb86-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">q.diffth  =</span> <span class="fl">0.2</span>,</span>
<span id="cb86-33"><a href="#cb86-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">q.dcgth   =</span> <span class="fl">0.2</span></span>
<span id="cb86-34"><a href="#cb86-34" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb86-35"><a href="#cb86-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-36"><a href="#cb86-36" aria-hidden="true" tabindex="-1"></a>params <span class="ot">&lt;-</span> params[<span class="fu">names</span>(params) <span class="sc">%in%</span> avail]</span>
<span id="cb86-37"><a href="#cb86-37" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">do.call</span>(fn, <span class="fu">c</span>(<span class="fu">list</span>(E_N, E_T), params))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Finished running comparecor.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Finished running coexpr.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>1644340 gene pairs remain after half thresholding.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>706111 DCLs identified.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>2016 DCGs identified.</code></pre>
</div>
</div>
<p>Get parameters for paper</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"number of DCLs:"</span>, <span class="fu">length</span>(<span class="fu">rownames</span>(res<span class="sc">$</span>DCLs)),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>number of DCLs: 706111 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"number of DCGs:"</span>, <span class="fu">length</span>(<span class="fu">rownames</span>(res<span class="sc">$</span>DCGs)),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>number of DCGs: 2016 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"number of same signed DCLs:"</span>,<span class="fu">length</span>(res<span class="sc">$</span>DCL<span class="sc">$</span>type[res<span class="sc">$</span>DCL<span class="sc">$</span>type <span class="sc">==</span> <span class="st">"same signed"</span>]), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>number of same signed DCLs: 547515 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"number of diff signed DCLs:"</span>,<span class="fu">length</span>(res<span class="sc">$</span>DCL<span class="sc">$</span>type[res<span class="sc">$</span>DCL<span class="sc">$</span>type <span class="sc">==</span> <span class="st">"diff signed"</span>]),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>number of diff signed DCLs: 158011 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"number of switched opposites DCLs:"</span>,<span class="fu">length</span>(res<span class="sc">$</span>DCL<span class="sc">$</span>type[res<span class="sc">$</span>DCL<span class="sc">$</span>type <span class="sc">==</span> <span class="st">"switched opposites"</span>]),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>number of switched opposites DCLs: 585 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="fu">max</span>(<span class="fu">abs</span>(res<span class="sc">$</span>DCLs<span class="sc">$</span>cor.diff))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1.207186</code></pre>
</div>
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">min</span>(res<span class="sc">$</span>DCLs<span class="sc">$</span>q.diffcor)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0</code></pre>
</div>
</div>
<p>COMPARISON DIFFCOEXP AND PAIRED WGCNA (pw) â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”-</p>
<p>â€œDCG-nessâ€ (DCL degree) with WGCNA hubness (kWithin)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a>dcl <span class="ot">&lt;-</span> res<span class="sc">$</span>DCLs</span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a>hubData2 <span class="ot">&lt;-</span> hubData[hubData<span class="sc">$</span>module<span class="sc">!=</span> <span class="st">"grey"</span>,]</span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a>genes_wgcna <span class="ot">&lt;-</span> hubData2<span class="sc">$</span>gene</span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a>dcl <span class="ot">&lt;-</span> dcl[dcl<span class="sc">$</span>Gene<span class="fl">.1</span> <span class="sc">%in%</span> genes_wgcna <span class="sc">&amp;</span> dcl<span class="sc">$</span>Gene<span class="fl">.2</span> <span class="sc">%in%</span> genes_wgcna, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb106-6"><a href="#cb106-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-7"><a href="#cb106-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-8"><a href="#cb106-8" aria-hidden="true" tabindex="-1"></a>dcg_degree_exp <span class="ot">&lt;-</span> <span class="fu">table</span>(<span class="fu">c</span>(dcl<span class="sc">$</span>Gene<span class="fl">.1</span>, dcl<span class="sc">$</span>Gene<span class="fl">.2</span>))</span>
<span id="cb106-9"><a href="#cb106-9" aria-hidden="true" tabindex="-1"></a>dcg_degree_exp <span class="ot">&lt;-</span> dcg_degree_exp[<span class="fu">names</span>(dcg_degree_exp) <span class="sc">%in%</span> genes_wgcna]</span>
<span id="cb106-10"><a href="#cb106-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-11"><a href="#cb106-11" aria-hidden="true" tabindex="-1"></a>deg_df_exp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb106-12"><a href="#cb106-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene =</span> <span class="fu">names</span>(dcg_degree_exp),</span>
<span id="cb106-13"><a href="#cb106-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">dcl_degree =</span> <span class="fu">as.numeric</span>(dcg_degree_exp),</span>
<span id="cb106-14"><a href="#cb106-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">row.names =</span> <span class="cn">NULL</span></span>
<span id="cb106-15"><a href="#cb106-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb106-16"><a href="#cb106-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-17"><a href="#cb106-17" aria-hidden="true" tabindex="-1"></a>comp_dcg_exp <span class="ot">&lt;-</span> <span class="fu">merge</span>(hubData2[, <span class="fu">c</span>(<span class="st">"gene"</span>,<span class="st">"kWithin"</span>,<span class="st">"module"</span>)], deg_df_exp, <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb106-18"><a href="#cb106-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-19"><a href="#cb106-19" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(comp_dcg_exp<span class="sc">$</span>kWithin, comp_dcg_exp<span class="sc">$</span>dcl_degree, <span class="at">method =</span> <span class="st">"spearman"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in cor.test.default(comp_dcg_exp$kWithin, comp_dcg_exp$dcl_degree, :
Cannot compute exact p-value with ties</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Spearman's rank correlation rho

data:  comp_dcg_exp$kWithin and comp_dcg_exp$dcl_degree
S = 4.718e+09, p-value &lt; 2.2e-16
alternative hypothesis: true rho is not equal to 0
sample estimates:
      rho 
0.2960477 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(comp_dcg_exp<span class="sc">$</span>kWithin, comp_dcg_exp<span class="sc">$</span>dcl_degree_exp, <span class="at">pch =</span> <span class="dv">20</span>,</span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"WGCNA kWithin"</span>, <span class="at">ylab =</span> <span class="st">"DiffCoExp DCL degree"</span>,</span>
<span id="cb109-3"><a href="#cb109-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"Are WGCNA hubs also DiffCoExp hubs?"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SD_WGCNA_comp_all_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Overlap of top hubs</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a>topK <span class="ot">&lt;-</span> <span class="dv">100</span></span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-3"><a href="#cb110-3" aria-hidden="true" tabindex="-1"></a>common_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(hubData2<span class="sc">$</span>gene, <span class="fu">names</span>(dcg_degree_exp))</span>
<span id="cb110-4"><a href="#cb110-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-5"><a href="#cb110-5" aria-hidden="true" tabindex="-1"></a><span class="co">#WGCNA hubs among genes that have DCL degree</span></span>
<span id="cb110-6"><a href="#cb110-6" aria-hidden="true" tabindex="-1"></a>hubs_wgcna_common <span class="ot">&lt;-</span> hubData2<span class="sc">$</span>gene[hubData2<span class="sc">$</span>gene <span class="sc">%in%</span> common_genes]</span>
<span id="cb110-7"><a href="#cb110-7" aria-hidden="true" tabindex="-1"></a>hubs_wgcna_common <span class="ot">&lt;-</span> hubs_wgcna_common[<span class="fu">order</span>(<span class="sc">-</span>hubData2<span class="sc">$</span>kWithin[<span class="fu">match</span>(hubs_wgcna_common, hubData2<span class="sc">$</span>gene)])]</span>
<span id="cb110-8"><a href="#cb110-8" aria-hidden="true" tabindex="-1"></a>hubs_wgcna_common <span class="ot">&lt;-</span> <span class="fu">head</span>(hubs_wgcna_common, topK)</span>
<span id="cb110-9"><a href="#cb110-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-10"><a href="#cb110-10" aria-hidden="true" tabindex="-1"></a><span class="co">#diffCoExp hubs </span></span>
<span id="cb110-11"><a href="#cb110-11" aria-hidden="true" tabindex="-1"></a>hubs_diff_common <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">sort</span>(dcg_degree_exp[common_genes], <span class="at">decreasing =</span> <span class="cn">TRUE</span>))[<span class="dv">1</span><span class="sc">:</span>topK]</span>
<span id="cb110-12"><a href="#cb110-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-13"><a href="#cb110-13" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">intersect</span>(hubs_wgcna_common, hubs_diff_common))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 4</code></pre>
</div>
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">intersect</span>(hubs_wgcna_common, hubs_diff_common), <span class="dv">50</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "C7"    "GSTP1" "FLI1"  "MFAP4"</code></pre>
</div>
</div>
<p>Are DCLs mostly within the same WGCNA module?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="co">#gene -&gt; module mapping</span></span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">setNames</span>(hubData2<span class="sc">$</span>module, hubData2<span class="sc">$</span>gene)</span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-5"><a href="#cb114-5" aria-hidden="true" tabindex="-1"></a>dcl<span class="sc">$</span>mod1 <span class="ot">&lt;-</span> mod[dcl<span class="sc">$</span>Gene<span class="fl">.1</span>]</span>
<span id="cb114-6"><a href="#cb114-6" aria-hidden="true" tabindex="-1"></a>dcl<span class="sc">$</span>mod2 <span class="ot">&lt;-</span> mod[dcl<span class="sc">$</span>Gene<span class="fl">.2</span>]</span>
<span id="cb114-7"><a href="#cb114-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-8"><a href="#cb114-8" aria-hidden="true" tabindex="-1"></a>dcls2 <span class="ot">&lt;-</span> <span class="fu">subset</span>(dcl, <span class="sc">!</span><span class="fu">is.na</span>(mod1) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(mod2))</span>
<span id="cb114-9"><a href="#cb114-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-10"><a href="#cb114-10" aria-hidden="true" tabindex="-1"></a>within <span class="ot">&lt;-</span> <span class="fu">mean</span>(dcls2<span class="sc">$</span>mod1 <span class="sc">==</span> dcls2<span class="sc">$</span>mod2)</span>
<span id="cb114-11"><a href="#cb114-11" aria-hidden="true" tabindex="-1"></a>within</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.4015465</code></pre>
</div>
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(dcls2<span class="sc">$</span>mod1 <span class="sc">==</span> dcls2<span class="sc">$</span>mod2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
 FALSE   TRUE 
273443 183473 </code></pre>
</div>
</div>
<p>Are DCLs â€œfocusedâ€ on the â€œtumourâ€ module?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a>tumMod <span class="ot">&lt;-</span> <span class="st">"green"</span></span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-3"><a href="#cb118-3" aria-hidden="true" tabindex="-1"></a><span class="co">#non-grey WGCNA genes</span></span>
<span id="cb118-4"><a href="#cb118-4" aria-hidden="true" tabindex="-1"></a>universe_genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(hubData2<span class="sc">$</span>gene)     </span>
<span id="cb118-5"><a href="#cb118-5" aria-hidden="true" tabindex="-1"></a>U <span class="ot">&lt;-</span> <span class="fu">length</span>(universe_genes)</span>
<span id="cb118-6"><a href="#cb118-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-7"><a href="#cb118-7" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">sum</span>(hubData2<span class="sc">$</span>module <span class="sc">==</span> tumMod)</span>
<span id="cb118-8"><a href="#cb118-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-9"><a href="#cb118-9" aria-hidden="true" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">nrow</span>(dcls2)</span>
<span id="cb118-10"><a href="#cb118-10" aria-hidden="true" tabindex="-1"></a>obs_touch    <span class="ot">&lt;-</span> <span class="fu">sum</span>(dcls2<span class="sc">$</span>mod1 <span class="sc">==</span> tumMod <span class="sc">|</span> dcls2<span class="sc">$</span>mod2 <span class="sc">==</span> tumMod)</span>
<span id="cb118-11"><a href="#cb118-11" aria-hidden="true" tabindex="-1"></a>obs_internal <span class="ot">&lt;-</span> <span class="fu">sum</span>(dcls2<span class="sc">$</span>mod1 <span class="sc">==</span> tumMod <span class="sc">&amp;</span> dcls2<span class="sc">$</span>mod2 <span class="sc">==</span> tumMod)</span>
<span id="cb118-12"><a href="#cb118-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-13"><a href="#cb118-13" aria-hidden="true" tabindex="-1"></a>p_touch_exp    <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">-</span> <span class="fu">choose</span>(U <span class="sc">-</span> m, <span class="dv">2</span>) <span class="sc">/</span> <span class="fu">choose</span>(U, <span class="dv">2</span>)</span>
<span id="cb118-14"><a href="#cb118-14" aria-hidden="true" tabindex="-1"></a>p_internal_exp <span class="ot">&lt;-</span> <span class="fu">choose</span>(m, <span class="dv">2</span>) <span class="sc">/</span> <span class="fu">choose</span>(U, <span class="dv">2</span>)</span>
<span id="cb118-15"><a href="#cb118-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-16"><a href="#cb118-16" aria-hidden="true" tabindex="-1"></a><span class="co">#enrichment (module-size adjusted)</span></span>
<span id="cb118-17"><a href="#cb118-17" aria-hidden="true" tabindex="-1"></a>enrich_touch    <span class="ot">&lt;-</span> (obs_touch <span class="sc">/</span> n) <span class="sc">/</span> p_touch_exp</span>
<span id="cb118-18"><a href="#cb118-18" aria-hidden="true" tabindex="-1"></a>enrich_internal <span class="ot">&lt;-</span> (obs_internal <span class="sc">/</span> n) <span class="sc">/</span> p_internal_exp</span>
<span id="cb118-19"><a href="#cb118-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-20"><a href="#cb118-20" aria-hidden="true" tabindex="-1"></a>enrich_touch</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.7523547</code></pre>
</div>
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a>enrich_internal</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2.487744</code></pre>
</div>
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="co">#significance vs module-size expectation</span></span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a><span class="fu">binom.test</span>(obs_touch,    n, p_touch_exp,    <span class="at">alternative =</span> <span class="st">"greater"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Exact binomial test

data:  obs_touch and n
number of successes = 64971, number of trials = 456916, p-value = 1
alternative hypothesis: true probability of success is greater than 0.1889995
95 percent confidence interval:
 0.1413454 1.0000000
sample estimates:
probability of success 
             0.1421946 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a><span class="fu">binom.test</span>(obs_internal, n, p_internal_exp, <span class="at">alternative =</span> <span class="st">"greater"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Exact binomial test

data:  obs_internal and n
number of successes = 11230, number of trials = 456916, p-value &lt;
2.2e-16
alternative hypothesis: true probability of success is greater than 0.00987956
95 percent confidence interval:
 0.02420219 1.00000000
sample estimates:
probability of success 
            0.02457782 </code></pre>
</div>
</div>
<p>NEW METHOD 4: GLASSO â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€“</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb126"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(huge)</span>
<span id="cb126-2"><a href="#cb126-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-3"><a href="#cb126-3" aria-hidden="true" tabindex="-1"></a><span class="co">#phenotype aligned to datExpr_ok</span></span>
<span id="cb126-4"><a href="#cb126-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-5"><a href="#cb126-5" aria-hidden="true" tabindex="-1"></a>pheno_graph <span class="ot">&lt;-</span> sample_traits[<span class="fu">rownames</span>(datExpr_ok), , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb126-6"><a href="#cb126-6" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">rownames</span>(pheno_graph), <span class="fu">rownames</span>(datExpr_ok)))</span>
<span id="cb126-7"><a href="#cb126-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-8"><a href="#cb126-8" aria-hidden="true" tabindex="-1"></a>pheno_graph<span class="sc">$</span>tumour <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.integer</span>(pheno_graph<span class="sc">$</span>tumour),</span>
<span id="cb126-9"><a href="#cb126-9" aria-hidden="true" tabindex="-1"></a>                            <span class="at">levels =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"normal"</span>,<span class="st">"tumour"</span>))</span>
<span id="cb126-10"><a href="#cb126-10" aria-hidden="true" tabindex="-1"></a>pheno_graph<span class="sc">$</span>patient_id <span class="ot">&lt;-</span> <span class="fu">as.character</span>(pheno_graph<span class="sc">$</span>patient_id)</span>
<span id="cb126-11"><a href="#cb126-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-12"><a href="#cb126-12" aria-hidden="true" tabindex="-1"></a>expr_graph  <span class="ot">&lt;-</span> datExpr_ok</span>
<span id="cb126-13"><a href="#cb126-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-14"><a href="#cb126-14" aria-hidden="true" tabindex="-1"></a><span class="co">#Gene filtering for feasibility</span></span>
<span id="cb126-15"><a href="#cb126-15" aria-hidden="true" tabindex="-1"></a>p_keep <span class="ot">&lt;-</span> <span class="dv">500</span>   <span class="co"># try 500â€“2000 first; 5000 is usually too big for glasso</span></span>
<span id="cb126-16"><a href="#cb126-16" aria-hidden="true" tabindex="-1"></a>vars <span class="ot">&lt;-</span> <span class="fu">apply</span>(expr_graph, <span class="dv">2</span>, var, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb126-17"><a href="#cb126-17" aria-hidden="true" tabindex="-1"></a>genes_keep <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">sort</span>(vars, <span class="at">decreasing =</span> <span class="cn">TRUE</span>))[<span class="dv">1</span><span class="sc">:</span><span class="fu">min</span>(p_keep, <span class="fu">ncol</span>(expr_graph))]</span>
<span id="cb126-18"><a href="#cb126-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-19"><a href="#cb126-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-20"><a href="#cb126-20" aria-hidden="true" tabindex="-1"></a><span class="co">#split samples by condition</span></span>
<span id="cb126-21"><a href="#cb126-21" aria-hidden="true" tabindex="-1"></a>sN <span class="ot">&lt;-</span> <span class="fu">rownames</span>(pheno_graph)[pheno_graph<span class="sc">$</span>tumour <span class="sc">==</span> <span class="st">"normal"</span>]</span>
<span id="cb126-22"><a href="#cb126-22" aria-hidden="true" tabindex="-1"></a>sT <span class="ot">&lt;-</span> <span class="fu">rownames</span>(pheno_graph)[pheno_graph<span class="sc">$</span>tumour <span class="sc">==</span> <span class="st">"tumour"</span>]</span>
<span id="cb126-23"><a href="#cb126-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-24"><a href="#cb126-24" aria-hidden="true" tabindex="-1"></a>X_N <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(expr_graph[sN, genes_keep, <span class="at">drop =</span> <span class="cn">FALSE</span>])  <span class="co"># samples x genes</span></span>
<span id="cb126-25"><a href="#cb126-25" aria-hidden="true" tabindex="-1"></a>X_T <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(expr_graph[sT, genes_keep, <span class="at">drop =</span> <span class="cn">FALSE</span>])</span>
<span id="cb126-26"><a href="#cb126-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-27"><a href="#cb126-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-28"><a href="#cb126-28" aria-hidden="true" tabindex="-1"></a>X_N <span class="ot">&lt;-</span> <span class="fu">scale</span>(X_N)</span>
<span id="cb126-29"><a href="#cb126-29" aria-hidden="true" tabindex="-1"></a>X_T <span class="ot">&lt;-</span> <span class="fu">scale</span>(X_T)</span>
<span id="cb126-30"><a href="#cb126-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-31"><a href="#cb126-31" aria-hidden="true" tabindex="-1"></a><span class="co">#fit sparse networks + StARS select lambda</span></span>
<span id="cb126-32"><a href="#cb126-32" aria-hidden="true" tabindex="-1"></a>nl <span class="ot">&lt;-</span> <span class="dv">10</span>  </span>
<span id="cb126-33"><a href="#cb126-33" aria-hidden="true" tabindex="-1"></a>fitN <span class="ot">&lt;-</span> <span class="fu">huge</span>(X_N, <span class="at">method =</span> <span class="st">"glasso"</span>, <span class="at">nlambda =</span> nl)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 9%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 19%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 30%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 40%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 50%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 60%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 70%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 80%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 90%
Conducting the graphical lasso (glasso)....done.                                          </code></pre>
</div>
<div class="sourceCode cell-code" id="cb128"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1" aria-hidden="true" tabindex="-1"></a>selN <span class="ot">&lt;-</span> <span class="fu">huge.select</span>(fitN, <span class="at">criterion =</span> <span class="st">"stars"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Conducting Subsampling....in progress:5% 
Conducting Subsampling....in progress:10% 
Conducting Subsampling....in progress:15% 
Conducting Subsampling....in progress:20% 
Conducting Subsampling....in progress:25% 
Conducting Subsampling....in progress:30% 
Conducting Subsampling....in progress:35% 
Conducting Subsampling....in progress:40% 
Conducting Subsampling....in progress:45% 
Conducting Subsampling....in progress:50% 
Conducting Subsampling....in progress:55% 
Conducting Subsampling....in progress:60% 
Conducting Subsampling....in progress:65% 
Conducting Subsampling....in progress:70% 
Conducting Subsampling....in progress:75% 
Conducting Subsampling....in progress:80% 
Conducting Subsampling....in progress:85% 
Conducting Subsampling....in progress:90% 
Conducting Subsampling....in progress:95% 
Conducting Subsampling....in progress:100% 
Conducting Subsampling....done.                  </code></pre>
</div>
<div class="sourceCode cell-code" id="cb130"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a>ThetaN <span class="ot">&lt;-</span> selN<span class="sc">$</span>opt.icov</span>
<span id="cb130-2"><a href="#cb130-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-3"><a href="#cb130-3" aria-hidden="true" tabindex="-1"></a>fitT <span class="ot">&lt;-</span> <span class="fu">huge</span>(X_T, <span class="at">method =</span> <span class="st">"glasso"</span>, <span class="at">nlambda =</span> nl)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 9%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 19%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 30%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 40%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 50%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 60%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 70%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 80%
Conducting the graphical lasso (glasso) wtih lossless screening....in progress: 90%
Conducting the graphical lasso (glasso)....done.                                          </code></pre>
</div>
<div class="sourceCode cell-code" id="cb132"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb132-1"><a href="#cb132-1" aria-hidden="true" tabindex="-1"></a>selT <span class="ot">&lt;-</span> <span class="fu">huge.select</span>(fitT, <span class="at">criterion =</span> <span class="st">"stars"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Conducting Subsampling....in progress:5% 
Conducting Subsampling....in progress:10% 
Conducting Subsampling....in progress:15% 
Conducting Subsampling....in progress:20% 
Conducting Subsampling....in progress:25% 
Conducting Subsampling....in progress:30% 
Conducting Subsampling....in progress:35% 
Conducting Subsampling....in progress:40% 
Conducting Subsampling....in progress:45% 
Conducting Subsampling....in progress:50% 
Conducting Subsampling....in progress:55% 
Conducting Subsampling....in progress:60% 
Conducting Subsampling....in progress:65% 
Conducting Subsampling....in progress:70% 
Conducting Subsampling....in progress:75% 
Conducting Subsampling....in progress:80% 
Conducting Subsampling....in progress:85% 
Conducting Subsampling....in progress:90% 
Conducting Subsampling....in progress:95% 
Conducting Subsampling....in progress:100% 
Conducting Subsampling....done.                  </code></pre>
</div>
<div class="sourceCode cell-code" id="cb134"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb134-1"><a href="#cb134-1" aria-hidden="true" tabindex="-1"></a>ThetaT <span class="ot">&lt;-</span> selT<span class="sc">$</span>opt.icov</span>
<span id="cb134-2"><a href="#cb134-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-3"><a href="#cb134-3" aria-hidden="true" tabindex="-1"></a><span class="co">#keep names for downstream comparisons</span></span>
<span id="cb134-4"><a href="#cb134-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(ThetaN) <span class="ot">&lt;-</span> <span class="fu">list</span>(genes_keep, genes_keep)</span>
<span id="cb134-5"><a href="#cb134-5" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(ThetaT) <span class="ot">&lt;-</span> <span class="fu">list</span>(genes_keep, genes_keep)</span>
<span id="cb134-6"><a href="#cb134-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-7"><a href="#cb134-7" aria-hidden="true" tabindex="-1"></a><span class="co">#adjacency (nonzero off-diagonal entries)</span></span>
<span id="cb134-8"><a href="#cb134-8" aria-hidden="true" tabindex="-1"></a>AdjN <span class="ot">&lt;-</span> (<span class="fu">abs</span>(ThetaN) <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">*</span> <span class="dv">1</span>; <span class="fu">diag</span>(AdjN) <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb134-9"><a href="#cb134-9" aria-hidden="true" tabindex="-1"></a>AdjT <span class="ot">&lt;-</span> (<span class="fu">abs</span>(ThetaT) <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">*</span> <span class="dv">1</span>; <span class="fu">diag</span>(AdjT) <span class="ot">&lt;-</span> <span class="dv">0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Parameters for paper</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb135"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" aria-hidden="true" tabindex="-1"></a>prec_to_pcor <span class="ot">&lt;-</span> <span class="cf">function</span>(Theta) {</span>
<span id="cb135-2"><a href="#cb135-2" aria-hidden="true" tabindex="-1"></a>  D <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="fu">diag</span>(Theta))</span>
<span id="cb135-3"><a href="#cb135-3" aria-hidden="true" tabindex="-1"></a>  P <span class="ot">&lt;-</span> <span class="sc">-</span>Theta <span class="sc">/</span> (D <span class="sc">%o%</span> D)</span>
<span id="cb135-4"><a href="#cb135-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">diag</span>(P) <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb135-5"><a href="#cb135-5" aria-hidden="true" tabindex="-1"></a>  P</span>
<span id="cb135-6"><a href="#cb135-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb135-7"><a href="#cb135-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-8"><a href="#cb135-8" aria-hidden="true" tabindex="-1"></a>network_summary <span class="ot">&lt;-</span> <span class="cf">function</span>(Theta, <span class="at">eps =</span> <span class="fl">1e-8</span>) {</span>
<span id="cb135-9"><a href="#cb135-9" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ncol</span>(Theta)</span>
<span id="cb135-10"><a href="#cb135-10" aria-hidden="true" tabindex="-1"></a>  Adj <span class="ot">&lt;-</span> (<span class="fu">abs</span>(Theta) <span class="sc">&gt;</span> eps) <span class="sc">*</span> <span class="dv">1</span></span>
<span id="cb135-11"><a href="#cb135-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">diag</span>(Adj) <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb135-12"><a href="#cb135-12" aria-hidden="true" tabindex="-1"></a>  E <span class="ot">&lt;-</span> <span class="fu">sum</span>(Adj) <span class="sc">/</span> <span class="dv">2</span></span>
<span id="cb135-13"><a href="#cb135-13" aria-hidden="true" tabindex="-1"></a>  density <span class="ot">&lt;-</span> E <span class="sc">/</span> (p <span class="sc">*</span> (p <span class="sc">-</span> <span class="dv">1</span>) <span class="sc">/</span> <span class="dv">2</span>)</span>
<span id="cb135-14"><a href="#cb135-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">p =</span> p, <span class="at">E =</span> E, <span class="at">density =</span> density, <span class="at">Adj =</span> Adj)</span>
<span id="cb135-15"><a href="#cb135-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb135-16"><a href="#cb135-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-17"><a href="#cb135-17" aria-hidden="true" tabindex="-1"></a>sumN <span class="ot">&lt;-</span> <span class="fu">network_summary</span>(ThetaN)</span>
<span id="cb135-18"><a href="#cb135-18" aria-hidden="true" tabindex="-1"></a>sumT <span class="ot">&lt;-</span> <span class="fu">network_summary</span>(ThetaT)</span>
<span id="cb135-19"><a href="#cb135-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-20"><a href="#cb135-20" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Normal edges:"</span>, sumN<span class="sc">$</span>E, <span class="st">" density:"</span>, <span class="fu">round</span>(sumN<span class="sc">$</span>density, <span class="dv">4</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Normal edges: 7012  density: 0.0562 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb137"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb137-1"><a href="#cb137-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Tumour edges:"</span>, sumT<span class="sc">$</span>E, <span class="st">" density:"</span>, <span class="fu">round</span>(sumT<span class="sc">$</span>density, <span class="dv">4</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Tumour edges: 7327  density: 0.0587 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb139"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb139-1"><a href="#cb139-1" aria-hidden="true" tabindex="-1"></a><span class="co">#gained/lost edges (tumour vs normal)</span></span>
<span id="cb139-2"><a href="#cb139-2" aria-hidden="true" tabindex="-1"></a>AdjN <span class="ot">&lt;-</span> sumN<span class="sc">$</span>Adj</span>
<span id="cb139-3"><a href="#cb139-3" aria-hidden="true" tabindex="-1"></a>AdjT <span class="ot">&lt;-</span> sumT<span class="sc">$</span>Adj</span>
<span id="cb139-4"><a href="#cb139-4" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">colnames</span>(X_N)   </span>
<span id="cb139-5"><a href="#cb139-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb139-6"><a href="#cb139-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(AdjN) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(AdjN) <span class="ot">&lt;-</span> genes</span>
<span id="cb139-7"><a href="#cb139-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(AdjT) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(AdjT) <span class="ot">&lt;-</span> genes</span>
<span id="cb139-8"><a href="#cb139-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb139-9"><a href="#cb139-9" aria-hidden="true" tabindex="-1"></a>gain_idx <span class="ot">&lt;-</span> <span class="fu">which</span>(AdjT <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> AdjN <span class="sc">==</span> <span class="dv">0</span>, <span class="at">arr.ind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb139-10"><a href="#cb139-10" aria-hidden="true" tabindex="-1"></a>loss_idx <span class="ot">&lt;-</span> <span class="fu">which</span>(AdjT <span class="sc">==</span> <span class="dv">0</span> <span class="sc">&amp;</span> AdjN <span class="sc">==</span> <span class="dv">1</span>, <span class="at">arr.ind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb139-11"><a href="#cb139-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb139-12"><a href="#cb139-12" aria-hidden="true" tabindex="-1"></a><span class="co">#keep upper triangle only</span></span>
<span id="cb139-13"><a href="#cb139-13" aria-hidden="true" tabindex="-1"></a>gain_idx <span class="ot">&lt;-</span> gain_idx[gain_idx[,<span class="dv">1</span>] <span class="sc">&lt;</span> gain_idx[,<span class="dv">2</span>], , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb139-14"><a href="#cb139-14" aria-hidden="true" tabindex="-1"></a>loss_idx <span class="ot">&lt;-</span> loss_idx[loss_idx[,<span class="dv">1</span>] <span class="sc">&lt;</span> loss_idx[,<span class="dv">2</span>], , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb139-15"><a href="#cb139-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb139-16"><a href="#cb139-16" aria-hidden="true" tabindex="-1"></a>gain_edges <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb139-17"><a href="#cb139-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">miRNA1 =</span> <span class="fu">rownames</span>(AdjT)[gain_idx[,<span class="dv">1</span>]],</span>
<span id="cb139-18"><a href="#cb139-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">miRNA2 =</span> <span class="fu">colnames</span>(AdjT)[gain_idx[,<span class="dv">2</span>]],</span>
<span id="cb139-19"><a href="#cb139-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">change =</span> <span class="st">"gained_in_tumour"</span></span>
<span id="cb139-20"><a href="#cb139-20" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb139-21"><a href="#cb139-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb139-22"><a href="#cb139-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb139-23"><a href="#cb139-23" aria-hidden="true" tabindex="-1"></a>loss_edges <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb139-24"><a href="#cb139-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">miRNA1 =</span> <span class="fu">rownames</span>(AdjN)[loss_idx[,<span class="dv">1</span>]],</span>
<span id="cb139-25"><a href="#cb139-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">miRNA2 =</span> <span class="fu">colnames</span>(AdjN)[loss_idx[,<span class="dv">2</span>]],</span>
<span id="cb139-26"><a href="#cb139-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">change =</span> <span class="st">"lost_in_tumour"</span></span>
<span id="cb139-27"><a href="#cb139-27" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb139-28"><a href="#cb139-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb139-29"><a href="#cb139-29" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Gained in tumour:"</span>, <span class="fu">nrow</span>(gain_edges), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Gained in tumour: 5608 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb141"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb141-1"><a href="#cb141-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Lost in tumour:"</span>, <span class="fu">nrow</span>(loss_edges), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Lost in tumour: 5293 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb143"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb143-1"><a href="#cb143-1" aria-hidden="true" tabindex="-1"></a><span class="co">#hubness: degree and strength </span></span>
<span id="cb143-2"><a href="#cb143-2" aria-hidden="true" tabindex="-1"></a><span class="co">#degree = number of non-zero neighbours</span></span>
<span id="cb143-3"><a href="#cb143-3" aria-hidden="true" tabindex="-1"></a>degN <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(AdjN)</span>
<span id="cb143-4"><a href="#cb143-4" aria-hidden="true" tabindex="-1"></a>degT <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(AdjT)</span>
<span id="cb143-5"><a href="#cb143-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-6"><a href="#cb143-6" aria-hidden="true" tabindex="-1"></a><span class="co">#strength = sum of abs(partial correlations) over neighbours</span></span>
<span id="cb143-7"><a href="#cb143-7" aria-hidden="true" tabindex="-1"></a>PcorN <span class="ot">&lt;-</span> <span class="fu">prec_to_pcor</span>(ThetaN)</span>
<span id="cb143-8"><a href="#cb143-8" aria-hidden="true" tabindex="-1"></a>PcorT <span class="ot">&lt;-</span> <span class="fu">prec_to_pcor</span>(ThetaT)</span>
<span id="cb143-9"><a href="#cb143-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(PcorN) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(PcorN) <span class="ot">&lt;-</span> genes</span>
<span id="cb143-10"><a href="#cb143-10" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(PcorT) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(PcorT) <span class="ot">&lt;-</span> genes</span>
<span id="cb143-11"><a href="#cb143-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-12"><a href="#cb143-12" aria-hidden="true" tabindex="-1"></a>strengthN <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">abs</span>(PcorN) <span class="sc">*</span> AdjN)   </span>
<span id="cb143-13"><a href="#cb143-13" aria-hidden="true" tabindex="-1"></a>strengthT <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">abs</span>(PcorT) <span class="sc">*</span> AdjT)</span>
<span id="cb143-14"><a href="#cb143-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-15"><a href="#cb143-15" aria-hidden="true" tabindex="-1"></a>hub_df <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb143-16"><a href="#cb143-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">miRNA =</span> genes,</span>
<span id="cb143-17"><a href="#cb143-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">degree_normal =</span> degN,</span>
<span id="cb143-18"><a href="#cb143-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">degree_tumour =</span> degT,</span>
<span id="cb143-19"><a href="#cb143-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">strength_normal =</span> strengthN,</span>
<span id="cb143-20"><a href="#cb143-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">strength_tumour =</span> strengthT,</span>
<span id="cb143-21"><a href="#cb143-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">degree_delta =</span> degT <span class="sc">-</span> degN,</span>
<span id="cb143-22"><a href="#cb143-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">strength_delta =</span> strengthT <span class="sc">-</span> strengthN</span>
<span id="cb143-23"><a href="#cb143-23" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb143-24"><a href="#cb143-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-25"><a href="#cb143-25" aria-hidden="true" tabindex="-1"></a><span class="co">#top hubs in each condition</span></span>
<span id="cb143-26"><a href="#cb143-26" aria-hidden="true" tabindex="-1"></a>top_n <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb143-27"><a href="#cb143-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-28"><a href="#cb143-28" aria-hidden="true" tabindex="-1"></a>top_deg_normal <span class="ot">&lt;-</span> hub_df <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(degree_normal)) <span class="sc">%&gt;%</span> <span class="fu">slice_head</span>(<span class="at">n =</span> top_n)</span>
<span id="cb143-29"><a href="#cb143-29" aria-hidden="true" tabindex="-1"></a>top_deg_tumour <span class="ot">&lt;-</span> hub_df <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(degree_tumour)) <span class="sc">%&gt;%</span> <span class="fu">slice_head</span>(<span class="at">n =</span> top_n)</span>
<span id="cb143-30"><a href="#cb143-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-31"><a href="#cb143-31" aria-hidden="true" tabindex="-1"></a>top_str_normal <span class="ot">&lt;-</span> hub_df <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(strength_normal)) <span class="sc">%&gt;%</span> <span class="fu">slice_head</span>(<span class="at">n =</span> top_n)</span>
<span id="cb143-32"><a href="#cb143-32" aria-hidden="true" tabindex="-1"></a>top_str_tumour <span class="ot">&lt;-</span> hub_df <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(strength_tumour)) <span class="sc">%&gt;%</span> <span class="fu">slice_head</span>(<span class="at">n =</span> top_n)</span>
<span id="cb143-33"><a href="#cb143-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-34"><a href="#cb143-34" aria-hidden="true" tabindex="-1"></a><span class="co">#hubs that increase most in tumour</span></span>
<span id="cb143-35"><a href="#cb143-35" aria-hidden="true" tabindex="-1"></a>top_degree_gain <span class="ot">&lt;-</span> hub_df <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(degree_delta)) <span class="sc">%&gt;%</span> <span class="fu">slice_head</span>(<span class="at">n =</span> top_n)</span>
<span id="cb143-36"><a href="#cb143-36" aria-hidden="true" tabindex="-1"></a>top_strength_gain <span class="ot">&lt;-</span> hub_df <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(strength_delta)) <span class="sc">%&gt;%</span> <span class="fu">slice_head</span>(<span class="at">n =</span> top_n)</span>
<span id="cb143-37"><a href="#cb143-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-38"><a href="#cb143-38" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(top_deg_normal)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 Ã— 7
  miRNA degree_normal degree_tumour strength_normal strength_tumour degree_delta
  &lt;chr&gt;         &lt;dbl&gt;         &lt;dbl&gt;           &lt;dbl&gt;           &lt;dbl&gt;        &lt;dbl&gt;
1 TSPAâ€¦            49            25            1.78           0.975          -24
2 SKAP1            49            36            1.38           0.868          -13
3 LARP6            46            38            1.16           1.18            -8
4 ANGPâ€¦            45            20            1.31           0.836          -25
5 LYVE1            43            28            1.50           0.939          -15
6 PLGLâ€¦            43            41            1.17           1.31            -2
# â„¹ 1 more variable: strength_delta &lt;dbl&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb145"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb145-1"><a href="#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(top_deg_tumour)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 Ã— 7
  miRNA degree_normal degree_tumour strength_normal strength_tumour degree_delta
  &lt;chr&gt;         &lt;dbl&gt;         &lt;dbl&gt;           &lt;dbl&gt;           &lt;dbl&gt;        &lt;dbl&gt;
1 RELN             28            52            1.18            1.33           24
2 VIL1             35            50            1.44            1.33           15
3 TESC             36            49            1.16            1.34           13
4 C8orâ€¦            36            48            1.31            1.47           12
5 CYP7â€¦            33            47            1.32            1.01           14
6 HAL              26            47            1.24            1.54           21
# â„¹ 1 more variable: strength_delta &lt;dbl&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb147"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb147-1"><a href="#cb147-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(top_degree_gain)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 Ã— 7
  miRNA degree_normal degree_tumour strength_normal strength_tumour degree_delta
  &lt;chr&gt;         &lt;dbl&gt;         &lt;dbl&gt;           &lt;dbl&gt;           &lt;dbl&gt;        &lt;dbl&gt;
1 RELN             28            52           1.18             1.33           24
2 C3orâ€¦            13            37           0.531            1.26           24
3 LCN2             20            43           0.964            1.46           23
4 PAGE4            19            40           0.856            1.32           21
5 HAL              26            47           1.24             1.54           21
6 HSD1â€¦            12            32           0.462            1.03           20
# â„¹ 1 more variable: strength_delta &lt;dbl&gt;</code></pre>
</div>
</div>
<p>COMPARISON GLASSO AND PAIRED DATA â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”â€”-</p>
<p>Do glasso edges fall inside wgcna modules?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb149"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb149-1"><a href="#cb149-1" aria-hidden="true" tabindex="-1"></a><span class="co">#WGCNA gene -&gt; module</span></span>
<span id="cb149-2"><a href="#cb149-2" aria-hidden="true" tabindex="-1"></a><span class="co">#take out grey module</span></span>
<span id="cb149-3"><a href="#cb149-3" aria-hidden="true" tabindex="-1"></a>moduleColors_wgcna <span class="ot">&lt;-</span> moduleColors[moduleColors <span class="sc">!=</span> <span class="st">"grey"</span>]</span>
<span id="cb149-4"><a href="#cb149-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb149-5"><a href="#cb149-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb149-6"><a href="#cb149-6" aria-hidden="true" tabindex="-1"></a><span class="co">#features used in glasso</span></span>
<span id="cb149-7"><a href="#cb149-7" aria-hidden="true" tabindex="-1"></a>featN <span class="ot">&lt;-</span> <span class="fu">colnames</span>(X_N)</span>
<span id="cb149-8"><a href="#cb149-8" aria-hidden="true" tabindex="-1"></a>featT <span class="ot">&lt;-</span> <span class="fu">colnames</span>(X_T)</span>
<span id="cb149-9"><a href="#cb149-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb149-10"><a href="#cb149-10" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(ThetaN) <span class="ot">&lt;-</span> <span class="fu">list</span>(featN, featN)</span>
<span id="cb149-11"><a href="#cb149-11" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(ThetaT) <span class="ot">&lt;-</span> <span class="fu">list</span>(featT, featT)</span>
<span id="cb149-12"><a href="#cb149-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb149-13"><a href="#cb149-13" aria-hidden="true" tabindex="-1"></a><span class="co">#compare only genes present in BOTH WGCNA modules and both glasso fits</span></span>
<span id="cb149-14"><a href="#cb149-14" aria-hidden="true" tabindex="-1"></a>genes_wgcna <span class="ot">&lt;-</span> <span class="fu">names</span>(moduleColors_wgcna)</span>
<span id="cb149-15"><a href="#cb149-15" aria-hidden="true" tabindex="-1"></a>common_glasso <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(intersect, <span class="fu">list</span>(genes_wgcna, featN, featT))</span>
<span id="cb149-16"><a href="#cb149-16" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(common_glasso)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 398</code></pre>
</div>
<div class="sourceCode cell-code" id="cb151"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb151-1"><a href="#cb151-1" aria-hidden="true" tabindex="-1"></a><span class="co">#named vector gene -&gt; module</span></span>
<span id="cb151-2"><a href="#cb151-2" aria-hidden="true" tabindex="-1"></a>mod2 <span class="ot">&lt;-</span> moduleColors_wgcna[common_glasso]  </span>
<span id="cb151-3"><a href="#cb151-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-4"><a href="#cb151-4" aria-hidden="true" tabindex="-1"></a>edge_frac_within <span class="ot">&lt;-</span> <span class="cf">function</span>(Theta, mod2, common) {</span>
<span id="cb151-5"><a href="#cb151-5" aria-hidden="true" tabindex="-1"></a>  Theta <span class="ot">&lt;-</span> Theta[common, common, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb151-6"><a href="#cb151-6" aria-hidden="true" tabindex="-1"></a>  A <span class="ot">&lt;-</span> <span class="fu">abs</span>(Theta) <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb151-7"><a href="#cb151-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">diag</span>(A) <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb151-8"><a href="#cb151-8" aria-hidden="true" tabindex="-1"></a>  idx <span class="ot">&lt;-</span> <span class="fu">which</span>(A <span class="sc">&amp;</span> <span class="fu">upper.tri</span>(A), <span class="at">arr.ind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb151-9"><a href="#cb151-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">nrow</span>(idx) <span class="sc">==</span> <span class="dv">0</span>) <span class="fu">return</span>(<span class="cn">NA_real_</span>)</span>
<span id="cb151-10"><a href="#cb151-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-11"><a href="#cb151-11" aria-hidden="true" tabindex="-1"></a>  m1 <span class="ot">&lt;-</span> mod2[<span class="fu">rownames</span>(Theta)[idx[,<span class="dv">1</span>]]]</span>
<span id="cb151-12"><a href="#cb151-12" aria-hidden="true" tabindex="-1"></a>  m2 <span class="ot">&lt;-</span> mod2[<span class="fu">colnames</span>(Theta)[idx[,<span class="dv">2</span>]]]</span>
<span id="cb151-13"><a href="#cb151-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mean</span>(m1 <span class="sc">==</span> m2, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb151-14"><a href="#cb151-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb151-15"><a href="#cb151-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-16"><a href="#cb151-16" aria-hidden="true" tabindex="-1"></a>f_within_N <span class="ot">&lt;-</span> <span class="fu">edge_frac_within</span>(ThetaN, mod2, common_glasso)</span>
<span id="cb151-17"><a href="#cb151-17" aria-hidden="true" tabindex="-1"></a>f_within_T <span class="ot">&lt;-</span> <span class="fu">edge_frac_within</span>(ThetaT, mod2, common_glasso)</span>
<span id="cb151-18"><a href="#cb151-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-19"><a href="#cb151-19" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="at">within_edge_fraction_normal =</span> f_within_N,</span>
<span id="cb151-20"><a href="#cb151-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">within_edge_fraction_tumour =</span> f_within_T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>within_edge_fraction_normal within_edge_fraction_tumour 
                  0.6178862                   0.6536038 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb153"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb153-1"><a href="#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="co">#expected within-module fraction under random edges given module sizes</span></span>
<span id="cb153-2"><a href="#cb153-2" aria-hidden="true" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">length</span>(mod2)</span>
<span id="cb153-3"><a href="#cb153-3" aria-hidden="true" tabindex="-1"></a>expected_within <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">sapply</span>(<span class="fu">split</span>(mod2, mod2), <span class="cf">function</span>(x) <span class="fu">choose</span>(<span class="fu">length</span>(x), <span class="dv">2</span>))) <span class="sc">/</span> <span class="fu">choose</span>(n, <span class="dv">2</span>)</span>
<span id="cb153-4"><a href="#cb153-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb153-5"><a href="#cb153-5" aria-hidden="true" tabindex="-1"></a>expected_within</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.3473033</code></pre>
</div>
<div class="sourceCode cell-code" id="cb155"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb155-1"><a href="#cb155-1" aria-hidden="true" tabindex="-1"></a>enrich_normal <span class="ot">&lt;-</span> f_within_N <span class="sc">/</span> expected_within</span>
<span id="cb155-2"><a href="#cb155-2" aria-hidden="true" tabindex="-1"></a>enrich_tumour <span class="ot">&lt;-</span> f_within_T <span class="sc">/</span> expected_within</span>
<span id="cb155-3"><a href="#cb155-3" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="at">enrich_normal =</span> enrich_normal, <span class="at">enrich_tumour =</span> enrich_tumour)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>enrich_normal enrich_tumour 
     1.779097      1.881940 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb157"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb157-1"><a href="#cb157-1" aria-hidden="true" tabindex="-1"></a><span class="co">#WGCNA gene -&gt; module (reference partition)</span></span>
<span id="cb157-2"><a href="#cb157-2" aria-hidden="true" tabindex="-1"></a>moduleColors_wgcna <span class="ot">&lt;-</span> moduleColors[moduleColors <span class="sc">!=</span><span class="dv">0</span>]</span>
<span id="cb157-3"><a href="#cb157-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-4"><a href="#cb157-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-5"><a href="#cb157-5" aria-hidden="true" tabindex="-1"></a><span class="co">#common genes used in both glasso fits and present in WGCNA, just to be safe</span></span>
<span id="cb157-6"><a href="#cb157-6" aria-hidden="true" tabindex="-1"></a>featN <span class="ot">&lt;-</span> <span class="fu">rownames</span>(ThetaN)</span>
<span id="cb157-7"><a href="#cb157-7" aria-hidden="true" tabindex="-1"></a>featT <span class="ot">&lt;-</span> <span class="fu">rownames</span>(ThetaT)</span>
<span id="cb157-8"><a href="#cb157-8" aria-hidden="true" tabindex="-1"></a>common_glasso <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(intersect, <span class="fu">list</span>(<span class="fu">names</span>(moduleColors_wgcna), featN, featT))</span>
<span id="cb157-9"><a href="#cb157-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-10"><a href="#cb157-10" aria-hidden="true" tabindex="-1"></a><span class="co">#module labels in the SAME order as common_glasso</span></span>
<span id="cb157-11"><a href="#cb157-11" aria-hidden="true" tabindex="-1"></a>mod_glasso <span class="ot">&lt;-</span> moduleColors_wgcna[common_glasso]</span>
<span id="cb157-12"><a href="#cb157-12" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">identical</span>(<span class="fu">names</span>(mod_glasso), common_glasso))</span>
<span id="cb157-13"><a href="#cb157-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-14"><a href="#cb157-14" aria-hidden="true" tabindex="-1"></a>count_edges_between_modules <span class="ot">&lt;-</span> <span class="cf">function</span>(Theta, mod_glasso, common_glasso) {</span>
<span id="cb157-15"><a href="#cb157-15" aria-hidden="true" tabindex="-1"></a>  Theta <span class="ot">&lt;-</span> Theta[common_glasso, common_glasso, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb157-16"><a href="#cb157-16" aria-hidden="true" tabindex="-1"></a>  A <span class="ot">&lt;-</span> <span class="fu">abs</span>(Theta) <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb157-17"><a href="#cb157-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">diag</span>(A) <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb157-18"><a href="#cb157-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-19"><a href="#cb157-19" aria-hidden="true" tabindex="-1"></a>  idx <span class="ot">&lt;-</span> <span class="fu">which</span>(A <span class="sc">&amp;</span> <span class="fu">upper.tri</span>(A), <span class="at">arr.ind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb157-20"><a href="#cb157-20" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">nrow</span>(idx) <span class="sc">==</span> <span class="dv">0</span>) <span class="fu">return</span>(<span class="fu">matrix</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>))</span>
<span id="cb157-21"><a href="#cb157-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-22"><a href="#cb157-22" aria-hidden="true" tabindex="-1"></a>  g1 <span class="ot">&lt;-</span> <span class="fu">rownames</span>(Theta)[idx[,<span class="dv">1</span>]]</span>
<span id="cb157-23"><a href="#cb157-23" aria-hidden="true" tabindex="-1"></a>  g2 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(Theta)[idx[,<span class="dv">2</span>]]</span>
<span id="cb157-24"><a href="#cb157-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-25"><a href="#cb157-25" aria-hidden="true" tabindex="-1"></a>  m1 <span class="ot">&lt;-</span> mod_glasso[g1]</span>
<span id="cb157-26"><a href="#cb157-26" aria-hidden="true" tabindex="-1"></a>  m2 <span class="ot">&lt;-</span> mod_glasso[g2]</span>
<span id="cb157-27"><a href="#cb157-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-28"><a href="#cb157-28" aria-hidden="true" tabindex="-1"></a>  tab <span class="ot">&lt;-</span> <span class="fu">table</span>(<span class="fu">pmin</span>(m1, m2), <span class="fu">pmax</span>(m1, m2))</span>
<span id="cb157-29"><a href="#cb157-29" aria-hidden="true" tabindex="-1"></a>  tab</span>
<span id="cb157-30"><a href="#cb157-30" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb157-31"><a href="#cb157-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-32"><a href="#cb157-32" aria-hidden="true" tabindex="-1"></a>tabN <span class="ot">&lt;-</span> <span class="fu">count_edges_between_modules</span>(ThetaN, mod_glasso, common_glasso)</span>
<span id="cb157-33"><a href="#cb157-33" aria-hidden="true" tabindex="-1"></a>tabT <span class="ot">&lt;-</span> <span class="fu">count_edges_between_modules</span>(ThetaT, mod_glasso, common_glasso)</span>
<span id="cb157-34"><a href="#cb157-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-35"><a href="#cb157-35" aria-hidden="true" tabindex="-1"></a>tabN</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           
            brown cyan green grey purple  tan turquoise
  brown      1330   26   172  745     90   12      1064
  cyan          0  164    33   62     17    3        68
  green         0    0    95  165      8    1       174
  grey          0    0     0  409    105   14       838
  purple        0    0     0    0     35    0       112
  tan           0    0     0    0      0    1         6
  turquoise     0    0     0    0      0    0      1263</code></pre>
</div>
<div class="sourceCode cell-code" id="cb159"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb159-1"><a href="#cb159-1" aria-hidden="true" tabindex="-1"></a>tabT</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           
            brown cyan green grey purple  tan turquoise
  brown      1557   61   186  818    173   22       871
  cyan          0  198    34   94     12    0        82
  green         0    0   110  130     11    0       178
  grey          0    0     0  407    119    5       676
  purple        0    0     0    0     69    0       117
  tan           0    0     0    0      0    1        12
  turquoise     0    0     0    0      0    0      1384</code></pre>
</div>
<div class="sourceCode cell-code" id="cb161"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb161-1"><a href="#cb161-1" aria-hidden="true" tabindex="-1"></a><span class="co">#difference table</span></span>
<span id="cb161-2"><a href="#cb161-2" aria-hidden="true" tabindex="-1"></a>mods <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">unique</span>(<span class="fu">c</span>(<span class="fu">rownames</span>(tabN), <span class="fu">colnames</span>(tabN), <span class="fu">rownames</span>(tabT), <span class="fu">colnames</span>(tabT))))</span>
<span id="cb161-3"><a href="#cb161-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb161-4"><a href="#cb161-4" aria-hidden="true" tabindex="-1"></a>make_square <span class="ot">&lt;-</span> <span class="cf">function</span>(tab, mods) {</span>
<span id="cb161-5"><a href="#cb161-5" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>L, <span class="at">nrow =</span> <span class="fu">length</span>(mods), <span class="at">ncol =</span> <span class="fu">length</span>(mods),</span>
<span id="cb161-6"><a href="#cb161-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">dimnames =</span> <span class="fu">list</span>(mods, mods))</span>
<span id="cb161-7"><a href="#cb161-7" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(tab) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb161-8"><a href="#cb161-8" aria-hidden="true" tabindex="-1"></a>    out[<span class="fu">rownames</span>(tab), <span class="fu">colnames</span>(tab)] <span class="ot">&lt;-</span> tab</span>
<span id="cb161-9"><a href="#cb161-9" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb161-10"><a href="#cb161-10" aria-hidden="true" tabindex="-1"></a>  out</span>
<span id="cb161-11"><a href="#cb161-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb161-12"><a href="#cb161-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb161-13"><a href="#cb161-13" aria-hidden="true" tabindex="-1"></a>Nmat <span class="ot">&lt;-</span> <span class="fu">make_square</span>(tabN, mods)</span>
<span id="cb161-14"><a href="#cb161-14" aria-hidden="true" tabindex="-1"></a>Tmat <span class="ot">&lt;-</span> <span class="fu">make_square</span>(tabT, mods)</span>
<span id="cb161-15"><a href="#cb161-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb161-16"><a href="#cb161-16" aria-hidden="true" tabindex="-1"></a>diff_mat <span class="ot">&lt;-</span> Tmat <span class="sc">-</span> Nmat</span>
<span id="cb161-17"><a href="#cb161-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb161-18"><a href="#cb161-18" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(diff_mat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 315</code></pre>
</div>
</div>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "î§‹";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>